<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.554">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>GD benchmark sets</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script type="module" src="../site_libs/quarto-ojs/quarto-ojs-runtime.js"></script>
<link href="../site_libs/quarto-ojs/quarto-ojs.css" rel="stylesheet">

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body>

<div id="quarto-search-results"></div>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#datasets-in-use" id="toc-datasets-in-use" class="nav-link active" data-scroll-target="#datasets-in-use"><span class="header-section-number">1</span> Datasets in use</a>
  <ul class="collapse">
  <li><a href="#classification-of-the-datasets" id="toc-classification-of-the-datasets" class="nav-link" data-scroll-target="#classification-of-the-datasets"><span class="header-section-number">1.1</span> Classification of the Datasets</a>
  <ul class="collapse">
  <li><a href="#sec-uniform-benchmark-datasets" id="toc-sec-uniform-benchmark-datasets" class="nav-link" data-scroll-target="#sec-uniform-benchmark-datasets">Uniform Benchmark datasets</a></li>
  <li><a href="#sec-established-network-repositories" id="toc-sec-established-network-repositories" class="nav-link" data-scroll-target="#sec-established-network-repositories">Established Network Repositories</a></li>
  <li><a href="#sec-single-graphs" id="toc-sec-single-graphs" class="nav-link" data-scroll-target="#sec-single-graphs">Single Graphs</a></li>
  <li><a href="#sec-aggregate-collections" id="toc-sec-aggregate-collections" class="nav-link" data-scroll-target="#sec-aggregate-collections">Aggregate Collections</a></li>
  <li><a href="#sec-lost" id="toc-sec-lost" class="nav-link" data-scroll-target="#sec-lost">Lost and unavailable datasets</a></li>
  <li><a href="#sec-custom-made-datasets" id="toc-sec-custom-made-datasets" class="nav-link" data-scroll-target="#sec-custom-made-datasets">Custom-made Datasets</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">




<section id="datasets-in-use" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Datasets in use</h1>
<p>The following chart shows how many times we found a dataset being used in the papers we collected. It excludes custom edits to the datasets, which are discussed later in this document.</p>
<div class="cell">
<div class="sourceCode cell-code hidden" id="cb1" data-startfrom="8" data-source-offset="0"><pre class="sourceCode js code-with-copy"><code class="sourceCode javascript" style="counter-reset: source-line 7;"><span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>{ </span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>    <span class="kw">let</span> d <span class="op">=</span> []</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>    <span class="kw">let</span> allbenchmarkdatasets <span class="op">=</span> [<span class="op">...</span><span class="kw">new</span> <span class="bu">Set</span>(literature<span class="op">.</span><span class="fu">map</span>(l <span class="kw">=&gt;</span> l[<span class="st">"Dataset tag relations"</span>]<span class="op">.</span><span class="fu">split</span>(<span class="st">"("</span>)[<span class="dv">0</span>]<span class="op">.</span><span class="fu">trim</span>()))]<span class="op">.</span><span class="fu">filter</span>(a <span class="kw">=&gt;</span> a <span class="op">!=</span> <span class="st">""</span>)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>    allbenchmarkdatasets <span class="op">=</span> allbenchmarkdatasets<span class="op">.</span><span class="fu">filter</span>(b <span class="kw">=&gt;</span> <span class="op">!</span>b<span class="op">.</span><span class="fu">includes</span>(<span class="st">"Custom"</span>))</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>    allbenchmarkdatasets <span class="op">=</span> allbenchmarkdatasets<span class="op">.</span><span class="fu">slice</span>(<span class="dv">0</span><span class="op">,</span> <span class="dv">20</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>    <span class="co">//console.log(allbenchmarkdatasets)</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (<span class="kw">let</span> dataset <span class="kw">of</span> allbenchmarkdatasets){</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>        <span class="kw">let</span> literature_entries_with_this_dataset <span class="op">=</span> literature<span class="op">.</span><span class="fu">filter</span>(l <span class="kw">=&gt;</span> l[<span class="st">"Dataset tag relations"</span>]<span class="op">.</span><span class="fu">includes</span>(dataset))</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>        d<span class="op">.</span><span class="fu">push</span>({<span class="st">"dataset"</span><span class="op">:</span> dataset<span class="op">,</span> <span class="st">"count"</span><span class="op">:</span> literature_entries_with_this_dataset<span class="op">.</span><span class="at">length</span>})</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> Plot<span class="op">.</span><span class="fu">plot</span>({</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>        <span class="dt">color</span><span class="op">:</span> {</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>      <span class="co">// legend: true,</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>      <span class="co">// type: "categorical",</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>      <span class="co">// domain: d3.range(10).map((d, i) =&gt; `Category ${i + 1}`),</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>        <span class="co">// range: ["steelblue"],</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>          <span class="dt">scheme</span><span class="op">:</span> <span class="st">"Tableau10"</span> <span class="co">// use the "accent" scheme</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>        }<span class="op">,</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>        <span class="dt">y</span><span class="op">:</span> {</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>      <span class="co">// tickFormat: "s"</span></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>        }<span class="op">,</span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>        <span class="dt">marks</span><span class="op">:</span> [</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>            Plot<span class="op">.</span><span class="fu">barX</span>(d<span class="op">,</span> {<span class="dt">x</span><span class="op">:</span> <span class="st">"count"</span><span class="op">,</span> <span class="dt">y</span><span class="op">:</span> <span class="st">"dataset"</span><span class="op">,</span> <span class="dt">fill</span><span class="op">:</span> <span class="st">"dataset"</span><span class="op">,</span> <span class="dt">inset</span><span class="op">:</span> <span class="dv">2</span><span class="op">,</span> <span class="dt">sort</span><span class="op">:</span> {<span class="dt">y</span><span class="op">:</span> <span class="st">"x"</span><span class="op">,</span> <span class="dt">reverse</span><span class="op">:</span> <span class="kw">true</span>}})<span class="op">,</span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>            Plot<span class="op">.</span><span class="fu">axisY</span>({<span class="dt">label</span><span class="op">:</span> <span class="kw">null</span><span class="op">,</span> <span class="dt">lineWidth</span><span class="op">:</span> <span class="dv">12</span><span class="op">,</span> <span class="dt">marginLeft</span><span class="op">:</span> <span class="dv">150</span>})<span class="op">,</span></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>        ]</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="fig-ojs-cell-1" class="cell-output cell-output-display quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ojs-cell-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div id="ojs-cell-1" data-nodetype="expression">

</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ojs-cell-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: This barchart shows, for the top 20 most used datasets, the number of times a dataset was used in collected papers.
</figcaption>
</figure>
</div>
</div>
<p>In the data we collected, the most used dataset is <a href="#Rome-Lib" class="dataset-link">Rome-Lib</a>, followed by <a href="#Assorted_collaboration_networks" class="dataset-link">assorted collaboration networks</a> (which in many cases refers to datasets of academic collaborations such as dblp or vispubdata). The third most used dataset is from <a href="#Walshaw" class="dataset-link">C.Walshaw</a> - it is important to note that the Walshaw dataset is available as part of other collections - for instance, its graphs are found in <a href="#SNAP" class="dataset-link">SNAP</a>. However, during the collection process, we preferred giving precedence to how the authors reported their own information. Thus, if the authors claimed the data was from the Walshaw collection, we reported it as such.</p>
<!-- We deem the popularity of Rome-Lib to be due to its simplicity, versatility, and most importantly the fact that it is linked on graphdrawing.org and offered as a benchmark dataset there. This is also true for the North DAGs. -->
<p>For each of the datasets collected as part of our process, we conducted a brief analysis of their contents. Where possible, the analysis includes information about the number of nodes per graph, the source of the dataset, which papers have used the dataset and what graph features they took into account.</p>
<p>In the following sections, the reader will find details about the classifications and datasets in detail. Each dataset gets a dedicated, collapsible section, that contains the following information:</p>
<ul>
<li>A brief summary description of the dataset, the reasons for its popularity, and notes about how it was created or collected.</li>
<li>Metadata containing:
<ul>
<li>The original source of the dataset.</li>
<li>Links to download the dataset in different formats, provided from our OSF repository.</li>
<li>The size of the dataset.</li>
<li>The original paper where the dataset was first introduced.</li>
<li>A list of papers that have used the dataset, together with what features their graph layout algorithm handles.</li>
</ul></li>
<li>Charts illustrating node and edge count distribution through the dataset, mean and maximum distribution.</li>
<li>Descriptions of the datasets lifted directly from the literature, which can help shine light on the content and properties of the dataset, how it was used in previous literature, or how it originated.</li>
<li>Figures collected from the literature, such as visualizations of the graphs, or other relevant information. The visualizations can help a reader looking for a specific type of graph to quickly identify if the dataset is relevant to their research.</li>
</ul>
<p><a name="acyclic"></a> <a name="bipartite"></a> <a name="bundled_edges_generated"></a> <a name="categorical_nodes"></a> <a name="clusters"></a> <a name="clusters_generated"></a> <a name="dense"></a> <a name="directed_edges"></a> <a name="dynamic"></a> <a name="dynamic_discrete"></a> <a name="edge_weighted"></a> <a name="generic"></a> <a name="hierarchical"></a> <a name="hypergraph"></a> <a name="known_crossing_number"></a> <a name="labeled_nodes"></a> <a name="large"></a> <a name="layered_graphs"></a> <a name="many_layers"></a> <a name="mesh"></a> <a name="multigraph"></a> <a name="node_weighted"></a> <a name="nonplanar"></a> <a name="partition"></a> <a name="planar"></a> <a name="signed"></a> <a name="sparse"></a> <a name="spatial"></a> <a name="temporal"></a> <a name="temporal_event_sequence"></a> <a name="tree-like"></a> <a name="trees"></a></p>
<section id="classification-of-the-datasets" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="classification-of-the-datasets"><span class="header-section-number">1.1</span> Classification of the Datasets</h2>
<p>The datasets we collected are divided in different categories: the categorization we used is the same as the one used in <span class="citation" data-cites="dibartolomeo2024evaluating">@dibartolomeo2024evaluating</span>. The following is a brief explanation of the categories used, which are explained in deeper detail in each one of the corresponding chapters.</p>
<ul>
<li><strong>Uniform Benchmark (<a href="#sec-uniform-benchmark-datasets" class="quarto-xref">Section&nbsp;1.1.1</a>):</strong> Datasets that are standalone widely used collections of graphs featuring uniform characteristics - usually simple, generic graphs, often used in evaluations that run over thousands of graphs to report average completion times, or other experiments where the reported metrics are usually aggregated.</li>
<li><strong>Established Network Repository (<a href="#sec-established-network-repositories" class="quarto-xref">Section&nbsp;1.1.2</a>):</strong> A collection of individual, or groups of graphs, part of a larger repository. These datasets are often used in evaluations that require a diverse set of graphs, or when the evaluation is focused on a specific type of graph. These collections usually also report on graphs with their own statistics or visualizations, providing a summary of the dataset. This category also includes <strong>Subsets (<a href="#sec-subsets" class="quarto-xref">Section&nbsp;1.1.2.10</a>)</strong>.</li>
<li><strong>Single Graphs (<a href="#sec-single-graphs" class="quarto-xref">Section&nbsp;1.1.3</a>):</strong> The data used is comprised of one individual graph, which often contains special characteristics making it particularly relevant for the evaluation at hand.</li>
<li><strong>Aggregate Collections (<a href="#sec-aggregate-collections" class="quarto-xref">Section&nbsp;1.1.4</a>):</strong> Collections of graphs focusing on one specific context (such as forum posts, or airline routes) due to the given context having a special set of features that is relevant for a graph layout algorithm.</li>
</ul>
<p>The type of collection informs various aspects of an evaluation, such as which results are relevant to report. For this discussion, we refer to <span class="citation" data-cites="dibartolomeo2024evaluating">@dibartolomeo2024evaluating</span>, where the authors link the type of collection in the context of graph layout evaluations.</p>
<p>The following visualization shows the amount of datasets we collected per category:</p>
<div class="cell">
<div class="sourceCode cell-code hidden" id="cb2" data-startfrom="110" data-source-offset="0"><pre class="sourceCode js code-with-copy"><code class="sourceCode javascript" style="counter-reset: source-line 109;"><span id="cb2-110"><a href="#cb2-110" aria-hidden="true" tabindex="-1"></a>{ </span>
<span id="cb2-111"><a href="#cb2-111" aria-hidden="true" tabindex="-1"></a>  <span class="kw">let</span> d <span class="op">=</span> []</span>
<span id="cb2-112"><a href="#cb2-112" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-113"><a href="#cb2-113" aria-hidden="true" tabindex="-1"></a>  <span class="kw">let</span> collection_type <span class="op">=</span> benchmark_datasets<span class="op">.</span><span class="fu">map</span>(a <span class="kw">=&gt;</span> a[<span class="st">"Type of Collection"</span>])<span class="op">;</span></span>
<span id="cb2-114"><a href="#cb2-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-115"><a href="#cb2-115" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (<span class="kw">let</span> c <span class="kw">of</span> [<span class="op">...</span> <span class="kw">new</span> <span class="bu">Set</span>(collection_type)]){</span>
<span id="cb2-116"><a href="#cb2-116" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (c <span class="op">==</span> <span class="st">"Skip"</span> <span class="op">||</span> c <span class="op">==</span> <span class="st">"Missed it"</span>) <span class="cf">continue</span><span class="op">;</span></span>
<span id="cb2-117"><a href="#cb2-117" aria-hidden="true" tabindex="-1"></a>    <span class="kw">let</span> count <span class="op">=</span> collection_type<span class="op">.</span><span class="fu">filter</span>(a <span class="kw">=&gt;</span> a <span class="op">==</span> c)<span class="op">.</span><span class="at">length</span><span class="op">;</span></span>
<span id="cb2-118"><a href="#cb2-118" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (c<span class="op">.</span><span class="fu">includes</span>(<span class="st">"No report"</span>)) c <span class="op">=</span> <span class="st">"Established Network Repository"</span><span class="op">;</span></span>
<span id="cb2-119"><a href="#cb2-119" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (c<span class="op">.</span><span class="fu">includes</span>(<span class="st">"Lost"</span>)) <span class="cf">continue</span><span class="op">;</span></span>
<span id="cb2-120"><a href="#cb2-120" aria-hidden="true" tabindex="-1"></a>    d<span class="op">.</span><span class="fu">push</span>({<span class="st">"collection_type"</span><span class="op">:</span> c<span class="op">,</span> <span class="st">"count"</span><span class="op">:</span> count})</span>
<span id="cb2-121"><a href="#cb2-121" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb2-122"><a href="#cb2-122" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-123"><a href="#cb2-123" aria-hidden="true" tabindex="-1"></a>  <span class="cf">return</span> Plot<span class="op">.</span><span class="fu">plot</span>({</span>
<span id="cb2-124"><a href="#cb2-124" aria-hidden="true" tabindex="-1"></a>    <span class="dt">color</span><span class="op">:</span> {</span>
<span id="cb2-125"><a href="#cb2-125" aria-hidden="true" tabindex="-1"></a>      <span class="dt">scheme</span><span class="op">:</span> <span class="st">"Tableau10"</span> </span>
<span id="cb2-126"><a href="#cb2-126" aria-hidden="true" tabindex="-1"></a>    }<span class="op">,</span></span>
<span id="cb2-127"><a href="#cb2-127" aria-hidden="true" tabindex="-1"></a>    <span class="dt">marks</span><span class="op">:</span> [</span>
<span id="cb2-128"><a href="#cb2-128" aria-hidden="true" tabindex="-1"></a>      Plot<span class="op">.</span><span class="fu">barX</span>(d<span class="op">,</span> {<span class="dt">x</span><span class="op">:</span> <span class="st">"count"</span><span class="op">,</span> <span class="dt">y</span><span class="op">:</span> <span class="st">"collection_type"</span><span class="op">,</span> <span class="dt">fill</span><span class="op">:</span> <span class="st">"collection_type"</span><span class="op">,</span> <span class="dt">inset</span><span class="op">:</span> <span class="dv">2</span><span class="op">,</span> <span class="dt">sort</span><span class="op">:</span> {<span class="dt">y</span><span class="op">:</span> <span class="st">"x"</span><span class="op">,</span> <span class="dt">reverse</span><span class="op">:</span> <span class="kw">true</span>}})<span class="op">,</span></span>
<span id="cb2-129"><a href="#cb2-129" aria-hidden="true" tabindex="-1"></a>      Plot<span class="op">.</span><span class="fu">axisY</span>({<span class="dt">label</span><span class="op">:</span> <span class="kw">null</span><span class="op">,</span> <span class="dt">lineWidth</span><span class="op">:</span> <span class="dv">12</span><span class="op">,</span> <span class="dt">marginLeft</span><span class="op">:</span> <span class="dv">150</span>})<span class="op">,</span></span>
<span id="cb2-130"><a href="#cb2-130" aria-hidden="true" tabindex="-1"></a>    ]</span>
<span id="cb2-131"><a href="#cb2-131" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb2-132"><a href="#cb2-132" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="fig-ojs-cell-2" class="cell-output cell-output-display quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ojs-cell-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div id="ojs-cell-2" data-nodetype="expression">

</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ojs-cell-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Occurrences of datasets we collected per category. Unform benchmark datasets were the majority.
</figcaption>
</figure>
</div>
</div>
<p>Unfortunately, some of the datasets used in layout algorithm evaluations are not accessible anymore. Where possible, we still collected and reported all the information contained in the papers about the datasets, so that we can conserve an idea of what they contained. This phenomenon, as long as the information we could collect, is discussed in <strong>Lost and Unavailable Datasets (<a href="#sec-lost" class="quarto-xref">Section&nbsp;1.1.5</a>)</strong>.</p>
<div class="cell">
<div class="sourceCode cell-code hidden" id="cb3" data-startfrom="140" data-source-offset="0"><pre class="sourceCode js code-with-copy"><code class="sourceCode javascript" style="counter-reset: source-line 139;"><span id="cb3-140"><a href="#cb3-140" aria-hidden="true" tabindex="-1"></a>{ </span>
<span id="cb3-141"><a href="#cb3-141" aria-hidden="true" tabindex="-1"></a>  <span class="kw">let</span> d <span class="op">=</span> []</span>
<span id="cb3-142"><a href="#cb3-142" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-143"><a href="#cb3-143" aria-hidden="true" tabindex="-1"></a>  <span class="kw">let</span> collection_type <span class="op">=</span> benchmark_datasets<span class="op">.</span><span class="fu">map</span>(a <span class="kw">=&gt;</span> a[<span class="st">"Type of Collection"</span>])<span class="op">;</span></span>
<span id="cb3-144"><a href="#cb3-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-145"><a href="#cb3-145" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (<span class="kw">let</span> c <span class="kw">of</span> [<span class="op">...</span> <span class="kw">new</span> <span class="bu">Set</span>(collection_type)]){</span>
<span id="cb3-146"><a href="#cb3-146" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (c <span class="op">==</span> <span class="st">"Skip"</span> <span class="op">||</span> c <span class="op">==</span> <span class="st">"Missed it"</span>) <span class="cf">continue</span><span class="op">;</span></span>
<span id="cb3-147"><a href="#cb3-147" aria-hidden="true" tabindex="-1"></a>    <span class="kw">let</span> count <span class="op">=</span> collection_type<span class="op">.</span><span class="fu">filter</span>(a <span class="kw">=&gt;</span> a <span class="op">==</span> c)<span class="op">.</span><span class="at">length</span><span class="op">;</span></span>
<span id="cb3-148"><a href="#cb3-148" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (c<span class="op">.</span><span class="fu">includes</span>(<span class="st">"No report"</span>)) c <span class="op">=</span> <span class="st">"Established Network Repository"</span><span class="op">;</span></span>
<span id="cb3-149"><a href="#cb3-149" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="op">!</span>c<span class="op">.</span><span class="fu">includes</span>(<span class="st">"Lost"</span>)) <span class="cf">continue</span><span class="op">;</span></span>
<span id="cb3-150"><a href="#cb3-150" aria-hidden="true" tabindex="-1"></a>    d<span class="op">.</span><span class="fu">push</span>({<span class="st">"collection_type"</span><span class="op">:</span> c<span class="op">,</span> <span class="st">"count"</span><span class="op">:</span> count})</span>
<span id="cb3-151"><a href="#cb3-151" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb3-152"><a href="#cb3-152" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-153"><a href="#cb3-153" aria-hidden="true" tabindex="-1"></a>  <span class="cf">return</span> Plot<span class="op">.</span><span class="fu">plot</span>({</span>
<span id="cb3-154"><a href="#cb3-154" aria-hidden="true" tabindex="-1"></a>    <span class="dt">color</span><span class="op">:</span> {</span>
<span id="cb3-155"><a href="#cb3-155" aria-hidden="true" tabindex="-1"></a>      <span class="dt">scheme</span><span class="op">:</span> <span class="st">"PiYg"</span> </span>
<span id="cb3-156"><a href="#cb3-156" aria-hidden="true" tabindex="-1"></a>    }<span class="op">,</span></span>
<span id="cb3-157"><a href="#cb3-157" aria-hidden="true" tabindex="-1"></a>    <span class="dt">marks</span><span class="op">:</span> [</span>
<span id="cb3-158"><a href="#cb3-158" aria-hidden="true" tabindex="-1"></a>      Plot<span class="op">.</span><span class="fu">barX</span>(d<span class="op">,</span> {<span class="dt">x</span><span class="op">:</span> <span class="st">"count"</span><span class="op">,</span> <span class="dt">y</span><span class="op">:</span> <span class="st">"collection_type"</span><span class="op">,</span> <span class="dt">fill</span><span class="op">:</span> <span class="st">"collection_type"</span><span class="op">,</span> <span class="dt">inset</span><span class="op">:</span> <span class="dv">2</span><span class="op">,</span> <span class="dt">sort</span><span class="op">:</span> {<span class="dt">y</span><span class="op">:</span> <span class="st">"x"</span><span class="op">,</span> <span class="dt">reverse</span><span class="op">:</span> <span class="kw">true</span>}})<span class="op">,</span></span>
<span id="cb3-159"><a href="#cb3-159" aria-hidden="true" tabindex="-1"></a>      Plot<span class="op">.</span><span class="fu">axisY</span>({<span class="dt">label</span><span class="op">:</span> <span class="kw">null</span><span class="op">,</span> <span class="dt">lineWidth</span><span class="op">:</span> <span class="dv">12</span><span class="op">,</span> <span class="dt">marginLeft</span><span class="op">:</span> <span class="dv">150</span>})<span class="op">,</span></span>
<span id="cb3-160"><a href="#cb3-160" aria-hidden="true" tabindex="-1"></a>    ]</span>
<span id="cb3-161"><a href="#cb3-161" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb3-162"><a href="#cb3-162" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="fig-ojs-cell-3" class="cell-output cell-output-display quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ojs-cell-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div id="ojs-cell-3" data-nodetype="expression">

</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ojs-cell-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Unfortunately, 9 of the datasets we found being used in the literature were either lost or no longer accessible.
</figcaption>
</figure>
</div>
</div>
<p>Finally, we include a discussion on <strong>Custom-made Datasets (<a href="#sec-custom-made-datasets" class="quarto-xref">Section&nbsp;1.1.6</a>)</strong>, synthetically generated datasets for the purpose of an experiment.</p>
<p>Each dataset also contains a number of graph features specific to the graphs contained in it. Authors might want to use a dataset that contains a specific feature, such as a graph with a high number of nodes, or a graph with a specific structure. The following interactive filtering systems allows for filtering of datasets (and papers) based on the features they contain.</p>
<div class="callout callout-style-simple callout-note callout-titled" title="Tag-filtered Navigation">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tag-filtered Navigation
</div>
</div>
<div class="callout-body-container callout-body">
<p>The following interactive filtering system is an initial overview over the contents of the paper, and can be used to navigate and identify datasets and papers easily. The left column contains all the properties used to tag graphs in the datasets (additional information about the nature of these tags can be found in <strong>?@sec-collection-process</strong> and <strong>?@sec-data-processing</strong>). Each tag has an associated checkbox: selecting a tag (or more than one) will filter the datasets and papers to only show those that contain the selected tags. The middle column contains the datasets, and the right column contains the papers. Clicking on a dataset or paper will scroll the page to the corresponding section in the paper. The datasets and papers are sorted alphabetically by name and citation name, respectively. The tags, datasets, and papers are generated from the data in the repository, and are updated automatically when the data changes.</p>
</div>
</div>
<div style="font-size: 0.9em">
<div class="row">
<div class="col-4 justify-content-center">
<b>Tags</b>
</div>
<div class="col-4 justify-content-center">
<b>Datsets</b>
</div>
<div class="col-4 justify-content-center">
<b>Papers</b>
</div>
</div>
<div class="row" style="max-height: 600px">
<div class="col-4" style="overflow-y:scroll; padding:2%">
<div id="checkbox-container" class="d-flex flex-column align-items-start">

</div>
</div>
<div class="col-4 d-flex flex-column align-items-start" style="overflow-y:scroll; padding:2%">
<div id="datasetlinkslist-container">
<div id="Airlines_Migrations_and_Air_Traffic-link">
<a href="#Airlines_Migrations_and_Air_Traffic" class="dataset-link">Airlines, Migrations, and Air Traffic</a>
</div>
<div id="Amazon-link">
<a href="#Amazon" class="dataset-link">Amazon</a>
</div>
<div id="Assorted_Collaboration_Network-link">
<a href="#Assorted_Collaboration_Network" class="dataset-link">Assorted Collaboration Network</a>
</div>
<div id="Assorted_Fiscal_Network-link">
<a href="#Assorted_Fiscal_Network" class="dataset-link">Assorted Fiscal Network</a>
</div>
<div id="ATT-link">
<a href="#ATT" class="dataset-link">AT&amp;T</a>
</div>
<div id="Autonomous_System_Network-link">
<a href="#Autonomous_System_Network" class="dataset-link">Autonomous System Network</a>
</div>
<div id="Biological_Pathways_KEGG-link">
<a href="#Biological_Pathways_KEGG" class="dataset-link">Biological Pathways (KEGG)</a>
</div>
<div id="Blogposts_Tweets_and_Forums-link">
<a href="#Blogposts_Tweets_and_Forums" class="dataset-link">Blogposts, Tweets, and Forums</a>
</div>
<div id="Business_Processes-link">
<a href="#Business_Processes" class="dataset-link">Business Processes</a>
</div>
<div id="California-link">
<a href="#California" class="dataset-link">California</a>
</div>
<div id="Car_Features-link">
<a href="#Car_Features" class="dataset-link">Car Features</a>
</div>
<div id="Chess_Games-link">
<a href="#Chess_Games" class="dataset-link">Chess Games</a>
</div>
<div id="World_Greenhouse_Gas_Emissions-link">
<a href="#World_Greenhouse_Gas_Emissions" class="dataset-link">World Greenhouse Gas Emissions</a>
</div>
<div id="Co-Phylogenetic_Trees-link">
<a href="#Co-Phylogenetic_Trees" class="dataset-link">Co-Phylogenetic Trees</a>
</div>
<div id="Code_Dependency_Graphs-link">
<a href="#Code_Dependency_Graphs" class="dataset-link">Code Dependency Graphs</a>
</div>
<div id="Complete_Graphs-link">
<a href="#Complete_Graphs" class="dataset-link">Complete Graphs</a>
</div>
<div id="World_Maps-link">
<a href="#World_Maps" class="dataset-link">World Maps</a>
</div>
<div id="Control_Flow_Graphs-link">
<a href="#Control_Flow_Graphs" class="dataset-link">Control Flow Graphs</a>
</div>
<div id="Debates-link">
<a href="#Debates" class="dataset-link">Debates</a>
</div>
<div id="Enron-link">
<a href="#Enron" class="dataset-link">Enron</a>
</div>
<div id="FM3-link">
<a href="#FM3" class="dataset-link">FM3</a>
</div>
<div id="GION-link">
<a href="#GION" class="dataset-link">GION</a>
</div>
<div id="Graphviz_Examples-link">
<a href="#Graphviz_Examples" class="dataset-link">Graphviz Examples</a>
</div>
<div id="Internet_Mapping_Project-link">
<a href="#Internet_Mapping_Project" class="dataset-link">Internet Mapping Project</a>
</div>
<div id="Investment_Interdependence-link">
<a href="#Investment_Interdependence" class="dataset-link">Investment Interdependence</a>
</div>
<div id="KnownCR-link">
<a href="#KnownCR" class="dataset-link">KnownCR</a>
</div>
<div id="Mathematics_Genealogy-link">
<a href="#Mathematics_Genealogy" class="dataset-link">Mathematics Genealogy</a>
</div>
<div id="Matrix_Market-link">
<a href="#Matrix_Market" class="dataset-link">Matrix Market</a>
</div>
<div id="Medical_Patient_Records-link">
<a href="#Medical_Patient_Records" class="dataset-link">Medical Patient Records</a>
</div>
<div id="Militarized_Interstate_Disputes_MID-link">
<a href="#Militarized_Interstate_Disputes_MID" class="dataset-link">Militarized Interstate Disputes (MID)</a>
</div>
<div id="Network_Repository-link">
<a href="#Network_Repository" class="dataset-link">Network Repository</a>
</div>
<div id="Neural_Network-link">
<a href="#Neural_Network" class="dataset-link">Neural Network</a>
</div>
<div id="North_DAGs-link">
<a href="#North_DAGs" class="dataset-link">North DAGs</a>
</div>
<div id="Padia_Stories-link">
<a href="#Padia_Stories" class="dataset-link">Padia Stories</a>
</div>
<div id="Pajek-link">
<a href="#Pajek" class="dataset-link">Pajek</a>
</div>
<div id="Pert_DAG-link">
<a href="#Pert_DAG" class="dataset-link">Pert DAG</a>
</div>
<div id="Protein_Interactions_Publications-link">
<a href="#Protein_Interactions_Publications" class="dataset-link">Protein Interactions (Publications)</a>
</div>
<div id="RandDAG-link">
<a href="#RandDAG" class="dataset-link">RandDAG</a>
</div>
<div id="Rome-Lib-link">
<a href="#Rome-Lib" class="dataset-link">Rome-Lib</a>
</div>
<div id="Scotch_Graph_Collection-link">
<a href="#Scotch_Graph_Collection" class="dataset-link">Scotch Graph Collection</a>
</div>
<div id="SNAP_Stanford_Network_Analysis_Platform-link">
<a href="#SNAP_Stanford_Network_Analysis_Platform" class="dataset-link">SNAP (Stanford Network Analysis Platform)</a>
</div>
<div id="Social_Networks-link">
<a href="#Social_Networks" class="dataset-link">Social Networks</a>
</div>
<div id="Stanford_GraphBase-link">
<a href="#Stanford_GraphBase" class="dataset-link">Stanford GraphBase</a>
</div>
<div id="SteinLib-link">
<a href="#SteinLib" class="dataset-link">SteinLib</a>
</div>
<div id="Storylines_Movie_Plots-link">
<a href="#Storylines_Movie_Plots" class="dataset-link">Storylines (Movie Plots)</a>
</div>
<div id="SuiteSparse_Matrix_Collection-link">
<a href="#SuiteSparse_Matrix_Collection" class="dataset-link">SuiteSparse Matrix Collection</a>
</div>
<div id="Toblers_Flow_Mapper-link">
<a href="#Toblers_Flow_Mapper" class="dataset-link">Tobler’s Flow Mapper</a>
</div>
<div id="Transportation_Networks-link">
<a href="#Transportation_Networks" class="dataset-link">Transportation Networks</a>
</div>
<div id="Tree_of_Life-link">
<a href="#Tree_of_Life" class="dataset-link">Tree of Life</a>
</div>
<div id="Walshaw-link">
<a href="#Walshaw" class="dataset-link">Walshaw</a>
</div>
<div id="WebCompute-link">
<a href="#WebCompute" class="dataset-link">WebCompute</a>
</div>
<div id="World_Bank_Trade_Data-link">
<a href="#World_Bank_Trade_Data" class="dataset-link">World Bank Trade Data</a>
</div>
</div>
</div>
<div class="col-4 d-flex flex-column align-items-start" style="overflow-y:scroll; padding:2%">
<div id="paperlinkslist-container">
<div id="Ahmed2020-link">
<span class="citation" data-cites="Ahmed2020">[@Ahmed2020]</span>
</div>
<div id="Alegra2021-link">
<span class="citation" data-cites="Alegra2021">[@Alegra2021]</span>
</div>
<div id="Angori2019-link">
<span class="citation" data-cites="Angori2019">[@Angori2019]</span>
</div>
<div id="Arleo2019-link">
<span class="citation" data-cites="Arleo2019">[@Arleo2019]</span>
</div>
<div id="Bach2014-link">
<span class="citation" data-cites="Bach2014">[@Bach2014]</span>
</div>
<div id="Ballweg2018-link">
<span class="citation" data-cites="Ballweg2018">[@Ballweg2018]</span>
</div>
<div id="Barnes1986-link">
<span class="citation" data-cites="Barnes1986">[@Barnes1986]</span>
</div>
<div id="Bartolomeo2021-link">
<span class="citation" data-cites="Bartolomeo2021">[@Bartolomeo2021]</span>
</div>
<div id="Bekos2007-link">
<span class="citation" data-cites="Bekos2007">[@Bekos2007]</span>
</div>
<div id="Bekos2018-link">
<span class="citation" data-cites="Bekos2018">[@Bekos2018]</span>
</div>
<div id="Bertault2000-link">
<span class="citation" data-cites="Bertault2000">[@Bertault2000]</span>
</div>
<div id="Bhatt1984-link">
<span class="citation" data-cites="Bhatt1984">[@Bhatt1984]</span>
</div>
<div id="Bhore2021-link">
<span class="citation" data-cites="Bhore2021">[@Bhore2021]</span>
</div>
<div id="Biedl2018-link">
<span class="citation" data-cites="Biedl2018">[@Biedl2018]</span>
</div>
<div id="Binucci2016-link">
<span class="citation" data-cites="Binucci2016">[@Binucci2016]</span>
</div>
<div id="Binucci2021-link">
<span class="citation" data-cites="Binucci2021">[@Binucci2021]</span>
</div>
<div id="Brsig2020-link">
<span class="citation" data-cites="Brsig2020">[@Brsig2020]</span>
</div>
<div id="Buchheim2006-link">
<span class="citation" data-cites="Buchheim2006">[@Buchheim2006]</span>
</div>
<div id="Buchheim2008-link">
<span class="citation" data-cites="Buchheim2008">[@Buchheim2008]</span>
</div>
<div id="Buchin2011-link">
<span class="citation" data-cites="Buchin2011">[@Buchin2011]</span>
</div>
<div id="Calamoneri2018-link">
<span class="citation" data-cites="Calamoneri2018">[@Calamoneri2018]</span>
</div>
<div id="Castermans2018-link">
<span class="citation" data-cites="Castermans2018">[@Castermans2018]</span>
</div>
<div id="Chen2019-link">
<span class="citation" data-cites="Chen2019">[@Chen2019]</span>
</div>
<div id="Chimani2008-link">
<span class="citation" data-cites="Chimani2008">[@Chimani2008]</span>
</div>
<div id="Chimani2012-link">
<span class="citation" data-cites="Chimani2012">[@Chimani2012]</span>
</div>
<div id="Chimani2021-link">
<span class="citation" data-cites="Chimani2021">[@Chimani2021]</span>
</div>
<div id="Clancy2019-link">
<span class="citation" data-cites="Clancy2019">[@Clancy2019]</span>
</div>
<div id="DIBATTISTA2000-link">
<span class="citation" data-cites="DIBATTISTA2000">[@DIBATTISTA2000]</span>
</div>
<div id="DRGraph21-link">
<span class="citation" data-cites="DRGraph21">[@DRGraph21]</span>
</div>
<div id="Dang2016-link">
<span class="citation" data-cites="dang2016">[@dang2016]</span>
</div>
<div id="Demel2018-link">
<span class="citation" data-cites="Demel2018">[@Demel2018]</span>
</div>
<div id="Devkota2019-link">
<span class="citation" data-cites="Devkota2019">[@Devkota2019]</span>
</div>
<div id="DiBattista1997b-link">
<span class="citation" data-cites="DiBattista1997b">[@DiBattista1997b]</span>
</div>
<div id="DiGiacomo2020-link">
<span class="citation" data-cites="DiGiacomo2020">[@DiGiacomo2020]</span>
</div>
<div id="Didimo2018-link">
<span class="citation" data-cites="Didimo2018">[@Didimo2018]</span>
</div>
<div id="Didimo2019-link">
<span class="citation" data-cites="Didimo2019">[@Didimo2019]</span>
</div>
<div id="Dwyer2013-link">
<span class="citation" data-cites="Dwyer2013">[@Dwyer2013]</span>
</div>
<div id="Eades2018-link">
<span class="citation" data-cites="Eades2018">[@Eades2018]</span>
</div>
<div id="Ersoy2011-link">
<span class="citation" data-cites="Ersoy2011">[@Ersoy2011]</span>
</div>
<div id="Firman2019-link">
<span class="citation" data-cites="Firman2019">[@Firman2019]</span>
</div>
<div id="Forster2002-link">
<span class="citation" data-cites="Forster2002">[@Forster2002]</span>
</div>
<div id="Forster2005-link">
<span class="citation" data-cites="Forster2005">[@Forster2005]</span>
</div>
<div id="Frishman2008-link">
<span class="citation" data-cites="Frishman2008">[@Frishman2008]</span>
</div>
<div id="Frster2021-link">
<span class="citation" data-cites="Frster2021">[@Frster2021]</span>
</div>
<div id="Fruchterman1991-link">
<span class="citation" data-cites="Fruchterman1991">[@Fruchterman1991]</span>
</div>
<div id="Fulek2018-link">
<span class="citation" data-cites="Fulek2018">[@Fulek2018]</span>
</div>
<div id="Gange2011-link">
<span class="citation" data-cites="Gange2011">[@Gange2011]</span>
</div>
<div id="Gansner1993-link">
<span class="citation" data-cites="Gansner1993">[@Gansner1993]</span>
</div>
<div id="Gansner2005-link">
<span class="citation" data-cites="Gansner2005">[@Gansner2005]</span>
</div>
<div id="Gansner2005b-link">
<span class="citation" data-cites="Gansner2005b">[@Gansner2005b]</span>
</div>
<div id="Gansner2006-link">
<span class="citation" data-cites="Gansner2006">[@Gansner2006]</span>
</div>
<div id="Gansner2013-link">
<span class="citation" data-cites="Gansner2013">[@Gansner2013]</span>
</div>
<div id="Gibson2012-link">
<span class="citation" data-cites="Gibson2012">[@Gibson2012]</span>
</div>
<div id="Giovannangeli2021-link">
<span class="citation" data-cites="Giovannangeli2021">[@Giovannangeli2021]</span>
</div>
<div id="Grabowicz2014-link">
<span class="citation" data-cites="Grabowicz2014">[@Grabowicz2014]</span>
</div>
<div id="Gronemann2016-link">
<span class="citation" data-cites="Gronemann2016">[@Gronemann2016]</span>
</div>
<div id="Gronemann2016b-link">
<span class="citation" data-cites="Gronemann2016b">[@Gronemann2016b]</span>
</div>
<div id="Gutwenger2004-link">
<span class="citation" data-cites="Gutwenger2004">[@Gutwenger2004]</span>
</div>
<div id="Hachul2005-link">
<span class="citation" data-cites="Hachul2005">[@Hachul2005]</span>
</div>
<div id="Hachul2006-link">
<span class="citation" data-cites="Hachul2006">[@Hachul2006]</span>
</div>
<div id="Heinsohn2018-link">
<span class="citation" data-cites="Heinsohn2018">[@Heinsohn2018]</span>
</div>
<div id="Henry2007-link">
<span class="citation" data-cites="Henry2007">[@Henry2007]</span>
</div>
<div id="Holten2006-link">
<span class="citation" data-cites="Holten2006">[@Holten2006]</span>
</div>
<div id="Holten2009-link">
<span class="citation" data-cites="Holten2009">[@Holten2009]</span>
</div>
<div id="Hong2019-link">
<span class="citation" data-cites="Hong2019">[@Hong2019]</span>
</div>
<div id="Huang2016-link">
<span class="citation" data-cites="Huang2016">[@Huang2016]</span>
</div>
<div id="Hummel2019-link">
<span class="citation" data-cites="Hummel2019">[@Hummel2019]</span>
</div>
<div id="Jabrayilov2016-link">
<span class="citation" data-cites="Jabrayilov2016">[@Jabrayilov2016]</span>
</div>
<div id="Jacomy2014-link">
<span class="citation" data-cites="Jacomy2014">[@Jacomy2014]</span>
</div>
<div id="Jnger2018-link">
<span class="citation" data-cites="Jnger2018">[@Jnger2018]</span>
</div>
<div id="KOREN20051867-link">
<span class="citation" data-cites="KOREN20051867">[@KOREN20051867]</span>
</div>
<div id="Kindermann2018-link">
<span class="citation" data-cites="Kindermann2018">[@Kindermann2018]</span>
</div>
<div id="Klawitter2018-link">
<span class="citation" data-cites="Klawitter2018">[@Klawitter2018]</span>
</div>
<div id="Klawitter2021-link">
<span class="citation" data-cites="Klawitter2021">[@Klawitter2021]</span>
</div>
<div id="Klemz2021-link">
<span class="citation" data-cites="Klemz2021">[@Klemz2021]</span>
</div>
<div id="Lim2021-link">
<span class="citation" data-cites="Lim2021">[@Lim2021]</span>
</div>
<div id="Lin2010-link">
<span class="citation" data-cites="Lin2010">[@Lin2010]</span>
</div>
<div id="Liu2017-link">
<span class="citation" data-cites="Liu2017">[@Liu2017]</span>
</div>
<div id="Mallach2019-link">
<span class="citation" data-cites="Mallach2019">[@Mallach2019]</span>
</div>
<div id="Marner2014-link">
<span class="citation" data-cites="Marner2014">[@Marner2014]</span>
</div>
<div id="Meidiana2019-link">
<span class="citation" data-cites="Meidiana2019">[@Meidiana2019]</span>
</div>
<div id="Meulemans2013-link">
<span class="citation" data-cites="Meulemans2013">[@Meulemans2013]</span>
</div>
<div id="Misue2019-link">
<span class="citation" data-cites="Misue2019">[@Misue2019]</span>
</div>
<div id="Nachmanson2017-link">
<span class="citation" data-cites="Nachmanson2017">[@Nachmanson2017]</span>
</div>
<div id="NanCao2011-link">
<span class="citation" data-cites="NanCao2011">[@NanCao2011]</span>
</div>
<div id="Ni2018-link">
<span class="citation" data-cites="Ni2018">[@Ni2018]</span>
</div>
<div id="Nickel2019-link">
<span class="citation" data-cites="Nickel2019">[@Nickel2019]</span>
</div>
<div id="Niedermann2020-link">
<span class="citation" data-cites="Niedermann2020">[@Niedermann2020]</span>
</div>
<div id="Nllenburg2010-link">
<span class="citation" data-cites="Nllenburg2010">[@Nllenburg2010]</span>
</div>
<div id="Nllenburg2021-link">
<span class="citation" data-cites="Nllenburg2021">[@Nllenburg2021]</span>
</div>
<div id="Nollenburg2011-link">
<span class="citation" data-cites="Nollenburg2011">[@Nollenburg2011]</span>
</div>
<div id="Ogawa2010-link">
<span class="citation" data-cites="Ogawa2010">[@Ogawa2010]</span>
</div>
<div id="Oikonomou2018-link">
<span class="citation" data-cites="Oikonomou2018">[@Oikonomou2018]</span>
</div>
<div id="Olszewski2018-link">
<span class="citation" data-cites="Olszewski2018">[@Olszewski2018]</span>
</div>
<div id="Ortali2018-link">
<span class="citation" data-cites="Ortali2018">[@Ortali2018]</span>
</div>
<div id="Ortmann2017-link">
<span class="citation" data-cites="Ortmann2017">[@Ortmann2017]</span>
</div>
<div id="Perri2020-link">
<span class="citation" data-cites="Perri2020">[@Perri2020]</span>
</div>
<div id="Purchase2020-link">
<span class="citation" data-cites="Purchase2020">[@Purchase2020]</span>
</div>
<div id="Qu2022-link">
<span class="citation" data-cites="Qu2022">[@Qu2022]</span>
</div>
<div id="Raj2018-link">
<span class="citation" data-cites="Raj2018">[@Raj2018]</span>
</div>
<div id="Regg2016-link">
<span class="citation" data-cites="Regg2016">[@Regg2016]</span>
</div>
<div id="Sander1996-link">
<span class="citation" data-cites="Sander1996">[@Sander1996]</span>
</div>
<div id="ShixiaLiu2013-link">
<span class="citation" data-cites="ShixiaLiu2013">[@ShixiaLiu2013]</span>
</div>
<div id="Simonetto2018-link">
<span class="citation" data-cites="Simonetto2018">[@Simonetto2018]</span>
</div>
<div id="Skambath2016-link">
<span class="citation" data-cites="Skambath2016">[@Skambath2016]</span>
</div>
<div id="Sugiyama1981-link">
<span class="citation" data-cites="Sugiyama1981">[@Sugiyama1981]</span>
</div>
<div id="Tanahashi2012-link">
<span class="citation" data-cites="Tanahashi2012">[@Tanahashi2012]</span>
</div>
<div id="Tanahashi2015-link">
<span class="citation" data-cites="Tanahashi2015">[@Tanahashi2015]</span>
</div>
<div id="Taurus2023-link">
<span class="citation" data-cites="Taurus2023">[@Taurus2023]</span>
</div>
<div id="Wallinger2022-link">
<span class="citation" data-cites="Wallinger2022">[@Wallinger2022]</span>
</div>
<div id="Walter2020-link">
<span class="citation" data-cites="Walter2020">[@Walter2020]</span>
</div>
<div id="WeiweiCui2008-link">
<span class="citation" data-cites="cui2008">[@cui2008]</span>
</div>
<div id="Wongsuphasawat2018-link">
<span class="citation" data-cites="Wongsuphasawat2018">[@Wongsuphasawat2018]</span>
</div>
<div id="Wu2018-link">
<span class="citation" data-cites="Wu2018">[@Wu2018]</span>
</div>
<div id="Wu2020-link">
<span class="citation" data-cites="Wu2020">[@Wu2020]</span>
</div>
<div id="Zhang2010-link">
<span class="citation" data-cites="Zhang2010">[@Zhang2010]</span>
</div>
<div id="Zhao2021-link">
<span class="citation" data-cites="Zhao2021">[@Zhao2021]</span>
</div>
<div id="arleo2022-link">
<span class="citation" data-cites="arleo2022">[@arleo2022]</span>
</div>
<div id="barth2002-link">
<span class="citation" data-cites="barth2002">[@barth2002]</span>
</div>
<div id="bast2020-link">
<span class="citation" data-cites="bast2020">[@bast2020]</span>
</div>
<div id="batik2022-link">
<span class="citation" data-cites="batik2022">[@batik2022]</span>
</div>
<div id="bekos2022-link">
<span class="citation" data-cites="bekos2022">[@bekos2022]</span>
</div>
<div id="binucci2022-link">
<span class="citation" data-cites="binucci2022">[@binucci2022]</span>
</div>
<div id="binucci2023-link">
<span class="citation" data-cites="binucci2023">[@binucci2023]</span>
</div>
<div id="brandes1998-link">
<span class="citation" data-cites="brandes1998">[@brandes1998]</span>
</div>
<div id="brandes2002-link">
<span class="citation" data-cites="brandes2002">[@brandes2002]</span>
</div>
<div id="brandes2006-link">
<span class="citation" data-cites="brandes2006">[@brandes2006]</span>
</div>
<div id="brandes2009-link">
<span class="citation" data-cites="brandes2009">[@brandes2009]</span>
</div>
<div id="burch2011-link">
<span class="citation" data-cites="burch2011">[@burch2011]</span>
</div>
<div id="burch2017-link">
<span class="citation" data-cites="burch2017">[@burch2017]</span>
</div>
<div id="chimani2006-link">
<span class="citation" data-cites="chimani2006">[@chimani2006]</span>
</div>
<div id="chimani2010-link">
<span class="citation" data-cites="chimani2010">[@chimani2010]</span>
</div>
<div id="chimani2016-link">
<span class="citation" data-cites="chimani2016">[@chimani2016]</span>
</div>
<div id="chimani2016b-link">
<span class="citation" data-cites="chimani2016b">[@chimani2016b]</span>
</div>
<div id="civril2006-link">
<span class="citation" data-cites="civril2006">[@civril2006]</span>
</div>
<div id="cohen1997-link">
<span class="citation" data-cites="cohen1997">[@cohen1997]</span>
</div>
<div id="cornel2016-link">
<span class="citation" data-cites="cornel2016">[@cornel2016]</span>
</div>
<div id="cypko2017-link">
<span class="citation" data-cites="cypko2017">[@cypko2017]</span>
</div>
<div id="davidson1996-link">
<span class="citation" data-cites="davidson1996">[@davidson1996]</span>
</div>
<div id="deCol2019-link">
<span class="citation" data-cites="deCol2019">[@deCol2019]</span>
</div>
<div id="devkota2018-link">
<span class="citation" data-cites="devkota2018">[@devkota2018]</span>
</div>
<div id="diBartolomeo2022-link">
<span class="citation" data-cites="diBartolomeo2022">[@diBartolomeo2022]</span>
</div>
<div id="diBartolomeo2022b-link">
<span class="citation" data-cites="diBartolomeo2022b">[@diBartolomeo2022b]</span>
</div>
<div id="eades1984-link">
<span class="citation" data-cites="eades1984">[@eades1984]</span>
</div>
<!-- <div id="elassady2017-link">[@elassady2017]</div> -->
<div id="felice2019-link">
<span class="citation" data-cites="felice2019">[@felice2019]</span>
</div>
<div id="fisheye2019-link">
<span class="citation" data-cites="fisheye2019">[@fisheye2019]</span>
</div>
<div id="frick1995-link">
<span class="citation" data-cites="frick1995">[@frick1995]</span>
</div>
<div id="frishman2004-link">
<span class="citation" data-cites="frishman2004">[@frishman2004]</span>
</div>
<div id="frishman2007-link">
<span class="citation" data-cites="frishman2007">[@frishman2007]</span>
</div>
<div id="giovannangeli2023-link">
<span class="citation" data-cites="giovannangeli2023">[@giovannangeli2023]</span>
</div>
<div id="gove2019-link">
<span class="citation" data-cites="gove2019">[@gove2019]</span>
</div>
<div id="gray2023-link">
<span class="citation" data-cites="gray2023">[@gray2023]</span>
</div>
<div id="greilich2009-link">
<span class="citation" data-cites="greilich2009">[@greilich2009]</span>
</div>
<div id="hacul2005-link">
<span class="citation" data-cites="hacul2005">[@hacul2005]</span>
</div>
<div id="harel2001-link">
<span class="citation" data-cites="harel2001">[@harel2001]</span>
</div>
<div id="hoffswell2018-link">
<span class="citation" data-cites="hoffswell2018">[@hoffswell2018]</span>
</div>
<div id="hofmann2017-link">
<span class="citation" data-cites="hofmann2017">[@hofmann2017]</span>
</div>
<div id="hu2005efficient-link">
<span class="citation" data-cites="hu2005efficient">[@hu2005efficient]</span>
</div>
<div id="junger1997-link">
<span class="citation" data-cites="junger1997">[@junger1997]</span>
</div>
<div id="kamada19897-link">
<span class="citation" data-cites="kamada19897">[@kamada19897]</span>
</div>
<div id="klammler2018-link">
<span class="citation" data-cites="klammler2018">[@klammler2018]</span>
</div>
<div id="koren2002-link">
<span class="citation" data-cites="koren2002">[@koren2002]</span>
</div>
<div id="kruiger2017-link">
<span class="citation" data-cites="kruiger2017">[@kruiger2017]</span>
</div>
<div id="lukasczyk2017-link">
<span class="citation" data-cites="lukasczyk2017">[@lukasczyk2017]</span>
</div>
<div id="meidiana2020-link">
<span class="citation" data-cites="meidiana2020">[@meidiana2020]</span>
</div>
<div id="meidiana2022-link">
<span class="citation" data-cites="meidiana2022">[@meidiana2022]</span>
</div>
<div id="meidiana2023-link">
<span class="citation" data-cites="meidiana2023">[@meidiana2023]</span>
</div>
<div id="mennens2019-link">
<span class="citation" data-cites="mennens2019">[@mennens2019]</span>
</div>
<div id="miller2023-link">
<span class="citation" data-cites="miller2023">[@miller2023]</span>
</div>
<div id="minghim2006-link">
<span class="citation" data-cites="minghim2006">[@minghim2006]</span>
</div>
<div id="mizuno2019-link">
<span class="citation" data-cites="mizuno2019">[@mizuno2019]</span>
</div>
<div id="muelder2008-link">
<span class="citation" data-cites="muelder2008">[@muelder2008]</span>
</div>
<div id="muelder2008b-link">
<span class="citation" data-cites="muelder2008b">[@muelder2008b]</span>
</div>
<div id="noack2004-link">
<span class="citation" data-cites="noack2004">[@noack2004]</span>
</div>
<div id="nocaj2015-link">
<span class="citation" data-cites="nocaj2015">[@nocaj2015]</span>
</div>
<div id="padia2018-link">
<span class="citation" data-cites="padia2018">[@padia2018]</span>
</div>
<div id="padia2019-link">
<span class="citation" data-cites="padia2019">[@padia2019]</span>
</div>
<div id="pawel2000-link">
<span class="citation" data-cites="pawel2000">[@pawel2000]</span>
</div>
<div id="preiner2020-link">
<span class="citation" data-cites="preiner2020">[@preiner2020]</span>
</div>
<div id="saket2016-link">
<span class="citation" data-cites="saket2016">[@saket2016]</span>
</div>
<div id="vanDijk2016-link">
<span class="citation" data-cites="vanDijk2016">[@vanDijk2016]</span>
</div>
<div id="vanDijk2018-link">
<span class="citation" data-cites="vanDijk2018">[@vanDijk2018]</span>
</div>
<div id="vandenElzen2013-link">
<span class="citation" data-cites="vandenElzen2013">[@vandenElzen2013]</span>
</div>
<div id="vonLandesberger2016-link">
<span class="citation" data-cites="vonLandesberger2016">[@vonLandesberger2016]</span>
</div>
<div id="wallinger2023-link">
<span class="citation" data-cites="wallinger2023">[@wallinger2023]</span>
</div>
<div id="walshaw2001-link">
<span class="citation" data-cites="walshaw2001">[@walshaw2001]</span>
</div>
<div id="wang2018-link">
<span class="citation" data-cites="wang2018">[@wang2018]</span>
</div>
<div id="wang2020-link">
<span class="citation" data-cites="wang2020">[@wang2020]</span>
</div>
<div id="welch2017-link">
<span class="citation" data-cites="welch2017">[@welch2017]</span>
</div>
<div id="zarate2018-link">
<span class="citation" data-cites="zarate2018">[@zarate2018]</span>
</div>
<div id="zeng2019-link">
<span class="citation" data-cites="zeng2019">[@zeng2019]</span>
</div>
<div id="zheng2019-link">
<span class="citation" data-cites="zheng2019">[@zheng2019]</span>
</div>
<div id="zhong2023-link">
<span class="citation" data-cites="zhong2023">[@zhong2023]</span>
</div>
</div>
</div>
</div>
</div>
<div class="cell hidden">
<div class="sourceCode cell-code hidden" id="cb4" data-startfrom="1430" data-source-offset="0"><pre class="sourceCode js code-with-copy"><code class="sourceCode javascript" style="counter-reset: source-line 1429;"><span id="cb4-1430"><a href="#cb4-1430" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb4-1431"><a href="#cb4-1431" aria-hidden="true" tabindex="-1"></a>    <span class="co">//console.log(benchmark_datasets)</span></span>
<span id="cb4-1432"><a href="#cb4-1432" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-1433"><a href="#cb4-1433" aria-hidden="true" tabindex="-1"></a>    <span class="co">// sorry for all the vars instead of let but this was split before into different cells</span></span>
<span id="cb4-1434"><a href="#cb4-1434" aria-hidden="true" tabindex="-1"></a>    <span class="kw">var</span> uniqueDataTagSet <span class="op">=</span> <span class="kw">new</span> <span class="bu">Set</span>()<span class="op">;</span> <span class="co">// set for the tags</span></span>
<span id="cb4-1435"><a href="#cb4-1435" aria-hidden="true" tabindex="-1"></a>    <span class="kw">var</span> uniquePaperTagSet <span class="op">=</span> <span class="kw">new</span> <span class="bu">Set</span>()<span class="op">;</span></span>
<span id="cb4-1436"><a href="#cb4-1436" aria-hidden="true" tabindex="-1"></a>    <span class="kw">var</span> dataSetNames <span class="op">=</span> []<span class="op">;</span> <span class="co">// list of names of datasets</span></span>
<span id="cb4-1437"><a href="#cb4-1437" aria-hidden="true" tabindex="-1"></a>    <span class="kw">var</span> paperNames <span class="op">=</span> []</span>
<span id="cb4-1438"><a href="#cb4-1438" aria-hidden="true" tabindex="-1"></a>    <span class="kw">var</span> dataMap <span class="op">=</span> {} <span class="co">// maps from the tags to the individual datasets</span></span>
<span id="cb4-1439"><a href="#cb4-1439" aria-hidden="true" tabindex="-1"></a>    <span class="kw">var</span> paperMap <span class="op">=</span> {} <span class="co">// maps from the tags to the individual papers</span></span>
<span id="cb4-1440"><a href="#cb4-1440" aria-hidden="true" tabindex="-1"></a>    <span class="kw">var</span> nameMap <span class="op">=</span> {} <span class="co">// maps from the cleaned name to the actual name containing potentially special characters</span></span>
<span id="cb4-1441"><a href="#cb4-1441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1442"><a href="#cb4-1442" aria-hidden="true" tabindex="-1"></a>    <span class="co">//console.log(literature)</span></span>
<span id="cb4-1443"><a href="#cb4-1443" aria-hidden="true" tabindex="-1"></a>    <span class="co">// "Citation name": "bekos2022"</span></span>
<span id="cb4-1444"><a href="#cb4-1444" aria-hidden="true" tabindex="-1"></a>    <span class="co">// "Graph feature": "hypergraphs, spatial"</span></span>
<span id="cb4-1445"><a href="#cb4-1445" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-1446"><a href="#cb4-1446" aria-hidden="true" tabindex="-1"></a>    <span class="kw">const</span> tagMapping <span class="op">=</span> {</span>
<span id="cb4-1447"><a href="#cb4-1447" aria-hidden="true" tabindex="-1"></a>    <span class="st">"hypergraphs"</span><span class="op">:</span> <span class="st">"Hypergraphs"</span><span class="op">,</span></span>
<span id="cb4-1448"><a href="#cb4-1448" aria-hidden="true" tabindex="-1"></a>    <span class="st">"spatial"</span><span class="op">:</span> <span class="st">"Spatial"</span><span class="op">,</span></span>
<span id="cb4-1449"><a href="#cb4-1449" aria-hidden="true" tabindex="-1"></a>    <span class="st">"layered graphs"</span><span class="op">:</span> <span class="st">"N-layers"</span><span class="op">,</span></span>
<span id="cb4-1450"><a href="#cb4-1450" aria-hidden="true" tabindex="-1"></a>    <span class="st">"generic"</span><span class="op">:</span> <span class="st">"Generic"</span><span class="op">,</span></span>
<span id="cb4-1451"><a href="#cb4-1451" aria-hidden="true" tabindex="-1"></a>    <span class="st">"large"</span><span class="op">:</span> <span class="st">"Large"</span><span class="op">,</span></span>
<span id="cb4-1452"><a href="#cb4-1452" aria-hidden="true" tabindex="-1"></a>    <span class="st">"dynamic (discrete)"</span><span class="op">:</span> <span class="st">"Dynamic"</span><span class="op">,</span></span>
<span id="cb4-1453"><a href="#cb4-1453" aria-hidden="true" tabindex="-1"></a>    <span class="st">"dynamic"</span><span class="op">:</span> <span class="st">"Dynamic"</span><span class="op">,</span></span>
<span id="cb4-1454"><a href="#cb4-1454" aria-hidden="true" tabindex="-1"></a>    <span class="st">"dynamic (continuous)"</span><span class="op">:</span> <span class="st">"Dynamic"</span><span class="op">,</span></span>
<span id="cb4-1455"><a href="#cb4-1455" aria-hidden="true" tabindex="-1"></a>    <span class="st">"clusters (generated)"</span><span class="op">:</span> <span class="st">"Clusters (generated)"</span><span class="op">,</span></span>
<span id="cb4-1456"><a href="#cb4-1456" aria-hidden="true" tabindex="-1"></a>    <span class="st">"clusters (pre-existing)"</span><span class="op">:</span> <span class="st">"Clusters (pre-existing)"</span><span class="op">,</span></span>
<span id="cb4-1457"><a href="#cb4-1457" aria-hidden="true" tabindex="-1"></a>    <span class="st">"bundled edges (generated)"</span><span class="op">:</span> <span class="st">"Bundled edges (generated)"</span><span class="op">,</span></span>
<span id="cb4-1458"><a href="#cb4-1458" aria-hidden="true" tabindex="-1"></a>    <span class="st">"directed edges"</span><span class="op">:</span> <span class="st">"Directed edges"</span><span class="op">,</span></span>
<span id="cb4-1459"><a href="#cb4-1459" aria-hidden="true" tabindex="-1"></a>    <span class="st">"weighted edges"</span><span class="op">:</span> <span class="st">"Weighted edges"</span><span class="op">,</span></span>
<span id="cb4-1460"><a href="#cb4-1460" aria-hidden="true" tabindex="-1"></a>    <span class="st">"weighted nodes"</span><span class="op">:</span> <span class="st">"Weighted nodes"</span><span class="op">,</span></span>
<span id="cb4-1461"><a href="#cb4-1461" aria-hidden="true" tabindex="-1"></a>    <span class="st">"hierarchical"</span><span class="op">:</span> <span class="st">"Hierarchical"</span><span class="op">,</span></span>
<span id="cb4-1462"><a href="#cb4-1462" aria-hidden="true" tabindex="-1"></a>    <span class="st">"categorical nodes"</span><span class="op">:</span> <span class="st">"Categorical nodes"</span><span class="op">,</span></span>
<span id="cb4-1463"><a href="#cb4-1463" aria-hidden="true" tabindex="-1"></a>    <span class="st">"trees"</span><span class="op">:</span> <span class="st">"Trees"</span><span class="op">,</span></span>
<span id="cb4-1464"><a href="#cb4-1464" aria-hidden="true" tabindex="-1"></a>    <span class="st">"labeled nodes"</span><span class="op">:</span> <span class="st">"Labeled nodes"</span><span class="op">,</span></span>
<span id="cb4-1465"><a href="#cb4-1465" aria-hidden="true" tabindex="-1"></a>    <span class="st">"bipartite"</span><span class="op">:</span> <span class="st">"Bipartite"</span><span class="op">,</span></span>
<span id="cb4-1466"><a href="#cb4-1466" aria-hidden="true" tabindex="-1"></a>    <span class="st">"tripartite"</span><span class="op">:</span> <span class="st">"Bipartite"</span><span class="op">,</span></span>
<span id="cb4-1467"><a href="#cb4-1467" aria-hidden="true" tabindex="-1"></a>    <span class="st">"labeled edges"</span><span class="op">:</span> <span class="st">"Labeled edges"</span><span class="op">,</span></span>
<span id="cb4-1468"><a href="#cb4-1468" aria-hidden="true" tabindex="-1"></a>    <span class="st">"planar"</span><span class="op">:</span> <span class="st">"Planar"</span><span class="op">,</span></span>
<span id="cb4-1469"><a href="#cb4-1469" aria-hidden="true" tabindex="-1"></a>    <span class="st">"sparse"</span><span class="op">:</span> <span class="st">"Sparse"</span><span class="op">,</span></span>
<span id="cb4-1470"><a href="#cb4-1470" aria-hidden="true" tabindex="-1"></a>    <span class="st">"K-planar"</span><span class="op">:</span> <span class="st">"Planar"</span><span class="op">,</span></span>
<span id="cb4-1471"><a href="#cb4-1471" aria-hidden="true" tabindex="-1"></a>    <span class="st">"2-planar"</span><span class="op">:</span> <span class="st">"Planar"</span><span class="op">,</span></span>
<span id="cb4-1472"><a href="#cb4-1472" aria-hidden="true" tabindex="-1"></a>    <span class="st">"acyclic"</span><span class="op">:</span> <span class="st">"Acyclic"</span><span class="op">,</span></span>
<span id="cb4-1473"><a href="#cb4-1473" aria-hidden="true" tabindex="-1"></a>}<span class="op">;</span></span>
<span id="cb4-1474"><a href="#cb4-1474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1475"><a href="#cb4-1475" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-1476"><a href="#cb4-1476" aria-hidden="true" tabindex="-1"></a>    benchmark_datasets<span class="op">.</span><span class="fu">forEach</span>((entry) <span class="kw">=&gt;</span> {</span>
<span id="cb4-1477"><a href="#cb4-1477" aria-hidden="true" tabindex="-1"></a>        <span class="kw">let</span> name <span class="op">=</span> <span class="fu">name_cleanup</span>(entry<span class="op">.</span><span class="at">Name</span>)</span>
<span id="cb4-1478"><a href="#cb4-1478" aria-hidden="true" tabindex="-1"></a>        nameMap[name] <span class="op">=</span> entry<span class="op">.</span><span class="at">Name</span></span>
<span id="cb4-1479"><a href="#cb4-1479" aria-hidden="true" tabindex="-1"></a>        dataSetNames<span class="op">.</span><span class="fu">push</span>(name)<span class="op">;</span></span>
<span id="cb4-1480"><a href="#cb4-1480" aria-hidden="true" tabindex="-1"></a>        <span class="kw">var</span> parts <span class="op">=</span> entry[<span class="st">"graph features handled"</span>]<span class="op">.</span><span class="fu">split</span>(<span class="st">","</span>)<span class="op">.</span><span class="fu">map</span>(item <span class="kw">=&gt;</span> item<span class="op">.</span><span class="fu">trim</span>())</span>
<span id="cb4-1481"><a href="#cb4-1481" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (<span class="kw">var</span> part <span class="kw">of</span> parts) {</span>
<span id="cb4-1482"><a href="#cb4-1482" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (part <span class="op">!==</span> <span class="st">""</span>) {</span>
<span id="cb4-1483"><a href="#cb4-1483" aria-hidden="true" tabindex="-1"></a>                uniqueDataTagSet<span class="op">.</span><span class="fu">add</span>(part)<span class="op">;</span></span>
<span id="cb4-1484"><a href="#cb4-1484" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> (<span class="op">!</span>dataMap[part]) {</span>
<span id="cb4-1485"><a href="#cb4-1485" aria-hidden="true" tabindex="-1"></a>                    dataMap[part] <span class="op">=</span> <span class="kw">new</span> <span class="bu">Set</span>()<span class="op">;</span></span>
<span id="cb4-1486"><a href="#cb4-1486" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb4-1487"><a href="#cb4-1487" aria-hidden="true" tabindex="-1"></a>                dataMap[part]<span class="op">.</span><span class="fu">add</span>(name)<span class="op">;</span></span>
<span id="cb4-1488"><a href="#cb4-1488" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb4-1489"><a href="#cb4-1489" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb4-1490"><a href="#cb4-1490" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb4-1491"><a href="#cb4-1491" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb4-1492"><a href="#cb4-1492" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-1493"><a href="#cb4-1493" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb4-1494"><a href="#cb4-1494" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-1495"><a href="#cb4-1495" aria-hidden="true" tabindex="-1"></a>    literature<span class="op">.</span><span class="fu">forEach</span>((paper) <span class="kw">=&gt;</span> {</span>
<span id="cb4-1496"><a href="#cb4-1496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1497"><a href="#cb4-1497" aria-hidden="true" tabindex="-1"></a>      paperNames<span class="op">.</span><span class="fu">push</span>(paper[<span class="st">"Citation name"</span>])<span class="op">;</span></span>
<span id="cb4-1498"><a href="#cb4-1498" aria-hidden="true" tabindex="-1"></a>      <span class="kw">var</span> parts <span class="op">=</span> paper[<span class="st">"Graph feature"</span>]<span class="op">.</span><span class="fu">split</span>(<span class="st">","</span>)<span class="op">.</span><span class="fu">map</span>(item <span class="kw">=&gt;</span> item<span class="op">.</span><span class="fu">trim</span>())</span>
<span id="cb4-1499"><a href="#cb4-1499" aria-hidden="true" tabindex="-1"></a>      <span class="cf">for</span> (<span class="kw">var</span> part <span class="kw">of</span> parts) {</span>
<span id="cb4-1500"><a href="#cb4-1500" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (part <span class="op">!==</span> <span class="st">""</span>) {</span>
<span id="cb4-1501"><a href="#cb4-1501" aria-hidden="true" tabindex="-1"></a>          uniquePaperTagSet<span class="op">.</span><span class="fu">add</span>(part)<span class="op">;</span></span>
<span id="cb4-1502"><a href="#cb4-1502" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb4-1503"><a href="#cb4-1503" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> (part <span class="kw">in</span> tagMapping) {</span>
<span id="cb4-1504"><a href="#cb4-1504" aria-hidden="true" tabindex="-1"></a>              part <span class="op">=</span> tagMapping[part]</span>
<span id="cb4-1505"><a href="#cb4-1505" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb4-1506"><a href="#cb4-1506" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> (<span class="op">!</span>paperMap[part]) {</span>
<span id="cb4-1507"><a href="#cb4-1507" aria-hidden="true" tabindex="-1"></a>              paperMap[part] <span class="op">=</span> <span class="kw">new</span> <span class="bu">Set</span>()<span class="op">;</span></span>
<span id="cb4-1508"><a href="#cb4-1508" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb4-1509"><a href="#cb4-1509" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb4-1510"><a href="#cb4-1510" aria-hidden="true" tabindex="-1"></a>          paperMap[part]<span class="op">.</span><span class="fu">add</span>(paper[<span class="st">"Citation name"</span>])<span class="op">;</span></span>
<span id="cb4-1511"><a href="#cb4-1511" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb4-1512"><a href="#cb4-1512" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb4-1513"><a href="#cb4-1513" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb4-1514"><a href="#cb4-1514" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-1515"><a href="#cb4-1515" aria-hidden="true" tabindex="-1"></a>    <span class="co">// Add keys not present in the paperTags </span></span>
<span id="cb4-1516"><a href="#cb4-1516" aria-hidden="true" tabindex="-1"></a>    <span class="bu">Object</span><span class="op">.</span><span class="fu">keys</span>(dataMap)<span class="op">.</span><span class="fu">forEach</span>((key) <span class="kw">=&gt;</span> {</span>
<span id="cb4-1517"><a href="#cb4-1517" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="op">!</span>paperMap[key]) {</span>
<span id="cb4-1518"><a href="#cb4-1518" aria-hidden="true" tabindex="-1"></a>        paperMap[key] <span class="op">=</span> <span class="kw">new</span> <span class="bu">Set</span>()<span class="op">;</span></span>
<span id="cb4-1519"><a href="#cb4-1519" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb4-1520"><a href="#cb4-1520" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb4-1521"><a href="#cb4-1521" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-1522"><a href="#cb4-1522" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb4-1523"><a href="#cb4-1523" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-1524"><a href="#cb4-1524" aria-hidden="true" tabindex="-1"></a>    <span class="kw">var</span> category_tags <span class="op">=</span> {</span>
<span id="cb4-1525"><a href="#cb4-1525" aria-hidden="true" tabindex="-1"></a>        <span class="dv">0</span><span class="op">:</span> <span class="st">"Layered"</span><span class="op">,</span></span>
<span id="cb4-1526"><a href="#cb4-1526" aria-hidden="true" tabindex="-1"></a>        <span class="dv">2</span><span class="op">:</span> <span class="st">"Hierarchical"</span><span class="op">,</span></span>
<span id="cb4-1527"><a href="#cb4-1527" aria-hidden="true" tabindex="-1"></a>        <span class="dv">4</span><span class="op">:</span> <span class="st">"Attributes"</span><span class="op">,</span></span>
<span id="cb4-1528"><a href="#cb4-1528" aria-hidden="true" tabindex="-1"></a>        <span class="dv">11</span><span class="op">:</span> <span class="st">"Structures"</span><span class="op">,</span></span>
<span id="cb4-1529"><a href="#cb4-1529" aria-hidden="true" tabindex="-1"></a>        <span class="dv">21</span><span class="op">:</span> <span class="st">"Other"</span><span class="op">,</span></span>
<span id="cb4-1530"><a href="#cb4-1530" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb4-1531"><a href="#cb4-1531" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-1532"><a href="#cb4-1532" aria-hidden="true" tabindex="-1"></a>    <span class="co">// if you change this ordering or move around values also change the category map above!!!</span></span>
<span id="cb4-1533"><a href="#cb4-1533" aria-hidden="true" tabindex="-1"></a>    <span class="co">//var uniqueDataTags = [...uniqueDataTagSet]</span></span>
<span id="cb4-1534"><a href="#cb4-1534" aria-hidden="true" tabindex="-1"></a>    <span class="kw">var</span> uniqueDataTags <span class="op">=</span> [</span>
<span id="cb4-1535"><a href="#cb4-1535" aria-hidden="true" tabindex="-1"></a>        <span class="co">// Layered</span></span>
<span id="cb4-1536"><a href="#cb4-1536" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Bipartite"</span><span class="op">,</span></span>
<span id="cb4-1537"><a href="#cb4-1537" aria-hidden="true" tabindex="-1"></a>        <span class="st">"N-layers"</span><span class="op">,</span></span>
<span id="cb4-1538"><a href="#cb4-1538" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb4-1539"><a href="#cb4-1539" aria-hidden="true" tabindex="-1"></a>        <span class="co">// Hierarchical</span></span>
<span id="cb4-1540"><a href="#cb4-1540" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Trees"</span><span class="op">,</span></span>
<span id="cb4-1541"><a href="#cb4-1541" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Hierarchical"</span><span class="op">,</span></span>
<span id="cb4-1542"><a href="#cb4-1542" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1543"><a href="#cb4-1543" aria-hidden="true" tabindex="-1"></a>        <span class="co">// Attributes</span></span>
<span id="cb4-1544"><a href="#cb4-1544" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Dynamic"</span><span class="op">,</span></span>
<span id="cb4-1545"><a href="#cb4-1545" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Weighted edges"</span><span class="op">,</span></span>
<span id="cb4-1546"><a href="#cb4-1546" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Weighted nodes"</span><span class="op">,</span></span>
<span id="cb4-1547"><a href="#cb4-1547" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Labeled nodes"</span><span class="op">,</span></span>
<span id="cb4-1548"><a href="#cb4-1548" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Categorical nodes"</span><span class="op">,</span></span>
<span id="cb4-1549"><a href="#cb4-1549" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Signed"</span><span class="op">,</span></span>
<span id="cb4-1550"><a href="#cb4-1550" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Spatial"</span><span class="op">,</span></span>
<span id="cb4-1551"><a href="#cb4-1551" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb4-1552"><a href="#cb4-1552" aria-hidden="true" tabindex="-1"></a>        <span class="co">// Structures</span></span>
<span id="cb4-1553"><a href="#cb4-1553" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Clusters (pre-existing)"</span><span class="op">,</span></span>
<span id="cb4-1554"><a href="#cb4-1554" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Clusters (generated)"</span><span class="op">,</span></span>
<span id="cb4-1555"><a href="#cb4-1555" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Directed edges"</span><span class="op">,</span></span>
<span id="cb4-1556"><a href="#cb4-1556" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Hypergraphs"</span><span class="op">,</span></span>
<span id="cb4-1557"><a href="#cb4-1557" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Bundled edges (generated)"</span><span class="op">,</span></span>
<span id="cb4-1558"><a href="#cb4-1558" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Partition"</span><span class="op">,</span></span>
<span id="cb4-1559"><a href="#cb4-1559" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Multigraph"</span><span class="op">,</span></span>
<span id="cb4-1560"><a href="#cb4-1560" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Acyclic"</span><span class="op">,</span></span>
<span id="cb4-1561"><a href="#cb4-1561" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Mesh"</span><span class="op">,</span></span>
<span id="cb4-1562"><a href="#cb4-1562" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1563"><a href="#cb4-1563" aria-hidden="true" tabindex="-1"></a>        <span class="co">// Other</span></span>
<span id="cb4-1564"><a href="#cb4-1564" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Generic"</span><span class="op">,</span></span>
<span id="cb4-1565"><a href="#cb4-1565" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Planar"</span><span class="op">,</span></span>
<span id="cb4-1566"><a href="#cb4-1566" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Nonplanar"</span><span class="op">,</span></span>
<span id="cb4-1567"><a href="#cb4-1567" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Known crossing number"</span><span class="op">,</span></span>
<span id="cb4-1568"><a href="#cb4-1568" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Large"</span><span class="op">,</span></span>
<span id="cb4-1569"><a href="#cb4-1569" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Dense"</span><span class="op">,</span></span>
<span id="cb4-1570"><a href="#cb4-1570" aria-hidden="true" tabindex="-1"></a>        <span class="st">"Sparse"</span><span class="op">,</span></span>
<span id="cb4-1571"><a href="#cb4-1571" aria-hidden="true" tabindex="-1"></a>    ]</span>
<span id="cb4-1572"><a href="#cb4-1572" aria-hidden="true" tabindex="-1"></a>    <span class="kw">var</span> uniquePaperTags <span class="op">=</span> [<span class="op">...</span>uniquePaperTagSet]  </span>
<span id="cb4-1573"><a href="#cb4-1573" aria-hidden="true" tabindex="-1"></a>    <span class="co">//console.log(uniqueDataTags)</span></span>
<span id="cb4-1574"><a href="#cb4-1574" aria-hidden="true" tabindex="-1"></a>    <span class="co">//console.log(uniquePaperTagSet)</span></span>
<span id="cb4-1575"><a href="#cb4-1575" aria-hidden="true" tabindex="-1"></a>    <span class="co">//console.log(dataMap)</span></span>
<span id="cb4-1576"><a href="#cb4-1576" aria-hidden="true" tabindex="-1"></a>    <span class="co">//console.log(paperMap)</span></span>
<span id="cb4-1577"><a href="#cb4-1577" aria-hidden="true" tabindex="-1"></a>    <span class="co">//console.log(dataSetNames)</span></span>
<span id="cb4-1578"><a href="#cb4-1578" aria-hidden="true" tabindex="-1"></a>    <span class="co">//console.log(paperNames)</span></span>
<span id="cb4-1579"><a href="#cb4-1579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1580"><a href="#cb4-1580" aria-hidden="true" tabindex="-1"></a>    <span class="co">// remove empty entries</span></span>
<span id="cb4-1581"><a href="#cb4-1581" aria-hidden="true" tabindex="-1"></a>    paperNames <span class="op">=</span> paperNames<span class="op">.</span><span class="fu">filter</span>(str <span class="kw">=&gt;</span> str<span class="op">.</span><span class="fu">trim</span>() <span class="op">!==</span> <span class="st">""</span>)</span>
<span id="cb4-1582"><a href="#cb4-1582" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb4-1583"><a href="#cb4-1583" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1584"><a href="#cb4-1584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1585"><a href="#cb4-1585" aria-hidden="true" tabindex="-1"></a>    <span class="kw">var</span> checkBoxContainer <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">getElementById</span>(<span class="st">'checkbox-container'</span>)<span class="op">;</span></span>
<span id="cb4-1586"><a href="#cb4-1586" aria-hidden="true" tabindex="-1"></a>    <span class="kw">var</span> linksListContainer <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">getElementById</span>(<span class="st">'datasetlinkslist-container'</span>)</span>
<span id="cb4-1587"><a href="#cb4-1587" aria-hidden="true" tabindex="-1"></a>    <span class="kw">var</span> paperLinksContainer <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">getElementById</span>(<span class="st">'paperlinkslist-container'</span>)</span>
<span id="cb4-1588"><a href="#cb4-1588" aria-hidden="true" tabindex="-1"></a>    <span class="kw">var</span> checkboxStates <span class="op">=</span> []<span class="op">;</span>  <span class="co">// Array to keep track of checkbox states (booleans)</span></span>
<span id="cb4-1589"><a href="#cb4-1589" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1590"><a href="#cb4-1590" aria-hidden="true" tabindex="-1"></a>    <span class="co">// needs to be in here since we access so many "global" variables</span></span>
<span id="cb4-1591"><a href="#cb4-1591" aria-hidden="true" tabindex="-1"></a>    <span class="kw">function</span> <span class="fu">displayDataSetLinksList</span>() {</span>
<span id="cb4-1592"><a href="#cb4-1592" aria-hidden="true" tabindex="-1"></a>        <span class="co">//console.log(checkboxStates); // current states to check which dataset links to display (index-based)</span></span>
<span id="cb4-1593"><a href="#cb4-1593" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1594"><a href="#cb4-1594" aria-hidden="true" tabindex="-1"></a>        <span class="kw">const</span> filteredTags <span class="op">=</span> uniqueDataTags<span class="op">.</span><span class="fu">map</span>((item<span class="op">,</span> index) <span class="kw">=&gt;</span> checkboxStates[index] <span class="op">?</span> item <span class="op">:</span> <span class="kw">undefined</span>)</span>
<span id="cb4-1595"><a href="#cb4-1595" aria-hidden="true" tabindex="-1"></a>            <span class="op">.</span><span class="fu">filter</span>(item <span class="kw">=&gt;</span> item <span class="op">!==</span> <span class="kw">undefined</span>)<span class="op">;</span></span>
<span id="cb4-1596"><a href="#cb4-1596" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1597"><a href="#cb4-1597" aria-hidden="true" tabindex="-1"></a>        <span class="co">//console.log(filteredTags)</span></span>
<span id="cb4-1598"><a href="#cb4-1598" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1599"><a href="#cb4-1599" aria-hidden="true" tabindex="-1"></a>        <span class="kw">var</span> final_datasets <span class="op">=</span> []</span>
<span id="cb4-1600"><a href="#cb4-1600" aria-hidden="true" tabindex="-1"></a>        <span class="kw">var</span> final_data_inverse <span class="op">=</span> []</span>
<span id="cb4-1601"><a href="#cb4-1601" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1602"><a href="#cb4-1602" aria-hidden="true" tabindex="-1"></a>        <span class="kw">var</span> final_papers <span class="op">=</span> []</span>
<span id="cb4-1603"><a href="#cb4-1603" aria-hidden="true" tabindex="-1"></a>        <span class="kw">var</span> final_paper_inverse <span class="op">=</span> []</span>
<span id="cb4-1604"><a href="#cb4-1604" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1605"><a href="#cb4-1605" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (filteredTags<span class="op">.</span><span class="at">length</span> <span class="op">==</span> <span class="dv">0</span>) {</span>
<span id="cb4-1606"><a href="#cb4-1606" aria-hidden="true" tabindex="-1"></a>            final_datasets <span class="op">=</span> dataSetNames<span class="op">;</span></span>
<span id="cb4-1607"><a href="#cb4-1607" aria-hidden="true" tabindex="-1"></a>            final_papers <span class="op">=</span> paperNames<span class="op">;</span></span>
<span id="cb4-1608"><a href="#cb4-1608" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb4-1609"><a href="#cb4-1609" aria-hidden="true" tabindex="-1"></a>            <span class="co">// Retrieve an array of sets from the provided keys</span></span>
<span id="cb4-1610"><a href="#cb4-1610" aria-hidden="true" tabindex="-1"></a>            <span class="kw">const</span> dataSets <span class="op">=</span> filteredTags<span class="op">.</span><span class="fu">map</span>(key <span class="kw">=&gt;</span> dataMap[key])<span class="op">;</span></span>
<span id="cb4-1611"><a href="#cb4-1611" aria-hidden="true" tabindex="-1"></a>            <span class="kw">const</span> paperSets <span class="op">=</span> filteredTags<span class="op">.</span><span class="fu">map</span>(key <span class="kw">=&gt;</span> paperMap[key])<span class="op">;</span></span>
<span id="cb4-1612"><a href="#cb4-1612" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb4-1613"><a href="#cb4-1613" aria-hidden="true" tabindex="-1"></a>            <span class="co">// Calculate the intersection of all sets</span></span>
<span id="cb4-1614"><a href="#cb4-1614" aria-hidden="true" tabindex="-1"></a>            <span class="kw">const</span> dataIntersection <span class="op">=</span> dataSets<span class="op">.</span><span class="fu">reduce</span>((acc<span class="op">,</span> <span class="kw">set</span>) <span class="kw">=&gt;</span> <span class="kw">new</span> <span class="bu">Set</span>([<span class="op">...</span>acc]<span class="op">.</span><span class="fu">filter</span>(x <span class="kw">=&gt;</span> <span class="kw">set</span><span class="op">.</span><span class="fu">has</span>(x))))<span class="op">;</span></span>
<span id="cb4-1615"><a href="#cb4-1615" aria-hidden="true" tabindex="-1"></a>            <span class="kw">const</span> paperIntersection <span class="op">=</span> paperSets<span class="op">.</span><span class="fu">reduce</span>((acc<span class="op">,</span> <span class="kw">set</span>) <span class="kw">=&gt;</span> <span class="kw">new</span> <span class="bu">Set</span>([<span class="op">...</span>acc]<span class="op">.</span><span class="fu">filter</span>(x <span class="kw">=&gt;</span> <span class="kw">set</span><span class="op">.</span><span class="fu">has</span>(x))))<span class="op">;</span></span>
<span id="cb4-1616"><a href="#cb4-1616" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb4-1617"><a href="#cb4-1617" aria-hidden="true" tabindex="-1"></a>            final_datasets <span class="op">=</span> [<span class="op">...</span>dataIntersection] <span class="co">// white list</span></span>
<span id="cb4-1618"><a href="#cb4-1618" aria-hidden="true" tabindex="-1"></a>            final_papers <span class="op">=</span> [<span class="op">...</span>paperIntersection]</span>
<span id="cb4-1619"><a href="#cb4-1619" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1620"><a href="#cb4-1620" aria-hidden="true" tabindex="-1"></a>            <span class="co">// Now, compute the difference between dataSetNames and final_datasets</span></span>
<span id="cb4-1621"><a href="#cb4-1621" aria-hidden="true" tabindex="-1"></a>            <span class="kw">const</span> dataDifference <span class="op">=</span> <span class="kw">new</span> <span class="bu">Set</span>(dataSetNames<span class="op">.</span><span class="fu">filter</span>(x <span class="kw">=&gt;</span> <span class="op">!</span>final_datasets<span class="op">.</span><span class="fu">includes</span>(x)))<span class="op">;</span></span>
<span id="cb4-1622"><a href="#cb4-1622" aria-hidden="true" tabindex="-1"></a>            <span class="kw">const</span> paperDifference <span class="op">=</span> <span class="kw">new</span> <span class="bu">Set</span>(paperNames<span class="op">.</span><span class="fu">filter</span>(x <span class="kw">=&gt;</span> <span class="op">!</span>final_papers<span class="op">.</span><span class="fu">includes</span>(x)))<span class="op">;</span></span>
<span id="cb4-1623"><a href="#cb4-1623" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1624"><a href="#cb4-1624" aria-hidden="true" tabindex="-1"></a>            final_data_inverse <span class="op">=</span> [<span class="op">...</span>dataDifference]<span class="op">;</span> <span class="co">// black list</span></span>
<span id="cb4-1625"><a href="#cb4-1625" aria-hidden="true" tabindex="-1"></a>            final_paper_inverse <span class="op">=</span> [<span class="op">...</span>paperDifference]</span>
<span id="cb4-1626"><a href="#cb4-1626" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb4-1627"><a href="#cb4-1627" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1628"><a href="#cb4-1628" aria-hidden="true" tabindex="-1"></a>        <span class="co">//console.log("show: " + final_datasets) // white list </span></span>
<span id="cb4-1629"><a href="#cb4-1629" aria-hidden="true" tabindex="-1"></a>        <span class="co">//console.log("show: " + final_papers)</span></span>
<span id="cb4-1630"><a href="#cb4-1630" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1631"><a href="#cb4-1631" aria-hidden="true" tabindex="-1"></a>        final_datasets<span class="op">.</span><span class="fu">forEach</span>((name) <span class="kw">=&gt;</span> {</span>
<span id="cb4-1632"><a href="#cb4-1632" aria-hidden="true" tabindex="-1"></a>            <span class="co">//console.log(name + '-link')</span></span>
<span id="cb4-1633"><a href="#cb4-1633" aria-hidden="true" tabindex="-1"></a>            <span class="kw">let</span> currLinkDiv <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">getElementById</span>(name <span class="op">+</span> <span class="st">'-link'</span>)</span>
<span id="cb4-1634"><a href="#cb4-1634" aria-hidden="true" tabindex="-1"></a>            currLinkDiv<span class="op">.</span><span class="at">style</span><span class="op">.</span><span class="at">display</span> <span class="op">=</span> <span class="st">'block'</span><span class="op">;</span>  <span class="co">// Clear the display property to default ('block')</span></span>
<span id="cb4-1635"><a href="#cb4-1635" aria-hidden="true" tabindex="-1"></a>        })</span>
<span id="cb4-1636"><a href="#cb4-1636" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1637"><a href="#cb4-1637" aria-hidden="true" tabindex="-1"></a>        final_data_inverse<span class="op">.</span><span class="fu">forEach</span>((name) <span class="kw">=&gt;</span> {</span>
<span id="cb4-1638"><a href="#cb4-1638" aria-hidden="true" tabindex="-1"></a>            <span class="co">//console.log(name + '-link')</span></span>
<span id="cb4-1639"><a href="#cb4-1639" aria-hidden="true" tabindex="-1"></a>            <span class="kw">let</span> currLinkDiv <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">getElementById</span>(name <span class="op">+</span> <span class="st">'-link'</span>)</span>
<span id="cb4-1640"><a href="#cb4-1640" aria-hidden="true" tabindex="-1"></a>            currLinkDiv<span class="op">.</span><span class="at">style</span><span class="op">.</span><span class="at">display</span> <span class="op">=</span> <span class="st">'none'</span><span class="op">;</span>  <span class="co">// Hide the div</span></span>
<span id="cb4-1641"><a href="#cb4-1641" aria-hidden="true" tabindex="-1"></a>        })</span>
<span id="cb4-1642"><a href="#cb4-1642" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1643"><a href="#cb4-1643" aria-hidden="true" tabindex="-1"></a>        final_papers<span class="op">.</span><span class="fu">forEach</span>((name) <span class="kw">=&gt;</span> {</span>
<span id="cb4-1644"><a href="#cb4-1644" aria-hidden="true" tabindex="-1"></a>            <span class="co">//console.log(name + '-link')</span></span>
<span id="cb4-1645"><a href="#cb4-1645" aria-hidden="true" tabindex="-1"></a>            <span class="kw">let</span> currLinkDiv <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">getElementById</span>(name <span class="op">+</span> <span class="st">'-link'</span>)</span>
<span id="cb4-1646"><a href="#cb4-1646" aria-hidden="true" tabindex="-1"></a>            currLinkDiv<span class="op">.</span><span class="at">style</span><span class="op">.</span><span class="at">display</span> <span class="op">=</span> <span class="st">'block'</span><span class="op">;</span>  <span class="co">// Clear the display property to default ('block')</span></span>
<span id="cb4-1647"><a href="#cb4-1647" aria-hidden="true" tabindex="-1"></a>        })</span>
<span id="cb4-1648"><a href="#cb4-1648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1649"><a href="#cb4-1649" aria-hidden="true" tabindex="-1"></a>        final_paper_inverse<span class="op">.</span><span class="fu">forEach</span>((name) <span class="kw">=&gt;</span> {</span>
<span id="cb4-1650"><a href="#cb4-1650" aria-hidden="true" tabindex="-1"></a>            <span class="co">//console.log(name + '-link')</span></span>
<span id="cb4-1651"><a href="#cb4-1651" aria-hidden="true" tabindex="-1"></a>            <span class="kw">let</span> currLinkDiv <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">getElementById</span>(name <span class="op">+</span> <span class="st">'-link'</span>)</span>
<span id="cb4-1652"><a href="#cb4-1652" aria-hidden="true" tabindex="-1"></a>            currLinkDiv<span class="op">.</span><span class="at">style</span><span class="op">.</span><span class="at">display</span> <span class="op">=</span> <span class="st">'none'</span><span class="op">;</span>  <span class="co">// Hide the div</span></span>
<span id="cb4-1653"><a href="#cb4-1653" aria-hidden="true" tabindex="-1"></a>        })</span>
<span id="cb4-1654"><a href="#cb4-1654" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1655"><a href="#cb4-1655" aria-hidden="true" tabindex="-1"></a>      <span class="co">/*</span></span>
<span id="cb4-1656"><a href="#cb4-1656" aria-hidden="true" tabindex="-1"></a><span class="co">      // this is how the it ran earlier and it was used to generate the static html links above this</span></span>
<span id="cb4-1657"><a href="#cb4-1657" aria-hidden="true" tabindex="-1"></a><span class="co">      </span></span>
<span id="cb4-1658"><a href="#cb4-1658" aria-hidden="true" tabindex="-1"></a><span class="co">      if (final_datasets.length == 0) {</span></span>
<span id="cb4-1659"><a href="#cb4-1659" aria-hidden="true" tabindex="-1"></a><span class="co">      // display message that no dataset contains all these tags</span></span>
<span id="cb4-1660"><a href="#cb4-1660" aria-hidden="true" tabindex="-1"></a><span class="co">      linksListContainer.innerHTML = '';</span></span>
<span id="cb4-1661"><a href="#cb4-1661" aria-hidden="true" tabindex="-1"></a><span class="co">      } else {</span></span>
<span id="cb4-1662"><a href="#cb4-1662" aria-hidden="true" tabindex="-1"></a><span class="co">      </span></span>
<span id="cb4-1663"><a href="#cb4-1663" aria-hidden="true" tabindex="-1"></a><span class="co">      linksListContainer.innerHTML = '';</span></span>
<span id="cb4-1664"><a href="#cb4-1664" aria-hidden="true" tabindex="-1"></a><span class="co">      </span></span>
<span id="cb4-1665"><a href="#cb4-1665" aria-hidden="true" tabindex="-1"></a><span class="co">      final_datasets.forEach((currDataName) =&gt; {</span></span>
<span id="cb4-1666"><a href="#cb4-1666" aria-hidden="true" tabindex="-1"></a><span class="co">      const anchor = document.createElement('a');</span></span>
<span id="cb4-1667"><a href="#cb4-1667" aria-hidden="true" tabindex="-1"></a><span class="co">      anchor.href = '#' + currDataName</span></span>
<span id="cb4-1668"><a href="#cb4-1668" aria-hidden="true" tabindex="-1"></a><span class="co">      anchor.classList.add('dataset-link')</span></span>
<span id="cb4-1669"><a href="#cb4-1669" aria-hidden="true" tabindex="-1"></a><span class="co">      anchor.textContent = nameMap[currDataName]</span></span>
<span id="cb4-1670"><a href="#cb4-1670" aria-hidden="true" tabindex="-1"></a><span class="co">      </span></span>
<span id="cb4-1671"><a href="#cb4-1671" aria-hidden="true" tabindex="-1"></a><span class="co">      const div = document.createElement('div');</span></span>
<span id="cb4-1672"><a href="#cb4-1672" aria-hidden="true" tabindex="-1"></a><span class="co">      div.id = currDataName + '-link'</span></span>
<span id="cb4-1673"><a href="#cb4-1673" aria-hidden="true" tabindex="-1"></a><span class="co">      div.appendChild(anchor);</span></span>
<span id="cb4-1674"><a href="#cb4-1674" aria-hidden="true" tabindex="-1"></a><span class="co">      </span></span>
<span id="cb4-1675"><a href="#cb4-1675" aria-hidden="true" tabindex="-1"></a><span class="co">      linksListContainer.appendChild(div);</span></span>
<span id="cb4-1676"><a href="#cb4-1676" aria-hidden="true" tabindex="-1"></a><span class="co">      })</span></span>
<span id="cb4-1677"><a href="#cb4-1677" aria-hidden="true" tabindex="-1"></a><span class="co">      }</span></span>
<span id="cb4-1678"><a href="#cb4-1678" aria-hidden="true" tabindex="-1"></a><span class="co">      </span></span>
<span id="cb4-1679"><a href="#cb4-1679" aria-hidden="true" tabindex="-1"></a><span class="co">      // and here for the paper names</span></span>
<span id="cb4-1680"><a href="#cb4-1680" aria-hidden="true" tabindex="-1"></a><span class="co">      paperNames = paperNames.filter(str =&gt; str.trim() !== "").sort();</span></span>
<span id="cb4-1681"><a href="#cb4-1681" aria-hidden="true" tabindex="-1"></a><span class="co">      paperNames.forEach((currPaperName) =&gt; {</span></span>
<span id="cb4-1682"><a href="#cb4-1682" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1683"><a href="#cb4-1683" aria-hidden="true" tabindex="-1"></a><span class="co">          const div = document.createElement('div');</span></span>
<span id="cb4-1684"><a href="#cb4-1684" aria-hidden="true" tabindex="-1"></a><span class="co">          div.id = currPaperName + '-link'</span></span>
<span id="cb4-1685"><a href="#cb4-1685" aria-hidden="true" tabindex="-1"></a><span class="co">          div.textContent = `[@${currPaperName}]`</span></span>
<span id="cb4-1686"><a href="#cb4-1686" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1687"><a href="#cb4-1687" aria-hidden="true" tabindex="-1"></a><span class="co">          paperLinksContainer.appendChild(div);</span></span>
<span id="cb4-1688"><a href="#cb4-1688" aria-hidden="true" tabindex="-1"></a><span class="co">      });*/</span></span>
<span id="cb4-1689"><a href="#cb4-1689" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb4-1690"><a href="#cb4-1690" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb4-1691"><a href="#cb4-1691" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb4-1692"><a href="#cb4-1692" aria-hidden="true" tabindex="-1"></a>  <span class="co">// Function to add a new checkbox with a styled span label</span></span>
<span id="cb4-1693"><a href="#cb4-1693" aria-hidden="true" tabindex="-1"></a>  <span class="kw">function</span> <span class="fu">addCheckbox</span>(index<span class="op">,</span> name) {</span>
<span id="cb4-1694"><a href="#cb4-1694" aria-hidden="true" tabindex="-1"></a>      <span class="kw">const</span> newCheckbox <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">createElement</span>(<span class="st">'input'</span>)<span class="op">;</span></span>
<span id="cb4-1695"><a href="#cb4-1695" aria-hidden="true" tabindex="-1"></a>      newCheckbox<span class="op">.</span><span class="at">type</span> <span class="op">=</span> <span class="st">'checkbox'</span><span class="op">;</span></span>
<span id="cb4-1696"><a href="#cb4-1696" aria-hidden="true" tabindex="-1"></a>      newCheckbox<span class="op">.</span><span class="at">id</span> <span class="op">=</span> <span class="st">'checkbox'</span> <span class="op">+</span> index<span class="op">;</span></span>
<span id="cb4-1697"><a href="#cb4-1697" aria-hidden="true" tabindex="-1"></a>      newCheckbox<span class="op">.</span><span class="at">style</span><span class="op">.</span><span class="at">marginRight</span> <span class="op">=</span> <span class="st">'8px'</span><span class="op">;</span> </span>
<span id="cb4-1698"><a href="#cb4-1698" aria-hidden="true" tabindex="-1"></a>      newCheckbox<span class="op">.</span><span class="at">classList</span><span class="op">.</span><span class="fu">add</span>(<span class="st">'form-check-input'</span>)<span class="op">;</span></span>
<span id="cb4-1699"><a href="#cb4-1699" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1700"><a href="#cb4-1700" aria-hidden="true" tabindex="-1"></a>      <span class="co">// Creating the span that will act as a label</span></span>
<span id="cb4-1701"><a href="#cb4-1701" aria-hidden="true" tabindex="-1"></a>      <span class="kw">const</span> span <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">createElement</span>(<span class="st">'span'</span>)<span class="op">;</span></span>
<span id="cb4-1702"><a href="#cb4-1702" aria-hidden="true" tabindex="-1"></a>      span<span class="op">.</span><span class="at">classList</span><span class="op">.</span><span class="fu">add</span>(<span class="st">'pill-tag'</span>)<span class="op">;</span></span>
<span id="cb4-1703"><a href="#cb4-1703" aria-hidden="true" tabindex="-1"></a>      span<span class="op">.</span><span class="at">classList</span><span class="op">.</span><span class="fu">add</span>(<span class="vs">`pill_color_</span><span class="sc">${</span><span class="fu">name_cleanup</span>(name)<span class="sc">}</span><span class="vs">`</span>)<span class="op">;</span></span>
<span id="cb4-1704"><a href="#cb4-1704" aria-hidden="true" tabindex="-1"></a>      <span class="co">//span.style.backgroundColor = get_css_color_for_tag(name)</span></span>
<span id="cb4-1705"><a href="#cb4-1705" aria-hidden="true" tabindex="-1"></a>      <span class="co">//span.style.backgroundColor = graph_feature_tags_colors[index % graph_feature_tags_colors.length];  </span></span>
<span id="cb4-1706"><a href="#cb4-1706" aria-hidden="true" tabindex="-1"></a>      span<span class="op">.</span><span class="at">textContent</span> <span class="op">=</span> name<span class="op">;</span></span>
<span id="cb4-1707"><a href="#cb4-1707" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1708"><a href="#cb4-1708" aria-hidden="true" tabindex="-1"></a>      <span class="kw">const</span> label <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">createElement</span>(<span class="st">'label'</span>)<span class="op">;</span></span>
<span id="cb4-1709"><a href="#cb4-1709" aria-hidden="true" tabindex="-1"></a>      label<span class="op">.</span><span class="at">className</span> <span class="op">=</span> <span class="st">'d-flex align-items-center justify-content-start'</span><span class="op">;</span></span>
<span id="cb4-1710"><a href="#cb4-1710" aria-hidden="true" tabindex="-1"></a>      label<span class="op">.</span><span class="at">htmlFor</span> <span class="op">=</span> newCheckbox<span class="op">.</span><span class="at">id</span><span class="op">;</span></span>
<span id="cb4-1711"><a href="#cb4-1711" aria-hidden="true" tabindex="-1"></a>      label<span class="op">.</span><span class="at">style</span><span class="op">.</span><span class="at">cursor</span> <span class="op">=</span> <span class="st">'pointer'</span><span class="op">;</span>  <span class="co">// Change the cursor to a pointer on hover</span></span>
<span id="cb4-1712"><a href="#cb4-1712" aria-hidden="true" tabindex="-1"></a>      label<span class="op">.</span><span class="fu">appendChild</span>(newCheckbox)<span class="op">;</span></span>
<span id="cb4-1713"><a href="#cb4-1713" aria-hidden="true" tabindex="-1"></a>      label<span class="op">.</span><span class="fu">appendChild</span>(span)<span class="op">;</span>  <span class="co">// Append the span to the label</span></span>
<span id="cb4-1714"><a href="#cb4-1714" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1715"><a href="#cb4-1715" aria-hidden="true" tabindex="-1"></a>      <span class="kw">const</span> div <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">createElement</span>(<span class="st">'div'</span>)<span class="op">;</span></span>
<span id="cb4-1716"><a href="#cb4-1716" aria-hidden="true" tabindex="-1"></a>      div<span class="op">.</span><span class="at">className</span> <span class="op">=</span> <span class="st">'form-check d-flex align-items-center'</span><span class="op">;</span>  <span class="co">// Ensure the checkbox and label are aligned</span></span>
<span id="cb4-1717"><a href="#cb4-1717" aria-hidden="true" tabindex="-1"></a>      div<span class="op">.</span><span class="fu">appendChild</span>(label)<span class="op">;</span></span>
<span id="cb4-1718"><a href="#cb4-1718" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1719"><a href="#cb4-1719" aria-hidden="true" tabindex="-1"></a>      checkBoxContainer<span class="op">.</span><span class="fu">appendChild</span>(div)<span class="op">;</span></span>
<span id="cb4-1720"><a href="#cb4-1720" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1721"><a href="#cb4-1721" aria-hidden="true" tabindex="-1"></a>      <span class="co">// Update state array and add event listener</span></span>
<span id="cb4-1722"><a href="#cb4-1722" aria-hidden="true" tabindex="-1"></a>      checkboxStates<span class="op">.</span><span class="fu">push</span>(<span class="kw">false</span>)<span class="op">;</span></span>
<span id="cb4-1723"><a href="#cb4-1723" aria-hidden="true" tabindex="-1"></a>      newCheckbox<span class="op">.</span><span class="fu">addEventListener</span>(<span class="st">'change'</span><span class="op">,</span> () <span class="kw">=&gt;</span> {</span>
<span id="cb4-1724"><a href="#cb4-1724" aria-hidden="true" tabindex="-1"></a>          checkboxStates[index] <span class="op">=</span> newCheckbox<span class="op">.</span><span class="at">checked</span><span class="op">;</span></span>
<span id="cb4-1725"><a href="#cb4-1725" aria-hidden="true" tabindex="-1"></a>          <span class="fu">displayDataSetLinksList</span>()</span>
<span id="cb4-1726"><a href="#cb4-1726" aria-hidden="true" tabindex="-1"></a>      })<span class="op">;</span></span>
<span id="cb4-1727"><a href="#cb4-1727" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-1728"><a href="#cb4-1728" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1729"><a href="#cb4-1729" aria-hidden="true" tabindex="-1"></a>  <span class="co">// Automatically add checkboxes on load for each tag, we cannot reorder after this!</span></span>
<span id="cb4-1730"><a href="#cb4-1730" aria-hidden="true" tabindex="-1"></a>  uniqueDataTags<span class="op">.</span><span class="fu">forEach</span>((entry<span class="op">,</span> index) <span class="kw">=&gt;</span> {</span>
<span id="cb4-1731"><a href="#cb4-1731" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (category_tags<span class="op">.</span><span class="fu">hasOwnProperty</span>(index)) {</span>
<span id="cb4-1732"><a href="#cb4-1732" aria-hidden="true" tabindex="-1"></a>          <span class="kw">const</span> span <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">createElement</span>(<span class="st">'span'</span>)<span class="op">;</span></span>
<span id="cb4-1733"><a href="#cb4-1733" aria-hidden="true" tabindex="-1"></a>          span<span class="op">.</span><span class="at">style</span><span class="op">.</span><span class="at">opacity</span> <span class="op">=</span> <span class="st">"50%"</span><span class="op">;</span>  <span class="co">// Setting it to be less obnoxious</span></span>
<span id="cb4-1734"><a href="#cb4-1734" aria-hidden="true" tabindex="-1"></a>          span<span class="op">.</span><span class="at">textContent</span> <span class="op">=</span> <span class="st">"</span><span class="sc">\u2012</span><span class="st"> "</span> <span class="op">+</span> category_tags[index] <span class="op">+</span> <span class="st">"</span><span class="sc">\u2012</span><span class="st"> "</span><span class="op">;</span></span>
<span id="cb4-1735"><a href="#cb4-1735" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1736"><a href="#cb4-1736" aria-hidden="true" tabindex="-1"></a>          <span class="kw">const</span> div <span class="op">=</span> <span class="bu">document</span><span class="op">.</span><span class="fu">createElement</span>(<span class="st">'div'</span>)<span class="op">;</span></span>
<span id="cb4-1737"><a href="#cb4-1737" aria-hidden="true" tabindex="-1"></a>          div<span class="op">.</span><span class="at">className</span> <span class="op">=</span> <span class="st">'d-flex align-items-center'</span><span class="op">;</span>  </span>
<span id="cb4-1738"><a href="#cb4-1738" aria-hidden="true" tabindex="-1"></a>          div<span class="op">.</span><span class="fu">appendChild</span>(span)<span class="op">;</span></span>
<span id="cb4-1739"><a href="#cb4-1739" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1740"><a href="#cb4-1740" aria-hidden="true" tabindex="-1"></a>          checkBoxContainer<span class="op">.</span><span class="fu">appendChild</span>(div)<span class="op">;</span></span>
<span id="cb4-1741"><a href="#cb4-1741" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb4-1742"><a href="#cb4-1742" aria-hidden="true" tabindex="-1"></a>      <span class="fu">addCheckbox</span>(index<span class="op">,</span> entry)<span class="op">;</span></span>
<span id="cb4-1743"><a href="#cb4-1743" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb4-1744"><a href="#cb4-1744" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1745"><a href="#cb4-1745" aria-hidden="true" tabindex="-1"></a>  <span class="co">// initial call to display all datasets (now a bit redundant but adds the "display" style-attribute)</span></span>
<span id="cb4-1746"><a href="#cb4-1746" aria-hidden="true" tabindex="-1"></a>  <span class="fu">displayDataSetLinksList</span>()</span>
<span id="cb4-1747"><a href="#cb4-1747" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-1748"><a href="#cb4-1748" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display hidden">
<div id="ojs-cell-4" data-nodetype="expression">

</div>
</div>
</div>
<section id="sec-uniform-benchmark-datasets" class="level3">
<h3 class="anchored" data-anchor-id="sec-uniform-benchmark-datasets">Uniform Benchmark datasets</h3>
<p><strong>Uniform Benchmark</strong> datasets are standalone widely used collections of graphs featuring uniform characteristics—usually simple, generic graphs, often used in evaluations that run over thousands of graphs to report average completion times, or other experiments where the reported metrics are usually aggregated.</p>
<p>Differently from the collections presented later in this section, an experiment that uses an Uniform Benchmark dataset aims to provide a general overview of the performance of a graph layout algorithm by testing on a large amount of graphs varying in size and density, rather than focusing on a specific type of graph.</p>
<p>The first of these collapsible sections is shown already expanded, to give an example of the content that can be found in each of them. The content is generated dynamically based on the data we collected.</p>
<!-- [http://graphdrawing.org describes it as: “Test suites of directed and undirected graphs from the http://www.dia.uniroma3.it/~gdt/ homepage (also known as the Rome Graphs)”.  -->
<!-- ,  and it is used in a large number of papers: [@Buchheim2008] [@chimani2010] [@chimani2006] [@diBartolomeo2022b] [@Clancy2019] [@DiBattista1997b] [@chimani2016b] [@Chimani2008] [@Bekos2018] [@Demel2018] [@Heinsohn2018] [@Kindermann2018] [@chimani2016] [@Chimani2012] [@Gansner2006] [@Gutwenger2004] [@klammler2018] [@Niedermann2020] [@Giovannangeli2021] [@Chimani2021]. -->
<p><a name="Rome-Lib"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Rome-Lib">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-2-contents" aria-controls="callout-2" aria-expanded="true" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Rome-Lib
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-2" class="callout-2-contents callout-collapse collapse show">
<div class="callout-body-container callout-body">
<p>Rome-Lib is, as previously discussed, the most common benchmark dataset, due to its already established popularity, its ease of use and access, and the many properties that are already known about it. It was first introduced in <span class="citation" data-cites="DiBattista1997b">[@DiBattista1997b]</span> and presented in their paper ”An experimental comparison of four graph drawing algorithms”. Mostly “real” networks such as software companies, government, software engineering books, database design, and journal articles on visualization. Originally sent as an extended abstract to Computational Geometry in 1995 <span class="citation" data-cites="DiBattista1995">[@DiBattista1995]</span>. It contains exclusively <a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>graphs (e.g., undirected, non-layered, without pre-established clusters…), although some of the research that uses it enhances with additional attributes (such as performing a rank assignment step). Information about optimal crossings can be found as part of <a href="#WebCompute">WebCompute</a>.</p>
<div>
<b>Originally found at: </b><a href="http://www.graphdrawing.org/data.html">http://www.graphdrawing.org/data.html</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d947f91101aa0dc66a0c4d" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96db0803e0c0cb7558c0f" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d971aa4cf748115a055897" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90ec4803e0c0b00558bc1" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>11534 graphs, 10 to 100 nodes, 9 to 158 edges
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="DiBattista1997b">[@DiBattista1997b]</span>
</div>
<div class="col-8">
An experimental comparison of four graph drawing algorithms
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="DiBattista1995">[@DiBattista1995]</span>
</div>
<div class="col-8">
An experimental comparison of three graph drawing algorithms (extended abstract)
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Buchheim2008">[@Buchheim2008]</span>
</div>
<div class="col-4">
A branch-and-cut approach to the crossing number problem
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="chimani2010">[@chimani2010]</span>
</div>
<div class="col-4">
Layer-Free Upward Crossing Minimization
</div>
<div class="col-4">
<a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="chimani2006">[@chimani2006]</span>
</div>
<div class="col-4">
Experiments on Exact Crossing Minimization Using Column Generation
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="diBartolomeo2022b">[@diBartolomeo2022b]</span>
</div>
<div class="col-4">
Stratisfimal Layout: A modular optimization model for laying out layered node-link network visualizations
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Other"><span class="pill-tag pill_color_other">Compound graphs</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="klammler2018">[@klammler2018]</span>
</div>
<div class="col-4">
Aesthetic Discrimination of Graph Layouts
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Chimani2008">[@Chimani2008]</span>
</div>
<div class="col-4">
A New Approach to Exact Crossing Minimization
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Gansner2006">[@Gansner2006]</span>
</div>
<div class="col-4">
Improved circular layouts
</div>
<div class="col-4">
<a href="#Other"><span class="pill-tag pill_color_other">Circular</span></a>
</div>
</div>
<div class="row">
<div class="col-4">

</div>
<div class="col-4">
Deep Neural Network for Drawing Networks
</div>
<div class="col-4">

</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Gutwenger2004">[@Gutwenger2004]</span>
</div>
<div class="col-4">
An Experimental Study of Crossing Minimization Heuristics
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="DiBattista1997b">[@DiBattista1997b]</span>
</div>
<div class="col-4">
An experimental comparison of four graph drawing algorithms
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="chimani2016">[@chimani2016]</span>
</div>
<div class="col-4">
A Note on the Practicality of Maximal Planar Subgraph Algorithms
</div>
<div class="col-4">
<a href="#Planar"><span class="pill-tag pill_color_Planar">Planar</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Kindermann2018">[@Kindermann2018]</span>
</div>
<div class="col-4">
Experimental Analysis of the Accessibility of Drawings with Few Segments
</div>
<div class="col-4">
<a href="#Sparse"><span class="pill-tag pill_color_Sparse">Sparse</span></a><a href="#Trees"><span class="pill-tag pill_color_Trees">Trees</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Heinsohn2018">[@Heinsohn2018]</span>
</div>
<div class="col-4">
An Interactive Tool to Explore and Improve the Ply Number of Drawings
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Demel2018">[@Demel2018]</span>
</div>
<div class="col-4">
A Greedy Heuristic for Crossing-Angle Maximization
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Bekos2018">[@Bekos2018]</span>
</div>
<div class="col-4">
A Heuristic Approach Towards Drawings of Graphs with High Crossing Resolution
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Niedermann2020">[@Niedermann2020]</span>
</div>
<div class="col-4">
An Integer-Linear Program for Bend-Minimization in Ortho-Radial Drawings
</div>
<div class="col-4">
<a href="#Other"><span class="pill-tag pill_color_other">Ortho</span></a><a href="#Other"><span class="pill-tag pill_color_other">Ortho-radial</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Chimani2021">[@Chimani2021]</span>
</div>
<div class="col-4">
Star-Struck by Fixed Embeddings: Modern Crossing Number Heuristics
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Clancy2019">[@Clancy2019]</span>
</div>
<div class="col-4">
An effective crossing minimisation heuristic based on star insertion
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Chimani2012">[@Chimani2012]</span>
</div>
<div class="col-4">
Advances in the Planarization Method: Effective Multiple Edge Insertions
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="chimani2016b">[@chimani2016b]</span>
</div>
<div class="col-4">
An ILP-based Proof System for the Crossing Number Problem
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<p><br></p>
<section id="statistics" class="level4">
<h4 class="anchored" data-anchor-id="statistics">Statistics</h4>
<div style="text-align: center">
<img src="data/Benchmark datasets/Rome-Lib c2f20984de724f4c89764b0bc494e99e/four_in_one.svg" width="80%">
</div>
</section>
<section id="descriptions-from-literature" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.1016/j.disopt.2007.05.006">A branch-and-cut approach to the crossing number problem</a>”:</p>
<blockquote class="blockquote">
<p>To test the performance of our new algorithm, we used a benchmark set of graphs of the University of Rome III, introduced in [11]. The set contains 11 389 graphs that consist of 10 to 100 nodes and 9 to 158 edges. These graphs were generated from a core set of 112 “real life” graphs used in database design and software engineering applications. Most of the graphs are sparse, which is a common property in most application areas of automatic graph drawing. The average ratio between the number of edges and the number of nodes of the graphs from the benchmark set is about 1.35.</p>
</blockquote>
<p>From “<a href="https://doi.org/10.1145/1671970.1671975">Layer-free upward crossing minimization</a>”:</p>
<blockquote class="blockquote">
<p>The Rome graphs [Di Battista et al.&nbsp;1997] are a widely used benchmark set in graph drawing, obtained from a basic set of 112 real-world graphs. It contains 11,528 instances with 10 through 100 nodes and 9 through 158 edges. Although the graphs are originally undirected, they have been used as directed graphs—by artificially directing the edges according to the node order given in the input files—for showing the performance of the mixed-upward planarization approach [Eiglsperger et al.&nbsp;2003]. In this case, all edges are directed and the graphs are acyclic; hence, the mixed-upward planarization approach turns into an upward planarization method.</p>
</blockquote>
<p>From “<a href="https://doi.org/10.1007/978-3-540-87744-8_24">A New Approach to Exact Crossing Minimization</a>”:</p>
<blockquote class="blockquote">
<p>We say graphs are trivial, if they are planar or if the heuristic achieves a planarization with only one crossing, as in these cases we need not prove optimality. The Rome library contains 7172 non-trivial graphs.</p>
</blockquote>
<p>From “<a href="https://doi.org/10.1016/s0925-7721(96)00005-3">An Experimental Comparison of Four Graph Drawing Algorithms</a>”</p>
<blockquote class="blockquote">
<p>Our test graph generation strategy is as follows. First, we have focused on the important application area of database and software visualization, where Entity-Relationship diagrams and Data-Flow diagrams are usually displayed with orthogonal drawings. Second, we have collected 112 “real life” graphs with number of vertices between 10 and 100, from now on called core graphs, from the following sources:</p>
</blockquote>
<blockquote class="blockquote">
<p>• 54% of the graphs have been obtained from major Italian software companies (especially from Database Informatica) and large government organization (including the Italian Internal Revenue Service and the Italian National Advisory Council for Computer Applications in the Government (Autorita’ per l’Informatica nella Pubblica Amministrazione)); • 33% of the graphs were taken from well-known reference books in software engineering [18] and database design [1], and from journal articles on software visualization in the recent issues of Information Systems and the IEEE Transactions on Software Engineering; • 13% of the graphs were extracted from theses in software and database visualization written by students at the University of Rome “La Sapienza”.</p>
</blockquote>
<blockquote class="blockquote">
<p>Our approach is based on the following scheme. We defined several primitive operations for updating graphs, which correspond to the typical operations performed by designers of Entity-Relationship and Data-Flow Diagrams, and attributed a certain probability to each of them. More specifically, the updating primitives we have used are the following: InsertEdge, which inserts a new edge between two existing vertices; DeleteEdge, which deletes an existing edge; InsertVertex, which splits an existing edge into two edges by inserting a new vertex; DeleteVertex, which deletes a vertex and all its incident edges; and MakeVertex, which creates a new vertex and connects it to a subset of vertices. The test graphs were then generated in several iterations starting from the core graphs by applying random sequences of operations with a “genetic” mechanism. Namely, at each iteration a new set of test graphs was obtained by applying a random sequence of operations to the current test set. Each new graph was then evaluated for “suitability”, and those found not suitable were discarded. The probability of each primitive operation was varied at the end of each iteration. The evaluation of the suitability of the generated graphs was conducted using both objective and subjective analyses. The objective analysis consisted of determining whether the new graph had similar structural properties with respect to the core graph it was derived from. We have taken into account parameters like the average ratio between number of vertices and number of edges and the average number of biconnected components. The subjective analysis consisted in a visual inspection of the new graph and an assessment by expert users of Entity-Relationship and Data-Flow diagrams of its similarity to a “real-life” diagram. For obvious reasons, the subjective analysis has been done on a randomly selected subset of the graphs.</p>
</blockquote>
</section>
<section id="example-figures" class="level4">
<h4 class="anchored" data-anchor-id="example-figures">Example Figures</h4>
<p>From ”<a href="https://doi.org/10.1016/s0925-7721(96)00005-3">An Experimental Comparison of Four Graph Drawing Algorithms</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Rome-Lib c2f20984de724f4c89764b0bc494e99e/Untitled.png" width="80%">
</div>
<p>From “<a href="https://doi.org/10.1007/978-3-319-73915-1_5">Experimental Analysis of the Accessibility of Drawings with Few Segments</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Rome-Lib c2f20984de724f4c89764b0bc494e99e/Untitled 1.png" width="80%">
</div>
<p>From: <a href="https://link.springer.com/chapter/10.1007/978-3-030-92931-2_27">Deep Neural Network for DrawiNg Networks, (DNN)2</a>:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Rome-Lib c2f20984de724f4c89764b0bc494e99e/Untitled 2.png" width="80%">
</div>
<p><strong>Fig 5.</strong> Layout examples for (DNN)^2, (DNN)^2, tsNET∗ and S_GD^2.</p>
</section>
</div>
</div>
</div>
<p>The following collections, together with Rome-Lib, can be easily accessed from the homepage of the <a href="http://www.graphdrawing.org/data.html">Graph Drawing Conference</a> website, and are therefore well known and widely used. <!-- AT&T is used in [@barth2002] [@DIBATTISTA2000] [@Jabrayilov2016] [@Chimani2012] [@Hachul2006] [@Hachul2005] [@Mallach2019] [@Jnger2018] [@hacul2005] [@welch2017] (and referred as AT&T), while the papers that use it but refer to it as North DAGs are [@chimani2010] [@chimani2016b] [@Demel2018] [@Binucci2016] [@Regg2016] [@chimani2016] [@klammler2018] [@Chimani2021] [@binucci2022]. --></p>
<p><a name="ATT"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="AT&amp;T">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
AT&amp;T
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Originally collected by Stephen North at AT&amp;T Bell Labs (see the descriptions from literature section below for more information). The original link from 1995 is broken: <a href="ftp://ftp.research.att.com/dist/drawdag. ">ftp://ftp.research.att.com/dist/drawdag.</a> Di Battista et al.&nbsp;modified the dataset by removing isomorphic graphs, connecting disconnected graphs, and removing cycles. Same as North DAG collection. It contains <a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a>and <a href="#Acyclic"><span class="pill-tag pill_color_Acyclic">Acyclic<span></span></span></a> graphs.</p>
<div>
<b>Originally found at: </b><a href="http://www.graphdrawing.org/data.html">http://www.graphdrawing.org/data.html</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d949574cf748107605564e" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96e3b1101aa0ea66a0c45" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d970d31101aa0ea36a0cb0" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90e87803e0c0b04558c1f" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>1277 graphs, 10 to 100 nodes, 9 to 241 edges
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div>
<a href="https://doi.org/10.1142/S0218195900000358"><span class="citation" data-cites="DIBATTISTA2000">[@DIBATTISTA2000]</span></a> &nbsp; <a style="cursor: pointer; text-decoration: underline; color: rgba(var(--bs-link-color-rgb)" onclick="citation(8)"></a> &nbsp;Drawing Directed Acyclic Graphs: An Experimental Study
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Hachul2005">[@Hachul2005]</span>
</div>
<div class="col-4">
Drawing Large Graphs with a Potential-Field-Based Multilevel Algorithm
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Hachul2006">[@Hachul2006]</span>
</div>
<div class="col-4">
An Experimental Comparison of Fast Algorithms for Drawing General Large Graphs
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">

</div>
<div class="col-4">
Large-Graph Layout with the Fast Multipole Multilevel Method
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Jabrayilov2016">[@Jabrayilov2016]</span>
</div>
<div class="col-4">
Compact Layered Drawings of General Directed Graphs
</div>
<div class="col-4">
<a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Jnger2018">[@Jnger2018]</span>
</div>
<div class="col-4">
A Flow Formulation for Horizontal Coordinate Assignment with Prescribed Width
</div>
<div class="col-4">
<a href="#Other"><span class="pill-tag pill_color_other">DAG</span></a><a href="#Hierarchical"><span class="pill-tag pill_color_Hierarchical">Hierarchical</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Mallach2019">[@Mallach2019]</span>
</div>
<div class="col-4">
A Natural Quadratic Approach to the Generalized Graph Layering Problem
</div>
<div class="col-4">
<a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Chimani2012">[@Chimani2012]</span>
</div>
<div class="col-4">
Advances in the Planarization Method: Effective Multiple Edge Insertions
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="barth2002">[@barth2002]</span>
</div>
<div class="col-4">
Simple and Efficient Bilayer Cross Counting
</div>
<div class="col-4">
<a href="#Bipartite"><span class="pill-tag pill_color_Bipartite">Bipartite</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="DIBATTISTA2000">[@DIBATTISTA2000]</span>
</div>
<div class="col-4">
Drawing Directed Acyclic Graphs: An Experimental Study
</div>
<div class="col-4">
<a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="welch2017">[@welch2017]</span>
</div>
<div class="col-4">
Measuring Symmetry in Drawings of Graphs
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-1" class="level4">
<h4 class="anchored" data-anchor-id="statistics-1">Statistics</h4>
<div style="text-align: center">
<img src="data/Benchmark datasets/AT&amp;T e86f130c42344169a9d75a61abc7e487/four_in_one.svg" width="80%">
</div>
</section>
<section id="descriptions-from-literature-1" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-1">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.1142/s0218195900000358">Drawing Directed Acyclic Graphs: An Experimental Study</a>”:</p>
<blockquote class="blockquote">
<p>The experimental study was performed on two different sets of DAGs, both with a strong connection to “real-life” applications. We considered two typical contexts where DAGs play a fundamental role, namely software engineering and project planning.</p>
</blockquote>
<blockquote class="blockquote">
<p>The first set of test DAGs are what we call the North DAGs. They are obtained from a collection of directed graphs [28], that North collected at AT&amp;T Bell Labs by running for two years <em>Draw DAG</em>, an e-mail graph drawing service that accepts directed graphs formatted as e-mail messages and returns messages with the corresponding drawings [27].</p>
</blockquote>
<blockquote class="blockquote">
<p>Originally, the North DAGs consisted of 5114 directed graphs, whose number of vertices varied in the range <em>1 … 7602</em>. However, the density of the directed graphs with a number of vertices that did not fall in the range <em>10 … 100</em> was very low (see also the statistics in [28]); since such directed graphs represent a very sparse statistical population we decided to discard them. Then we noted that many directed graphs were isomorphic; since the vertices of the directed graphs have labels associated with them, the problem is tractable. For each isomorphism class, we kept only one representative directed graph. Also, we deleted the directed graphs where subgraphs were specified as clusters, to be drawn in their own distinct rectangular region of the layout, because constrained algorithms are beyond the scope of this study. This filtering left us with 1277 directed graphs.</p>
</blockquote>
<blockquote class="blockquote">
<p>Still, 491 directed graphs were not connected and this was a problem for running algorithms implemented in <em>G D W</em> (they assume input directed graphs to be connected). Instead of discarding the directed graphs, we followed a more practical approach, by randomly adding a minimum set of directed edges that makes each directed graph connected. Finally, we made the directed graph acyclic, where necessary, by applying some heuristics for inverting the direction of a small subset of edges.</p>
</blockquote>
<blockquote class="blockquote">
<p>We then ran a first set of experiments and produced the statistics by grouping the DAGs by number of vertices. Although the comparison among the algorithms looked consistent (the produced plots never oddly overlapped), each single plot was not satisfactory, because it showed peaks and valleys. We went back to study the test suite and observed that grouping them by number of vertices was not the best approach. In fact, the North DAGs come from very heterogeneous sources, mainly representing different phases of various software engineering projects; as a result, directed graphs with more or less the same number of vertices may be either very dense or very sparse.</p>
</blockquote>
<blockquote class="blockquote">
<p>Since most of the analyzed quality measures strongly depend on the number of edges of the DAG (e.g.&nbsp;area, number of bends, and number of crossings), we decided that a better approach was to group the DAGs by number of edges. After some tests, we clustered the DAGs into nine groups, each with at least 40 DAGs, so that the number of edges in the DAGs belonging to group <em>i, 1 ≤ i ≤ 9</em>, is in the range <em>10 i … 10 i+9</em> (see Fig. 3). The resulting test suite consists of 1158 DAGs, each with edges in the range <em>10 … 99</em>.</p>
</blockquote>
<p>From “<a href="https://doi.org/10.1145/1671970.1671975">Layer-Free Upward Crossing Minimization</a>”:</p>
<blockquote class="blockquote">
<p><em>North DAGs.</em> The North DAGs have been introduced in an experimental comparison of algorithms for drawing DAGs [Di Battista et al.&nbsp;2000]. The benchmark set contains 1,158 DAGs collected by Stephen North, which were slightly modified by Di Battista et al.&nbsp;The graphs are grouped into nine sets, where set <em>i</em> contains graphs with <em>10 i</em> to <em>10 i+9</em> arcs for <em>i=1, …, 9</em>.</p>
</blockquote>
</section>
<section id="example-figures-1" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-1">Example Figures</h4>
<p>From “<a href="https://doi.org/10.1142/s0218195900000358">Drawing Directed Acyclic Graphs: An Experimental Study</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/AT&amp;T e86f130c42344169a9d75a61abc7e487/Untitled.png" width="80%">
</div>
<div style="text-align: center">
<img src="data/Benchmark datasets/AT&amp;T e86f130c42344169a9d75a61abc7e487/Untitled 1.png" width="80%">
</div>
<p>From “<a href="https://doi.org/10.1007/978-3-030-35802-0_40">A Natural Quadratic Approach to the Generalized Graph Layering Problem</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/AT&amp;T e86f130c42344169a9d75a61abc7e487/Screen_Shot_2023-08-09_at_4.26.57_PM.png" width="80%">
</div>
<p>From “<a href="https://doi.org/10.1007/978-3-030-04414-5_13">A Flow Formulation for Horizontal Coordinate Assignment with Prescribed Width</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/AT&amp;T e86f130c42344169a9d75a61abc7e487/Screen_Shot_2023-08-09_at_4.24.19_PM.png" width="80%">
</div>
</section>
</div>
</div>
</div>
<p><a name="North_DAGs"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="North DAGs" data-collapse-id="North_DAGs">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-4-contents" aria-controls="callout-4" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
North DAGs
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-4" class="callout-4-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<!-- ```{ojs}
//| echo: false

make_sparkline("North_DAGs")
```

<div id="named-list-North_DAGs" data-bs-spy="scroll"  data-bs-target="db-nav-list" data-bs-offset="20" tabindex="0"></div> -->
<p>This is the same collection as AT&amp;T, but referred to as North DAGs in the papers we analyzed. We kept the two collections separate, based on how they are referred to in the papers we collected.</p>
<div>
<b>Originally found at: </b>See AT&amp;T
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d949574cf748107605564e" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96e3b1101aa0ea66a0c45" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d970d31101aa0ea36a0cb0" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90e87803e0c0b04558c1f" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>See AT&amp;T
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="DIBATTISTA2000">[@DIBATTISTA2000]</span>
</div>
<div class="col-8">
Drawing Directed Acyclic Graphs: An Experimental Study
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="chimani2010">[@chimani2010]</span>
</div>
<div class="col-4">
Layer-Free Upward Crossing Minimization
</div>
<div class="col-4">
<a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="chimani2016">[@chimani2016]</span>
</div>
<div class="col-4">
A Note on the Practicality of Maximal Planar Subgraph Algorithms
</div>
<div class="col-4">
<a href="#Planar"><span class="pill-tag pill_color_Planar">Planar</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Regg2016">[@Regg2016]</span>
</div>
<div class="col-4">
A Generalization of the Directed Graph Layering Problem
</div>
<div class="col-4">
<a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Demel2018">[@Demel2018]</span>
</div>
<div class="col-4">
A Greedy Heuristic for Crossing-Angle Maximization
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Chimani2021">[@Chimani2021]</span>
</div>
<div class="col-4">
Star-Struck by Fixed Embeddings: Modern Crossing Number Heuristics
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="chimani2016b">[@chimani2016b]</span>
</div>
<div class="col-4">
An ILP-based Proof System for the Crossing Number Problem
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="DIBATTISTA2000">[@DIBATTISTA2000]</span>
</div>
<div class="col-4">
Drawing Directed Acyclic Graphs: An Experimental Study
</div>
<div class="col-4">
<a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Binucci2016">[@Binucci2016]</span>
</div>
<div class="col-4">
Placing Arrows in Directed Graph Drawings
</div>
<div class="col-4">
<a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="klammler2018">[@klammler2018]</span>
</div>
<div class="col-4">
Aesthetic Discrimination of Graph Layouts
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="binucci2022">[@binucci2022]</span>
</div>
<div class="col-4">
Placing Arrows in Directed Graph Layouts: Algorithms and Experiments
</div>
<div class="col-4">
<a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a>
</div>
</div>
<p><br></p>
<section id="descriptions-from-literature-2" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-2">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.1145/1671970.1671975">Layer-Free Upward Crossing Minimization</a>”:</p>
<p><strong>North DAGs.</strong> The North DAGs have been introduced in an experimental comparison of algorithms for drawing DAGs [Di Battista et al.&nbsp;2000]. The benchmark set contains 1,158 DAGs collected by Stephen North, which were slightly modified by Di Battista et al.&nbsp;The graphs are grouped into nine sets, where set <span class="math inline">\(i\)</span> contains graphs with <span class="math inline">\(10 i\)</span> to <span class="math inline">\(10 i+9\)</span> arcs for <span class="math inline">\(i=1, \ldots, 9\)</span>.</p>
<!-- === STOP RENDERING ===

is this the AT&T collection? [http://www.graphdrawing.org/data/](http://www.graphdrawing.org/data/)

from [Drawing Directed Acyclic Graphs: An Experimental Study](../Benchmark%20sets%200cc6b5e454304aec98f3b59b1a720476/Literature%20ad87f14e7097454fb2f784e2c8a2797a/Literature%20-%20Algorithm%2012e01bfc60a84007aa7d2d34293e123d/Drawing%20Directed%20Acyclic%20Graphs%20An%20Experimental%20St%201677531652194663b7fdf25025c61cc6.md) 

 

<div style='text-align: center'><img src='data/Benchmark datasets/North%20DAGs%20a58f7143ef524c8a8c737df90162d3fb/Untitled.png' width='80%'></div>

from [Layer-Free Upward Crossing Minimization](../Benchmark%20sets%200cc6b5e454304aec98f3b59b1a720476/Literature%20ad87f14e7097454fb2f784e2c8a2797a/Literature%20-%20Algorithm%2012e01bfc60a84007aa7d2d34293e123d/Layer-Free%20Upward%20Crossing%20Minimization%20676f8a507ff343b5ae83c34446796fb5.md) 

<div style='text-align: center'><img src='data/Benchmark datasets/North%20DAGs%20a58f7143ef524c8a8c737df90162d3fb/Untitled%201.png' width='80%'></div> -->
</section>
</div>
</div>
</div>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="RandDAG or DAGMar" data-collapse-id="RandDAG">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-5-contents" aria-controls="callout-5" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
RandDAG or DAGMar
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-5" class="callout-5-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>The randDAG collection concludes the collections that can be easily accessed from graphdrawing.org. http://<a href="http://graphdrawing.org">graphdrawing.org</a> highlights the DAGmar graph generator, and provides a benchmark set of randomly generated directed acyclic graphs. While not explicit, it is implied this benchmark comes from the DAGmar generator. The collection is uniformly sampled from set of level graphs with certain graph parameters, such as number of nodes and number of edges. These graphs have no particular features, thus they are classified as <a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a></p>
<!-- ```{ojs}
//| echo: false

make_sparkline("RandDAG")
```

<div id="named-list-RandDAG" data-bs-spy="scroll"  data-bs-target="db-nav-list" data-bs-offset="20" tabindex="0"></div> -->
<div>
<b>Originally found at: </b><a href="http://www.graphdrawing.org/data.html">http://www.graphdrawing.org/data.html</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d948061101aa0dc66a0c55" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96dec94a6be112512e706" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d97088803e0c0cb6558d7e" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90eff803e0c0b04558ca3" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>10-100 nodes, 15-167 edges
</div>
<br>
<div>
<b>Number of Graphs: </b>909
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Bachmaier_Gleißner_Hofmeier_2012">[@Bachmaier_Gleißner_Hofmeier_2012]</span>
</div>
<div class="col-8">
DAGmar: Library for DAGs
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="klammler2018">[@klammler2018]</span>
</div>
<div class="col-4">
Aesthetic Discrimination of Graph Layouts
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-2" class="level4">
<h4 class="anchored" data-anchor-id="statistics-2">Statistics</h4>
<div style="text-align: center">
<img src="data/Benchmark datasets/RandDAG 54b6ed8fdecb4256b2cca6298f09965e/four_in_one.svg" width="80%">
</div>
</section>
<section id="descriptions-from-the-literature" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-the-literature">Descriptions from the Literature</h4>
<p>From “<a href="https://link.springer.com/chapter/10.1007/978-3-030-04414-5_12">Aesthetic Discrimination of Graph Layouts</a>”:</p>
<blockquote class="blockquote">
<p>We have assembled such a dataset using two types of sources. First, we used the collections of the well-known graph archives ROME, NORTH and RANDDAG which are published on <a href="http://www.graphdrawing.org/">graphdrawing.org</a> as well as the NIST’s “Matrix Market” [<a href="https://link.springer.com/chapter/10.1007/978-3-030-04414-5_12#ref-CR2">2</a>].</p>
</blockquote>
</section>
<section id="example-figures-2" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-2">Example Figures</h4>
<p>From “<a href="https://link.springer.com/chapter/10.1007/978-3-030-04414-5_12">Aesthetic Discrimination of Graph Layouts</a> (<a href="https://arxiv.org/abs/1809.01017">Appendix</a>)”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/RandDAG 54b6ed8fdecb4256b2cca6298f09965e/Untitled.png" width="80%">
</div>
<p><strong>Fig. 5.</strong> (cropped)… All graphs are visualized using the <span class="math inline">\(FM^3\)</span> algorithm.</p>
<!-- === STOP RENDERING ===

Might be same as DAGmar graphs? looked at internet archive to see what the GD data website looked like in 2018 and there is no mention of RAndDAG as a thing, but there are the randomly generated DAGs which makes sense tto believe it is those. -->
</section>
</div>
</div>
</div>
<p>A very common problem in graph drawing is approximating or computing the exact <strong>crossing number</strong> of a graph. The datasets listed below are particularly useful when researchers are particularly useful for this task, as they contain graphs with <a href="#Known_crossing_number"><span class="pill-tag pill_color_Known_crossing_number">Known crossing numbers</span></a>, so that other algorithms’ results can be tested against them.</p>
<p><a name="KnownCR"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="KnownCR" data-collapse-id="KnownCR">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-6-contents" aria-controls="callout-6" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
KnownCR
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-6" class="callout-6-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p><strong>KnownCR</strong> is a collection of graphs with <a href="#Known_crossing_number"><span class="pill-tag pill_color_Known_crossing_number">Known crossing numbers</span></a>, and it is used to test the performance of algorithms that approximate the crossing number of a graph. The dataset is used in papers that aim to approximate the crossing number of a graph, and it is used to compare the results of the approximation algorithms to the known crossing numbers of the graphs. A good resource for this is also the survey provided by <span class="citation" data-cites="Clancy2019ASO">[@Clancy2019ASO]</span>. The graphs themselves do not appear to exist online elsewhere, but the method by which they can be created is fully described in <span class="citation" data-cites="gutwenger_phd">[@gutwenger_phd]</span>.</p>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d948121101aa0dc76a0c94" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96dce0c2b4d0f653862d7" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d9715c0c2b4d0f65386577" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90ef60c2b4d0d2e3862be" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>9-250 nodes, 18-750 edges
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="gutwenger_phd">[@gutwenger_phd]</span>
</div>
<div class="col-8">
Application of SPQR-Trees in the Planarization Approach for Drawing Graphs
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Chimani2021">[@Chimani2021]</span>
</div>
<div class="col-4">
Star-Struck by Fixed Embeddings: Modern Crossing Number Heuristics
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Known_crossing_number"><span class="pill-tag pill_color_Known_crossing_number">Known crossing number</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Clancy2019">[@Clancy2019]</span>
</div>
<div class="col-4">
An effective crossing minimisation heuristic based on star insertion
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Known_crossing_number"><span class="pill-tag pill_color_Known_crossing_number">Known crossing number</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Chimani2012">[@Chimani2012]</span>
</div>
<div class="col-4">
Advances in the Planarization Method: Effective Multiple Edge Insertions
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Known_crossing_number"><span class="pill-tag pill_color_Known_crossing_number">Known crossing number</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-3" class="level4">
<h4 class="anchored" data-anchor-id="statistics-3">Statistics</h4>
<div style="text-align: center">
<img src="data/Benchmark datasets/KnownCR ed33d600c65e4c52a1506a905b774335/four_in_one.svg" width="80%">
</div>
</section>
<section id="description-from-literature" class="level4">
<h4 class="anchored" data-anchor-id="description-from-literature">Description from Literature</h4>
<p>From “<a href="https://dx.doi.org/10.7155/jgaa.00487">An effective crossing minimisation heuristic based on star insertion</a>”:</p>
<blockquote class="blockquote">
<p>KnownCR graphs- these are a set of instances containing between 9 and 250 vertices, first collected by Gutwenger [22], which can be further partitioned into four families of graphs as follows: <span class="math inline">\(C_i \times C_j\)</span>: The Cartesian product of the cycle on <span class="math inline">\(i\)</span> vertices with the cycle on <span class="math inline">\(j\)</span> vertices. The instances contain graphs with <span class="math inline">\(3 \leq i \leq 7\)</span> and <span class="math inline">\(j \geq i\)</span> such that <span class="math inline">\(ij \leq 250\)</span>. <span class="math inline">\(G_i \times P_j\)</span>: The Cartesian product of the path on <span class="math inline">\(j+1\)</span> vertices with one of the 21 non-isomorphic connected graphs on 5 vertices, where <span class="math inline">\(i\)</span> denotes which of the 21. The instances contain graphs with <span class="math inline">\(3 \leq j \leq 49\)</span>. <span class="math inline">\(G_i \times C_j\)</span>: The Cartesian product of the cycle on <span class="math inline">\(j\)</span> vertices with one of the 21 non-isomorphic connected graphs on 5 vertices, where <span class="math inline">\(i\)</span> denotes which of the 21. The crossing number of these graphs is only known for some of the <span class="math inline">\(G_i\)</span> and only these cases are included. The instances contain graphs with <span class="math inline">\(3 \leq j \leq 50\)</span>. The Generalised Petersen graphs <span class="math inline">\(P(j,2)\)</span> and <span class="math inline">\(P(j,3)\)</span>, on <span class="math inline">\(2j\)</span> vertices. We shall only use those of type <span class="math inline">\(P(j,3)\)</span> as <span class="math inline">\(P(j,2)\)</span> (studied in [20]) are easy for heuristics to solve, as has already been observed in [9]. The instances contain graphs with <span class="math inline">\(9 \leq j \leq 125\)</span>.</p>
</blockquote>
<p>From “<a href="https://doi.org/10.1007/978-3-642-25878-7_10">Advances in the Planarization Method: Effective Multiple Edge Insertions</a>”:</p>
<blockquote class="blockquote">
<p>Finally, the KnownCR graphs [11] are a collection of 1946 graphs with known crossing numbers (by proofs), consisting of generalized Petersen graphs (P(m, 2), P(m, 3)) and products of cycles <span class="math inline">\(C_n\)</span>, paths <span class="math inline">\(P_n\)</span>, and 5-vertex graphs <span class="math inline">\(G_i (C_m \times C_n, G_i \times P_n, G_i \times C_n)\)</span>; these graphs have between 9 and 250 nodes.</p>
</blockquote>
</section>
<section id="example-figures-3" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-3">Example Figures</h4>
<p>From “<a href="https://eldorado.tu-dortmund.de/handle/2003/27430">Application of SPQR-Trees in the Planarization Approach for Drawing Graphs</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/KnownCR ed33d600c65e4c52a1506a905b774335/Screen_Shot_2023-01-28_at_2.42.42_PM.png" width="80%">
</div>
<!-- === STOP RENDERING ===

[Eldorado: Application of SPQR-trees in the planarization approach for drawing graphs](https://eldorado.tu-dortmund.de/handle/2003/27430)

From *An effective crossing minimisation heuristic based on star insertion,* “KnownCR graphs- these are a set of instances containing between 9 and 250 vertices, first collected by Gutwenger [22], which can be further partitioned into four families of graphs as follows:

- Ci ×Cj: the Cartesian product of the cycle on i vertices with the cycle on j vertices. The instances contain graphs with 3 ≤ i ≤ 7 and j ≥i such that ij ≤ 250.
- Gi×Pj: the cartesian product of the path on j+1 vertices with one of the 21 non-isomorphic connected graphs on 5 vertices, where i denotes which of the 21. The instances contain graphs with 3 ≤ j ≤ 49.
- Gi ×Cj: the cartesian product of the cycle on j vertices with one of the 21 non-isomorphic connected graphs on 5 vertices, where i denotes which of the 21. The crossing number of these graphs are only known for some of the Gi and only these cases are included. The instances contain graphs with 3 ≤ j ≤ 50
- The Generalised Petersen graphs P(j,2) and P(j,3), on 2j vertices. We shall only use those of type P(j,3) as P(j,2) (studied in [20]) are154 Clancy et al. An effective crossing minimisation heuristic easy for heuristics to solve as has already been observed in [9]. The instances contain graphs with 9 ≤ j ≤ 125.” from **(An effective crossing minimisation heuristic based on star insertion, Clancy et al. )**

How to construct a cartesian product graph:

<div style='text-align: center'><img src='data/Benchmark datasets/KnownCR%20ed33d600c65e4c52a1506a905b774335/Screen_Shot_2023-01-28_at_2.42.31_PM.png' width='80%'></div>

Example of construction:

Images showing each Gi, and the known crossing number of Gi x Pn:

<div style='text-align: center'><img src='data/Benchmark datasets/KnownCR%20ed33d600c65e4c52a1506a905b774335/Screen_Shot_2023-01-28_at_2.50.06_PM.png' width='80%'></div> -->
</section>
</div>
</div>
</div>
<p><a name="WebCompute"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="WebCompute" data-callout-id="WebCompute">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-7-contents" aria-controls="callout-7" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
WebCompute
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-7" class="callout-7-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>The dataset is comprised of instances of graphs uploaded to Crossing Number WebCompute, attributed to <span class="citation" data-cites="chimani2016c">@chimani2016c</span>. Their website also features proofs of the crossing numbers of many of the Rome-Lib graphs. Newest version of database specifies non-planar graphs, but older versions do have some planar graphs.</p>
<div>
<b>Originally found at: </b><a href="https://tcs.uos.de/doku.php?id=research/cr">https://tcs.uos.de/doku.php?id=research/cr</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d94828803e0c0bf5558bab" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d94d7d803e0c0c1f558c9c" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d9709994a6be112912e9ec" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90ea01101aa0c9e6a0c74" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>3-112 nodes, 3-366 edges
</div>
<br>
<div>
<b>Number of Graphs: </b>507
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="chimani2016b">[@chimani2016b]</span>
</div>
<div class="col-8">
An ILP-based Proof System for the Crossing Number Problem
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="chimani2016b">[@chimani2016b]</span>
</div>
<div class="col-4">
An ILP-based Proof System for the Crossing Number Problem
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Known_crossing_number"><span class="pill-tag pill_color_Known_crossing_number">Known crossing number</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-4" class="level4">
<h4 class="anchored" data-anchor-id="statistics-4">Statistics</h4>
<div style="text-align: center">
<img src="data/Benchmark datasets/WebCompute 7a10c41801914e6faa5a6cc1b1ef0b77/four_in_one.svg" width="80%">
</div>
</section>
<section id="descriptions-from-literature-3" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-3">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.4230/LIPIcs.ESA.2016.29">An ILP-based Proof System for the Crossing Number Problem</a>”:</p>
<blockquote class="blockquote">
<p>To determine the applicability of the proof system, we tested the algorithms on three benchmark sets: the 3110 non-trivial Rome graphs [14], the 1277 North graphs [15], and the145 non-planar graphs (<a href="http://crossings.uos.de/instances">http://crossings.uos.de/instances</a>) collected by our crossing number web-service.</p>
</blockquote>
<!-- === STOP RENDERING ===

Paper attached to his dataset provides no visuliaztion of the graph, [https://arxiv.org/abs/2108.11443](https://arxiv.org/abs/2108.11443) also uses the dataset and provides no visualizations of the graphs themselves -->
</section>
</div>
</div>
</div>
<p><a name="Complete_Graphs"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Complete Graphs" data-callout-id="Complete_Graphs">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-8-contents" aria-controls="callout-8" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Complete Graphs
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-8" class="callout-8-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p><strong>Complete graphs</strong> refers to a generic collection of graphs that are fully connected, i.e., each node is connected to every other node, up to any number of nodes that are needed for the purpose of the experiment. The linked dataset includes both the complete graphs <span class="math inline">\(K_n\)</span> for <span class="math inline">\(5≤n≤50\)</span> and the complete bipartite graphs <span class="math inline">\(K_{n_1,n_2}\)</span> for <span class="math inline">\(5≤n1,n2≤40\)</span>. Crossing number is conjectured for most of these, and while not proven, we found these used in papers to validate minimum crossing numbers. We also note that the papers in our literature review did not provide example figures.</p>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d94965803e0c0c10558bc8" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d94db50c2b4d0ea1386352" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d972030c2b4d0f5d3862dc" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90ebb94a6be0ec012e7e6" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>5-80 nodes, 10-1600 edges
</div>
<br>
<div>
<b>Number of Graphs: </b>1342
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Clancy2019">[@Clancy2019]</span>
</div>
<div class="col-4">
An effective crossing minimisation heuristic based on star insertion
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-5" class="level4">
<h4 class="anchored" data-anchor-id="statistics-5">Statistics</h4>
<div style="text-align: center">
<img src="data/Benchmark datasets/Complete Graphs 202f26621cf34604935433a41b130f10/four_in_one.svg" width="80%">
</div>
</section>
<section id="descriptions-from-literature-4" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-4">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.7155/jgaa.00487">An effective crossing minimisation heuristic based on star insertion</a>”:</p>
<blockquote class="blockquote">
<p><strong>Complete graphs</strong> - Although the crossing number of the complete graph <span class="math inline">\(K_n\)</span> is not known for for <span class="math inline">\(n ≥ 13\)</span>, the value is conjectured, and typically assumed to be correct. We compare the nine combinations of schemes to see how close to the conjectured value each of them is able to get, for various sizes of complete graphs up to <span class="math inline">\(n = 50\)</span>. We indicate how many crossings are obtained after the initial embedding, as well as at the conclusion of the heuristic. We also provide the runtimes, again separated into time spent producing the initial embedding, and time spent in the main loop of the heuristic.</p>
</blockquote>
<blockquote class="blockquote">
<p><strong>Complete bipartite graphs</strong> - Much like the complete graphs, the crossing number of the complete bipartite graph <span class="math inline">\(K_{n1,n2}\)</span> is only known in general for <span class="math inline">\(n_1 ≤ 6\)</span>, but the value is conjectured and typically assumed to be correct. Again, we compare the nine combinations of schemes to see how close to the conjectured value they can get for values up to <span class="math inline">\(n_1,n_2 = 40\)</span>, and report the same data as for the Complete graphs.</p>
</blockquote>
<blockquote class="blockquote">
<p>The crossing number of the complete graph <span class="math inline">\(K_n\)</span> is conjectured (e.g.&nbsp;see Guy [25]) to be equal to <span class="math inline">\(H(n) := 1/4 \left\lfloor n/2\right\rfloor \left\lfloor (n − 1)/2\right\rfloor \left\lfloor(n − 2)/2\right\rfloor \left\lfloor (n − 3)/2\right\rfloor\)</span>. Although this conjecture is widely believed to be correct, it has only been confirmed for <span class="math inline">\(n ≤ 12\)</span> despite considerable effort to extend the results further [30]. We ran the graphs <span class="math inline">\(K_n\)</span> for <span class="math inline">\(5 ≤ n ≤ 50\)</span>. Each graph was run with 100 random permutations and the minimum found solution was compared to <span class="math inline">\(H(n)\)</span> by computing the percent relative deviation from <span class="math inline">\(H(n)\)</span>… For these graphs, we observe that when <span class="math inline">\(n\)</span> was odd, every scheme combination was able to obtain a drawing with <span class="math inline">\(H(n)\)</span> crossings. However, when <span class="math inline">\(n\)</span> was even, each scheme reached a value which was usually very close but not equal to <span class="math inline">\(H(n)\)</span>.</p>
</blockquote>
<blockquote class="blockquote">
<p>The crossing number of the complete bipartite graph <span class="math inline">\(K_{n_1,n_2}\)</span> is conjectured (e.g.&nbsp;see Zarankiewicz [35]) to be equal to <span class="math inline">\(Z(n_1, n_2) := \left\lfloor n_1/2\right\rfloor \left\lfloor (n_1 − 1)/2\right\rfloor \left\lfloor n_2/2\right\rfloor \left\lfloor (n_2 − 1)/2\right\rfloor\)</span>. We ran the graphs <span class="math inline">\(K_{n_1,n_2}\)</span> for <span class="math inline">\(5 ≤ n1 ≤ n2 ≤ 40\)</span>. Each graph was run with 100 random permutations and the minimum found solution was compared to <span class="math inline">\(Z(n_1,n_2)\)</span>… As can be seen in Table 6, QuickCross was successful in obtaining the conjectured optimum in all cases and for all scheme combinations, except <span class="math inline">\(K_{30,30}\)</span> and <span class="math inline">\(K_{40,40}\)</span></p>
</blockquote>
</section>
</div>
</div>
</div>
<p>In some cases, authors of algorithms that deal with particular graph types might prefer to focus on collections with specific attributes, that might be attributes of the nodes, the edges or the entire graph. This is the case of <a href="#Storylines_Movie_Plots" class="dataset-link">Storylines (Movie_Plots)</a> and <a href="#Militarized_Interstate_Disputes_MID" class="dataset-link">Militarized Interstate_Disputes (MID)</a> for <a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Temporal event sequences</span></a> and <a href="#World_Maps" class="dataset-link">World Maps</a> for <a href="#Spatial"><span class="pill-tag pill_color_spatial">Geographical</span></a>data:</p>
<p><a name="Storylines_Movie_Plots"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Storylines (Movie Plots)" data-collapse-id="Storylines_Movie_Plots">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-9-contents" aria-controls="callout-9" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Storylines (Movie Plots)
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-9" class="callout-9-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>The storylines dataset is particularly useful for <a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Temporal event sequence</span></a>visualization because of its <a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a>aspect and <a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters</span></a> (which also evolve through time). It is a collection of graphs that represent movie plots, and the nodes are the characters in the movie. The edges represent the interactions between the characters. This dataset was initially collected by Yuzuru Tanahashi, then stored on his homepage at UC Davis, which, unfortunately, got taken down after some time from its publication. This caused the dataset to be lost for a period of time, as none of the papers that used it provided a link to the dataset or stored it as part of their supplemental material. The experiments contained in such papers were then rendered non-reproducible because of the missing dataset. Through the help of personal connections at UC Davis, we were luckily able to recover the dataset.</p>
<div>
<b>Originally found at: </b>UC Davis’ website, personal page which is now unavailable. Currently distributed here.
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d948c6803e0c0bf6558be7" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96e0b1101aa0ea36a0bd4" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d971960c2b4d0f64386417" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90e9794a6be0ec012e7cb" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>698-24056 nodes, 357-23088 edges
</div>
<br>
<div>
<b>Number of Graphs: </b>6
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Tanahashi2012">[@Tanahashi2012]</span>
</div>
<div class="col-8">
Design Considerations for Optimizing Storyline Visualizations
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Tanahashi2015">[@Tanahashi2015]</span>
</div>
<div class="col-4">
An Efficient Framework for Generating Storyline Visualizations from Streaming Data
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="ShixiaLiu2013">[@ShixiaLiu2013]</span>
</div>
<div class="col-4">
StoryFlow: Tracking the Evolution of Stories
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Tanahashi2012">[@Tanahashi2012]</span>
</div>
<div class="col-4">
Design Considerations for Optimizing Storyline Visualizations
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Gronemann2016b">[@Gronemann2016b]</span>
</div>
<div class="col-4">
Crossing Minimization in Storyline Visualization
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="vanDijk2018">[@vanDijk2018]</span>
</div>
<div class="col-4">
Computing Storyline Visualizations with Few Block Crossings
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Bartolomeo2021">[@Bartolomeo2021]</span>
</div>
<div class="col-4">
Sequence Braiding: Visual Overviews of Temporal Event Sequences and Attributes
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="diBartolomeo2022b">[@diBartolomeo2022b]</span>
</div>
<div class="col-4">
Stratisfimal Layout: A modular optimization model for laying out layered node-link network visualizations
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Other"><span class="pill-tag pill_color_other">Compound graphs</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-6" class="level4">
<h4 class="anchored" data-anchor-id="statistics-6">Statistics</h4>
<div style="text-align: center">
<img src="data/Benchmark datasets/Storylines (Movie Plots) 5d9ea7cbfc2c4a6c88a7ca1bf32c67d7/four_in_one.svg" width="80%">
</div>
</section>
<section id="descriptions-from-literature-5" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-5">Descriptions from Literature</h4>
<p>From “An efficient Framework for Generating Storyline Visualizations from Streaming Data”:</p>
<blockquote class="blockquote">
<p>The datasets we used for this study were the three movie datasets (Star Wars, The Matrix, and Inception) [25] which have been used in previous research as benchmark datasets for measuring layout legibilities. Star Wars consists of 14 entities and 200 time steps. The Matrix consists of 14 entities and 100 time steps. Inception consists of 8 entities and 490 time steps.</p>
</blockquote>
<p>From “Design Considerations for Optimizing Storyline Visualizations” (origin paper):</p>
<blockquote class="blockquote">
<p>The basic form of the data applied in this visualization is simply a chronological list of interactions between various characters. These interactions between characters can be separated into a series of interaction sessions, each session representing the time span of a set of interacting characters. More precisely, we define an interaction session as a unit which holds three attributes: • initiation time, • duration, and • members. Initiation time corresponds to the time when the interaction session begins. Duration corresponds to the duration of the interaction session. Members corresponds to the set of characters involved in the interaction session. Each interaction session represents a time slot in the data where its members are interacting. Note that any change in the formation of members (e.g., a character leaving or joining) indicates a separate interaction session.</p>
</blockquote>
</section>
<section id="example-figures-4" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-4">Example Figures</h4>
<p>From “<a href="https://doi.org/10.1109/TVCG.2015.2392771">An Efficient Framework for Generating Storyline Visualizations from Streaming Data</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Storylines (Movie Plots) 5d9ea7cbfc2c4a6c88a7ca1bf32c67d7/Untitled.png" width="80%">
</div>
<p>From “<a href="https://doi.org/10.1109/TVCG.2013.196">StoryFlow: Tracking the Evolution of Stories</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Storylines (Movie Plots) 5d9ea7cbfc2c4a6c88a7ca1bf32c67d7/Untitled 1.png" width="80%">
</div>
<p>From “<a href="https://doi.org/10.1007/978-3-319-50106-2_29">Crossing Minimization in Storyline Visualizations</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Storylines (Movie Plots) 5d9ea7cbfc2c4a6c88a7ca1bf32c67d7/Untitled 2.png" width="80%">
</div>
</section>
</div>
</div>
</div>
<p><a name="Militarized_Interstate_Disputes_MID"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Militarized Interstate Disputes (MID)" data-callout-id="Militarized_Interstate_Disputes_MID">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-10-contents" aria-controls="callout-10" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Militarized Interstate Disputes (MID)
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-10" class="callout-10-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Collected as part of the <a href="https://correlatesofwar.org/data-sets/MIDs/">Correlates Of War</a> project, the dataset contains two disconnected <a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Temporal event sequences</span></a></p>
<div>
<b>Originally found at: </b><a href="https://correlatesofwar.org/data-sets/MIDs/">https://correlatesofwar.org/data-sets/MIDs/</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d9481e4cf7481059055704" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96e2a1101aa0e9f6a0bd5" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d971de0c2b4d0f6438644d" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90e8f0c2b4d0d2e38626d" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>The dataset consists of 2 disconnected temporal event sequence graphs, one with 11,766 nodes and 5,883 edges, and the other with 9,619 nodes and 8,017 edges.
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="palmer2022">[@palmer2022]</span>
</div>
<div class="col-8">
The MID5 Dataset, 2011–2014: Procedures, coding rules, and description.
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Tanahashi2012">[@Tanahashi2012]</span>
</div>
<div class="col-4">
Design Considerations for Optimizing Storyline Visualizations
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="ShixiaLiu2013">[@ShixiaLiu2013]</span>
</div>
<div class="col-4">
StoryFlow: Tracking the Evolution of Stories
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<p><br></p>
<section id="description-from-literature-1" class="level4">
<h4 class="anchored" data-anchor-id="description-from-literature-1">Description From Literature</h4>
<p>From <a href="https://ieeexplore.ieee.org/document/6327274">Design Considerations for Optimizing Storyline Visualizations:</a></p>
<blockquote class="blockquote">
<p>For the time-varying network, we use the Militarized Interstate Dispute (MID) network dataset [19].</p>
</blockquote>
<blockquote class="blockquote">
<p>For example, in the MID network dataset, which consists of 196 nations and 740 time steps, the genome length was reduced from 1714 to 432, allowing the algorithm to gain a considerable speedup.</p>
</blockquote>
<p>From <a href="https://ieeexplore.ieee.org/document/6634164">StoryFlow: Tracking the Evolution of Stories</a>:</p>
<blockquote class="blockquote">
<p>We conducted the experiment based on three movie datasets and a network dataset: Star Wars, Inception, The Matrix, and the MID network dataset (from 1817 to 1947).</p>
</blockquote>
</section>
<section id="example-figures-5" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-5">Example Figures</h4>
<p>From <a href="https://ieeexplore.ieee.org/document/6327274">Design Considerations for Optimizing Storyline Visualization</a>:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Militarized Interstate Disputes (MID) 27d9df7d11e44a9596267bb930bc6f03/Screen_Shot_2023-01-21_at_2.41.34_PM.png" width="80%">
</div>
<!-- == STOP RENDERING ==

Earliest Citation on Website for version 2.1: [Jones, Daniel M., Stuart A. Bremer and J. David Singer. 1996 .”Militarized Interstate Disputes, 1816-1992: Rationale, Coding Rules, and Empirical Patterns.” *Conflict Management and Peace Science 15*
:163-213](https://doi.org/10.1177/073889429601500203)

Was originally here:

[http://psfaculty.ucdavis.edu/zmaoz/dyadmid.html](http://psfaculty.ucdavis.edu/zmaoz/dyadmid.html)

[**Design Considerations for Optimizing Storyline Visualizations**](../Benchmark%20sets%200cc6b5e454304aec98f3b59b1a720476/Literature%20ad87f14e7097454fb2f784e2c8a2797a/Literature%20-%20Algorithm%2012e01bfc60a84007aa7d2d34293e123d/Design%20Considerations%20for%20Optimizing%20Storyline%20Vis%200a518ed402154bdcb136759d50e09b1b.md) 

Or here: [https://correlatesofwar.org/COW2 Data/MIDs/MID310.html](https://correlatesofwar.org/COW2%20Data/MIDs/MID310.html)
from [**StoryFlow: Tracking the Evolution of Stories**](../Benchmark%20sets%200cc6b5e454304aec98f3b59b1a720476/Literature%20ad87f14e7097454fb2f784e2c8a2797a/Literature%20-%20Algorithm%2012e01bfc60a84007aa7d2d34293e123d/StoryFlow%20Tracking%20the%20Evolution%20of%20Stories%2085d6352e70a54f3e8840adcfea70363a.md) 

**Now it’s here**: [https://correlatesofwar.org/data-sets/MIDs/](https://correlatesofwar.org/data-sets/MIDs/)

And linked to by Z. Maoz’s website: [https://zmaoz.faculty.ucdavis.edu/data-sets/](https://zmaoz.faculty.ucdavis.edu/data-sets/)

StoryFlow: Tracking Evolution of Stories had no visualization-

“We failed to generate the storyline layout of the MID
data, so we used the results provided by Tanahashi.”

Version 5 Bibliography entry:

Palmer, Glenn, Roseanne W. McManus, Vito D’Orazio, Michael R. Kenwick, Mikaela Karstens, Chase Bloch, Nick Dietrich, Kayla Kahn, Kellan Ritter, Michael J. Soules. 2020. “The MID5 Dataset, 2011-2014: Procedures, Coding Rules, and Description.” *Conflict Management and Peace Science*
, 39(4): 470-482. DOI: [https://journals.sagepub.com/doi/full/10.1177/0738894221995743](https://doi.org/10.1177/0738894221995743)

Part of Storylines data -->
</section>
</div>
</div>
</div>
<p><a name="World_Maps"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="World Maps" data-collapse-id="World_Maps">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-11-contents" aria-controls="callout-11" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
World Maps
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-11" class="callout-11-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Datasets contain various attributes for a few locations and their geographical adjacency, namely neighboring states, countries, and municipalities. World Bank country information transformed into a weight-vectors dataset. Some of the incomplete data was filled from disparate sources mentioned in the paper’s supplemental materials: <a href="https://doi.org/10.48550/arXiv.1908.07291">https://doi.org/10.48550/arXiv.1908.07291</a>. The collection contains 14 graphs with <a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a>and <a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>attributes, <a href="#Categorical_nodes"><span class="pill-tag pill_color_Categorical_nodes">Categorical nodes</span></a> and <a href="#Weighted_nodes"><span class="pill-tag pill_color_Weighted_nodes">Weighted nodes</span></a></p>
<div>
<b>Originally found at: </b><a href="https://data.worldbank.org/indicator
">https://data.worldbank.org/indicator </a><a href="https://github.com/loizuf/StableDemersLP/tree/main/code/StableDemers_TVCG/data">https://github.com/loizuf/StableDemersLP/tree/main/code/StableDemers_TVCG/data</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d948f10c2b4d0e8d386217" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96dde0c2b4d0f653862e3" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d971f194a6be112a12eaaa" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d94aca94a6be102f12e736" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>48-514 nodes, 205-2428 edges
</div>
<br>
<div>
<b>Number of Graphs: </b>14
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Nickel2019">[@Nickel2019]</span>
</div>
<div class="col-8">
Computing Stable Demers Cartograms
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Nickel2019">[@Nickel2019]</span>
</div>
<div class="col-4">
Computing Stable Demers Cartograms
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-7" class="level4">
<h4 class="anchored" data-anchor-id="statistics-7">Statistics</h4>
<div style="text-align: center">
<img src="data/Benchmark datasets/World Maps 99689f52cc854daba9c25be42de1cbe3/four_in_one.svg" width="80%">
</div>
</section>
<section id="descriptions-from-literature-6" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-6">Descriptions from Literature</h4>
<p>From <a href="https://link.springer.com/chapter/10.1007/978-3-030-35802-0_4">Computing Stable Demers Cartograms</a>”:</p>
<blockquote class="blockquote">
<p>We run experiments on real-world datasets. For time-series data, we expect a gradual change and strong correlation between the different values. For weight-vectors data, we expect more erratic changes and less correlation. We use two maps with rather different geographic structures: the first (<strong>World</strong>) is a map of world countries, having mixed region (country) sizes in a rather unstructured manner; the second (<strong>US</strong>) is a map of the 48 contiguous US states, having relatively high structure in sizes of its states, with large states in the middle and along the west coast and many smaller states along the east coast.</p>
</blockquote>
<p>From Supplemental Materials, Appendix C of *<a href="https://doi.org/10.48550/arXiv.1908.07291">“Computing Stable Demers Cartograms*”:</a></p>
<div style="text-align: center">
<img src="data/Benchmark datasets/World Maps 99689f52cc854daba9c25be42de1cbe3/Untitled.png" width="80%">
</div>
</section>
<section id="example-figures-6" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-6">Example Figures</h4>
<p>From <a href="https://link.springer.com/chapter/10.1007/978-3-030-35802-0_4">Computing Stable Demers Cartograms</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/World Maps 99689f52cc854daba9c25be42de1cbe3/Untitled 1.png" width="80%">
</div>
<!-- === STOP RENDERING ===

from [Computing Stable Demers Cartograms](../Benchmark%20sets%200cc6b5e454304aec98f3b59b1a720476/Literature%20ad87f14e7097454fb2f784e2c8a2797a/Literature%20-%20Algorithm%2012e01bfc60a84007aa7d2d34293e123d/Computing%20Stable%20Demers%20Cartograms%20fb418899e13f45f4b6b3c6296e07a599.md) 

World bank - bunch of references - all temporal datasets!

Missing Data from the world bank was filled with following linked sources:

https://en.wikipedia.org/wiki/Economy of North Korea#cite note-2

https://tradingeconomics.com/kuwait/rural-population-wb-data.html

http://www.efgs.info/wp-content/uploads/conferences/efgs/2016/S8-1
presentationV1 IdrizShala EFGS2016.pdf

https://tradingeconomics.com/eritrea/rural-population-wb-data.html

https://tradingeconomics.com/kuwait/forest-area-percent-of-land-area-wb-data.
html

https://tradingeconomics.com/kosovo/forest-area-percent-of-land-area-wb-data.
html -->
</section>
</div>
</div>
</div>
<p>The collections presented here are particularly varied in features:</p>
<p><a name="Graphviz_Examples"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Graphviz Examples">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-12-contents" aria-controls="callout-12" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Graphviz Examples
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-12" class="callout-12-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>While graphviz is a graph visualization software, its example gallery has proven useful to many researchers as a source of benchmark datasets. The graphs have various origins, most of which are described on the Graphviz https://www.graphviz.org/gallery/. Example graphs used to generate images with the https://www.graphviz.org for their https://www.graphviz.org/gallery/. These graphs are picked as graphviz examples because they are diverse in types of features they contain: together with generic graphs, there are also graphs with <a href="#clusters"><span class="pill-tag pill_color_clusters">clusters</span></a> <a href="#layered_graphs"><span class="pill-tag pill_color_layered_graphs">layers</span></a> <a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a>and <a href="#Labeled_nodes"><span class="pill-tag pill_color_Labeled_nodes">Labeled nodes</span></a></p>
<div>
<b>Originally found at: </b><a href="https://gitlab.com/graphviz/graphviz/-/tree/main/tests/graphs">https://gitlab.com/graphviz/graphviz/-/tree/main/tests/graphs</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d948d60c2b4d0e8d38620a" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96ce394a6be111712e991" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d970e60c2b4d0f65386519" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90ea91101aa0c9e6a0c97" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>1-1464 nodes, 0-5806 edges
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="gansner2000">[@gansner2000]</span>
</div>
<div class="col-8">
An open graph visualization system and its applications to software engineering
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Gange2011">[@Gange2011]</span>
</div>
<div class="col-4">
Optimal k-level planarization and crossing minimization
</div>
<div class="col-4">
<a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Nachmanson2017">[@Nachmanson2017]</span>
</div>
<div class="col-4">
Node Overlap Removal by Growing a Tree
</div>
<div class="col-4">
<a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="giovannangeli2023">[@giovannangeli2023]</span>
</div>
<div class="col-4">
FORBID: Fast Overlap Removal by&nbsp;Stochastic GradIent Descent for&nbsp;Graph Drawing
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-8" class="level4">
<h4 class="anchored" data-anchor-id="statistics-8">Statistics</h4>
<div style="text-align: center">
<img src="data/Benchmark datasets/Graphviz Examples e90aec42f02c4e8b8a6b8727fe2f86ae/four_in_one.svg" width="80%">
</div>
</section>
<section id="descriptions-from-literature-7" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-7">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.7155/jgaa.00442">Node Overlap Removal by Growing a Tree</a>”:</p>
<blockquote class="blockquote">
<p>Our data includes the same set of graphs that was used by the authors of PRISM to compare it with other algorithms [6]. The dataset is available in the Graphviz open source package.</p>
</blockquote>
<p>From “<a href="https://doi.org/10.1007/978-3-642-18469-7_22">Optimal k-level planarization and crossing minimization</a>”:</p>
<blockquote class="blockquote">
<p>The first set of graphs are all the hierarchical network diagrams appearing in the GraphViz gallery [3]… Table 1 shows the results of minimizing edge crossings and maximizing planar subgraphs with MIP and SAT solvers, as well as the crossings resulting in the Graphviz heuristic layout for graphs from the GraphViz gallery.</p>
</blockquote>
</section>
<section id="example-figures-7" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-7">Example Figures</h4>
<p>From “<a href="https://doi.org/10.7155/jgaa.00442">Node Overlap Removal by Growing a Tree</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Graphviz Examples e90aec42f02c4e8b8a6b8727fe2f86ae/Screen_Shot_2023-08-16_at_3.44.08_PM.png" width="80%">
</div>
<div style="text-align: center">
<img src="data/Benchmark datasets/Graphviz Examples e90aec42f02c4e8b8a6b8727fe2f86ae/Screen_Shot_2023-08-16_at_3.44.43_PM.png" width="80%">
</div>
<p>From “<a href="https://link.springer.com/chapter/10.1007/978-3-031-22203-0_6">FORBID: Fast Overlap Removal by Stochastic GradIent Descent for Graph Drawing</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Graphviz Examples e90aec42f02c4e8b8a6b8727fe2f86ae/Screen_Shot_2023-08-16_at_3.50.39_PM.png" width="80%">
</div>
<p>From “<a href="https://doi.org/10.1007/978-3-642-18469-7_22">Optimal k-level planarization and crossing minimization</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Graphviz Examples e90aec42f02c4e8b8a6b8727fe2f86ae/Screen_Shot_2023-08-16_at_3.35.58_PM.png" width="80%">
</div>
</section>
</div>
</div>
</div>
<p><a name="Scotch_Graph_Collection"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Scotch Graph Collection" data-callout-id="Scotch_GraphCollection">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-13-contents" aria-controls="callout-13" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Scotch Graph Collection
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-13" class="callout-13-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Collection consisting of graphs from various sources including topological meshes, meshes related to physical problems (fluid dynamics, structural mechanics, combinatorial optimization), and interprocess communication graphs for a parallel computing implementation of a factorization solver. The <a href="https://gitlab.inria.fr/scotch/scotch">SCOTCH software package</a> is produced by the <a href="https://www.labri.fr/perso/pelegrin/scotch/">SCOTCH project</a> whose goal is to study static mapping by the means of graph theory, using “divide and conquer’’ graph bipartitioning heuristics. The original link to the data (http://www.labri.u-bordeaux.fr/Equipe/PARADIS/Member/pelegrin/graph) is broken.</p>
<div>
<b>Originally found at: </b><a href="https://wotug.org/parallel/libraries/communication/scotch/Graphs/">https://wotug.org/parallel/libraries/communication/scotch/Graphs/</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d94897803e0c0bfb558d06" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://osf.io/download/w37bv/" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d971ca94a6be112a12ea97" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90eda803e0c0b00558bc8" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>The graphs range from 7 to 144649 nodes, and 10 to 1074393 edges.
</div>
<br>
<div>
<b>Origin Notes: </b>The <a href="https://gitlab.inria.fr/scotch/scotch ">https://gitlab.inria.fr/scotch/scotch </a>is produced by the <a href="https://www.labri.fr/perso/pelegrin/scotch/ ">https://www.labri.fr/perso/pelegrin/scotch/ </a>whose goal is to study static mapping by the means of graph theory, using “divide and conquer’’ graph bipartitioning heuristics. The original link to the data <a href="http://www.labri.u-bordeaux.fr/Equipe/PARADIS/Member/pelegrin/graph ">http://www.labri.u-bordeaux.fr/Equipe/PARADIS/Member/pelegrin/graph </a>is broken.
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="pellegrini:hal-00770422">[@pellegrini:hal-00770422]</span>
</div>
<div class="col-8">
Scotch and PT-Scotch Graph Partitioning Software: An Overview
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="chevalier:hal-00402893">[@chevalier:hal-00402893]</span>
</div>
<div class="col-8">
PT-Scotch: A tool for efficient parallel graph ordering
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="harel2001">[@harel2001]</span>
</div>
<div class="col-4">
A Fast Multi-Scale Method for Drawing Large Graphs
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="koren2002">[@koren2002]</span>
</div>
<div class="col-4">
ACE: A Fast Multiscale Eigenvectors Computation for Drawing Huge Graphs
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-9" class="level4">
<h4 class="anchored" data-anchor-id="statistics-9">Statistics</h4>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Scotch Graph Collection cfe6f105da3a4c699fa3e02ca1f6e88a/four_in_one.svg" width="80%">
</div>
</section>
<section id="descriptions-from-literature-8" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-8">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.1109/INFVIS.2002.1173159">ACE: A Fast Multiscale Eigenvectors Computation for Drawing Huge Graphs</a>”:</p>
<blockquote class="blockquote">
<p>Figure 10c, the 4elt graph, is yet another example of a graph with inherent diversity of distance scales.</p>
</blockquote>
</section>
<section id="example-figures-8" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-8">Example Figures</h4>
<p>From “<a href="https://doi.org/10.1007/3-540-44541-2_18">A Fast Multi-Scale Method for Drawing Large Graphs</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Scotch Graph Collection cfe6f105da3a4c699fa3e02ca1f6e88a/Screen_Shot_2023-08-04_at_8.14.10_AM.png" width="80%">
</div>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Scotch Graph Collection cfe6f105da3a4c699fa3e02ca1f6e88a/Screen_Shot_2023-08-04_at_8.14.29_AM.png" width="80%">
</div>
<p>From “<a href="https://doi.org/10.1109/INFVIS.2002.1173159">ACE: A Fast Multiscale Eigenvectors Computation for Drawing Huge Graphs</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Scotch Graph Collection cfe6f105da3a4c699fa3e02ca1f6e88a/Screen_Shot_2023-08-04_at_8.30.54_AM.png" width="80%">
</div>
<!-- ===  STOP RENDERING ===

```

            ===========================

            The SCOTCH graph collection

            ===========================

The graphs that comprise this collection have been gathered from various
sites and people (Steve Hammond, the Harwell-Boeing collection, Bruce
Hendrickson, and Horst Simon). They are all stored in SCOTCH source graph
format (.src), and are accompanied with their geometry file (.xyz) whenever
available.

This directory contains:

ar_*        : Architecture graphs for classical topologies, mostly meshes.

fe_*        : Triangular and quadrangular unstructured meshes related to
          fluid dynamics, structural mechanics, or combinatorial
          optimization problems.

nd_*        : Valuated interprocess communication graphs issued from a
          parallel implementation of a sparse block Cholesky
          factorization solver, which represent partitions of the
          unknowns induced by a nested dissection method.
          The nd_ref0, nd_ref1, nd_ref2, nd_ref3, and nd_ref4 graphs
          are aliases for the nd_2047.7750, nd_2453.47659,
          nd_2815.84406, nd_3093.105713, and nd_3470.135148 graphs,
          respectively.
```

Author: Francois Pellegrini (*[pelegrin@labri.u-bordeaux.fr](mailto:pelegrin@labri.u-bordeaux.fr)*)
_site:150:21 -->
</section>
</div>
</div>
</div>
<p>Otherwise, one might decide to focus on a particular domain:</p>
<p><a name="Biological_Pathways_KEGG"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Biological Pathways (KEGG)">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-14-contents" aria-controls="callout-14" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Biological Pathways (KEGG)
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-14" class="callout-14-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>According to <a href="https://www.kegg.jp/kegg-bin/show_pathway?map01100">kegg.jp</a>, “the KEGG PATHWAY database is a collection of manually drawn graphical diagrams, called KEGG pathway maps, for metabolic pathways, signaling pathways, pathways involved in various cellular processes and organismal systems, and perturbed pathways associated with human diseases”. The pathway data is separated by module, but the individual modules can be combined into a large graph as visualized at <a href="https://www.kegg.jp/kegg-bin/show_pathway?map01100">https://www.kegg.jp/kegg-bin/show_pathway?map01100</a></p>
<div>
<b>Originally found at: </b><a href="https://www.kegg.jp/kegg/pathway.html#global">https://www.kegg.jp/kegg/pathway.html#global</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d948fe803e0c0c0c558b95" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96d6e1101aa0ea76a0be4" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d970ab4cf748115a055802" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90e80803e0c0b04558c14" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>47-292 nodes, 41-327 edges
</div>
<br>
<div>
<b>Number of Graphs: </b>13
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="kegg">[@kegg]</span>
</div>
<div class="col-8">
KEGG: Kyoto Encyclopedia of Genes and Genomes
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Wu2020">[@Wu2020]</span>
</div>
<div class="col-4">
Multi-Level Area Balancing of Clustered Graphs
</div>
<div class="col-4">
<a href="#Categorical_nodes"><span class="pill-tag pill_color_Categorical_nodes">Categorical nodes</span></a><a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Labeled_nodes"><span class="pill-tag pill_color_Labeled_nodes">Labeled nodes</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-10" class="level4">
<h4 class="anchored" data-anchor-id="statistics-10">Statistics</h4>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Biological Pathways (KEGG) 6d302b16e7af4982a57c5b9aeecfe3b8/four_in_one.svg" width="80%">
</div>
</section>
<section id="descriptions-from-literature-9" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-9">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.1109/TVCG.2020.3038154">Multi-Level Area Balancing of Clustered Graphs</a>”:</p>
<blockquote class="blockquote">
<p>KEGG Overview Pathway. Fig. 6 is the result of reproducing the KEGG overview pathway map using our approach. The color coding of the category here is directly retrieved from the original KEGG database [43], as similarly incorporated in Metabopolis [64] (see figures in supplementary materials, available online). We also set the same threshold for specifying unimportant vertices as in Metabopolis, so that readers can refer to the paper for comparison (Fig. 6). The advantage of this technique allows us to arrange the vertexts in a balanced fashion by pushing vertexts away from each other. This initially gives users an idea of how big each category is, and explicitly shows which reaction is classified under which category. Users can also identify sub-components effectively since those components with identical topological structures are aligned as neighbors. This also helps users to comprehend which structures are associated with certain types of pathways, such as small chains, stars, etc.</p>
</blockquote>
</section>
<section id="example-figures-9" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-9">Example Figures</h4>
<p>From “<a href="https://doi.org/10.1109/TVCG.2020.3038154">Multi-Level Area Balancing of Clustered Graphs</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Biological Pathways (KEGG) 6d302b16e7af4982a57c5b9aeecfe3b8/Screen_Shot_2023-01-25_at_10.50.53_AM.png" width="80%">
</div>
<!-- === STOP RENDERING ===

<div style='text-align: center'><img src='../data/Benchmark datasets/Biological%20Pathways%20(KEGG)%206d302b16e7af4982a57c5b9aeecfe3b8/Screen_Shot_2023-01-25_at_10.52.55_AM.png' width='80%'></div>

<div style='text-align: center'><img src='../data/Benchmark datasets/Biological%20Pathways%20(KEGG)%206d302b16e7af4982a57c5b9aeecfe3b8/Untitled.png' width='80%'></div>
_site:150:21 -->
</section>
</div>
</div>
</div>
<p><a name="Co-Phylogenetic_Trees"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Co-Phylogenetic Trees" data-collapse-id="Co-Phylogenetic_Trees">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-15-contents" aria-controls="callout-15" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Co-Phylogenetic Trees
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-15" class="callout-15-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Collected by the authors of “https://almob.biomedcentral.com/articles/10.1186/s13015-014-0031-3. Each element of the data set is two binary co-phylogentic trees. “Caryophyllaceae &amp; Microbotryum” and “Stinkbugs &amp; Bacteria” are missing but our dataset includes all other files from “https://doi.org/10.1007/978-3-319-73915-1_27”. An additional dataset from “https://almob.biomedcentral.com/articles/10.1186/s13015-014-0031-3”, called “Wolbachia”, is included. Naturally, the data contains <a href="#Trees"><span class="pill-tag pill_color_Trees">Trees</span></a></p>
<div>
<b>Originally found at: </b><a href="https://gitlab.inria.fr/erable/eucalypt/-/blob/master/html/PaperDatasets.zip">https://gitlab.inria.fr/erable/eucalypt/-/blob/master/html/PaperDatasets.zip</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d948a24cf748105905574b" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://osf.io/download/m9wf2/" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d970c00c2b4d0f5d3862a4" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90ee40c2b4d0d2e3862b7" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>13-773 nodes, 12-772 edges
</div>
<br>
<div>
<b>Number of Graphs: </b>30
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Donati2015">[@Donati2015]</span>
</div>
<div class="col-8">
EUCALYPT: efficient tree reconciliation enumerator
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Calamoneri2018">[@Calamoneri2018]</span>
</div>
<div class="col-4">
Visualizing Co-phylogenetic Reconciliations
</div>
<div class="col-4">
<a href="#Trees"><span class="pill-tag pill_color_Trees">Binary trees</span></a><a href="#Trees"><span class="pill-tag pill_color_Trees">Trees</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-11" class="level4">
<h4 class="anchored" data-anchor-id="statistics-11">Statistics</h4>
<div style="text-align: center">
<img src="data/Benchmark datasets/Co-Phylogenetic Trees 2440f3927d254148abd93168c1399b43/four_in_one.svg" width="80%">
</div>
</section>
<section id="descriptions-from-literature-10" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-10">Descriptions from Literature</h4>
<p>From “<a href="https://almob.biomedcentral.com/articles/10.1186/s13015-014-0031-3">EUCALYPT: efficient tree reconciliation enumerator</a>”:</p>
<blockquote class="blockquote">
<p>To test EUCALYPT, we selected 12 datasets from the literature. As we are mostly interested in host-parasite systems, the first 10 datasets concern such relations: EC Encyrtidae &amp; Coccidae [23], GL Gopher &amp; Lice [24], SC Seabirds &amp; Chewing Lice [25], RP Rodents &amp; Pinworms [26], SCF Smut Fungi &amp; Caryophillaceus plants [27], PLML Pelican Lice ML [28] (the trees are generated through a maximum likelihood approach), PLMP Pelican Lice MP [28] (the trees are generated through a maximum parsimony approach), RH Rodents &amp; Hantaviruses [29], PP Primates &amp; Pinworm [30], and FD Fishs and Dactylogyrus [31].</p>
</blockquote>
<blockquote class="blockquote">
<p>In addition, we used a dataset of our own which corresponds to arthropod hosts and a bacterium genus, Wolbachia, living inside the cells of their hosts [32,33]. The datasets were chosen to provide a variety in terms of size of the host and parasite trees: those from the literature are relatively small (from 7 to 100 leaves), while our own data provide an example of much bigger host and parasite trees, each having 387 leaves. Moreover, we were careful that the selected datasets cover, as much as possible, a range of situations in terms of coevolution and of the expected frequencies of each event</p>
</blockquote>
</section>
<section id="example-figures-10" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-10">Example Figures</h4>
<p>From “<a href="https://doi.org/10.1007/978-3-319-73915-1_27">Visualizing Co-phylogenetic Reconciliations</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Co-Phylogenetic Trees 2440f3927d254148abd93168c1399b43/Screen_Shot_2023-08-15_at_10.50.19_AM.png" width="80%">
</div>
<div style="text-align: center">
<img src="data/Benchmark datasets/Co-Phylogenetic Trees 2440f3927d254148abd93168c1399b43/Screen_Shot_2023-08-15_at_10.50.09_AM.png" width="80%">
</div>
<div style="text-align: center">
<img src="data/Benchmark datasets/Co-Phylogenetic Trees 2440f3927d254148abd93168c1399b43/Screen_Shot_2023-08-15_at_10.49.57_AM.png" width="80%">
</div>
<!-- === STOP RENDERING ===

from [Visualizing Co-phylogenetic Reconciliations](../Benchmark%20sets%200cc6b5e454304aec98f3b59b1a720476/Literature%20ad87f14e7097454fb2f784e2c8a2797a/Literature%20-%20Algorithm%2012e01bfc60a84007aa7d2d34293e123d/Visualizing%20Co-phylogenetic%20Reconciliations%20fd4b98c0071f4e96924bacd938e706b5.md) 
Eucalypt tool (Donati, B., Baudet, C., Sinaimeri, B., Crescenzi, P., Sagot, M.F.: EUCALYPT:
efficient tree reconciliation enumerator. Algorithms Mol. Biol. 10(1), 3 (2015)), trees from a bioinfromatics journal (Wieseke, N., Hartmann, T., Bernt, M., Middendorf, M.: Cophylogenetic reconcil-
iation with ILP. IEEE/ACM Trans. Comput. Biol. Bioinform. 12(6), 1227–1235
(2015))

Figure below: An HP-drawing of a reconciliation of Gopher & Lice drawn by SearchMaximalPlanar. (2) The same instance drawn by ShortenHostSwitch.

<div style='text-align: center'><img src='data/Benchmark datasets/Co-Phylogenetic%20Trees%202440f3927d254148abd93168c1399b43/Untitled.png' width='80%'></div>

<div style='text-align: center'><img src='data/Benchmark datasets/Co-Phylogenetic%20Trees%202440f3927d254148abd93168c1399b43/evolu.png' width='80%'></div> -->
</section>
</div>
</div>
</div>
<p>And finally, a collection of graph problems:</p>
<p><a name="SteinLib"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="SteinLib" data-collapse-id="SteinLib">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-16-contents" aria-controls="callout-16" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
SteinLib
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-16" class="callout-16-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p><strong>SteinLib</strong> is a collection of Steiner tree problems in graphs and variants. Their <a href="https://steinlib.zib.de/steinlib.php">website</a> has additional information about each problem in the dataset, including whether or not it has been solved. We omit two sets, <a href="https://steinlib.zib.de/showset.php?Relay">Relay</a> and <a href="https://steinlib.zib.de/showset.php?EFST">EFST</a>, from our provided data due to size constraints.</p>
<div>
<b>Originally found at: </b><a href="https://steinlib.zib.de/steinlib.php">https://steinlib.zib.de/steinlib.php</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d948684cf74810580556ac" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://osf.io/download/zbrf5/" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href=" https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d970300c2b4d0f61386324" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90e5e1101aa0c9f6a0c0f" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>6-38,418 nodes, 9-221,445 edges
</div>
<br>
<div>
<b>Number of Graphs: </b>1207
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Koch2001">[@Koch2001]</span>
</div>
<div class="col-8">
SteinLib: An Updated Library on Steiner Tree Problems in Graphs
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="chimani2016">[@chimani2016]</span>
</div>
<div class="col-4">
A Note on the Practicality of Maximal Planar Subgraph Algorithms
</div>
<div class="col-4">
<a href="#Planar"><span class="pill-tag pill_color_Planar">Planar</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-12" class="level4">
<h4 class="anchored" data-anchor-id="statistics-12">Statistics</h4>
<div style="text-align: center">
<img src="data/Benchmark datasets/SteinLib 4b766ffe22de40ecaa9307671862ea96/four_in_one.svg" width="80%">
</div>
</section>
<section id="descriptions-from-literature-11" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-11">Descriptions from Literature</h4>
<p>From <a href="https://doi.org/10.1007/978-3-319-50106-2_28">A Note on the Practicality of Maximal Planar Subgraph Algorithms</a>:</p>
<blockquote class="blockquote">
<p>We use the non-planar graphs of the established benchmark sets North [12] (423 instances), Rome [11] (8249), and <strong>SteinLib</strong> [16] (586), all of which include real-world instances. In our plots, we group instances according to |V| rounded to the nearest multiple of 10; for Rome we only consider graph with ≥ 25 vertices.</p>
</blockquote>
</section>
<section id="example-figures-11" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-11">Example Figures</h4>
<p>From <a href="https://doi.org/10.1007/978-1-4613-0255-1_9">SteinLib: An Updated Library on Steiner Tree Problems in Graphs</a>:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/SteinLib 4b766ffe22de40ecaa9307671862ea96/Screen_Shot_2023-08-09_at_4.09.50_PM.png" width="80%">
</div>
<!-- === STOP RENDERING ===

Is it lost? -->
</section>
</div>
</div>
</div>
<p>Additional interesting graphs can be found in the <a href="https://mozart.diei.unipg.it/gdcontest/history/">Graph Drawing Contest website</a>.</p>
</section>
<section id="sec-established-network-repositories" class="level3">
<h3 class="anchored" data-anchor-id="sec-established-network-repositories">Established Network Repositories</h3>
<p>A popular choice is to use datasets from <strong>Established Network Repositories</strong>. These are ample collections, often organized in dedicated websites which also offer a few stats about the contained graphs. These collections are particularly useful when trying to locate a graph with a specific structure or property.</p>
<p>Because their hosts are already dedicated to the maintaining and reporting of information on these collections, we do not include here any storage of the data (which would be redundant) or report statistics on them. Rather, our analysis here is focused on highlighting their properties, origins, and ways in which they have been used.</p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Matrix Market">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-17-contents" aria-controls="callout-17" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Matrix Market
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-17" class="callout-17-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Compiled by the <a href="https://math.nist.gov/mcsd/">Mathematical and Computational Sciences Division</a>&nbsp;of the&nbsp;<a href="http://www.itl.nist.gov/">Information Technology Laboratory</a>&nbsp;of the&nbsp;<a href="http://www.nist.gov/">National Institute of Standards and Technology</a>, the <strong>Matrix Market</strong> is a repository of test data for use in comparative studies of algorithms for numerical linear algebra, featuring nearly 500 sparse matrices from a variety of applications, as well as matrix generation tools and services. It has been used for experiments with <a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>graphs, <a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>graphs, and graphs with <a href="#Weighted_edges"><span class="pill-tag pill_color_Weighted_edges">Weighted edges</span></a> Each matrix entry is accompanied by metadata that includes the matrix name and identifier, dimensions (number of rows and columns), number of non-zero elements, symmetry type (symmetric, skew-symmetric, or general) and data type (real, complex, integer, or pattern). The website also provides visualizations of the matrices, helping users understand their structure and distribution of non-zero elements. Downloads are also provided in a variety of formats, including their own Matrix Market Exchange (MME) format, Harwell-Boeing, and MATLAB.</p>
<!-- <div id="named-list-Matrix_Market" data-bs-spy="scroll"  data-bs-target="db-nav-list" data-bs-offset="20" tabindex="0"></div> -->
<div class="col">
<div>
<b>Originally found at: </b><a href="https://math.nist.gov/MatrixMarket/index.html">https://math.nist.gov/MatrixMarket/index.html</a>
</div>
<br>
<div>
<b>Download formatted data: </b>
</div>
<br>
<div>
<b>Note: </b>A repository of test data for use in comparative studies of algorithms for numerical linear algebra, featuring nearly 500 sparse matrices from a variety of applications, as well as matrix generation tools and services.
</div>
<br><br>
<div>
<b>Origin Notes: </b>Compiled by the <a href="https://math.nist.gov/mcsd/&nbsp;">https://math.nist.gov/mcsd/&nbsp;</a>of the&nbsp;<a href="http://www.itl.nist.gov/&nbsp;">http://www.itl.nist.gov/&nbsp;</a>of the&nbsp;<a href="http://www.nist.gov/">http://www.nist.gov/</a>
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div>
Matrix Market: a web resource for test matrix collections
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-8">
<span class="citation" data-cites="klammler2018"><a href="#ref-klammler2018" role="doc-biblioref" aria-expanded="false">[link]</a></span> &nbsp; Aesthetic Discrimination of Graph Layouts
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-8">
<span class="citation" data-cites="KOREN20051867"><a href="#ref-KOREN20051867" role="doc-biblioref" aria-expanded="false">[link]</a></span> &nbsp; Drawing graphs by eigenvectors: theory and practice
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-8">
<span class="citation" data-cites="brandes2006"><a href="#ref-brandes2006" role="doc-biblioref" aria-expanded="false">[link]</a></span> &nbsp; Eigensolver methods for progressive multidimensional scaling of large data
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-8">
<span class="citation" data-cites="Gansner2005"><a href="#ref-Gansner2005" role="doc-biblioref" aria-expanded="false">[link]</a></span> &nbsp; Graph Drawing by Stress Majorization
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Weighted_edges"><span class="pill-tag pill_color_Weighted_edges">Weighted edges</span></a>
</div>
</div>
<div class="row">
<div class="col-8">
<span class="citation" data-cites="meidiana2020"><a href="#ref-meidiana2020" role="doc-biblioref" aria-expanded="false">[link]</a></span> &nbsp; Sublinear Time Force Computation for Big Complex Network Visualization
</div>
<div class="col-4">
<a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<br>
<div>

</div>
<div>
<h3 id="descriptionsfromliterature" class="anchored">
Descriptions from Literature
</h3>
</div>
<div>

</div>
<div>
<p>
From “<a href="https://doi.org/10.1007/978-3-540-31843-9_25">Graph Drawing by Stress Majorization</a>”:
</p>
</div>
<div>

</div>
<div>
<blockquote class="blockquote">
<p>
The first example is the 1138Bus graph (|V|=1138, |E|=1458) from the Matrix Market repository [1]. This graph models a network of high-voltage power distribution lines. Figure 4 shows two layouts of this graph.
</p>
</blockquote>
</div>
<div>

</div>
<div>
<h3 id="examplefigures" class="anchored">
Example Figures
</h3>
</div>
<div>

</div>
<div>
<p>
From “<a href="https://doi.org/10.1016/j.camwa.2004.08.015">Drawing graphs by eigenvectors: theory and practice</a>” (top two layouts are a Matrix Market graph):
</p>
</div>
<div>

</div>
<div style="text-align: center;">
<img src="data/Benchmark datasets/Matrix Market 80bd4320417342458aefa031cf65db1f/Screen_Shot_2023-08-17_at_9.59.19_AM.png" width="80%">
</div>
<div>

</div>
<div>
<p>
From “<a href="https://doi.org/10.1007/978-3-540-31843-9_25">Graph Drawing by Stress Majorization</a>”:
</p>
</div>
<div>

</div>
<div style="text-align: center;">
<img src="data/Benchmark datasets/Matrix Market 80bd4320417342458aefa031cf65db1f/Screen_Shot_2023-08-17_at_11.13.32_AM.png" width="80%">
</div>
</div>
</div>
</div>
</div>
<p><a name="Network_Repository"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Network Repository">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-18-contents" aria-controls="callout-18" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Network Repository
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-18" class="callout-18-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>The <strong>Network Repository</strong> was proposed in 2015 by Rossi and Ahmed of Purdue University as a visually interactive real world graph tool and database. Graphs in this repository are tagged with their real-world source, have in-depth analysis, and even a preview visualization of each individual graph. Types of graphs include social networks, infrastructure networks, biological networks, and many others. The repository also offers sources for individual graphs. It contains many graphs from other benchmark sets described here.</p>
<div>
<b>Originally found at: </b><a href="https://networkrepository.com/">https://networkrepository.com/</a>
</div>
<br> <!-- <div><b>Note: </b>Graphs in this repository are tagged with their real-world source, have in-depth analysis, and even a preview visualization of each individual graph. Types of graphs include social networks, infrastructure networks, biological networks, and many others. The repository also offers sources for individual graphs. It contains many graphs from other benchmark sets described here.</div><br>
<br>
<div><b>Origin Notes: </b>This repository was proposed in 2015 by Rossi and Ahmed of Purdue University as a visually interactive real world graph tool and database.</div><br> -->
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="nr">[@nr]</span>
</div>
<div class="col-8">
The Network Data Repository with Interactive Graph Analytics and Visualization
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Eades2018">[@Eades2018]</span>
</div>
<div class="col-4">
Drawing Big Graphs Using Spectral Sparsification
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Arleo2019">[@Arleo2019]</span>
</div>
<div class="col-4">
A Distributed Multilevel Force-Directed Algorithm
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<p><br></p>
<section id="descriptions-from-literature-12" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-12">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.1109/TPDS.2018.2869805">A Distributed Multilevel Force-Directed Algorithm</a>”:</p>
<blockquote class="blockquote">
<p>To evaluate the running time of MULTI-GILA, we used the REALGRAPHS and BIGGRAPHS benchmarks, which contain larger graphs extracted from both real-world applications and synthetic generators; see Table 4. The REALGRAPHS set includes the 5 largest real-world graphs (mainly scale-free graphs) used in the experimental study of GILA [5]. These graphs are taken from the Stanford Large Networks Dataset Collection [3] and from the <strong>Network Data Repository</strong> [46], and their number of edges is between 121523 and 1541514.</p>
</blockquote>
<p>From “<a href="https://link.springer.com/chapter/10.1007/978-3-319-73915-1_22">Drawing Big Graphs Using Spectral Sparsification</a>”:</p>
<blockquote class="blockquote">
<p>We used three data sets. The first set of graphs is taken from “defacto-benchmark” graphs, including the Hachul library, Walshaw’s Graph Partitioning Archive, the sparse matrices collection [<a href="https://link.springer.com/chapter/10.1007/978-3-319-73915-1_22#ref-CR6">6</a>] and the network repository [<a href="https://link.springer.com/chapter/10.1007/978-3-319-73915-1_22#ref-CR24">24</a>]. These include two types of graphs that have been extensively studied in graph drawing research: grid-like graphs and scale-free graphs.</p>
</blockquote>
</section>
</div>
</div>
</div>
<p><a name="Pajek"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Pajek">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-19-contents" aria-controls="callout-19" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Pajek
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-19" class="callout-19-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>The <strong>Pajek</strong> Program for Large Network Analysis is a tool developed and hosted by Andrej Vlado and some of their colleagues. As part of this program, they later compiled relevant graphs and links to other datasets, which we call today the Pajek collection. As a curiosity, pajek means spider in Slovenian. Many of Pajek graphs have been included as part of the <a href="#SuiteSparse_Matrix_Collection">SuiteSparse Matrix Collection</a>.</p>
<div>
<b>Originally found at: </b><a href="http://vlado.fmf.uni-lj.si/pub/networks/data/">http://vlado.fmf.uni-lj.si/pub/networks/data/</a>
</div>
<br> <!-- <div><b>Download formatted data: </b></div><br> --> <!-- <div><b>Note: </b>Pajek means spider in Slovenian.</div><br>
<br>
<div><b>Origin Notes: </b>The Pajek Program for Large Network Analysis is a tool developed and hosted by Andrej Vlado and some of their colleges. As part of this program, they later compiled relevant graphs and links to other datasets, which we call today the Pajek collection.</div><br> -->
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="pajek">[@pajek]</span>
</div>
<div class="col-8">
Pajek datasets
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="noack2004">[@noack2004]</span>
</div>
<div class="col-4">
Energy Models for Graph Clustering
</div>
<div class="col-4">
<a href="#Clusters_generated"><span class="pill-tag pill_color_Clusters_generated">Clusters (generated)</span></a><a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Meidiana2019">[@Meidiana2019]</span>
</div>
<div class="col-4">
A Quality Metric for Visualization of Clusters in Graphs
</div>
<div class="col-4">
<a href="#Clusters_generated"><span class="pill-tag pill_color_Clusters_generated">Clusters (generated)</span></a>
</div>
</div>
<p><br></p>
<section id="descriptions-from-literature-13" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-13">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.1007/978-3-540-24595-7_40">Energy Models for Graph Clustering</a>”:</p>
<blockquote class="blockquote">
<p><strong>Airline Routing (Figure 5):</strong> The nodes of this graph represent US airports, and the (unweighted) edges represent direct flights. The probability that two airports are connected by a direct flight is strongly related to their geographical distance: Most direct flights are relatively short, and only few large hub airports are connected by direct long-range flights. The distances in the edge-repulsion LinLog layout resemble the relative geographical distances of the airports remarkably closely, given that the graph does not contain any explicit information about geographical distances.</p>
</blockquote>
<blockquote class="blockquote">
<p><strong>Dictionary (Figure 7):</strong> The nodes represent terms in the Online Dictionary of Library and Information Science (ODLIS), and the edges represent hyperlinks. A hyperlink between two terms exists if one term is used to describe the meaning of the other term, and thus connects semantically related terms. The edge-repulsion LinLog layout indeed groups semantically related terms, which is better reflected in the VRML file on the supplementary web page than in Figure 7(c). Such a grouping is useful, for example, for discovering the global topic areas (like publishing, printing, information technology, etc.), for identifying entry points for the exploration of topics, or for finding semantically related terms even if they are not explicitly linked.</p>
</blockquote>
<p>From “<a href="https://link.springer.com/chapter/10.1007/978-3-030-35802-0_10">A Quality Metric for Visualization of Clusters in Graphs</a>”:</p>
<blockquote class="blockquote">
<p>We re-used some datasets from the validation experiments and created some new ones, listed in Table <a href="https://link.springer.com/chapter/10.1007/978-3-030-35802-0_10#Tab2">2</a>. We also selected real world graph datasets with existing vertex categorization, which are listed under the double line in Table <a href="https://link.springer.com/chapter/10.1007/978-3-030-35802-0_10#Tab2">2</a>. The datasets were taken from Pajek [<a href="https://link.springer.com/chapter/10.1007/978-3-030-35802-0_10#ref-CR2">2</a>] and Stanford Network Analysis Project’s (SNAP) repository [<a href="https://link.springer.com/chapter/10.1007/978-3-030-35802-0_10#ref-CR23">23</a>, <a href="https://link.springer.com/chapter/10.1007/978-3-030-35802-0_10#ref-CR40">40</a>].</p>
</blockquote>
</section>
<section id="example-figures-12" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-12">Example Figures</h4>
<p>From “<a href="https://doi.org/10.1007/978-3-540-24595-7_40">Energy Models for Graph Clustering</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Pajek e323e05a735d4151a0bccfec08aa3bd6/Screen_Shot_2023-08-17_at_4.05.25_PM.png" width="80%">
</div>
<div style="text-align: center">
<img src="data/Benchmark datasets/Pajek e323e05a735d4151a0bccfec08aa3bd6/Screen_Shot_2023-08-17_at_4.06.19_PM.png" width="80%">
</div>
<!-- === STOP RENDERING ===

[http://vlado.fmf.uni-lj.si/pub/networks/data/](http://vlado.fmf.uni-lj.si/pub/networks/data/)

Direct flights between US airports (332 nodes, 2126 edges). Some

distant airports in Alaska and the South Sea (e.g. Guam) are omitted to improve

readability.

[](http://vlado.fmf.uni-lj.si/pub/networks/doc/)

Collection of other dataset that might not be linked into paper

There is another link to data not mentioned in their repo. (Also part of SuiteSpase Matrix Collection) -->
</section>
</div>
</div>
</div>
<p><a name="SNAP"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="SNAP (Stanford Network Analysis Platform)" data-collapse-id="SNAP">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-20-contents" aria-controls="callout-20" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
SNAP (Stanford Network Analysis Platform)
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-20" class="callout-20-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>The <strong>SNAP</strong> repository is a collection of datasets assembled as part of the Stanford Network Analysis Platform, which started in 2004 and launched their website in 2009. Well-known, widely used graph repository. <!-- There also exists BioSNAP, which consists of similar datasets for medical applications under the same umbrella project. While BioSNAP was not as relevant in the literature, it may still prove useful for future researchers.  --> A number of graphs that became relevant individually are included in SNAP, such as <a href="#Enron" class="dataset-link">Enron</a>, <a href="#amazon">Amazon</a>, <a href="#protein-interaction">Protein Interactions</a> datasets and various <a href="#social-network">Social Network</a> graphs. SNAP contains <a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>graphs, <a href="#directed_edges"><span class="pill-tag pill_color_directed_edges">directed</span></a>and <a href="#Other"><span class="pill-tag pill_color_other">Undirected</span></a>graphs, <a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a>graphs and more. Graphs are presented with name and descriptions and a few statistics such as a general description, numbers of nodes and edges, source and reference information.</p>
<div>
<b>Originally found at: </b><a href="http://snap.stanford.edu/data/index.html">http://snap.stanford.edu/data/index.html</a>
</div>
<br> <!-- <div><b>Note: </b>Well-known, widely used graph repository.
Note that BioSNAP, which consists of similar datasets for medical applications, exists under the same umbrella project. While BioSNAP was not as relevant in the literature, it may still prove useful for future researchers.</div><br> -->
<div>
<b>Size: </b>1008 unique graphs (excluding graph classification tasks consisting of 352,536 graphs)
</div>
<br> <!-- <div><b>Origin Notes: </b>The SNAP library is the collection of datasets assembled as part of the Stanford Network Analysis Platform, which started in 2004 and launched their website in 2009. The group attributes the dataset preparation to Ana Pavlisic. </div><br> -->
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="snapnets">[@snapnets]</span>
</div>
<div class="col-8">
SNAP Datasets: Stanford Large Network Dataset Collection
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Arleo2019">[@Arleo2019]</span>
</div>
<div class="col-4">
A Distributed Multilevel Force-Directed Algorithm
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Meidiana2019">[@Meidiana2019]</span>
</div>
<div class="col-4">
A Quality Metric for Visualization of Clusters in Graphs
</div>
<div class="col-4">
<a href="#Clusters_generated"><span class="pill-tag pill_color_Clusters_generated">Clusters (generated)</span></a>
</div>
</div>
<div class="row">
<div class="col-4">

</div>
<div class="col-4">
ForceAtlas2
</div>
<div class="col-4">

</div>
</div>
<div class="row">
<div class="col-4">

</div>
<div class="col-4">
a Continuous Graph Layout Algorithm for Handy Network Visualization Designed for the Gephi Software
</div>
<div class="col-4">

</div>
</div>
<div class="row">
<div class="col-4">

</div>
<div class="col-4">
A Random Sampling O
</div>
<div class="col-4">

</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="zhong2023">[@zhong2023]</span>
</div>
<div class="col-4">
Force-Directed Graph Layouts Revisited: A New Force Based on the T-Distribution
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="meidiana2023">[@meidiana2023]</span>
</div>
<div class="col-4">
Shape-Faithful Graph Drawings
</div>
<div class="col-4">
<a href="#Mesh"><span class="pill-tag pill_color_Mesh">Mesh</span></a>
</div>
</div>
<p><br></p>
<section id="descriptions-from-literature-14" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-14">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.1371/journal.pone.0098679">ForceAtlas2, a Continuous Graph Layout Algorithm for Handy Network Visualization Designed for the Gephi Software</a>”:</p>
<blockquote class="blockquote">
<p>We benchmarked our algorithm with a dataset of 68 networks from 5 to 23,133 nodes. We tried to gather varied networks corresponding to the actual use of Gephi (a lot of social networks, and scale-free networks in general). Most of these networks are from the Stanford Large Network Dataset Collection (<a href="http://snap/">http://snap</a>. <a href="http://stanford.edu/data/">stanford.edu/data/</a>) and include social networks (Facebook and Twitter ego-networks), collaboration networks (from Arxiv) and autonomous systems (peering information inferred from Oregon route-views).</p>
</blockquote>
</section>
<section id="example-figures-13" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-13">Example Figures</h4>
<p>From “<a href="https://doi.org/10.1371/journal.pone.0098679">ForceAtlas2, a Continuous Graph Layout Algorithm for Handy Network Visualization Designed for the Gephi Software</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/SNAP (Stanford Network Analysis Platform) 1cd77eaee57147ce8263b2f9eaa2589c/Screen_Shot_2023-08-02_at_11.20.12_AM.png" width="80%">
</div>
</section>
</div>
</div>
</div>
<p><a name="Stanford_GraphBase"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Stanford GraphBase">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-21-contents" aria-controls="callout-21" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Stanford GraphBase
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-21" class="callout-21-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Compiled by Donald Knuth, the <strong>Stanford Graphbase</strong> consists of programs and datasets for network analysis. The datasets accompany a textbook, “The Stanford GraphBase: A Platform for Combinatorial Computing”.</p>
<div>
<b>Originally found at: </b><a href="https://www-cs-faculty.stanford.edu/~knuth/sgb.html">https://www-cs-faculty.stanford.edu/~knuth/sgb.html</a>
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Knuth_2009">[@Knuth_2009]</span>
</div>
<div class="col-8">
The Stanford GraphBase: A Platform for Combinatorial Computing
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="kruiger2017">[@kruiger2017]</span>
</div>
<div class="col-4">
Graph Layouts by t-SNE
</div>
<div class="col-4">
<a href="#Clusters_generated"><span class="pill-tag pill_color_Clusters_generated">Clusters (generated)</span></a><a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Planar"><span class="pill-tag pill_color_Planar">Planar</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Raj2018">[@Raj2018]</span>
</div>
<div class="col-4">
Anisotropic Radial Layout for Visualizing Centrality and Structure in Graphs
</div>
<div class="col-4">
<a href="#Clusters_generated"><span class="pill-tag pill_color_Clusters_generated">Clusters (generated)</span></a><a href="#Weighted_edges"><span class="pill-tag pill_color_Weighted_edges">Weighted edges</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Gronemann2016b">[@Gronemann2016b]</span>
</div>
<div class="col-4">
Crossing Minimization in Storyline Visualization
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="junger1997">[@junger1997]</span>
</div>
<div class="col-4">
2-Layer Straightline Crossing Minimization
</div>
<div class="col-4">
<a href="#Bipartite"><span class="pill-tag pill_color_Bipartite">Bipartite</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="barth2002">[@barth2002]</span>
</div>
<div class="col-4">
Simple and Efficient Bilayer Cross Counting
</div>
<div class="col-4">
<a href="#Bipartite"><span class="pill-tag pill_color_Bipartite">Bipartite</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a>
</div>
</div>
<div class="row">
<div class="col-4">

</div>
<div class="col-4">
A Random Sampling O
</div>
<div class="col-4">

</div>
</div>
<p><br></p>
<section id="descriptions-from-literature-15" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-15">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.1007/978-3-319-50106-2_29">Crossing Minimization in Storyline Visualizations</a>”:</p>
<blockquote class="blockquote">
<p>Since the instances “Anna Karenina” and “Les Miserables” are very big, we have split them into chapters and sequences of chapters. The resulting test-bed is made of eight chapters, seven pairs of chapters, six triples of chapters and five quadruples of chapters from “Anna Karenina”, and five chapters, four pairs of chapters and three triples of chapters from “Les Mis ́erables”, plus the entire “Adventures of Huckleberry Finn”, “Inception-sf”, “Inception”, “Star Wars”, “The Matrix-sf”, and “The Matrix”.</p>
</blockquote>
<p>From “<a href="https://doi.org/10.1007/978-3-319-73915-1_28">Anisotropic Radial Layout for Visualizing Centrality and Structure in Graphs</a>”:</p>
<blockquote class="blockquote">
<p>The third dataset is a graph of character associations in the famous French novel Les Miserables (Fig. 5) [18]. This graph consists of 77 nodes, each representing a character in the novel, and 254 weighted edges where the weights represent the number of chapters that feature both characters associated with an edge. We see the that the main protagonist Valjean (marked in red) is placed prominently in all three visualizations (Fig. 5). However, other key characters in the plot such as Inspector Javert (blue) and Cosett (orange), who do not appear often with characters other than the protagonist (and thus have low betweenness centrality) are treated differently. While the radial layout relegates them to the periphery (far from Valjean) (Fig. 5b), MDS (Fig. 5a) paints a conflicting picture with regard to their centrality, e.g., Cosett’s node almost overlaps with Valjean despite its low centrality. In contrast, the proposed ARL (Fig.5c) is able to coherently convey the low centrality of the Inspector Javert and Cosett, as well as, their closeness to Valjean. The above issue of distance distortion appears to be a frequent occurrence in the radial layout due to many characters who have a low centrality value causing them to end up being packed in the outer periphery. A case of contrast is that of the character Bishop Myriel (green) who despite being associated with several characters, is only seen with Valjean once.</p>
</blockquote>
</section>
<section id="example-figures-14" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-14">Example Figures</h4>
<p>From “<a href="https://doi.org/10.1007/978-3-319-50106-2_29">Crossing Minimization in Storyline Visualizations</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Stanford GraphBase effff4b40e9d4a79b6f33825ccca7401/Screen_Shot_2023-08-17_at_4.29.43_PM.png" width="80%">
</div>
<p>From “<a href="https://doi.org/10.1111/cgf.13187">Graph Layouts by t-SNE</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Stanford GraphBase effff4b40e9d4a79b6f33825ccca7401/Screen_Shot_2023-08-17_at_4.21.24_PM.png" width="80%">
</div>
<p>lesmis is the GraphBase Les Miserables graph</p>
<p>From “<a href="https://doi.org/10.1007/978-3-319-73915-1_28">Anisotropic Radial Layout for Visualizing Centrality and Structure in Graphs</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Stanford GraphBase effff4b40e9d4a79b6f33825ccca7401/Screen_Shot_2023-08-17_at_4.23.56_PM.png" width="80%">
</div>
</section>
</div>
</div>
</div>
<p><a name="SuiteSparse_Matrix_Collection"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="SuiteSparse Matrix Collection">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-22-contents" aria-controls="callout-22" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
SuiteSparse Matrix Collection
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-22" class="callout-22-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<!-- 
<div id="named-list-SuiteSparse_MatrixCollection" data-bs-spy="scroll"  data-bs-target="db-nav-list" data-bs-offset="20" tabindex="0"></div> -->
<p>The <strong>SuiteSparse Matrix Collection</strong>, formerly known as the <strong>University of Florida Sparse Matrix Collection</strong>, is a comprehensive repository of 2893 sparse matrices. All graphs in SuiteSparse belong to groups which will have more information about the graphs and the sub-collections they belong to. In our Descriptions from the Literature section we also highlight a few tables with the specific graphs used in a couple of papers. From “The university of Florida Sparse Matrix Collection”, Davis and Hu describe the origin of this network repository. Namely they cite the Harwell-Boeing collection as the starting point for SuiteSparse, then called the University of Florida (UF) Sparse matrix collection, back in 1991. Other groups, or collections, have then been added to SuitseSparse through the years, mainly focusing on real-world matrices and other relevant problems related to them.</p>
<div>
<b>Originally found at: </b><a href="https://sparse.tamu.edu/">https://sparse.tamu.edu/</a>
</div>
<br>
<div>
<b>Size: </b>2893
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="davis2011">[@davis2011]</span>
</div>
<div class="col-8">
The University of Florida Sparse Matrix Collection
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Gansner2013">[@Gansner2013]</span>
</div>
<div class="col-4">
A Maxent-Stress Model for Graph Layout
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Ortmann2017">[@Ortmann2017]</span>
</div>
<div class="col-4">
A Sparse Stress Model
</div>
<div class="col-4">
<a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a><a href="#Weighted_edges"><span class="pill-tag pill_color_Weighted_edges">Weighted edges</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Eades2018">[@Eades2018]</span>
</div>
<div class="col-4">
Drawing Big Graphs Using Spectral Sparsification
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Hong2019">[@Hong2019]</span>
</div>
<div class="col-4">
Multi-level Graph Drawing Using Infomap Clustering
</div>
<div class="col-4">
<a href="#Other"><span class="pill-tag pill_color_Other">Multilevel</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Brsig2020">[@Brsig2020]</span>
</div>
<div class="col-4">
Stochastic Gradient Descent Works Really Well for Stress Minimization
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="kruiger2017">[@kruiger2017]</span>
</div>
<div class="col-4">
Graph Layouts by t-SNE
</div>
<div class="col-4">
<a href="#Clusters_generated"><span class="pill-tag pill_color_Clusters_generated">Clusters (generated)</span></a><a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Planar"><span class="pill-tag pill_color_Planar">Planar</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="zheng2019">[@zheng2019]</span>
</div>
<div class="col-4">
Graph Drawing by Stochastic Gradient Descent
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="DRGraph21">[@DRGraph21]</span>
</div>
<div class="col-4">
DRGraph: An Efficient Graph Layout Algorithm for Large-scale Graphs by Dimensionality Reduction
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="zhong2023">[@zhong2023]</span>
</div>
<div class="col-4">
Force-Directed Graph Layouts Revisited: A New Force Based on the T-Distribution
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="miller2023">[@miller2023]</span>
</div>
<div class="col-4">
Spherical Graph Drawing by&nbsp;Multi-dimensional Scaling
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="meidiana2023">[@meidiana2023]</span>
</div>
<div class="col-4">
Shape-Faithful Graph Drawings
</div>
<div class="col-4">
<a href="#Mesh"><span class="pill-tag pill_color_Mesh">Mesh</span></a>
</div>
</div>
<p><br></p>
<section id="descriptions-from-literature-16" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-16">Descriptions From Literature</h4>
<p>From “<a href="https://dx.doi.org/10.7155/jgaa.00440">A Sparse Stress Model</a>”:</p>
<blockquote class="blockquote">
<p>We conducted our experiments on a series of different graphs, see Tab. 1, most of them taken from the sparse matrix collection [9]. We selected these graphs as they differ in their structure and size, and are large enough to compare the results of different techniques. Two of the graphs, LeHavre and commanche, have predefined edge lengths that were derived from the node coordinates. We did not modify the graphs in any way, except for those that were disconnected, in which case we only kept the largest component.</p>
</blockquote>
<div style="text-align: center">
<img src="data/Benchmark datasets/SuiteSparse Matrix Collection b8772d6a2cbb456894b4673e32c6f956/Untitled.png" width="80%">
</div>
<p>From “<a href="https://doi.org/10.1109/TVCG.2012.299">A Maxent-Stress Model for Graph Layout</a>”:</p>
<blockquote class="blockquote">
<p>With the exception of graph gd, which is an author collaboration graph of the International Symposium on Graph Drawing between 1994 and 2007, the graphs used are from the University of Florida Sparse Matrix Collection [9]. Our selection covers a range of graph sizes, and includes mesh-like and other nonmesh graphs, and graphs from Brandes and Pich’s experimental study of distance scaling [5].</p>
</blockquote>
<div style="text-align: center">
<img src="data/Benchmark datasets/SuiteSparse Matrix Collection b8772d6a2cbb456894b4673e32c6f956/Untitled 1.png" width="80%">
</div>
<p><strong>Table 2.</strong> Test Graphs</p>
<p>From “<a href="https://doi.org/10.1109/TVCG.2020.3030447">DRGraph: An Efficient Graph Layout Algorithm for Large-scale Graphs by Dimensionality Reduction</a>”:</p>
<blockquote class="blockquote">
<p>We perform experiments on a broad range of datasets selected from the University of Florida Sparse Matrix Collection [10] and tsNET [34] (Table 1).</p>
</blockquote>
<div style="text-align: center">
<img src="data/Benchmark datasets/SuiteSparse Matrix Collection b8772d6a2cbb456894b4673e32c6f956/Untitled 2.png" width="80%">
</div>
<p><strong>Table 1.</strong> Test Datasets</p>
</section>
<section id="example-figures-15" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-15">Example Figures</h4>
<p>From “<a href="https://doi.org/10.1145/2049662.2049663">The university of Florida sparse matrix collection</a><strong>”:</strong></p>
<div style="text-align: center">
<img src="data/Benchmark datasets/SuiteSparse Matrix Collection b8772d6a2cbb456894b4673e32c6f956/Untitled 3.png" width="80%">
</div>
<p><strong>Fig. 13.</strong> A sample of matrices from the Collection, for the purpose of illustrating the complexity and diversity of matrices arising in real applications</p>
<p>From “<a href="https://doi.org/10.1109/TVCG.2020.3030447">DRGraph: An Efficient Graph Layout Algorithm for Large-scale Graphs by Dimensionality Reduction</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/SuiteSparse Matrix Collection b8772d6a2cbb456894b4673e32c6f956/Untitled 4.png" width="80%">
</div>
<p><strong>Fig. 8.</strong> Visualizations of selected graph datasets using FR, KK, S.M., FM3, SFDP, PMDS, tsNET and DRGraph.</p>
</section>
</div>
</div>
</div>
<p>One more of such collections is <a href="http://konect.cc/">Konect</a>. At the time of writing, though, the website for Konect has been down for a while. Both the data and the website are still accessible through <a href="https://web.archive.org/web/20231201123711/http://www.konect.cc/">web archive</a>—thus we do not consider this a lost collection.</p>
<section id="sec-subsets" class="level4">
<h4 class="anchored" data-anchor-id="sec-subsets">Subsets of other collections</h4>
<p>Some of the datasets mentioned in the papers have, after a certain time, been integrated into other, larger collections. This is a phenomenon that can happen through the years, through the redistribution and through the merging of different sources: the <a href="#Walshaw" class="dataset-link">Walshaw</a> dataset, for instance, was and still is distributed and cited as its own standalone dataset, but its graphs can be now found as part of many other larger collections. In order to keep precise records about the data, we kept these instances separated, in order to link papers to the individual dataset they used if they explicitly mention using one of these smaller collections. We classified these datasets as <strong>Subsets</strong>.</p>
<p><a name="Walshaw"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Walshaw" data-collapse-id="Walshaw">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-23-contents" aria-controls="callout-23" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Walshaw
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-23" class="callout-23-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>C. Walshaw, collected for an archive of graph partitions which has been maintained since 2000. These are multilevel mesh partitions of heterogeneous networks from a variety of domains.</p>
<p>We do not provide the data ourselves. Walshaw requests to be emailed for the distribution of the data, and so we link to their site and to the graphs found in the <a href="#SuiteSparse_Matrix_Collection">SuiteSparse Matrix Collection</a>.</p>
<p>It has <a href="#Weighted_nodes"><span class="pill-tag pill_color_Weighted_nodes">Weighted nodes</span></a>and <a href="#Weighted_edges"><span class="pill-tag pill_color_Weighted_edges">Weighted edges</span></a>graphs, and the partitions of the 34 graphs are given with various partition sizes and imbalance percentages.</p>
<div>
<p><b>Originally found at: </b><a href="https://chriswalshaw.co.uk/partition/ ">https://chriswalshaw.co.uk/partition/ </a>where C. Walshaw requests to be emailed for the dataset at mailto:c.walshaw@gre.ac.uk. the individual graphs can also be found on the SuiteSparse Matrix Collection under different groups.</p>
</div>
<br>
<div>
<b>Size: </b>34 graphs, 2395-448695 nodes, 6837-3314611 edges. Node-weighted, edge-weighted. Partitions of the 34 graphs are given with various partition sizes and imbalance percentages.
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="soper_combined_2004">[@soper_combined_2004]</span>
</div>
<div class="col-8">
A Combined Evolutionary Search and Multilevel Optimisation Approach to Graph Partitioning
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="walshaw2000">[@walshaw2000]</span>
</div>
<div class="col-8">
Mesh Partitioning: a Multilevel Balancing and Refinement Algorithm
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="walshaw2001">[@walshaw2001]</span>
</div>
<div class="col-4">
A Multilevel Algorithm for Force-Directed Graph Drawing
</div>
<div class="col-4">
<a href="#Clusters_generated"><span class="pill-tag pill_color_Clusters_generated">Clusters (generated)</span></a><a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="frishman2007">[@frishman2007]</span>
</div>
<div class="col-4">
Multi-Level Graph Layout on the GPU
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="koren2002">[@koren2002]</span>
</div>
<div class="col-4">
ACE: A Fast Multiscale Eigenvectors Computation for Drawing Huge Graphs
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="KOREN20051867">[@KOREN20051867]</span>
</div>
<div class="col-4">
Drawing graphs by eigenvectors: theory and practice
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="civril2006">[@civril2006]</span>
</div>
<div class="col-4">
SDE: Graph Drawing Using Spectral Distance Embedding
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">

</div>
<div class="col-4">
Efficient and High Quality Force-Directed Graph Drawing
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">

</div>
<div class="col-4">
Large-Graph Layout with the Fast Multipole Multilevel Method
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Hachul2005">[@Hachul2005]</span>
</div>
<div class="col-4">
Drawing Large Graphs with a Potential-Field-Based Multilevel Algorithm
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Eades2018">[@Eades2018]</span>
</div>
<div class="col-4">
Drawing Big Graphs Using Spectral Sparsification
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Hachul2006">[@Hachul2006]</span>
</div>
<div class="col-4">
An Experimental Comparison of Fast Algorithms for Drawing General Large Graphs
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="brandes2006">[@brandes2006]</span>
</div>
<div class="col-4">
Eigensolver methods for progressive multidimensional scaling of large data
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Gansner2005b">[@Gansner2005b]</span>
</div>
<div class="col-4">
Topological fisheye views for visualizing large graphs
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Other"><span class="pill-tag pill_color_Other">High degree</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Frishman2008">[@Frishman2008]</span>
</div>
<div class="col-4">
Online Dynamic Graph Drawing
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Hong2019">[@Hong2019]</span>
</div>
<div class="col-4">
Multi-level Graph Drawing Using Infomap Clustering
</div>
<div class="col-4">
<a href="#Other"><span class="pill-tag pill_color_Other">Multilevel</span></a>
</div>
</div>
<div class="row">
<div class="col-4">

</div>
<div class="col-4">
A Random Sampling O
</div>
<div class="col-4">

</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="meidiana2020">[@meidiana2020]</span>
</div>
<div class="col-4">
Sublinear Time Force Computation for Big Complex Network Visualization
</div>
<div class="col-4">
<a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<p><br></p>
<section id="descriptions-from-literature-17" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-17">Descriptions from Literature</h4>
<p>From “<a href="https://link.springer.com/chapter/10.1007/11618058_22">An Experimental Comparison of Fast Algorithms for Drawing General Large Graphs</a>”:</p>
<blockquote class="blockquote">
<p>The rest of the test graphs are taken from real-world applications. In particular, we selected graphs from the AT&amp;T graph library [1], from C. Walshaw’s graph collection [20], and a graph that describes a social network of 2113 people that we obtained from C. Lipp.</p>
</blockquote>
<p>From “<a href="https://doi.org/10.1023/B:JOGO.0000042115.44455.f3">A Combined Evolutionary Search and Multilevel Optimisation Approach to Graph-Partitioning</a>”:</p>
<blockquote class="blockquote">
<p>The test graphs have been chosen to be a representative sample of small to medium scale real-life problems (mostly mesh-based) and include both 2D and 3D examples of nodal graphs (where the mesh nodes are partitioned) and dual graphs (where the mesh elements are partitioned). In addition there is a 3D semi-structured graph, cti, which is unstructured in the x−y plane but extended regularly along the z-axis. Finally the test suite includes three non mesh-based graphs (add32, vibrobox, bcsstk32) which arise from various scientific computing applications2. None of the graphs have either vertex or edge weights; such graphs are not widely available since most applications do not accurately instrument costs and it is difficult to draw meaningful conclusions from the few examples that we have access to.</p>
</blockquote>
<div style="text-align: center">
<img src="data/Benchmark datasets/Walshaw e40b37a1147942d89ff1d8dfad285256/Untitled.png" width="80%">
</div>
</section>
<section id="example-figures-16" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-16">Example Figures</h4>
<p>From “<a href="https://link.springer.com/chapter/10.1007/11618058_22">An Experimental Comparison of Fast Algorithms for Drawing General Large Graphs</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Walshaw e40b37a1147942d89ff1d8dfad285256/Untitled 1.png" width="80%">
</div>
<p>From “<a href="https://doi.org/10.1109/TVCG.2007.70580">Multi-Level Graph Layout on the GPU</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Walshaw e40b37a1147942d89ff1d8dfad285256/Untitled 2.png" width="80%">
</div>
<p><strong>Fig. 11.</strong> 4elt. Red: our layout, black: Kamada-Kawai layout</p>
</section>
<section id="other" class="level4">
<h4 class="anchored" data-anchor-id="other">Other</h4>
<p>From <em><a href="https://sparse.tamu.edu/DIMACS10">Group DIMACS10 in the SuiteSparse Matrix Collection</a></em>:</p>
<pre><code>Chris Walshaw's graph partitioning archive contains 34 graphs that
    have been very popular as benchmarks for graph partitioning algorithms
    ( http://staffweb.cms.gre.ac.uk/~wc06/partition/ ).

    17 of them are already in the UF Collection.  Only the 17 new graphs
    not yet in the collection are added here in the DIMACS10 set.

DIMACS10 graph:                 new?   UF matrix:
    ---------------                 ----   -------------
    walshaw/144                      *     DIMACS10/144
    walshaw/3elt                           AG-Monien/3elt
    walshaw/4elt                           Pothen/barth5
    walshaw/598a                     *     DIMACS10/598a
    walshaw/add20                          Hamm/add20
    walshaw/add32                          Hamm/add32
    walshaw/auto                     *     DIMACS10/auto
    walshaw/bcsstk29                       HB/bcsstk29
    walshaw/bcsstk30                       HB/bcsstk30
    walshaw/bcsstk31                       HB/bcsstk31
    walshaw/bcsstk32                       HB/bcsstk32
    walshaw/bcsstk33                       HB/bcsstk33
    walshaw/brack2                         AG-Monien/brack2
    walshaw/crack                          AG-Monient/crack
    walshaw/cs4                      *     DIMACS10/cs4
    walshaw/cti                      *     DIMACS10/cti
    walshaw/data                     *     DIMACS10/data
    walshaw/fe_4elt2                 *     DIMACS10/fe_4elt2
    walshaw/fe_body                  *     DIMACS10/fe_body
    walshaw/fe_ocean                 *     DIMACS10/fe_ocean
    walshaw/fe_pwt                         Pothen/pwt
    walshaw/fe_rotor                 *     DIMACS10/fe_rotor
    walshaw/fe_sphere                *     DIMACS10/fe_sphere
    walshaw/fe_tooth                 *     DIMACS10/fe_tooth
    walshaw/finan512                       Mulvey/finan512
    walshaw/m14b                     *     DIMACS10/m14b
    walshaw/memplus                        Hamm/memplus
    walshaw/t60k                     *     DIMACS10/t60k
    walshaw/uk                       *     DIMACS10/uk
    walshaw/vibrobox                       Cote/vibrobox
    walshaw/wave                           AG-Monien/wave
    walshaw/whitaker3                      AG-Monien/whitaker3
    walshaw/wing                     *     DIMACS10/wing
    walshaw/wing_nodal               *     DIMACS10/wing_nodal</code></pre>
<!-- === STOP RENDERING ===

Made explicit that they want people to reach out to them for the dataset so we do not collect it. Also in Suite Sparse, and many of the graphs are included in the Scotch graph collection.

From: [An Experimental Comparison of Fast Algorithms for Drawing General Large Graphs](https://link.springer.com/chapter/10.1007/11618058_22)

<div style='text-align: center'><img src='data/Benchmark datasets/Walshaw%20e40b37a1147942d89ff1d8dfad285256/Untitled%203.png' width='80%'></div>

Graphs found on the Sparse Matrix Collection -->
</section>
</div>
</div>
</div>
<!-- ::: {.callout-note title="Pert DAG" collapse=true appearance="minimal"}




:::{.cell}

```{.js .cell-code .hidden startFrom="3025" source-offset="0"}
make_sparkline("Pert_DAG")
```

:::{.cell-output .cell-output-display}

:::{#ojs-cell-5 nodetype="expression"}
:::
:::
:::



<div id="named-list-Pert_DAG" data-bs-spy="scroll"  data-bs-target="db-nav-list" data-bs-offset="20" tabindex="0"></div>

:::  -->
<p><a name="amazon"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Amazon">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-24-contents" aria-controls="callout-24" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Amazon
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-24" class="callout-24-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Wallinger et al.&nbsp;randomly filter data from the original dataset collected by Leskovec et al.&nbsp;in The dynamics of viral marketing. Leskovec et al.&nbsp;collected 15 million recommendations made from June 2001, to May 2003 by looking at the “Customers who Bought This Item Also Bought” feature in the Amazon online shopping platform. This dataset is a subset of SNAP and can be found there. It has <a href="#directed_edges"><span class="pill-tag pill_color_directed_edges">directed</span></a>edges and <a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>graphs.</p>
<div>
<b>Originally found at: </b> <a href="http://snap.stanford.edu/data/#amazon">http://snap.stanford.edu/data/#amazon</a>
</div>
<br>
<div>
<b>Size: </b>1 graph, 192k vertices and 269k edges
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="snapnets">[@snapnets]</span>
</div>
<div class="col-8">
SNAP Datasets: Stanford Large Network Dataset Collection
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="leskovec2007">[@leskovec2007]</span>
</div>
<div class="col-8">
The dynamics of viral marketing
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Wallinger2022">[@Wallinger2022]</span>
</div>
<div class="col-4">
Edge-Path Bundling: A Less Ambiguous Edge Bundling Approach
</div>
<div class="col-4">
<a href="#bundled_edges_generated"><span class="pill-tag pill_color_bundled_edges_generated">bundled edges (generated)</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<p><br></p>
<section id="description-from-literature-2" class="level4">
<h4 class="anchored" data-anchor-id="description-from-literature-2">Description from Literature</h4>
<p>From <a href="https://ieeexplore.ieee.org/document/9552919">Edge-Path Bundling: A Less Ambiguous Edge Bundling Approach:</a></p>
<blockquote class="blockquote">
<p><strong>Amazon Subset</strong> This dataset consists of products with edges indicating that they are commonly co-purchased. We randomly filtered edges from the original graph [33], [34]. It has |V|=192k vertices and |E|=269k edges.</p>
</blockquote>
</section>
<section id="example-figures-17" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-17">Example Figures</h4>
<p>From <a href="https://ieeexplore.ieee.org/document/9552919">Edge-Path Bundling: A Less Ambiguous Edge Bundling Approach:</a></p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Amazon 1b39a766deb747b8a3b51fc41cfc2e20/Untitled.png" width="80%">
</div>
<!-- == STOP RENDERING ==

look here: [https://github.com/mwallinger-tu/edge-path-bundling/tree/master/data](https://github.com/mwallinger-tu/edge-path-bundling/tree/master/data) from [**Edge-Path Bundling: A Less Ambiguous Edge Bundling Approach**](../Benchmark%20sets%200cc6b5e454304aec98f3b59b1a720476/Literature%20ad87f14e7097454fb2f784e2c8a2797a/Literature%20-%20Algorithm%2012e01bfc60a84007aa7d2d34293e123d/Edge-Path%20Bundling%20A%20Less%20Ambiguous%20Edge%20Bundling%20%20fcfa5882e692481298cbee906e005c30.md) 

Note: Amazon is the only dataset not included in their GitHub. They randomly sampled a graph with 192k vertices and 269k edges from the data collected by Leskovec et al in the SNAP dataset

Original dataset: [http://snap.stanford.edu/data/#amazon](http://snap.stanford.edu/data/#amazon)

 doi: 10.1145/1134707.1134732 (subgraph of a SNAP collection graph)
_site:150:21 -->
</section>
</div>
</div>
</div>
<p><a name="Autonomous_System_Network"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Autonomous System Network">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-25-contents" aria-controls="callout-25" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Autonomous System Network
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-25" class="callout-25-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>The 733-AS network is a subset of SNAP. We provide the link containing the original project, https://www.routeviews.org/routeviews/, and redirect to SNAP, <a href="https://snap.stanford.edu/data/as-733.html ">https://snap.stanford.edu/data/as-733.html </a>for the network data. In the dataset description they highlight how the graphs exhibit node and edge addition and deletion over time.</p>
<p>The data was originally part of the University of Oregon’s RouteViews project. It was then used by Jure Leskovec et al.&nbsp;and eventually stored in SNAP. It contains <a href="#large"><span class="pill-tag pill_color_large">dynamic</span></a>graphs.</p>
<div>
<b>Originally found at: </b><a href="https://www.routeviews.org/routeviews/
">https://www.routeviews.org/routeviews/ </a><a href="https://snap.stanford.edu/data/as-733.html
">https://snap.stanford.edu/data/as-733.html </a><a href="https://github.com/csuvis/MCGS/blob/master/dataset/AS-733_edge.csv">https://github.com/csuvis/MCGS/blob/master/dataset/AS-733_edge.csv</a>
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Zhao2021">[@Zhao2021]</span>
</div>
<div class="col-8">
Preserving Minority Structures in Graph Sampling
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="leskovec2005">[@leskovec2005]</span>
</div>
<div class="col-8">
Graphs over Time: Densification Laws, Shrinking Diameters and Possible Explanations
</div>
</div>
<br>
<div>
<b>Number of Graphs: </b>733
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Zhao2021">[@Zhao2021]</span>
</div>
<div class="col-4">
Preserving Minority Structures in Graph Sampling
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<p><br></p>
<section id="descriptions-from-literature-18" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-18">Descriptions from Literature</h4>
<p>From “<a href="https://ieeexplore.ieee.org/document/9222065">Preserving Minority Structures in Graph Sampling</a>”</p>
<blockquote class="blockquote">
<p>The AS-733 graph data set [48] is an autonomous systems network on the Internet with 6,474 nodes and 13,895 edges. The original graph and samples obtained by RDN, SST, MCGS are shown in Figure 5.</p>
</blockquote>
<p>From “<a href="https://dl.acm.org/doi/10.1145/1081870.1081893">Graphs over time: densification laws, shrinking diameters and possible explanations</a>”</p>
<blockquote class="blockquote">
<p>The graph of routers comprising the Internet can be organized into sub-graphs called Autonomous Systems (AS). Each AS exchanges traffic flows with some neighbors (peers). We can construct a communication network of who-talks-towhom from the BGP (Border Gateway Protocol) logs. We use the the Autonomous Systems (AS) dataset from [26]. The dataset contains 735 daily instances which span an interval of 785 days from November 8 1997 to January 2 2000.In contrast to citation networks, where nodes and edges only get added (not deleted) over time, the AS dataset also exhibits both the addition and deletion of the nodes and edges over time.</p>
</blockquote>
</section>
<section id="example-figures-18" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-18">Example Figures</h4>
<p>From “<a href="https://ieeexplore.ieee.org/document/9222065">Preserving Minority Structures in Graph Sampling</a>”</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Autonomous System Network 96d24eb9c7f94ab6a6b38e6a3af816ae/Untitled.png" width="80%">
</div>
<p><strong>Fig. 5.</strong> Case analysis of the AS-733 graph data set (a) and three samples generated by RDN (b), SST(c), and MCGS (d) with a sampling rate of 30%.</p>
<!-- == STOP RENDERING ==

Possibly related to something like this:

[https://www.caida.org/catalog/datasets/ipv4_routed_24_topology_dataset/](https://www.caida.org/catalog/datasets/ipv4_routed_24_topology_dataset/) -->
</section>
</div>
</div>
</div>
<p><a name="Control_Flow_Graphs"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Control Flow Graphs">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-26-contents" aria-controls="callout-26" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Control Flow Graphs
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-26" class="callout-26-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>14 graphs from the GraphViz test suite collected by Gansner and Hu to evaluate their PRISM algorithm in ”https://dx.doi.org/10.7155/jgaa.00198”. This is just a subset of <a href="#Graphviz_Examples">GraphViz</a>, so we link to that dataset instead.</p>
<div>
<b>Size: </b>14 graphs, from 36 to 1463 nodes
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Nachmanson2017">[@Nachmanson2017]</span>
</div>
<div class="col-4">
Node Overlap Removal by Growing a Tree
</div>
<div class="col-4">
<a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<p><br></p>
<section id="descriptions-from-literature-19" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-19">Descriptions from Literature</h4>
<p>From “<a href="https://dx.doi.org/10.7155/jgaa.00442">Node Overlap Removal by Growing a Tree</a>”:</p>
<blockquote class="blockquote">
<p>Our data includes the same set of graphs that was used by the authors of PRISM to compare it with other algorithms [6]. The set is available in the Graphviz open source package.</p>
</blockquote>
</section>
<section id="example-figures-19" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-19">Example Figures</h4>
<p>From “<a href="https://dx.doi.org/10.7155/jgaa.00442">Node Overlap Removal by Growing a Tree</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Control Flow Graphs af079f199a2c402d8633f6d41101934b/Untitled.png" width="80%">
</div>
</section>
</div>
</div>
</div>
</section>
</section>
<section id="sec-single-graphs" class="level3">
<h3 class="anchored" data-anchor-id="sec-single-graphs">Single Graphs</h3>
<p>A number of papers used individual, <strong>Single Graphs</strong> for their experiments instead of a collection. These graphs have become popular because of their properties as individual graphs - see, for example, the <a href="#Enron" class="dataset-link">Enron</a> dataset below, a huge individual graph that has a large variance in node degree distribution. Many of these graphs are also available in other repositories - their locations are noted wherever known.</p>
<p><a name="Enron"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Enron">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-27-contents" aria-controls="callout-27" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Enron
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-27" class="callout-27-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Data made public by the Federal Energy Regulatory Commission when investigating Enron. It has had a few changes over time documented in the link provided as its source. The data is currently hosted by William W. Cohen from CMU on a webiste, and it is also hosted on SNAP. SNAP asks to cite the paper linked in Origin paper. Very high variance in node degrees. Data can be considered threaded and directed, although <a href="#SNAP" class="dataset-link">SNAP</a> provides a version of the network that is explicitly undirected. This dataset proves useful for evaluating algorithms that work on very large graphs, due to its very large size.</p>
<div>
<b>Originally found at: </b><a href="https://www.cs.cmu.edu/~enron/">https://www.cs.cmu.edu/~enron/</a>
</div>
<br>
<div>
<b>Size: </b>The data consists of 150 Enron executives, who sent 500,000 messages between themselves.
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Klimt2004IntroducingTE">[@Klimt2004IntroducingTE]</span>
</div>
<div class="col-8">
Introducing the Enron Corpus
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Tanahashi2015">[@Tanahashi2015]</span>
</div>
<div class="col-4">
An Efficient Framework for Generating Storyline Visualizations from Streaming Data
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="vandenElzen2013">[@vandenElzen2013]</span>
</div>
<div class="col-4">
Reordering massive sequence views: Enabling temporal and structural analysis of dynamic networks
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<p><br></p>
<section id="descriptions-from-literature-20" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-20">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.1109/PacificVis.2013.6596125">Reordering massive sequence views: Enabling temporal and structural analysis of dynamic networks</a>”:</p>
<blockquote class="blockquote">
<p>We cleaned the data set by removing duplicates, spam and only to contain internal communication between Enron employees annotated with employee function leaving us with 151 nodes (employees) and 21374 edges (emails). Visualization using a node-link diagram enables the identification of stronger connections (see Figure 9(a)). However, temporal patterns and the evolution of the network cannot be analysed. From the standard MSV (Figure 9(b)) it becomes somewhat clear that transaction density increases over time and we can distinguish between different phases. We cannot, however, see features and identify communities due to visual clutter</p>
</blockquote>
</section>
<section id="example-figures-20" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-20">Example Figures</h4>
<p>From “<a href="https://doi.org/10.1109/TVCG.2015.2392771">An Efficient Framework for Generating Storyline Visualizations from Streaming Data</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Enron ed3c62b92cf14a7b86c691ac3651dab3/Untitled.png" width="80%">
</div>
<p>From “<a href="https://doi.org/10.1109/PacificVis.2013.6596125">Reordering massive sequence views: Enabling temporal and structural analysis of dynamic networks</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Enron ed3c62b92cf14a7b86c691ac3651dab3/Untitled 1.png" width="80%">
</div>
<!-- === STOP RENDERING ===

[https://www.cs.cmu.edu/~enron/](https://www.cs.cmu.edu/~enron/)

150 enron executives (users), 500k messages - many more employees

some have attachments

case nodes are users: could be a very connected dataset (perhaps too few nodes) - very high variance in node degree

case nodes are messages: could be a sparsely connected graph

there’s dates → can be layered

This dataset was collected and prepared by the

[CALO Project](http://www.ai.sri.com/project/CALO)

(A Cognitive Assistant that Learns and Organizes). It contains data from about 150 users, mostly senior management of Enron, organized into folders. The corpus contains a total of about 0.5M messages. This data was originally

[made public, and posted to the web](http://www.salon.com/news/feature/2003/10/14/enron/index_np.html)

, by the

[Federal Energy Regulatory Commission](http://www.ferc.gov/)

during its investigation.

The email dataset was later purchased by [Leslie Kaelbling](http://www.ai.mit.edu/people/lpk/lpk.html) at MIT, and turned out to have a number of integrity problems. A number of folks at SRI, notably [Melinda Gervasio](http://www.ai.sri.com/people/gervasio), worked hard to correct these problems, and it is thanks to them (not me) that the dataset is available. The dataset here does not include attachments, and some messages have been deleted "as part of a redaction effort due to requests from affected employees". Invalid email addresses were converted to something of the form user@enron.com whenever possible (i.e., recipient is specified in some parse-able format like "Doe, John" or "Mary K. Smith") and to no_address@enron.com when no recipient was specified.

Visualizations from [https://www.kaggle.com/jamestollefson/enron-network-analysis](https://www.kaggle.com/jamestollefson/enron-network-analysis):

<div style='text-align: center'><img src='data/Benchmark datasets/Enron%20ed3c62b92cf14a7b86c691ac3651dab3/Untitled%202.png' width='80%'></div>

<div style='text-align: center'><img src='data/Benchmark datasets/Enron%20ed3c62b92cf14a7b86c691ac3651dab3/Untitled%203.png' width='80%'></div>

This paper definitely uses this as a layered, categorical graph - could even be a hypergraph


→ unclear. Layered? -->
</section>
</div>
</div>
</div>
<p><a name="California"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="California">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-28-contents" aria-controls="callout-28" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
California
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-28" class="callout-28-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Data contains links related to an online search of the word “California” as nodes - the edges are links between webpages. We found this dataset on the webpage of a 2002 computer science course at Cornell, instructed by Jon Kleinberg. The original name was Pages that match the word “California”. They mention how the data was queried from a search engine, and that many of the original links are broken. It is not clear whether Kleinberg collected the data, or compiled it for the course.</p>
<div>
<b>Originally found at: </b><a href="https://www.cs.cornell.edu/courses/cs685/2002fa/data/gr0.California">https://www.cs.cornell.edu/courses/cs685/2002fa/data/gr0.California</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d9492e803e0c0c0f558b9d" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d94da70c2b4d0ea138634b" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d9700094a6be112912e9a5" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90e48803e0c0b00558baa" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>4772 nodes, 8965 edges
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Kleinberg_2002">[@Kleinberg_2002]</span>
</div>
<div class="col-8">
The Structure of Information Networks
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="muelder2008b">[@muelder2008b]</span>
</div>
<div class="col-4">
A Treemap Based Method for Rapid Layout of Large Graphs
</div>
<div class="col-4">
<a href="#Clusters_generated"><span class="pill-tag pill_color_Clusters_generated">Clusters (generated)</span></a><a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="muelder2008">[@muelder2008]</span>
</div>
<div class="col-4">
Rapid Graph Layout Using Space Filling Curves
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-13" class="level4">
<h4 class="anchored" data-anchor-id="statistics-13">Statistics</h4>
<div style="text-align: center">
<img src="data/Benchmark datasets/California e0a841bc57c045b5a8514110cfe85635/degree_distr.svg" width="80%">
</div>
</section>
<section id="descriptions-from-literature-21" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-21">Descriptions from Literature</h4>
<p>From <a href="https://ieeexplore.ieee.org/document/4475481">“A Treemap Based Method for Rapid Layout of Large Graphs</a>”:</p>
<blockquote class="blockquote">
<p>This particular graph is a non-weighted graph of links between search results for the word “California” (also in Figures 6, 8, and 9, |V|=6107,|E|=15160).</p>
</blockquote>
<p>From “<a href="https://ieeexplore.ieee.org/document/4658143">Rapid Graph Layout Using Space Filling Curves</a>”:</p>
<blockquote class="blockquote">
<p>The “california” dataset (shown in Figures 5 and 8) consists of the links between the webpages found from a search for the word ‘California’ [4].</p>
</blockquote>
</section>
<section id="example-figures-21" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-21">Example Figures</h4>
<p>From <a href="https://ieeexplore.ieee.org/document/4475481">“A Treemap Based Method for Rapid Layout of Large Graphs</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/California e0a841bc57c045b5a8514110cfe85635/Untitled.png" width="80%">
</div>
<p><strong>Fig. 1.</strong> <em>A graph laid out using our treemap based approach</em>. This graph portrays the links between websites that came from a search on the word “California” [7]. Nodes are clustered into a hierarchy, and laid out by applying a treemap to this hierarchy. Levels of the hierarchy below a threshold are clustered together into larger nodes. It can very easily be seen that there are three primary groups of websites that link to each other, and a plethora of others that are not as tightly linked.</p>
<p>From “<a href="https://ieeexplore.ieee.org/document/4658143">Rapid Graph Layout Using Space Filling Curves</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/California e0a841bc57c045b5a8514110cfe85635/Untitled 1.png" width="80%">
</div>
<p><strong>Fig. 5</strong> <em>Separating clusters.</em> By adjusting the spacing between nodes according to the clustering information, clusters can be separated.</p>
<!-- == STOP RENDERING ==

[https://www.cs.cornell.edu/courses/cs685/2002fa/data/gr0.California](https://www.cs.cornell.edu/courses/cs685/2002fa/data/gr0.California) -->
</section>
</div>
</div>
</div>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="World Greenhouse Gas Emissions">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-29-contents" aria-controls="callout-29" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
World Greenhouse Gas Emissions
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-29" class="callout-29-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Data collected by the <a href="https://www.wri.org/">WRI</a>. Linked data is from a different year than the data used by <a href="https://doi.org/10.1109/PacificVis.2018.00025">Optimal Sankey Diagrams Via Integer Programming</a>, for which data could not be found. Contains <a href="#Weighted_edges"><span class="pill-tag pill_color_Weighted_edges">Weighted edges</span></a> <a href="#Weighted_nodes"><span class="pill-tag pill_color_Weighted_nodes">Weighted nodes</span></a> <a href="#Categorical_nodes"><span class="pill-tag pill_color_Categorical_nodes">Categorical nodes</span></a> and <a href="#layered_graphs"><span class="pill-tag pill_color_layered_graphs">layers</span></a></p>
<div>
<b>Originally found at: </b><a href="https://www.wri.org/data/world-greenhouse-gas-emissions-2005
">https://www.wri.org/data/world-greenhouse-gas-emissions-2005 </a><a href="https://www.wri.org/search?keys=world%20greenhouse%20gas%20emissions&amp;f%5B0%5D=content_type%3Adata
">https://www.wri.org/search?keys=world%20greenhouse%20gas%20emissions&amp;f%5B0%5D=content_type%3Adata </a><a href="https://github.com/lhc70000/sankey/tree/master">https://github.com/lhc70000/sankey/tree/master</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d9493c803e0c0c0c558ba6" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d94d8b803e0c0c1f558cb4" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d971484cf748115a055875" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90eb2803e0c0b03558c69" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>40 nodes, 85 edges
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="navigating_numbers">[@navigating_numbers]</span>
</div>
<div class="col-8">
Navigating the Numbers: Greenhouse Gas Data and International Climate Policy
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="zarate2018">[@zarate2018]</span>
</div>
<div class="col-4">
Optimal Sankey Diagrams via Integer Programming
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-14" class="level4">
<h4 class="anchored" data-anchor-id="statistics-14">Statistics</h4>
<div style="text-align: center">
<img src="data/Benchmark datasets/World Greenhouse Gas Emissions 859aab42a55e43ac9dad1eae1a91a534/degree_distr.svg" width="80%">
</div>
</section>
<section id="descriptions-from-literature-22" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-22">Descriptions from Literature</h4>
<blockquote class="blockquote">
<p>We have tested our method on the “World Greenhouse Gas Emissions” data from the World Resources Institute [8]. After transforming the “long” edges of the graph into “short” edges and adding the corresponding dummy nodes, as described in Section 3, this example has 4 layers, 55 nodes and 100 edges. The node ordering in the layout of the Sankey diagram shown in Figure 1 has been computed using Sugiyama’s heuristic method [13].</p>
</blockquote>
</section>
<section id="example-figures-22" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-22">Example Figures</h4>
<p>From <a href="https://doi.org/10.1109/PacificVis.2018.00025">Optimal Sankey Diagrams via Integer Linear Programming</a>:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/World Greenhouse Gas Emissions 859aab42a55e43ac9dad1eae1a91a534/Screen_Shot_2023-08-09_at_11.13.23_AM.png" width="80%">
</div>
<div style="text-align: center">
<img src="data/Benchmark datasets/World Greenhouse Gas Emissions 859aab42a55e43ac9dad1eae1a91a534/Screen_Shot_2023-01-21_at_2.59.58_PM.png" width="80%">
</div>
<div style="text-align: center">
<img src="data/Benchmark datasets/World Greenhouse Gas Emissions 859aab42a55e43ac9dad1eae1a91a534/Screen_Shot_2023-01-21_at_3.00.19_PM.png" width="80%">
</div>
<p>127.0.0.1:5501:150:21</p>
</section>
</div>
</div>
</div>
<p><a name="Chess_Games"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Chess Games">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-30-contents" aria-controls="callout-30" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Chess Games
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-30" class="callout-30-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>20,058 chess openings from games in a database of played online chess games. The authors of “Sequence Braiding: Visual Overviews of Temporal Event Sequences and Attributes” collected the dataset online from the data science platform kaggle. The dataset was collected by user Mitchell J. and collects data from the chess website Lichess.org. The data used in the literature focuses on the 200 most common openings. These are all sequences of 6 moves and are topologically equivalent. It is a layered graph with <a href="#Categorical_nodes"><span class="pill-tag pill_color_Categorical_nodes">Categorical nodes</span></a>and <a href="#N-layers"><span class="pill-tag pill_color_N-layers">Layers</span></a> and can be seen as a <a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Temporal event sequence</span></a></p>
<div>
<b>Originally found at: </b><a href="https://www.kaggle.com/datasets/datasnaek/chess?resource=download
">https://www.kaggle.com/datasets/datasnaek/chess?resource=download </a><br><a href="https://github.com/VisDunneRight/sequence_braiding/blob/master/data/chess.json">https://github.com/VisDunneRight/sequence_braiding/blob/master/data/chess.json</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d948ae4cf74810580556b8" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96e1b0c2b4d0f65386312" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d9705494a6be112912e9d0" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90eed4cf7480eef055740" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>The data consists of 20058 sequences, which result in a disconnected graph of 118164 nodes, and 98106 edges.
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Jolly_2017">[@Jolly_2017]</span>
</div>
<div class="col-8">
Chess Game Dataset (Lichess)
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Bartolomeo2021">[@Bartolomeo2021]</span>
</div>
<div class="col-4">
Sequence Braiding: Visual Overviews of Temporal Event Sequences and Attributes
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<p><br></p>
<section id="descriptions-from-literature-23" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-23">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.1109/TVCG.2020.3030442">Sequence Braiding: Visual Overviews of Temporal Event Sequences and Attributes</a>”:</p>
<blockquote class="blockquote">
<p>200 Chess openings displayed with Sequence Braiding. Each line represents a sequence of moves of the white player, each group is a chess piece type. Most openings start with a pawn, and very little with the knight.</p>
</blockquote>
</section>
<section id="example-figures-23" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-23">Example Figures</h4>
<p>From “<a href="https://doi.org/10.1109/TVCG.2020.3030442">Sequence Braiding: Visual Overviews of Temporal Event Sequences and Attributes</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Chess Games 023a47d6ae914567a05132ac1dfbe972/Screen_Shot_2023-01-21_at_2.15.02_PM.png" width="80%">
</div>
<p><strong>Fig. 7.</strong> 200 Chess openings displayed with Sequence Braiding. Each line represents a sequence of moves of the white player, each group is a chess piece type. Most openings start with a pawn, and very little with the knight. After moving a pawn, it is common to move a knight or a pawn, it is a little less common to move a bishop, and only a little number of openings move the queen on the second move.</p>
</section>
</div>
</div>
</div>
<p><a name="Tree_ofLife"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Tree of Life">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-31-contents" aria-controls="callout-31" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tree of Life
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-31" class="callout-31-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Part of the <a href="http://tolweb.org/tree/to ">http://tolweb.org/tree/to </a>characterize information about biodiversity and their evolutionary genealogy. It is a tree <a href="#trees"><span class="pill-tag pill_color_trees">tree</span></a>with <a href="#Categorical_nodes"><span class="pill-tag pill_color_Categorical_nodes">Categorical nodes</span></a></p>
<div>
<b>Originally found at: </b><a href="https://www.kaggle.com/datasets/konivat/tree-of-life?resource=download">https://www.kaggle.com/datasets/konivat/tree-of-life?resource=download</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d9492294a6be103312e71d" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96dfb4cf748115a0556b0" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d971350c2b4d0f65386558" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90e404cf7480eec0555fd" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>35,960 nodes, 35,959 edges
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="gkab301">[@gkab301]</span>
</div>
<div class="col-8">
Interactive Tree Of Life (iTOL) v5: an online tool for phylogenetic tree display and annotation
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="maddison2007">[@maddison2007]</span>
</div>
<div class="col-8">
The Tree of Life Web Project
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="gray2023">[@gray2023]</span>
</div>
<div class="col-4">
Visualizing Evolving Trees
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a>
</div>
</div>
<p><br></p>
<section id="descriptions-from-literature-24" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-24">Descriptions from Literature</h4>
<p>From “<a href="https://link.springer.com/chapter/10.1007/978-3-031-22203-0_23">Visualizing Evolving Trees</a>”:</p>
<blockquote class="blockquote">
<p><strong>The Tree of Life:</strong> captures the evolutionary progression of life on Earth [<a href="https://link.springer.com/chapter/10.1007/978-3-031-22203-0_23#ref-CR37">37</a>]. The underlying data is a tree structure with a natural time component. As a new species evolves, a new node in the tree is added. The edges give the parent-child relation of the nodes, where the parent is the original species, and the child is the new species. We use a subset of this graph with 500 nodes. The maximum node degree of this tree is 5, and the radius is 24</p>
</blockquote>
</section>
<section id="example-figures-24" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-24">Example Figures</h4>
<p>From “<a href="https://link.springer.com/chapter/10.1007/978-3-031-22203-0_23">Visualizing Evolving Trees</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Tree of Life cb0493d6b6da4a73a979f06225983011/Untitled.png" width="80%">
</div>
<p><strong>Fig. 4.</strong> Layouts obtained by the seven methods for the tree of life dataset.</p>
<!-- == STOP RENDERING ==

From:

[http://tolweb.org/tree/](http://tolweb.org/tree/) -->
</section>
</div>
</div>
</div>
<p><a name="World_BankTradeData"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="World Bank Trade Data">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-32-contents" aria-controls="callout-32" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
World Bank Trade Data
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-32" class="callout-32-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Bilateral trade value in 1999 (total imports+exports), from <a href="https://wits.worldbank.org/module/ALL/sub-module/ALL/reporter/ALL/year/ALL/tradeflow/ALL/pagesize/50/page/1">wits.worldbank.org</a>. It is <a href="#dense"><span class="pill-tag pill_color_dense">dense</span></a>and has <a href="#Categorical_nodes"><span class="pill-tag pill_color_Categorical_nodes">Categorical nodes</span></a></p>
<div>
<b>Originally found at: </b><a href="https://wits.worldbank.org/module/ALL/sub-module/ALL/reporter/ALL/year/ALL/tradeflow/ALL/pagesize/50/page/1">https://wits.worldbank.org/module/ALL/sub-module/ALL/reporter/ALL/year/ALL/tradeflow/ALL/pagesize/50/page/1</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d949494cf7481075055615" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96d610c2b4d0f6438622a" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d970654cf748115a0557d4" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90e390c2b4d0d2e386256" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br> <br>
<div>
<b>Size: </b>194 nodes, 10080 edges
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="SUBRAMANIAN2007151">[@SUBRAMANIAN2007151]</span>
</div>
<div class="col-8">
The WTO promotes trade strongly but unevenly
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="noack2004">[@noack2004]</span>
</div>
<div class="col-4">
Energy Models for Graph Clustering
</div>
<div class="col-4">
<a href="#Clusters_generated"><span class="pill-tag pill_color_Clusters_generated">Clusters (generated)</span></a><a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="nocaj2015">[@nocaj2015]</span>
</div>
<div class="col-4">
Untangling the Hairballs of Multi-Centered Small-World Online Social Media Networks
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-15" class="level4">
<h4 class="anchored" data-anchor-id="statistics-15">Statistics</h4>
<div style="text-align: center">
<img src="data/Benchmark datasets/World Bank Trade Data e6e54a00bc574c639cfa165458c71232/degree_distr.svg" width="80%">
</div>
</section>
<section id="descriptions-from-literature-25" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-25">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.1007/978-3-540-24595-7_40">Energy Models for Graph Clustering</a>”:</p>
<blockquote class="blockquote">
<p>The difference between conventional energy models, node-repulsion LinLog, and edge-repulsion LinLog can be illustrated with a model of the trade between ten North American and European countries. The nodes of the graph correspond to the countries, and the edge weights specify the trade volume between each pair of countries. Because of geographical closeness and free trade agreements, countries on the same continent trade more intensively than countries on different continents. Figure 1 shows the minimum energy layouts of the trade graph for the three force and energy models. The layout of the widely used Fruchterman-Reingold model [20] does not show any clear groups at all. The layout of the node-repulsion LinLog energy model groups the countries (nodes) primarily according to their total trade volume (degree). Only the layout of the edge-repulsion LinLog energy model shows the expected grouping according to continents.</p>
</blockquote>
</section>
<section id="example-figures-25" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-25">Example Figures</h4>
<p>From “<a href="https://doi.org/10.7155/jgaa.00370">Untangling the Hairballs of Multi-Centered Small-World Online Social Media Networks</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/World Bank Trade Data e6e54a00bc574c639cfa165458c71232/Screen_Shot_2023-08-17_at_5.17.54_PM.png" width="80%">
</div>
<p>From “<a href="https://doi.org/10.1007/978-3-540-24595-7_40">Energy Models for Graph Clustering</a>”:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/World Bank Trade Data e6e54a00bc574c639cfa165458c71232/Screen_Shot_2023-08-17_at_5.13.41_PM.png" width="80%">
</div>
<!-- === STOP RENDERING ===

world bank: [www.worldbank.org](http://www.worldbank.org/) -->
</section>
</div>
</div>
</div>
<p><a name="Investment_Interdependence"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Investment Interdependence">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-33-contents" aria-controls="callout-33" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Investment Interdependence
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-33" class="callout-33-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Data collected in a survey of Ohio State University students by D. W. Malone. Students were presented with two potential obstacles to investment in Columbus, Ohio’s business district and asked, “does obstacle A aggravate or intensify obstacle B?”. Contains <a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a>and is <a href="#Hierarchical"><span class="pill-tag pill_color_Hierarchical">Hierarchical</span></a></p>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d949164cf7481076055631" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96e4d803e0c0caf558b9f" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d97042803e0c0cb3558c35" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90e6794a6be0ebd12e705" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>25 nodes, 30 edges
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="malone1975">[@malone1975]</span>
</div>
<div class="col-8">
An introduction to the application of interpretive structural modeling
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Sugiyama1981">[@Sugiyama1981]</span>
</div>
<div class="col-4">
Methods for Visual Understanding of Hierarchical System Structures
</div>
<div class="col-4">
<a href="#Hierarchical"><span class="pill-tag pill_color_Hierarchical">Hierarchical</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-16" class="level4">
<h4 class="anchored" data-anchor-id="statistics-16">Statistics</h4>
<div style="text-align: center">
<img src="data/Benchmark datasets/Investment Interdependence 141028dac1c34b6dbb69710ea5c7ee10/two_in_one.svg" width="80%">
</div>
</section>
<section id="description-from-literature-3" class="level4">
<h4 class="anchored" data-anchor-id="description-from-literature-3">Description from Literature</h4>
<p>From <a href="https://ieeexplore.ieee.org/document/4308636">Methods for Visual Understanding of Hierarchical System Structures</a>:</p>
<blockquote class="blockquote">
<p>Presented in Fig. 5 are drawings of the nine-level hierarchy which appeared in Malone [11] and represent the interdependence of obstacles to investment in the Columbus, Ohio, central business district.</p>
</blockquote>
</section>
<section id="example-figures-26" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-26">Example Figures</h4>
<p>From <a href="https://ieeexplore.ieee.org/document/4308636">Methods for Visual Understanding of Hierarchical System Structures</a>:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Investment Interdependence 141028dac1c34b6dbb69710ea5c7ee10/Untitled.png" width="80%">
</div>
<p><strong>Fig. 5. (cropped):</strong> Maps of the nine-level hierarchy [11]. (a) Initial. (b) BC + QP methods (c = 1.0). (c) BC + QP methods (cL= 0.5). (d) BC + QP methods (c = 0.05). (e) BC + PR methods.</p>
</section>
</div>
</div>
</div>
<p><a name="Protein_Interactions"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Protein Interactions">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-34-contents" aria-controls="callout-34" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Protein Interactions
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-34" class="callout-34-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Protein Interaction Publications were collected from the <a href="http://www.pathwaycommons.org/">Biological Pathway Commons Database</a>. Various types of protein interaction graphs are recorded in other network collections. We highlight two of them from the surveyed papers. First, we have the temporal Protein Interaction Publications, showing the history of how protein interactions were described in the literature. Second, the Protein Homology graph was provided by the Large Graph Layout project, their links are now broken and the dataset lost. Nonetheless, <a href="@SNAP">SNAP</a> and <a href="http://konect.cc/networks/">Konect</a> both also have various dataset of human/other species protein interactions.</p>
<div>
<b>Originally found at: </b><a href="https://github.com/CreativeCodingLab/TimeArcs/tree/master/IndexCards">https://github.com/CreativeCodingLab/TimeArcs/tree/master/IndexCards </a><a href="https://lgl.sourceforge.net/ ">https://lgl.sourceforge.net/ </a>(download link broken)
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d9487494a6be101c12e7a5" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96cd4803e0c0ca9558d2b" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d971704cf748115a055882" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d94abc0c2b4d0e8d38629c" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>2961 nodes, 5267 edges
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="ADAI2004179">[@ADAI2004179]</span>
</div>
<div class="col-8">
LGL: Creating a Map of Protein Function with an Algorithm for Visualizing Very Large Biological Networks
</div>
</div>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="muelder2008">[@muelder2008]</span>
</div>
<div class="col-4">
Rapid Graph Layout Using Space Filling Curves
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="dang2016">[@dang2016]</span>
</div>
<div class="col-4">
TimeArcs: Visualizing Fluctuations in Dynamic Networks
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<p><br></p>
<section id="protein-interaction-publications" class="level4">
<h4 class="anchored" data-anchor-id="protein-interaction-publications">Protein Interaction Publications</h4>
</section>
<section id="statistics-17" class="level4">
<h4 class="anchored" data-anchor-id="statistics-17">Statistics</h4>
<div style="text-align: center">
<img src="data/Benchmark datasets/Protein Interactions (Publications) fb5bc3bb0c5d40468da36d279a114a78/degree_distr.svg" width="80%">
</div>
</section>
<section id="description-from-the-literature" class="level4">
<h4 class="anchored" data-anchor-id="description-from-the-literature">Description from the Literature</h4>
<p>From <a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.12882">TimeArcs: Visualizing Fluctuations in Dynamic Networks</a>:</p>
<blockquote class="blockquote">
<p>The data contains the publication information (such as publication year, author, and textual evidence) of interactions between pairs of proteins, as well as their specific interaction type.</p>
</blockquote>
<blockquote class="blockquote">
<p>When there are multiple arcs connecting two proteins, it falls into one of the two circumstances. If they have the same color, these arcs indicate that there are supporting evidences in different publications which <em>confirm</em> the interaction between two elements. On the other hand, if they have the different colors, the more recent appearance provides either more detailed knowledge about the interaction or shows a <em>conflict</em> between different articles regarding the way in which these proteins interact.</p>
</blockquote>
</section>
<section id="example-figures-27" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-27">Example Figures</h4>
<p>From <a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.12882">TimeArcs: Visualizing Fluctuations in Dynamic Networks</a>:</p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Protein Interactions (Publications) fb5bc3bb0c5d40468da36d279a114a78/Untitled.png" width="80%">
</div>
<p><strong>Fig. 10</strong>: TimeArcs visualization for interactions around PCAF protein. (1), (2), and (3) in the figure are supporting evidences in literature of “PCAF binds MAML”.</p>
</section>
<section id="protein-homology-lost" class="level4">
<h4 class="anchored" data-anchor-id="protein-homology-lost">Protein Homology (Lost)</h4>
</section>
<section id="description-from-the-literature-1" class="level4">
<h4 class="anchored" data-anchor-id="description-from-the-literature-1">Description from the Literature</h4>
<p>From [Rapid Graph Layout Using Space Filling Curves:](https://ieeexplore.ieee.org/document/4658143)</p>
<blockquote class="blockquote">
<p>The “pgraph” dataset (shown in Figures 2 and 6) is a protein homology graph, which is relatively dense [7].</p>
</blockquote>
<p>Consists of 28, 854 vertices and 1,180, 816 edges, found in Table 1 of paper above.</p>
</section>
<section id="example-figures-28" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-28">Example Figures</h4>
<p>From <a href="https://ieeexplore.ieee.org/document/4658143">Rapid Graph Layout Using Space Filling Curves:</a></p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Protein Interactions (Publications) fb5bc3bb0c5d40468da36d279a114a78/Untitled 1.png" width="80%">
</div>
<p><strong>Fig. 2.</strong> <em>A protein homology graph laid out with our space filling curve based approach.</em> Color corresponds to depth in the clustering hierarchy. |<em>V</em>| = 28, 854,|<em>E</em>| = 1, 180, 816</p>
<p>From <a href="https://www.sciencedirect.com/science/article/pii/S0022283604004851">LGL: Creating a Map of Protein Function with an Algorithm for Visualizing Very Large Biological Networks</a></p>
<div style="text-align: center">
<img src="data/Benchmark datasets/Protein Interactions (Publications) fb5bc3bb0c5d40468da36d279a114a78/Untitled 2.png" width="80%">
</div>
<p><strong>Fig. 3.</strong> The complete protein homology map. A layout of the entire protein homology map; a total of 11,516 connected sets containing 111,604 proteins (vertices) with 1,912,684 edges. The largest connected set is shown more clearly in the inset and is enlarged further in <a href="https://www.sciencedirect.com/science/article/pii/S0022283604004851?via%3Dihub#FIG4">Figure 4</a>.</p>
<!-- == STOP RENDERING ==
broken link referenced in [**Rapid Graph Layout Using Space Filling Curves**](../Benchmark%20sets%200cc6b5e454304aec98f3b59b1a720476/Literature%20ad87f14e7097454fb2f784e2c8a2797a/Literature%20-%20Algorithm%2012e01bfc60a84007aa7d2d34293e123d/Rapid%20Graph%20Layout%20Using%20Space%20Filling%20Curves%2010601cd6078a4ea18b17c7d40eda0041.md) : 

Same link referenced in original paper, 

I also found [https://lgl.sourceforge.net/](https://lgl.sourceforge.net/) which contains the link to supposedly download the dataset but it is also broken

Reached out to author - Alex Adai 

[](http://www.marcottelab.org/index.php/OldBioinformaticsServerlgl/) -->
</section>
</div>
</div>
</div>
</section>
<section id="sec-aggregate-collections" class="level3">
<h3 class="anchored" data-anchor-id="sec-aggregate-collections">Aggregate Collections</h3>
<p>Many papers use graphs from specific domains that contain particular characteristics, such as geographical coordinates often found in airline data. Instead of collecting each of these individual, contextual datasets, we aggregated them into subcategories called <strong>Aggregate collections</strong>. These collections group together datasets from the same real-life context, which may have multiple sources. Real-life contexts are distinguishable by specific properties or requirements found in graphs from those sources; for instance, collaboration networks can be represented as hypergraphs, while air traffic routes have nodes (airports) with fixed spatial coordinates and numerous edges connecting the nodes, making them suitable for edge bundling algorithms. Individual information about each aggregate collection can be found in the papers that contain them.</p>
<p><a name="Assorted_Collaboration_Network"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Assorted Collaboration Network">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-35-contents" aria-controls="callout-35" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Assorted Collaboration Network
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-35" class="callout-35-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Contains IEEEVis Publication Data, DBLP collaborations, and CPAN, which contains collaboration information among developers using Perl. The VIS publication data was found online in the following website: <a href="https://sites.google.com/site/vispubdata. ">https://sites.google.com/site/vispubdata. </a>However, we also downloaded the text file fromhttps://www.cc.gatech.edu/gvu/ii/citevis/, a tool/poster associated to the publication: http://vispubdata.org/: A Metadata Collection about IEEE Visualization (VIS) Publications.</p>
<p>Assorted collaboration networks contain <a href="#Categorical_nodes"><span class="pill-tag pill_color_Categorical_nodes">Categorical nodes</span></a> <a href="#directed_edges"><span class="pill-tag pill_color_directed_edges">directed</span></a>edges, and <a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a>graphs.</p>
<div>
<b>Originally found at: </b><a href="https://www.cc.gatech.edu/gvu/ii/citevis/
">https://www.cc.gatech.edu/gvu/ii/citevis/ </a><a href="https://sites.google.com/site/vispubdata/home?authuser=0
">https://sites.google.com/site/vispubdata/home?authuser=0 </a><a href="https://dblp.uni-trier.de/
">https://dblp.uni-trier.de/ </a><a href="https://github.com/csuvis/MCGS/tree/master/dataset">https://github.com/csuvis/MCGS/tree/master/dataset</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d948ba94a6be101d12e880" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96d7d0c2b4d0f65386287" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d971834cf748115b055998" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90e6f4cf7480eef0556cd" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>3503-7102 nodes, 2112-25677 edges
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Isenberg:2017:VMC">[@Isenberg:2017:VMC]</span>
</div>
<div class="col-8">
vispubdata.org: A Metadata Collection about IEEE Visualization VIS Publications
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="heymanncpan2009">[@heymanncpan2009]</span>
</div>
<div class="col-8">
CPAN-explorer
</div>
</div>
<div class="row">
<div class="col-4">
[link]
</div>
<div class="col-8">
an interactive exploration of the Perl Ecosystem
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Zhao2021">[@Zhao2021]</span>
</div>
<div class="col-8">
Preserving Minority Structures in Graph Sampling
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Angori2019">[@Angori2019]</span>
</div>
<div class="col-4">
ChordLink: A New Hybrid Visualization Model0
</div>
<div class="col-4">
<a href="#Categorical_nodes"><span class="pill-tag pill_color_Categorical_nodes">Categorical nodes</span></a><a href="#Other"><span class="pill-tag pill_color_Other">High degree</span></a><a href="#Labeled_nodes"><span class="pill-tag pill_color_Labeled_nodes">Labeled nodes</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="minghim2006">[@minghim2006]</span>
</div>
<div class="col-4">
Content-based text mapping using multi-dimensional projections for exploration of document collections
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Ogawa2010">[@Ogawa2010]</span>
</div>
<div class="col-4">
Software evolution storylines
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Bach2014">[@Bach2014]</span>
</div>
<div class="col-4">
GraphDiaries: Animated Transitions and Temporal Navigation for Dynamic Networks
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Simonetto2018">[@Simonetto2018]</span>
</div>
<div class="col-4">
Drawing Dynamic Graphs Without Timeslices
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (continuous)</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Henry2007">[@Henry2007]</span>
</div>
<div class="col-4">
NodeTrix: a Hybrid Visualization of Social Networks
</div>
<div class="col-4">
<a href="#Clusters_generated"><span class="pill-tag pill_color_Clusters_generated">Clusters (generated)</span></a><a href="#Other"><span class="pill-tag pill_color_Other">High degree</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="DiGiacomo2020">[@DiGiacomo2020]</span>
</div>
<div class="col-4">
Storyline Visualizations with Ubiquitous Actors
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="dang2016">[@dang2016]</span>
</div>
<div class="col-4">
TimeArcs: Visualizing Fluctuations in Dynamic Networks
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Qu2022">[@Qu2022]</span>
</div>
<div class="col-4">
Automatic Polygon Layout for Primal-Dual Visualization of Hypergraphs
</div>
<div class="col-4">
<a href="#Clusters_generated"><span class="pill-tag pill_color_Clusters_generated">Clusters (generated)</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="diBartolomeo2022">[@diBartolomeo2022]</span>
</div>
<div class="col-4">
Six methods for transforming layered hypergraphs to apply layered graph layout algorithms
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#hypergraph"><span class="pill-tag pill_color_hypergraph">hypergraph</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Gansner2013">[@Gansner2013]</span>
</div>
<div class="col-4">
A Maxent-Stress Model for Graph Layout
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Zhao2021">[@Zhao2021]</span>
</div>
<div class="col-4">
Preserving Minority Structures in Graph Sampling
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Grabowicz2014">[@Grabowicz2014]</span>
</div>
<div class="col-4">
Fast filtering and animation of large dynamic networks
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (continuous)</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="mizuno2019">[@mizuno2019]</span>
</div>
<div class="col-4">
Optimizing Stepwise Animation in Dynamic Set Diagrams
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a>
</div>
</div>
<div class="row">
<div class="col-4">

</div>
<div class="col-4">
A Random Sampling O
</div>
<div class="col-4">

</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="arleo2022">[@arleo2022]</span>
</div>
<div class="col-4">
Event-based Dynamic Graph Drawing without the Agonizing Pain
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="burch2011">[@burch2011]</span>
</div>
<div class="col-4">
Parallel Edge Splatting for Scalable Dynamic Graph Visualization
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a><a href="#Weighted_edges"><span class="pill-tag pill_color_Weighted_edges">Weighted edges</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Lin2010">[@Lin2010]</span>
</div>
<div class="col-4">
ContexTour: Contextual Contour Visual Analysis on Dynamic Multi- Relational Clustering
</div>
<div class="col-4">
<a href="#Clusters_generated"><span class="pill-tag pill_color_Clusters_generated">Clusters (generated)</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (continuous)</span></a><a href="#Bipartite"><span class="pill-tag pill_color_Bipartite">Tripartite</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="saket2016">[@saket2016]</span>
</div>
<div class="col-4">
Comparing Node-Link and Node-Link-Group Visualizations From An Enjoyment Perspective
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-18" class="level4">
<h4 class="anchored" data-anchor-id="statistics-18">Statistics</h4>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Assorted Collaboration Network 6062ff126f474a50b5f3dc9b945d43da/four_in_one.svg" width="80%">
</div>
</section>
<section id="description-from-papers" class="level4">
<h4 class="anchored" data-anchor-id="description-from-papers">Description from papers</h4>
<p>From “<a href="https://link.springer.com/chapter/10.1007/978-3-319-73915-1_31">Drawing Dynamic Graphs Without Timeslices</a>”</p>
<blockquote class="blockquote">
<p><em>InfoVis Co-Authorship (Discrete):</em> a co-authorship network for papers published in the InfoVis conference from 1995 to 2015 [<a href="https://link.springer.com/chapter/10.1007/978-3-319-73915-1_31#ref-CR1">1</a>]. Authors collaborating on a paper are connected in a clique at the time of publication of the paper. Note this is not a cumulative network as authors can appear, disappear, and appear again. The data is of discrete nature with exactly 21 timeslices (one per year).</p>
</blockquote>
<p>From “<a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.14615">Event-based Dynamic Graph Drawing without the Agonizing Pain</a>”</p>
<blockquote class="blockquote">
<p>InfoVis is a co-authorship network for papers published in the InfoVis conference from 1995 to 2015 [<a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.14615#cgf14615-bib-0035">IHK*16</a>]. Authors on a paper are connected in a clique at the time of publication. This is not a cumulative network as authors can appear, disappear and appear again. The dataset has 21 timeslices (one per year).</p>
</blockquote>
</section>
<section id="example-figures-29" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-29">Example Figures</h4>
<p>From “<a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.14538">Six methods for transforming layered hypergraphs to apply layered graph layout algorithms</a>”</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Assorted Collaboration Network 6062ff126f474a50b5f3dc9b945d43da/Untitled.png" width="80%">
</div>
<p><strong>Fig. 7.</strong> Collaborations in papers published at VIS between a set of universities—in particular, these are the collaborators of Harvard University and the collaborators of the collaborators (up to 2 degrees of separation). This figure shows the result of the application of aggregate-collapse.</p>
<p>From “<a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.12882">TimeArcs: Visualizing Fluctuations in Dynamic Networks</a>”</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Assorted Collaboration Network 6062ff126f474a50b5f3dc9b945d43da/Untitled 1.png" width="80%">
</div>
<p><strong>Figure 5.</strong> The TimeArcs visualization applied to the IEEE VIS publication co-authorship network of the top 50 researchers from 2010 to 2014 (i.e., the same data in Fig. <a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.12882#cgf12882-fig-0004">4</a>).</p>
</section>
<section id="cpan-graph-dataset" class="level4">
<h4 class="anchored" data-anchor-id="cpan-graph-dataset">CPAN Graph Dataset</h4>
</section>
<section id="descriptions-from-literature-26" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-26">Descriptions from Literature</h4>
<p>From <a href="https://ieeexplore.ieee.org/document/9222065"><strong>Preserving Minority Structures in Graph Sampling</strong></a></p>
<blockquote class="blockquote">
<p><strong>6.3.2 Cpan Graph Data Set</strong> The Cpan data set is a collaboration network with 839 nodes and 2,127 edges [1]. It depicts the relationships between the developers using the same Perl modules. The original graph and samples obtained by FF, TIES, and MCGS are shown in Figure 6. This case focused on the preservation of parachute-like rims at marginal areas.</p>
</blockquote>
</section>
<section id="example-figures-30" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-30">Example Figures</h4>
<p>From <a href="https://ieeexplore.ieee.org/document/9222065"><strong>Preserving Minority Structures in Graph Sampling</strong></a></p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Assorted Collaboration Network 6062ff126f474a50b5f3dc9b945d43da/Untitled 2.png" width="80%">
</div>
<p><strong>Fig. 6.</strong> Visual illustration of the cpan graph data set (a) and three samples generated by ff (b), ties (c), and mcgs (d) with a sampling rate of 30%.</p>
</section>
<section id="dblp" class="level4">
<h4 class="anchored" data-anchor-id="dblp">DBLP</h4>
</section>
<section id="descriptions-from-the-literature-1" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-the-literature-1">Descriptions from the Literature</h4>
<p>From <a href="https://link.springer.com/chapter/10.1007/978-3-030-35802-0_22">ChordLink: A New Hybrid Visualization Model</a></p>
<blockquote class="blockquote">
<p>The second case study considers co-authorship networks extracted from the DBLP dataset [<a href="https://link.springer.com/chapter/10.1007/978-3-030-35802-0_22#ref-CR30">30</a>], which contains publication data in computer science. Through a query consisting of keywords and Boolean operators, one can retrieve a set of publications on a desired topic. We use the results returned by DBLP to construct networks where nodes are authors and edges indicate co-authorships, weighted by the number of papers shared by their end-nodes. Nodes are labeled with authors’ names and edges with the titles of the corresponding publications. We performed the query “network AND visualization” and limited to 500 the number of search results (i.e., publications) to be returned. The resulting network consists of 1766 nodes, 3780 edges, and 382 connected components. The largest of these components contains 118 nodes and 322 edges.</p>
</blockquote>
</section>
<section id="example-figures-31" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-31">Example Figures</h4>
<p>From Parallel Edge Splatting for Scalable dynamic Graph Visualization</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Assorted Collaboration Network 6062ff126f474a50b5f3dc9b945d43da/Untitled 3.png" width="80%">
</div>
<p><strong>Fig. 8.</strong> The evolution of the word graph generated from paper titles containing the words “vis” and “web”. 21 graphs are shown for the years 1990 until 2010. A total of 264, 311 edges with weights of more than five are displayed.</p>
<p>From <a href="https://link.springer.com/chapter/10.1007/978-3-030-35802-0_22">ChordLink: A New Hybrid Visualization Model</a></p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Assorted Collaboration Network 6062ff126f474a50b5f3dc9b945d43da/Untitled 4.png" width="80%">
</div>
<p><strong>Fig. 1.</strong> A CHORDLINK visualization of a co-authorship network. The drawing has four clusters, represented as chord diagrams. In each chord diagram, circular arcs of the same color are copies of the same author. For example, in the smallest cluster, F. Montecchiani has two (green) copies, each connected to some nodes external to the cluster.</p>
<p>From Automatic Polygon Layout for Primal-Dual Visualization of Hypergraphs</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Assorted Collaboration Network 6062ff126f474a50b5f3dc9b945d43da/Untitled 5.png" width="80%">
</div>
<p><strong>Fig. 10.</strong> Paper and authorship data from the online database DBLP [26] for publications from 2013 to 2015 in IEEE Transactions on Pattern Analysis and Machine Intelligence. Each N-ary relationship is either a paper with N authors (left: the primal view) or an author with N papers (right: the dual view).</p>
<!-- == STOP RENDERING ==

from [Drawing Dynamic Graphs Without Timeslices](../Benchmark%20sets%200cc6b5e454304aec98f3b59b1a720476/Literature%20ad87f14e7097454fb2f784e2c8a2797a/Literature%20-%20Algorithm%2012e01bfc60a84007aa7d2d34293e123d/Drawing%20Dynamic%20Graphs%20Without%20Timeslices%204da7155eac4b4979ae165f3b0f1d2b1b.md) 

InfoVis Co-Authorship (Discrete): a co-authorship network for papers published

in the InfoVis conference from 1995 to 2015 [1]. Authors collaborating on a paper

are connected in a clique at the time of publication of the paper. Note this is

not a cumulative network as authors can appear, disappear, and appear again.

The data is of discrete nature with exactly 21 timeslices (one per year).

[https://cs.swan.ac.uk/~dynnoslice/software.html](https://cs.swan.ac.uk/~dynnoslice/software.html)

<div style='text-align: center'><img src='../data/Benchmark datasets/Assorted%20Collaboration%20Network%206062ff126f474a50b5f3dc9b945d43da/Untitled%206.png' width='80%'></div>

^ not really a description of the graph 

from **A Maxent-Stress Model for Graph Layout**

<div style='text-align: center'><img src='../data/Benchmark datasets/Assorted%20Collaboration%20Network%206062ff126f474a50b5f3dc9b945d43da/Untitled%207.png' width='80%'></div>

ith the exception of graph gd, which is an author collaboration graph of the International Symposium on Graph Drawing between 1994 and 2007, the graphs used are from the University of Florida Sparse Matrix Collection [9]. Our selection covers a range of graph sizes, and includes mesh-like and other nonmesh graphs, and graphs from Brandes and Pich's experimental study of distance scaling

## CPAN-Explorer graphs

Link works, but subpages do not

[https://gephi.wordpress.com/2009/06/25/cpan-explorer-an-interactive-exploration-of-the-perl-ecosystem/](https://gephi.wordpress.com/2009/06/25/cpan-explorer-an-interactive-exploration-of-the-perl-ecosystem/)
_site:150:21 -->
</section>
</div>
</div>
</div>
<p><a name="Airlines_Migrations_and_Air_Traffic"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Airlines, Migrations, and Air Traffic">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-36-contents" aria-controls="callout-36" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Airlines, Migrations, and Air Traffic
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-36" class="callout-36-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Airlines, migrations and air traffic data can be interesting for its <a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>properties and <a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a> The data is often used for edge bundling algorithms, which group edges together to reduce visual clutter.</p>
<p>The data below was collected from the Edge-Path bundling paper by Markus Wallinger et al.&nbsp;They cite the Airlines dataset to Force Directed Edge Bundling for Graph Visualization and the migrations dataset to Geometry-Based Edge Clustering for Graph Visualization.</p>
<div>
<b>Originally found at: </b><a href="https://github.com/mwallinger-tu/edge-path-bundling/tree/master/data">https://github.com/mwallinger-tu/edge-path-bundling/tree/master/data</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d9490a1101aa0dcf6a0ba5" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96d8c0c2b4d0f64386243" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d971241101aa0e9f6a0c0b" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90e774cf7480ef0055631" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>235-1702 nodes, 2101-16494 edges
</div>
<br>
<div>
<b>Number of Graphs: </b>3
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="holten2009">[@holten2009]</span>
</div>
<div class="col-8">
Force-Directed Edge Bundling for Graph Visualization
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="cui2008">[@cui2008]</span>
</div>
<div class="col-8">
Geometry-Based Edge Clustering for Graph Visualization
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Wallinger2022">[@Wallinger2022]</span>
</div>
<div class="col-8">
Edge-Path Bundling: A Less Ambiguous Edge Bundling Approach
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Ersoy2011">[@Ersoy2011]</span>
</div>
<div class="col-4">
Skeleton-Based Edge Bundling for Graph Visualization
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Holten2009">[@Holten2009]</span>
</div>
<div class="col-4">
Force-directed edge bundling for graph visualization
</div>
<div class="col-4">
<a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="vonLandesberger2016">[@vonLandesberger2016]</span>
</div>
<div class="col-4">
MobilityGraphs: Visual Analysis of Mass Mobility Dynamics via Spatio-Temporal Graphs and Clustering
</div>
<div class="col-4">
<a href="#Clusters_generated"><span class="pill-tag pill_color_Clusters_generated">Clusters (generated)</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="cui2008">[@cui2008]</span>
</div>
<div class="col-4">
Geometry-Based Edge Clustering for Graph Visualization
</div>
<div class="col-4">
<a href="#bundled_edges_generated"><span class="pill-tag pill_color_bundled_edges_generated">bundled edges (generated)</span></a><a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Wu2018">[@Wu2018]</span>
</div>
<div class="col-4">
MLSEB: Edge Bundling Using Moving Least Squares Approximation
</div>
<div class="col-4">
<a href="#bundled_edges_generated"><span class="pill-tag pill_color_bundled_edges_generated">bundled edges (generated)</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Perri2020">[@Perri2020]</span>
</div>
<div class="col-4">
HOTVis: Higher-Order Time-Aware Visualisation of Dynamic Graphs
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Wallinger2022">[@Wallinger2022]</span>
</div>
<div class="col-4">
Edge-Path Bundling: A Less Ambiguous Edge Bundling Approach
</div>
<div class="col-4">
<a href="#bundled_edges_generated"><span class="pill-tag pill_color_bundled_edges_generated">bundled edges (generated)</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="wang2020">[@wang2020]</span>
</div>
<div class="col-4">
Interactive Structure-aware Blending of Diverse Edge Bundling Visualizations
</div>
<div class="col-4">
<a href="#bundled_edges_generated"><span class="pill-tag pill_color_bundled_edges_generated">bundled edges (generated)</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="wallinger2023">[@wallinger2023]</span>
</div>
<div class="col-4">
Faster Edge-Path Bundling through Graph Spanners
</div>
<div class="col-4">
<a href="#bundled_edges_generated"><span class="pill-tag pill_color_bundled_edges_generated">bundled edges (generated)</span></a><a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-19" class="level4">
<h4 class="anchored" data-anchor-id="statistics-19">Statistics</h4>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Airlines, Migrations, and Air Traffic 45ffa6caedd141b082e163d7da6eb895/four_in_one.svg" width="80%">
</div>
</section>
<section id="descriptions-from-literature-27" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-27">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.1109/TVCG.2011.233">Skeleton-Based Edge Bundling for Graph Visualization</a>”:</p>
<blockquote class="blockquote">
<p>Figure 7 illustrates the SBEB and compares it with several existing bundling methods. Note that in all images here generated with our method, we used simple additive edge blending only, as our focus here is the layout, not the rendering. Images (a,b) show an air traffic graph (nodes are city locations, edges are interconnecting flights)… Images (e-h) show the US migrations graph bundled with the WR, GBEB, FDEB, and our method (SBEB) respectively. Overall, SBEB produces stronger bundling, due to the many iterations I = 10 being used), and emphasizes the structure of connections between groups of close cities (due to the skeleton layout cues). If less bundling is desired, fewer iterations can be used (Fig. 4). Adjusting the postprocessing smoothing and relaxation parameters, SBEB can create bundling styles similar to either GBEB (higher bundle curvatures, more emphasis on the graph structure) or FDEB (smoother bundles). Finally, images (i,j) show the US airlines graph bundled with the FDEB and SBEB respectively. SBEB generates stronger bundling (more overdraw) but arguably less clutter. Note also that SBEB generates treelike bundle structures which is useful when the exploration task at hand has an inherent (local) hierarchical nature, e.g.&nbsp;see how traffic connections merge into and/or split from main traffic routes.</p>
</blockquote>
</section>
<section id="example-figures-32" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-32">Example Figures</h4>
<p>From “<a href="https://ieeexplore.ieee.org/document/9552919">Edge-Path Bundling: A Less Ambiguous Edge Bundling Approach</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Airlines, Migrations, and Air Traffic 45ffa6caedd141b082e163d7da6eb895/Untitled.png" width="80%">
</div>
<p><strong>Fig. 1.</strong> Edge bundling of the migrations dataset. (a) Straight line drawing. (b) Force-directed edge bundling aggregates edges well, but overaggregates at the centre of the drawing making it difficult to see patterns in the east-west flow (red bundle). (c) Cubu has a similar drawback at the centre of the map to a lesser degree. (d) Winding roads divides this structure into several smaller flows, but they may not be necessarily related to graph structure. (e) Edge-path bundling is able to distinguish between several flows that reflect paths in the underlying graph. (f) When edge direction is considered, the algorithm is able to further subdivide these flows based on direction.</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Airlines, Migrations, and Air Traffic 45ffa6caedd141b082e163d7da6eb895/Untitled 1.png" width="80%">
</div>
<p><strong>Fig. 8.</strong> Airlines (undirected). (a) Input drawing. (b) Force-directed bundling is able to cluster edges into the major flows, but some overaggregation prevents details from being visualised. (c) Cubu provides a good bundling, but also has overaggregation. (c) Confluent drawings can be imposed on the layout, but as the approach cannot layout the graph bicliques can be distantly located, resulting in suboptimal performance. (e) Winding roads divides the flows into many streams, but these streams can be unfaithful to graph structure. (f) Edge-path bundling aggregates edges using weighted paths. The four prominent bundle intersections on the east coast correspond to major airports: Atlanta, Detroit, Minneapolis, and Dallas.</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Airlines, Migrations, and Air Traffic 45ffa6caedd141b082e163d7da6eb895/Untitled 2.png" width="80%">
</div>
<p><strong>Fig. 10.</strong> Results of four algorithms on the air traffic network. (a) Force-directed bundling is able to recover the major trajectories but does not strongly bundle the network. (b) Cubu strongly bundles the main flows of airtraffic, but can suffer from overaggregation. (c) Winding roads divides the traffic into many smaller bundles, but this may not be reflective of underlying graph structure. (d) With edge-path bundling, each bundle necessarily reflects a path in the network. There are separate flows across the atlantic and asia that correspond to paths through the network.</p>
<p>From “<a href="https://doi.org/10.1109/TVCG.2011.233">Skeleton-Based Edge Bundling for Graph Visualization</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Airlines, Migrations, and Air Traffic 45ffa6caedd141b082e163d7da6eb895/Screen_Shot_2023-08-14_at_3.42.37_PM.png" width="80%">
</div>
<!-- == STOP RENDERING ==

gd contest: [http://www.research.att.com/conf/gd96/contest.html](http://www.research.att.com/conf/gd96/contest.html)

look here: [https://github.com/mwallinger-tu/edge-path-bundling/tree/master/data](https://github.com/mwallinger-tu/edge-path-bundling/tree/master/data) from [**Edge-Path Bundling: A Less Ambiguous Edge Bundling Approach**](../Benchmark%20sets%200cc6b5e454304aec98f3b59b1a720476/Literature%20ad87f14e7097454fb2f784e2c8a2797a/Literature%20-%20Algorithm%2012e01bfc60a84007aa7d2d34293e123d/Edge-Path%20Bundling%20A%20Less%20Ambiguous%20Edge%20Bundling%20%20fcfa5882e692481298cbee906e005c30.md) -->
</section>
</div>
</div>
</div>
<p><a name="Code_Dependency_Graphs"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Code Dependency Graphs">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-37-contents" aria-controls="callout-37" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Code Dependency Graphs
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-37" class="callout-37-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Various software call dependency graphs have been used in the literature. We provide the benchmarks used in <a href="https://onlinelibrary.wiley.com/doi/full/10.1111/cgf.13724">A Random Sampling O(n) Force-calculation Algorithm for Graph Layouts</a>, as these posed the most comprehensive and available collection. <a href="https://ieeexplore.ieee.org/document/6065001">Parallel Edge Splatting for Scalable Dynamic Graph Visualization</a> used a software call graph from a device called a “Phillips Medical System Eindhoven medical scanner”, which is lost. Python and Eclipse subversion control logs are also currently lost. The authors of <a href="https://ieeexplore.ieee.org/document/6065001">Parallel Edge Splatting for Scalable Dynamic Graph Visualization</a> used <a href="https://depfind.sourceforge.io/ ">https://depfind.sourceforge.io/</a> to extract the commit history.</p>
<div>
<b>Originally found at: </b><a href="http://vis.cs.ucdavis.edu/~ogawa/datasets/ ">http://vis.cs.ucdavis.edu/~ogawa/datasets/ </a>(broken) <a href="https://osf.io/dcz5h ">https://osf.io/dcz5h </a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d94840803e0c0bf6558bd1" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96dc01101aa0ea66a0bee" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d970111101aa0ea36a0c62" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90f0994a6be0ec112e76b" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>128-2956 nodes, 310-10845 edges
</div>
<br>
<div>
<b>Number of Graphs: </b>11
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="gove2019">[@gove2019]</span>
</div>
<div class="col-8">
A Random Sampling O(n) Force-calculation Algorithm for Graph Layouts
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="burch2011">[@burch2011]</span>
</div>
<div class="col-4">
Parallel Edge Splatting for Scalable Dynamic Graph Visualization
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a><a href="#Weighted_edges"><span class="pill-tag pill_color_Weighted_edges">Weighted edges</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Holten2006">[@Holten2006]</span>
</div>
<div class="col-4">
Hierarchical Edge Bundles: Visualization of Adjacency Relations in Hierarchical Data
</div>
<div class="col-4">
<a href="#bundled_edges_generated"><span class="pill-tag pill_color_bundled_edges_generated">bundled edges (generated)</span></a><a href="#Other"><span class="pill-tag pill_color_other">Compound graphs</span></a><a href="#Hierarchical"><span class="pill-tag pill_color_Hierarchical">Hierarchical</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Tanahashi2015">[@Tanahashi2015]</span>
</div>
<div class="col-4">
An Efficient Framework for Generating Storyline Visualizations from Streaming Data
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="devkota2018">[@devkota2018]</span>
</div>
<div class="col-4">
CFGExplorer: Designing a Visual Control Flow Analytics System around Basic Program Analysis Operations
</div>
<div class="col-4">
<a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-20" class="level4">
<h4 class="anchored" data-anchor-id="statistics-20">Statistics</h4>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Code Dependency Graphs 3b6beec0413c486184cee398056e6c87/four_in_one.svg" width="80%">
</div>
</section>
<section id="software-dependency-calls" class="level4">
<h4 class="anchored" data-anchor-id="software-dependency-calls">Software Dependency Calls</h4>
</section>
<section id="descriptions-from-literature-28" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-28">Descriptions from Literature</h4>
<p>From “<a href="https://onlinelibrary.wiley.com/doi/full/10.1111/cgf.13724">A Random Sampling O(n) Force-calculation Algorithm for Graph Layouts</a>”:</p>
<blockquote class="blockquote">
<p>The software graphs show dependencies in several software packages [<strong><a href="https://onlinelibrary.wiley.com/doi/full/10.1111/cgf.13724#cgf13724-bib-0076">vB11</a></strong>, <strong><a href="https://onlinelibrary.wiley.com/doi/full/10.1111/cgf.13724#cgf13724-bib-0075">vB02</a></strong>, <strong><a href="https://onlinelibrary.wiley.com/doi/full/10.1111/cgf.13724#cgf13724-bib-0077">vB12</a></strong>, <strong><a href="https://onlinelibrary.wiley.com/doi/full/10.1111/cgf.13724#cgf13724-bib-0078">vBB13</a></strong>, <strong><a href="https://onlinelibrary.wiley.com/doi/full/10.1111/cgf.13724#cgf13724-bib-0082">vvBB14</a></strong>].</p>
</blockquote>
<p>From “<a href="https://ieeexplore.ieee.org/document/6065001">Parallel Edge Splatting for Scalable Dynamic Graph Visualization</a>”:</p>
<blockquote class="blockquote">
<p>We collected the 21 available releases of JUnit as compiled Java bytecode and extracted the method call dependencies using <em>DependencyFinder</em> (depfind. sourceforge.net). The package and class structure provides a natural hierarchy on the methods of the system. The resulting dynamic graph consists of 2,817 vertices connected by 15,339 edges, where each edge is weighted by 1.</p>
</blockquote>
</section>
<section id="example-figures-33" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-33">Example Figures</h4>
<p>From <a href="https://ieeexplore.ieee.org/document/6065001">Parallel Edge Splatting for Scalable Dynamic Graph Visualization</a>:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Code Dependency Graphs 3b6beec0413c486184cee398056e6c87/Untitled.png" width="80%">
</div>
<p><strong>Fig. 6.</strong> Call dependencies between methods in 21 releases of the JUnit project: all (top), added (left), and removed (right) dependencies.</p>
</section>
<section id="phillips-medical-software-call-graph-lost" class="level4">
<h4 class="anchored" data-anchor-id="phillips-medical-software-call-graph-lost">Phillips Medical Software Call Graph (Lost)</h4>
</section>
<section id="descriptions-from-literature-29" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-29">Descriptions from Literature</h4>
<p>From “<a href="https://ieeexplore.ieee.org/document/4015425">Hierarchical Edge Bundles: Visualization of Adjacency Relations in Hierarchical Data</a>”:</p>
<blockquote class="blockquote">
<p>The software is part of a medical scanner and was provided by Philips Medical Systems Eindhoven. Three hierarchy levels – layers, units, and modules – consisting of 284 nodes are used together with the associated call graph for the elements at the lowest level of the hierarchy, i.e., 1,011 adjacency relations representing module-to-module calls.</p>
</blockquote>
</section>
<section id="example-figures-34" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-34">Example Figures</h4>
<p>From “<a href="https://ieeexplore.ieee.org/document/4015425">Hierarchical Edge Bundles: Visualization of Adjacency Relations in Hierarchical Data</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Code Dependency Graphs 3b6beec0413c486184cee398056e6c87/Untitled 1.png" width="80%">
</div>
<p><strong>Fig. 13.</strong> A software system and its associated call graph (caller = green, callee = red). (a) and (b) show the system with bundling strength β = 0.85 using a balloon layout (node labels disabled) and a radial layout, respectively. Bundling reduces visual clutter, making it easier to perceive the actual connections than when compared to the non-bundled versions (figures 2a and 11a). Bundled visualizations also show relations between sparsely connected systems more clearly (encircled regions); these are almost completely obscured in the non-bundled versions. The encircled regions highlight identical parts of the system for (a), (b), and figure 15.</p>
</section>
<section id="subversion-control-logs-lost" class="level4">
<h4 class="anchored" data-anchor-id="subversion-control-logs-lost">Subversion Control Logs (Lost)</h4>
<p>From “<a href="https://ieeexplore.ieee.org/document/7015617">An Efficient Framework for Generating Storyline Visualizations from Streaming Data”:</a></p>
<blockquote class="blockquote">
<p>The datasets we used for this study were: … <em>Python Subversion Control Logs</em> (<em>Python</em>) [30] E<em>clipse Subversion Control Logs</em> (<em>Eclipse</em>) [30] … These datasets all contained temporal logs of events in which entities interact.</p>
</blockquote>
<!-- === STOP RENDERING ==

I cannot find the Phillips medical software call graphs - the paper implies they were directly given to the author so it might be hard to get. 

Subversion control logs also lost. Broken link from UCDavis. 

M. Ogawa. Software development history logs [Online]. Available:
[http://vis.cs.ucdavis.edu/~ogawa/datasets/](http://vis.cs.ucdavis.edu/~ogawa/datasets/), Jan. 2015. -->
</section>
</div>
</div>
</div>
<p><a name="Blogposts_Tweets_and_Forums"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Blogposts, Tweets, and Forums">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-38-contents" aria-controls="callout-38" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Blogposts, Tweets, and Forums
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-38" class="callout-38-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>In the surveyed papers we found various instances of blogposts, tweets, and forum posts. Many of these are timestamped posts that connect user mentions, textually derived entities, hashtags, or user actions on posts. We provide a selection of Blogposts, Tweets, and MOOC (a student forum) as found in the literature. Time Arcs uses AMERICAblog, Huffington Post, and other sources (corpus_ner_geo) data which can be found at <a href="https://github.com/CreativeCodingLab/TimeArcs/tree/master/Text/data.">https://github.com/CreativeCodingLab/TimeArcs/tree/master/Text/data.</a></p>
<p>Fast filtering uses the Twitter gardenhouse streaming API to collect data on the 2013 Super Bowl and the announcement of Osama Bin Laden’s death, both can be found <a href="https://github.com/WICI/fastviz/tree/master/data ">https://github.com/WICI/fastviz/tree/master/data </a>(shared hashtags).</p>
<p>Event-based Dynamic Graph Drawing without the Agonizing Pain uses the rugby tweet dataset (pro12_mentions) consisting of mentions among the members of the GuinessPro12 competition, and the MOOC dataset from the <a href="http://moocdata.cn/challenges/kdd-cup-2015">KDD Cup 2015</a> (A data challenge).</p>
<p>This kind of data can be interesting because it might be interpreted as <a href="#hypergraph"><span class="pill-tag pill_color_hypergraph">hypergraph</span></a>data, where nodes are connected by multiple edges, or as <a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a>data, where the edges change over time.</p>
<div>
<b>Originally found at: </b><a href="http://moocdata.cn/data/user-activity
">http://moocdata.cn/data/user-activity </a><a href="https://github.com/CreativeCodingLab/TimeArcs
">https://github.com/CreativeCodingLab/TimeArcs </a><a href="https://github.com/WICI/fastviz/tree/master/data
">https://github.com/WICI/fastviz/tree/master/data </a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d9497a0c2b4d0e8c386228" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96da194a6be112a12e740" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d971111101aa0ea36a0cb6" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d94add4cf7481071055619" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>12-386412 nodes, 3151-556820 edges
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="feng2019dropout">[@feng2019dropout]</span>
</div>
<div class="col-8">
Understanding Dropouts in MOOCs
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="dang2016">[@dang2016]</span>
</div>
<div class="col-8">
TimeArcs: Visualizing Fluctuations in Dynamic Networks
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Grabowicz2014">[@Grabowicz2014]</span>
</div>
<div class="col-8">
Fast filtering and animation of large dynamic networks
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Frishman2008">[@Frishman2008]</span>
</div>
<div class="col-4">
Online Dynamic Graph Drawing
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Purchase2020">[@Purchase2020]</span>
</div>
<div class="col-4">
The Turing Test for Graph Drawing Algorithms
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Grabowicz2014">[@Grabowicz2014]</span>
</div>
<div class="col-4">
Fast filtering and animation of large dynamic networks
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (continuous)</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="dang2016">[@dang2016]</span>
</div>
<div class="col-4">
TimeArcs: Visualizing Fluctuations in Dynamic Networks
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="arleo2022">[@arleo2022]</span>
</div>
<div class="col-4">
Event-based Dynamic Graph Drawing without the Agonizing Pain
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a>
</div>
</div>
<p><br></p>
<section id="statistics-21" class="level4">
<h4 class="anchored" data-anchor-id="statistics-21">Statistics</h4>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Blogposts, Tweets, and Forums e315a82238dc40c5a3559c81ef7c57c8/four_in_one.svg" width="80%">
</div>
</section>
<section id="blogsmooc" class="level4">
<h4 class="anchored" data-anchor-id="blogsmooc">Blogs/MOOC</h4>
</section>
<section id="descriptions-from-literature-30" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-30">Descriptions from Literature</h4>
<p>From <a href="https://doi.org/10.1111/cgf.12882">TimeArcs: Visualizing Fluctuations in Dynamic Networks</a>:</p>
<blockquote class="blockquote">
<p>We collected 90,811 political blog posts over a ten-year period from 2005 to 2015 from seven different sources, including AMERICAblog, Huffington Post, and ProPublica. We then ran text analyses on these blogs and generated terms that were classified into four different categories. These terms were then input into <em>TimeArcs</em>.</p>
</blockquote>
<p>From <a href="https://onlinelibrary.wiley.com/doi/full/10.1111/cgf.14615">Event-based Dynamic Graph Drawing without the Agonizing Pain</a>:</p>
<blockquote class="blockquote">
<p>MOOC represents the actions (e.g.&nbsp;viewing a video, submitting an answer, etc.) taken by users on a popular massive open online class platform [<a href="https://onlinelibrary.wiley.com/doi/full/10.1111/cgf.14615#cgf14615-bib-0037">KZL19</a>]. The nodes represent users and course activities (targets), and temporal edges represent the actions by users on the targets. We pick and elaborate the first 15 thousands events.</p>
</blockquote>
</section>
<section id="example-figures-35" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-35">Example Figures</h4>
<p>From <a href="https://doi.org/10.1111/cgf.12882">TimeArcs: Visualizing Fluctuations in Dynamic Networks</a>:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Blogposts, Tweets, and Forums e315a82238dc40c5a3559c81ef7c57c8/Untitled.png" width="80%">
</div>
<p><strong>Fig. 7.</strong> Overview of political events in the past 10 years using TimeArcs. The top 100 terms were selected based on their sudden attention and degree centrality. Terms are color-coded by category: green for person, red for location, blue for organization, yellow for miscellaneous category.</p>
<p>From <a href="https://onlinelibrary.wiley.com/doi/full/10.1111/cgf.14615">Event-based Dynamic Graph Drawing without the Agonizing Pain</a>:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Blogposts, Tweets, and Forums e315a82238dc40c5a3559c81ef7c57c8/Untitled 1.png" width="80%">
</div>
<p><strong>Table 2.</strong> Flattened snapshots of the network evolution over time taken at regular intervals. Twenty artificial time slices were inserted for temporal graphs</p>
</section>
<section id="tweets" class="level4">
<h4 class="anchored" data-anchor-id="tweets">Tweets</h4>
</section>
<section id="descriptions-from-literature-31" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-31">Descriptions from Literature</h4>
<p>From <a href="https://epjdatascience.springeropen.com/articles/10.1140/epjds/s13688-014-0027-8">Fast filtering and animation of large dynamic networks</a>:</p>
<blockquote class="blockquote">
<p>We use data obtained through the Twitter gardenhose streaming API, which covers around 10% of the tweet volume.We focus on two events: the announcement of Osama bin Laden’s death and the 2013 Super Bowl. We consider user mentions and hashtags as entities and their co-occurrence in the same tweet as interactions between them.</p>
</blockquote>
<p>From <a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.14615">Event-based Dynamic Graph Drawing without the Agonizing Pain</a>:</p>
<blockquote class="blockquote">
<p><strong>Rugby</strong> is a network derived from over 3000 tweets involving teams in the ‘Guinness Pro12’ rugby competition. The tweets were posted between 1 September 2014 and 23 October 2015. Each tweet contains information about the involved teams and the time of publication with a precision down to the second.</p>
</blockquote>
</section>
<section id="example-figures-36" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-36">Example Figures</h4>
<p>From <a href="https://epjdatascience.springeropen.com/articles/10.1140/epjds/s13688-014-0027-8">Fast filtering and animation of large dynamic networks</a></p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Blogposts, Tweets, and Forums e315a82238dc40c5a3559c81ef7c57c8/Untitled 2.png" width="80%">
</div>
<!-- == STOP RENDERING ==

**Online dynamic -**  cannot find code online, paper shows two links of where they collected the data. Both broken [http://www.dailytech.com](http://www.dailytech.com/), [http://www.rimzu.com](http://www.rimzu.com/) - paper says to look in [https://www.computer.org/csdl/journal/tg/2008/04/ttg2008040727/13rRUxBJhvo](https://www.computer.org/csdl/journal/tg/2008/04/ttg2008040727/13rRUxBJhvo) but it is asking me to pay

**Turing Test -** Uses the Zachary Karate dataset (they link to the suiteSparse Matrix collection, but I also know it is in sparse, and one of the Pajek subcollections). Also uses a timeslice of the same dataset used by Online dynamic (we could reconstruct this from their images since it is also only 85 nodes, but we really loose all the info we want)
_site:150:21 -->
</section>
</div>
</div>
</div>
<p><a name="Social_Networks"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Social Networks">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-39-contents" aria-controls="callout-39" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Social Networks
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-39" class="callout-39-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Various social network datasets are used in the literature and here we provide a collection of some of the ones found in our survey. Namely, the Zachary Karate Club, Facebook100, Infectious SocioPatterns dynamic contact networks, and the Van De Brunt Fraternity Data. Note that Facebook1684 refers to the ego network for user 1684, and this along various other users can be found in SNAP. Unfortunately, we are unable to provide this dataset at this time. Social networks can be useful for their <a href="#clusters"><span class="pill-tag pill_color_clusters">clustering</span></a>opportunities and <a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a>properties.</p>
<div>
<b>Originally found at: </b><a href="https://snap.stanford.edu/data/ego-Facebook.html
">https://snap.stanford.edu/data/ego-Facebook.html </a><a href="https://cs.swan.ac.uk/~dynnoslice/software.html
">https://cs.swan.ac.uk/~dynnoslice/software.html </a><a href="https://archive.org/details/oxford-2005-facebook-matrix 
">https://archive.org/details/oxford-2005-facebook-matrix </a><a href="https://osf.io/dcz5h
">https://osf.io/dcz5h </a><a href="http://www.sociopatterns.org/datasets/infectious-sociopatterns-dynamic-contact-networks/
">http://www.sociopatterns.org/datasets/infectious-sociopatterns-dynamic-contact-networks/ </a><a href="http://www-personal.umich.edu/~mejn/netdata/">http://www-personal.umich.edu/~mejn/netdata/</a>
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="zachary1977">[@zachary1977]</span>
</div>
<div class="col-8">
An Information Flow Model for Conflict and Fission in Small Groups
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="mcauley2012">[@mcauley2012]</span>
</div>
<div class="col-8">
Learning to discover social circles in ego networks
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="ISELLA2011166">[@ISELLA2011166]</span>
</div>
<div class="col-8">
What’s in a crowd? analysis of face-to-face behavioral networks
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="van_de_bunt_friendship_1999">[@van_de_bunt_friendship_1999]</span>
</div>
<div class="col-8">
Friendship Networks Through Time: An Actor-Oriented Dynamic Statistical Network Model
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="facebooklong">[@facebooklong]</span>
</div>
<div class="col-8">
Social Structure of Facebook Networks
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="red2011">[@red2011]</span>
</div>
<div class="col-8">
Comparing Community Structure to Characteristics in Online Collegiate Social Networks
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="nocaj2015">[@nocaj2015]</span>
</div>
<div class="col-4">
Untangling the Hairballs of Multi-Centered Small-World Online Social Media Networks
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Zhao2021">[@Zhao2021]</span>
</div>
<div class="col-4">
Preserving Minority Structures in Graph Sampling
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Simonetto2018">[@Simonetto2018]</span>
</div>
<div class="col-4">
Drawing Dynamic Graphs Without Timeslices
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (continuous)</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Raj2018">[@Raj2018]</span>
</div>
<div class="col-4">
Anisotropic Radial Layout for Visualizing Centrality and Structure in Graphs
</div>
<div class="col-4">
<a href="#Clusters_generated"><span class="pill-tag pill_color_Clusters_generated">Clusters (generated)</span></a><a href="#Weighted_edges"><span class="pill-tag pill_color_Weighted_edges">Weighted edges</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Ni2018">[@Ni2018]</span>
</div>
<div class="col-4">
Network Alignment by Discrete Ollivier-Ricci Flow
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Purchase2020">[@Purchase2020]</span>
</div>
<div class="col-4">
The Turing Test for Graph Drawing Algorithms
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Hachul2006">[@Hachul2006]</span>
</div>
<div class="col-4">
An Experimental Comparison of Fast Algorithms for Drawing General Large Graphs
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Qu2022">[@Qu2022]</span>
</div>
<div class="col-4">
Automatic Polygon Layout for Primal-Dual Visualization of Hypergraphs
</div>
<div class="col-4">
<a href="#Clusters_generated"><span class="pill-tag pill_color_Clusters_generated">Clusters (generated)</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="hoffswell2018">[@hoffswell2018]</span>
</div>
<div class="col-4">
SetCoLa: High-Level Constraints for Graph Layout
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Labeled_nodes"><span class="pill-tag pill_color_Labeled_nodes">Labeled nodes</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">

</div>
<div class="col-4">
A Random Sampling O
</div>
<div class="col-4">

</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="fisheye2019">[@fisheye2019]</span>
</div>
<div class="col-4">
Structure-aware Fisheye Views for Efficient Large Graph Exploration
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="wang2018">[@wang2018]</span>
</div>
<div class="col-4">
Revisiting Stress Majorization as a Unified Framework for Interactive Constrained Graph Visualization
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a>
</div>
</div>
<p><br></p>
<section id="zachary-karate-club-dataset" class="level4">
<h4 class="anchored" data-anchor-id="zachary-karate-club-dataset">Zachary Karate Club Dataset</h4>
</section>
<section id="descriptions-from-literature-32" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-32">Descriptions from Literature</h4>
<p><em><a href="https://link.springer.com/chapter/10.1007/978-3-030-68766-3_36">The Turing Test for Graph Drawing Algorithms</a></em> describes the dataset simply as “social network” in their Table 1.</p>
<p>From <a href="https://link.springer.com/chapter/10.1007/978-3-319-73915-1_28">Anisotropic Radial Layout for Visualizing Centrality and Structure in Graphs</a>:</p>
<blockquote class="blockquote">
<p>The Zachary’s karate club graph is a well known data set that is a social network of friendships in a karate club at a US university, as recorded during a study&nbsp;[<a href="https://link.springer.com/chapter/10.1007/978-3-319-73915-1_28#ref-CR32">32</a>]. This graph contains 34 nodes, each representing an individual, and 78 unweighted edges that represent a friendship between the associated individuals (Fig.&nbsp;<a href="https://link.springer.com/chapter/10.1007/978-3-319-73915-1_28#Fig1">1</a>). During the period of observation, a conflict between two key members, identified as the “administrator” and “instructor”, leads to a split in the club, giving it an interesting two cluster structure. In Fig.&nbsp;<a href="https://link.springer.com/chapter/10.1007/978-3-319-73915-1_28#Fig1">1</a>, nodes representing members who are part of the instructor’s and administrator’s groups are drawn in green and blue, respectively.</p>
</blockquote>
</section>
<section id="example-figures-37" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-37">Example Figures</h4>
<p>From “<a href="https://arxiv.org/abs/2008.04869">The Turing Test for Graph Drawing Algorithms</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Social Networks 90888b3285d042d49072985b05f95203/Untitled.png" width="80%">
</div>
<p>Figure from <a href="https://www.dcs.gla.ac.uk/~hcp/GD2020/">supplementary materials</a> using a force directed layout.</p>
<p>From “<a href="https://link.springer.com/chapter/10.1007/978-3-319-73915-1_28">Anisotropic Radial Layout for Visualizing Centrality and Structure in Graphs</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Social Networks 90888b3285d042d49072985b05f95203/Untitled 1.png" width="80%">
</div>
<p><strong>Fig. 1.</strong> Visualization of Zachary’s karate club social network using (a) MDS, (b) radial layout, and (c) anisotropic radial layout. Node sizes encode betweenness centrality.</p>
<p>****From “<a href="https://link.springer.com/chapter/10.1007/978-3-030-04414-5_32">Network Alignment by Discrete Ollivier-Ricci Flow</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Social Networks 90888b3285d042d49072985b05f95203/Untitled 2.png" width="80%">
</div>
<p><strong>Fig. 1.</strong> An example of Ricci curvature on the karate club graph before and after Ricci flow. The colors represent the Ricci curvature while the thickness represents the edge weight. Ricci flow deformed the edge weights until Ricci curvatures converged (−0.0027 in this case).</p>
</section>
<section id="facebook100" class="level4">
<h4 class="anchored" data-anchor-id="facebook100">Facebook100</h4>
</section>
<section id="description-from-literature-4" class="level4">
<h4 class="anchored" data-anchor-id="description-from-literature-4">Description from Literature</h4>
<p>From <a href="https://jgaa.info/getPaper?id=370">Untangling the Hairballs of Multi-Centered, Small-World Online Social Media Networks</a>:</p>
<blockquote class="blockquote">
<p>As real world samples, we use the Facebook100 dataset [40], which contains social relations of 100 higher educational institutes in the US. The network size varies from 762 to 41K vertices and from 16K to 1.6M edges. Thedataset is directly from Facebook, not sampled, and thus very complete in terms of capturing the social relations according to a widely used service at that time. Additional attributes obtained from the Facebook profiles are gender, expected year of graduation, dormitory, etc. Due to incomplete profiles, a number of attribute values are missing.</p>
</blockquote>
</section>
<section id="example-figures-38" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-38">Example Figures</h4>
<p>From <a href="https://jgaa.info/getPaper?id=370">Untangling the Hairballs of Multi-Centered, Small-World Online Social Media Networks</a></p>
<p><strong>Fig. 1.</strong> Facebook friendships at California Institute of Technology (Caltech36). Vertex color corresponds to dormitory (gray for missing values), but has not been utilized by the layout algorithm. The layout in (a) is based on the entire graph, whereas (b)-(f) use edge embeddedness, which spreads the graph while keeping cohesive groups together. Embeddedness mapped to edge color; backbone edges dark gray.</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Social Networks 90888b3285d042d49072985b05f95203/Untitled 3.png" width="80%">
</div>
</section>
<section id="facebook-ego-networks-facebook-1864" class="level4">
<h4 class="anchored" data-anchor-id="facebook-ego-networks-facebook-1864">Facebook Ego Networks (Facebook 1864)</h4>
</section>
<section id="description-from-literature-5" class="level4">
<h4 class="anchored" data-anchor-id="description-from-literature-5">Description from Literature</h4>
<p>From ****<a href="https://ieeexplore.ieee.org/document/9222065">Preserving Minority Structures in Graph Sampling</a></p>
<blockquote class="blockquote">
<p>The Facebook1684 graph data set is an online social network with 775 nodes and 14,006 edges. The original graph and six samples are shown in&nbsp;Figure 4. This data set is an unbalanced graph in which two large communities include 705 nodes and two small communities/cliques contain only 70 nodes.</p>
</blockquote>
<p>The <a href="https://github.com/saibalmars/RicciFlow-SampleGraphs/tree/61f9c2c48d0de368369e232da959c4f996122e3a">supplemental</a> material for *****<a href="https://link.springer.com/chapter/10.1007/978-3-030-04414-5_32">Network Alignment by Discrete Ollivier-Ricci Flow</a>* also provides user 1864 of the Facebook ego networks in their repository.</p>
<p>From <a href="https://osf.io/dcz5h">A Random Sampling O(n) Force-calculation Algorithm for Graph Layouts</a>:</p>
<blockquote class="blockquote">
<p>The ego graphs are Facebook ego graphs [<strong><a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.13724#cgf13724-bib-0060">ML12</a></strong>] with the ego vertex re-added to the graph.</p>
</blockquote>
</section>
<section id="example-figures-39" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-39">Example Figures</h4>
<p>From Preserving Minority Structures in Graph Sampling](https://ieeexplore.ieee.org/document/9222065)</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Social Networks 90888b3285d042d49072985b05f95203/Untitled 4.png" width="80%">
</div>
<p><strong>Fig. 4.</strong>&nbsp;Interface for subjective assessment experiment with the Facebook1684 data set (775 nodes and 14,006 edges). The leftmost graph is the original graph. The six randomly arranged samples are generated by SST (a), FF (b), MCGS (c), RW (d), TIES (e), and RDN (f) with a sampling rate of 30%.</p>
</section>
<section id="van-de-bunt-fraternity" class="level4">
<h4 class="anchored" data-anchor-id="van-de-bunt-fraternity">Van De Bunt Fraternity</h4>
</section>
<section id="description-from-the-literature-2" class="level4">
<h4 class="anchored" data-anchor-id="description-from-the-literature-2">Description from the Literature</h4>
<p>From <a href="https://ieeexplore.ieee.org/document/9222065">Preserving Minority Structures in Graph Sampling</a></p>
<blockquote class="blockquote">
<p><em>Van De Bunt (Discrete):</em>&nbsp;shows the relationships between 32 freshmen at seven different time points. A discrete dynamic graph is built using the method of Brandes&nbsp;<em>et al.</em>&nbsp;&nbsp;[<a href="https://link.springer.com/chapter/10.1007/978-3-319-73915-1_31#ref-CR9">9</a>], with an undirected edge inserted into a timeslice if the participants reciprocally report “best friendship” or “friendship” at that time.</p>
</blockquote>
</section>
<section id="syphilis-social-network" class="level4">
<h4 class="anchored" data-anchor-id="syphilis-social-network"><strong>Syphilis Social Network</strong></h4>
</section>
<section id="description-from-the-literature-3" class="level4">
<h4 class="anchored" data-anchor-id="description-from-the-literature-3">Description from the Literature</h4>
<p>From <a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.13440">SetCoLa: High-Level Constraints for Graph Layout</a>:</p>
<blockquote class="blockquote">
<p>Social networks can be a powerful way to understand inter-personal relationships and are useful for tracking the spread of diseases that result from personal contact [<strong><a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.13440#cgf13440-bib-0045">RST*98</a></strong>,&nbsp;<strong><a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.13440#cgf13440-bib-0022">FHH*01</a></strong>,&nbsp;<strong><a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.13440#cgf13440-bib-0043">MRV*03</a></strong>,&nbsp;<strong><a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.13440#cgf13440-bib-0021">FHD*11</a></strong>]. The ability to track and identify at risk individuals can lead to treatment and help manage the spread of the disease. In addition to the links between individuals, structuring the layout by node properties such as the social or ethnographically-identified group may reveal additional details about how the disease is spread.</p>
</blockquote>
</section>
<section id="example-figures-40" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-40">Example Figures</h4>
<p>From <a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.13440">SetCoLa: High-Level Constraints for Graph Layout</a>:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Social Networks 90888b3285d042d49072985b05f95203/Untitled 5.png" width="80%">
</div>
<p><strong>Fig. 4.</strong> The layout for the syphilis social network from (a) Rothenberg et al.&nbsp;[<a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.13440#cgf13440-bib-0045">RST*98</a>]. (b) We recreated and improved the layout in SetCoLa by introducing additional padding, alignment, and circle constraints to further highlight the relative number of interactions among the different groups. For both figures, the nodes are split into three groups, from left to right: young affluent white men, younger white women, and young African-American men. Individuals not associated with any of these “core” groups are positioned above the others. In both figures, individuals diagnosed with syphilis during the outbreak are labeled with an “S” on the node label.</p>
<!-- === STOP RENDERING ===

Contact Network

This dataset contains the daily dynamic contact networks collected during the [Infectious SocioPatterns](http://www.sociopatterns.org/deployments/infectious-sociopatterns/) event that took place at the [Science Gallery](http://www.sciencegallery.com/) in Dublin, Ireland, during the artscience exhibition [INFECTIOUS: STAY AWAY](http://www.sciencegallery.com/infectious). Each file in the downloadable package contains a tab-separated list representing the active contacts during 20-second intervals of one day of data collection.  Each line has the form “t i j“, where i and j are the anonymous IDs of the persons in contact, and the interval during which this contact was active is [ t – 20s, t ]. If multiple contacts are active in a given interval, you will see multiple lines starting with the same value of t. Time is measured in seconds and expressed in UNIX ctime.

“This evaluation uses 109 common graphs from network science and graph algorithm research (note: many graphs from older research are no longer available online).”

# C. Lipp?

> The rest of the test graphs are taken from real-world applications. In particular, we selected graphs from the AT&T graph library [1], from C. Walshaw’s graph collection [20], and a graph that describes a social network of 2113 people that we obtained from C. Lipp.
> 

Last author of original paper,

→ facebook100 network: [https://epubs.siam.org/doi/10.1137/080734315](https://epubs.siam.org/doi/10.1137/080734315)

[https://archive.org/details/oxford-2005-facebook-matrix](https://archive.org/details/oxford-2005-facebook-matrix) ← pretty sure the internet archive is reliable? 

→ dead link in this paper

→ [http://masonporter.blogspot.com/2011/02/facebook100-data-set.html](http://masonporter.blogspot.com/2011/02/facebook100-data-set.html)
also dead link

→ here? [https://networkrepository.com/socfb.php](https://networkrepository.com/socfb.php)

from [Drawing Dynamic Graphs Without Timeslices](../Benchmark%20sets%200cc6b5e454304aec98f3b59b1a720476/Literature%20ad87f14e7097454fb2f784e2c8a2797a/Literature%20-%20Algorithm%2012e01bfc60a84007aa7d2d34293e123d/Drawing%20Dynamic%20Graphs%20Without%20Timeslices%204da7155eac4b4979ae165f3b0f1d2b1b.md) -->
</section>
</div>
</div>
</div>
<p><a name="Transportation_Networks"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Transportation Networks">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-40-contents" aria-controls="callout-40" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Transportation Networks
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-40" class="callout-40-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Transportation networks are interesting because of their <a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>nature, and feautre <a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a> DIMACS USA Road Network source notes that the data has errors, namely missing major highways and bridges. We currently include only the subway layout networks in our files.</p>
<div>
<b>Originally found at: </b>http://www.diag.uniroma1.it//challenge9/download.shtml <a href="https://osf.io/dcz5h">https://osf.io/dcz5h</a>
</div>
<br>
<div>
<b>Download formatted data: </b><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d9498894a6be103212e712" class="btn btn-secondary btn-sm skinny-button"> GEXF</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d94d990c2b4d0e9c386231" class="btn btn-secondary btn-sm skinny-button"> GML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d970f894a6be112a12ea21" class="btn btn-secondary btn-sm skinny-button"> GRAPHML</a><a href="https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d90f1394a6be0ec012e916" class="btn btn-secondary btn-sm skinny-button"> JSON</a>
</div>
<br>
<div>
<b>Size: </b>82-433 nodes, 158-950 edges
</div>
<br>
<div>
<b>Number of Graphs: </b>15
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="gove2019">[@gove2019]</span>
</div>
<div class="col-8">
A Random Sampling O(n) Force-calculation Algorithm for Graph Layouts
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="muelder2008b">[@muelder2008b]</span>
</div>
<div class="col-4">
A Treemap Based Method for Rapid Layout of Large Graphs
</div>
<div class="col-4">
<a href="#Clusters_generated"><span class="pill-tag pill_color_Clusters_generated">Clusters (generated)</span></a><a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="muelder2008">[@muelder2008]</span>
</div>
<div class="col-4">
Rapid Graph Layout Using Space Filling Curves
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Meulemans2013">[@Meulemans2013]</span>
</div>
<div class="col-4">
KelpFusion: A Hybrid Set Visualization Technique
</div>
<div class="col-4">
<a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Nollenburg2011">[@Nollenburg2011]</span>
</div>
<div class="col-4">
Drawing and Labeling High-Quality Metro Maps by Mixed-Integer Programming
</div>
<div class="col-4">
<a href="#Labeled_nodes"><span class="pill-tag pill_color_Labeled_nodes">Labeled nodes</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Huang2016">[@Huang2016]</span>
</div>
<div class="col-4">
TrajGraph: A Graph-Based Visual Analytics Approach to Studying Urban Network Centralities Using Taxi Trajectory Data
</div>
<div class="col-4">
<a href="#Labeled_nodes"><span class="pill-tag pill_color_Labeled_nodes">Labeled nodes</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="greilich2009">[@greilich2009]</span>
</div>
<div class="col-4">
Visualizing the evolution of compound digraphs with TimeArcTrees
</div>
<div class="col-4">
<a href="#Other"><span class="pill-tag pill_color_other">Compound graphs</span></a><a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a><a href="#Hierarchical"><span class="pill-tag pill_color_Hierarchical">Hierarchical</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a><a href="#Weighted_edges"><span class="pill-tag pill_color_Weighted_edges">Weighted edges</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="brandes1998">[@brandes1998]</span>
</div>
<div class="col-4">
Using Graph Layout to Visualize Train Interconnection Data
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="zeng2019">[@zeng2019]</span>
</div>
<div class="col-4">
Route-Aware Edge Bundling for Visualizing Origin-Destination Trails in Urban Traffic
</div>
<div class="col-4">
<a href="#bundled_edges_generated"><span class="pill-tag pill_color_bundled_edges_generated">bundled edges (generated)</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<div class="row">
<div class="col-4">

</div>
<div class="col-4">
A Random Sampling O
</div>
<div class="col-4">

</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="preiner2020">[@preiner2020]</span>
</div>
<div class="col-4">
Augmenting Node-Link Diagrams with Topographic Attribute Maps
</div>
<div class="col-4">
<a href="#Categorical_nodes"><span class="pill-tag pill_color_Categorical_nodes">Categorical nodes</span></a><a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Other"><span class="pill-tag pill_color_Other">Topological</span></a><a href="#Trees"><span class="pill-tag pill_color_Trees">Trees</span></a><a href="#Weighted_edges"><span class="pill-tag pill_color_Weighted_edges">Weighted edges</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="bast2020">[@bast2020]</span>
</div>
<div class="col-4">
Metro Maps on Octilinear Grid Graphs
</div>
<div class="col-4">
<a href="#Other"><span class="pill-tag pill_color_Other">Octilinear</span></a><a href="#Other"><span class="pill-tag pill_color_Other">Port constraints</span></a><a href="#Other"><span class="pill-tag pill_color_Other">Terminals</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="zheng2019">[@zheng2019]</span>
</div>
<div class="col-4">
Graph Drawing by Stochastic Gradient Descent
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="fisheye2019">[@fisheye2019]</span>
</div>
<div class="col-4">
Structure-aware Fisheye Views for Efficient Large Graph Exploration
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="wang2018">[@wang2018]</span>
</div>
<div class="col-4">
Revisiting Stress Majorization as a Unified Framework for Interactive Constrained Graph Visualization
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="batik2022">[@batik2022]</span>
</div>
<div class="col-4">
Shape-Guided Mixed Metro Map Layout
</div>
<div class="col-4">
<a href="#Categorical_nodes"><span class="pill-tag pill_color_Categorical_nodes">Categorical nodes</span></a>
</div>
</div>
<p><br></p>
<section id="subways" class="level4">
<h4 class="anchored" data-anchor-id="subways">Subways</h4>
</section>
<section id="statistics-22" class="level4">
<h4 class="anchored" data-anchor-id="statistics-22">Statistics</h4>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Transportation Networks 720a658bb1914b51910c480d86943e80/four_in_one.svg" width="80%">
</div>
</section>
<section id="description-from-literature-6" class="level4">
<h4 class="anchored" data-anchor-id="description-from-literature-6">Description from Literature</h4>
<blockquote class="blockquote">
<p>Apart from the example of Sydney presented in Section 6, we have evaluated our method for two additional real-world networks: Vienna, which is a rather small network, and London, the oldest and still one of the most complex metro systems in the world. These networks have not been used as examples for previous metro map layout methods, so we can compare only against the official maps of Vienna and London. The size of the metro graphs (including Sydney) is given in Table 3 and ranges from 84 vertices and 8 faces (Vienna) to 308 vertices and 55 faces (London). The table further shows how the removal of degree-2 vertices described in Section 5.1 effectively reduces the number of vertices and edges.</p>
</blockquote>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Transportation Networks 720a658bb1914b51910c480d86943e80/Untitled.png" width="80%">
</div>
</section>
<section id="example-figures-41" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-41">Example Figures</h4>
<p>From “<a href="https://ieeexplore.ieee.org/document/5473229">Drawing and Labeling High-Quality Metro Maps by Mixed-Integer Programming</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Transportation Networks 720a658bb1914b51910c480d86943e80/Untitled 1.png" width="80%">
</div>
<p><strong>Fig. 16.</strong> Unlabeled layout of the London Underground network produced by our method.</p>
</section>
<section id="dimacs-usa-road-networks" class="level4">
<h4 class="anchored" data-anchor-id="dimacs-usa-road-networks">DIMACS USA Road Networks</h4>
</section>
<section id="description-from-literature-7" class="level4">
<h4 class="anchored" data-anchor-id="description-from-literature-7">Description from Literature</h4>
<p>From “<a href="https://ieeexplore.ieee.org/document/4475481">A Treemap Based Method for Rapid Layout of Large Graphs</a>”:</p>
<blockquote class="blockquote">
<p>The other graphs used in this paper are: a graph of network scans, which is a complete graph with edge weights between 0 and 1, but for clarity, edges with weights less than a certain threshold are not shown (Figures 3 and 4, |V|=878,|E|=385003, [19]), a small artificial graph of a grid topology (Figure 5, |V|=16,|E|=24), a large graph of streets in the San Francisco Bay Area (Figure 7, |V|=321,270,|E|=800,172, [8]).</p>
</blockquote>
<p>From “<a href="https://ieeexplore.ieee.org/document/4658143">Rapid Graph Layout Using Space Filling Curves</a>”:</p>
<blockquote class="blockquote">
<p>Finally, the “usafla” dataset (shown in Figure 7) is of the intersections and the streets between them in the state of Florida</p>
</blockquote>
</section>
<section id="example-figures-42" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-42">Example Figures</h4>
<p>From “<a href="https://ieeexplore.ieee.org/document/4475481">A Treemap Based Method for Rapid Layout of Large Graphs</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Transportation Networks 720a658bb1914b51910c480d86943e80/Untitled 2.png" width="80%">
</div>
<p><strong>Fig. 7.</strong> <em>Scalability</em>. Our approach can scale to very large networks while still maintaining interactivity. |V|=321,270,|E|=800,172</p>
<p>From “<a href="https://ieeexplore.ieee.org/document/4658143">Rapid Graph Layout Using Space Filling Curves</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Transportation Networks 720a658bb1914b51910c480d86943e80/Untitled 3.png" width="80%">
</div>
<p><strong>Fig. 7.</strong> Scalability. Our approach can scale to large graphs. This graph is of the streets in the state of Florida, |V| = 1,070,376,|E| = 2,712,798 One small region is expanded to show detail.</p>
<!-- === STOP RENDERING ===

[http://www.diag.uniroma1.it//challenge9/download.shtml](http://www.diag.uniroma1.it//challenge9/download.shtml)
_site:150:21
 -->
</section>
</div>
</div>
</div>
</section>
<section id="sec-lost" class="level3">
<h3 class="anchored" data-anchor-id="sec-lost">Lost and unavailable datasets</h3>
<p>Unfortunately, some of the datasets that were used in the papers in our corpus are lost, or not available anymore. A dataset that is not anymore accessible renders all the papers using it unfortunately not reproducible. Most of the instances we found became lost because of lack of maintenance of the original storage locations: websites, servers, or repositories. This is, unfortunately, a significant issue in science, which is now hopefully gaining more and more attention.</p>
<p>“Loss” of a dataset does not exclusively mean that the entire, original dataset became lost, but also that a paper using a dataset with particular edits—such as interpreting a debate as a graph between the participants—has not documented well enough their process and did not link to the edited dataset. This makes it impossible to reproduce the results of the paper, and is further discussed in <a href="#sec-custom-made-datasets" class="quarto-xref">Section&nbsp;1.1.6</a>.</p>
<p>While we did go through the effort, for each one of them, to recover them and store them on OSF, we could not find anywhere the following list of datasets:</p>
<p><a name="GION"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="GION">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-41-contents" aria-controls="callout-41" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
GION
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-41" class="callout-41-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>The data was originally collected for a Bioinformatics paper, <a href="https://doi.org/10.1186/1471-2105-11-378">Graph-based clustering and characterization of repetitive sequences in next-generation sequencing data</a>. It was then used in <a href="https://link.springer.com/chapter/10.1007/978-3-662-45803-7_10">GION: Interactively Untangling Large Graphs on Wall-Sized Displays</a> as a benchmark for their layout and experiment. However, the links to the original data no longer work. It had <a href="#Sparse"><span class="pill-tag pill_color_Sparse">Sparse</span></a> <a href="#dense"><span class="pill-tag pill_color_dense">dense</span></a>and <a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>graphs.</p>
<div>
<b>Originally found at: </b><a href="http://wcl.ml.unisa.edu.au/graph-untangling/graphs.zip">http://wcl.ml.unisa.edu.au/graph-untangling/graphs.zip</a>
</div>
<br>
<div>
<b>Number of Graphs: </b>8
</div>
<br>
<div>
<b>Size: </b>8 graphs, 1159-7885 nodes, 6424-427406 edges
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Marner2014">[@Marner2014]</span>
</div>
<div class="col-8">
GION: Interactively Untangling Large Graphs on Wall-Sized Displays
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="novak_graph-based_2010">[@novak_graph-based_2010]</span>
</div>
<div class="col-8">
Graph-based clustering and characterization of repetitive sequences in next-generation sequencing data
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Eades2018">[@Eades2018]</span>
</div>
<div class="col-4">
Drawing Big Graphs Using Spectral Sparsification
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Marner2014">[@Marner2014]</span>
</div>
<div class="col-4">
GION: Interactively Untangling Large Graphs on Wall-Sized Displays
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<p><br></p>
<section id="descriptions-from-literature-33" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-33">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.1007/978-3-319-73915-1_22">Drawing Big Graphs Using Spectral Sparsification</a>”:</p>
<blockquote class="blockquote">
<p>this consists of RNA sequence graphs that are used for the analysis of repetitive sequences in sequencing data; these graphs have been used in previous experiments. They are locally dense and globally sparse, and generally have distinctive shapes.</p>
</blockquote>
<p>From “<a href="https://link.springer.com/chapter/10.1007/978-3-662-45803-7_10">GION: Interactively Untangling Large Graphs on Wall-Sized Displays</a>”:</p>
<blockquote class="blockquote">
<p>We selected graphs that come from a real world application where graph visualization is used for data analysis and the graph size and layout quality requirements pose a challenge for state-of-the-art layout methods. Our graph set consists of RNA sequence graphs that are used for the analysis of repetitive sequences in sequencing data [11].</p>
</blockquote>
<blockquote class="blockquote">
<p>Graph properties are provided in Table 1. The set of graphs can be downloaded from http://wcl.ml.unisa.edu.au/graph-untangling/graphs.zip. All graphs have high local density, and a sparse global structure that allows to create layouts far from hairballs that are showing the structure well.</p>
</blockquote>
<div style="text-align: center">
<img src="../data/Benchmark datasets/GION fb235f160e504395a884f5a18fca6f17/Untitled.png" width="80%">
</div>
</section>
<section id="example-figures-43" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-43">Example Figures</h4>
<p>From “<a href="https://link.springer.com/chapter/10.1007/978-3-662-45803-7_10">GION: Interactively Untangling Large Graphs on Wall-Sized Displays</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/GION fb235f160e504395a884f5a18fca6f17/Untitled 1.png" width="80%">
</div>
<p>From “<a href="https://link.springer.com/article/10.1186/1471-2105-11-378">Graph-based clustering and characterization of repetitive sequences in next-generation sequencing data</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/GION fb235f160e504395a884f5a18fca6f17/Untitled 2.png" width="80%">
</div>
<p><strong>Fig. 4. Examples of graph layouts derived from clusters of repetitive sequences</strong>. Graph layouts were calculated using the 3D version of Fruchterman and Reingold algorithm from which a 2D projection is shown. Individual reads are represented by vertices and similar reads are connected by edges. Individual clusters are described further in Table 1.</p>
<!-- === STOP RENDERING ===

**NOTE: eliminate the relationships between different tables**

<div style='text-align: center'><img src='../data/Benchmark datasets/GION%20fb235f160e504395a884f5a18fca6f17/Untitled%203.png' width='80%'></div>

From [GION: Interactively Untangling Large Graphs on Wall-Sized Displays](../Benchmark%20sets%200cc6b5e454304aec98f3b59b1a720476/Literature%20ad87f14e7097454fb2f784e2c8a2797a/Literature%20-%20Algorithm%2012e01bfc60a84007aa7d2d34293e123d/GION%20Interactively%20Untangling%20Large%20Graphs%20on%20Wall%20774568ebdd114a8790c026bfa0b125f0.md) 
Graph properties are provided in Table 1. The set of graphs can be downloaded from

http://wcl.ml.unisa.edu.au/graph-untangling/graphs.zip. All graphs have high local density, and a sparse global structure that allows to create layouts far from hairballs that are showing the structure well

Dead link! [http://wcl.ml.unisa.edu.au/graph-untangling/graphs.zip](http://wcl.ml.unisa.edu.au/graph-untangling/graphs.zip) ← Emailed [michael@20papercups.net](mailto:michael@20papercups.net) (seems to have left academia and this is the email under his personal website)
_site:150:21 -->
</section>
</div>
</div>
</div>
<p><a name="FM3"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="FM3">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-42-contents" aria-controls="callout-42" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
FM3
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-42" class="callout-42-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>FM3 contains trees, path graphs, cycle graphs, scale-free graphs, caterpillar graphs, planar graphs, and generic graphs (connected simple graphs, generated with uniform probability distribution). Nonetheless, download links to data are broken.</p>
<div>
<b>Originally found at: </b><a href="http://www.felicedeluca.com/ply/data.html ">http://www.felicedeluca.com/ply/data.html </a>(broken)
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="felice2017">[@felice2017]</span>
</div>
<div class="col-8">
An Experimental Study on the Ply Number of Straight-Line Drawings”
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Heinsohn2018">[@Heinsohn2018]</span>
</div>
<div class="col-4">
An Interactive Tool to Explore and Improve the Ply Number of Drawings
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a>
</div>
</div>
<p><br></p>
<section id="descriptions-from-literature-34" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-34">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.1007/978-3-319-53925-6_11">An Experimental Study on the Ply Number of Straight-Line Drawings</a>”:</p>
<blockquote class="blockquote">
<p><strong>Graph Datasets.</strong> To understand whether the experimental results are influenced by the structure of the graph we considered several graph families. All graphs are of small or medium size, expressed as the number of their vertices. In some cases, the size and the number of instances used for each graph family depends on the type of question we want to answer (see Sect. 5 for details). We used the following datasets: <strong>Trees.</strong> Generated with uniform probability distribution using Pru ̈fer sequences [28]. <strong>Planar.</strong> Connected simple planar graphs, generated with the OGDF library [6]. <strong>General.</strong> Connected simple graphs, generated with uniform probability distribution. <strong>Scale-free.</strong> Scale-free graphs, generated according to the Barabasi-Albert model [2]. <strong>Caterpillars.</strong> Each caterpillar of n vertices is generated by first creating a path (spine of the caterpillar) of length <span class="math inline">\(k\in [ \frac{n}{4} , \frac{n}{2} ]\)</span> (randomly chosen), and attaching each remaining vertex to a randomly selected vertex of the spine. <strong>Paths, Cycles.</strong> For each desired number of vertices n, there is only one (unlabeled) path and one (unlabeled) cycle of n vertices. <strong>k-ary Trees.</strong> Rooted trees where each node has either 0 or k children. Each tree is generated by starting with a single vertex and then creating k children of a randomly selected leaf, until the desired number n of vertices is achieved. When n cannot be obtained, we use a value close to it.</p>
</blockquote>
<p>From “<a href="https://doi.org/10.1007/978-3-319-73915-1_4">An Interactive Tool to Explore and Improve the Ply Number of Drawings</a>”:</p>
<blockquote class="blockquote">
<p><strong>FM3data</strong> contains 50 caterpillars with 250 to 450 vertices… <strong>FM3data</strong> also contains planar graphs ranging from 250 to 400 vertices and a density ranging from 1.5 to 2… The remaining subset of <strong>FM3data</strong> consists of general graphs with 250 to 450 vertices and the densities 1.5 and 2.5.</p>
</blockquote>
</section>
<section id="example-figures-44" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-44">Example Figures</h4>
<p>From “<a href="https://doi.org/10.1007/978-3-319-53925-6_11">An Experimental Study on the Ply Number of Straight-Line Drawings</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/FM3 b3dc638f5d7541b8b01105483c356e11/Screen_Shot_2023-08-16_at_3.26.01_PM.png" width="80%">
</div>
<!-- === STOP RENDERING ===

Download links seem to be broken…  in website
_site:150:21 -->
</section>
</div>
</div>
</div>
<p><a name="Padia_Stories"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Padia Stories">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-43-contents" aria-controls="callout-43" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Padia Stories
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-43" class="callout-43-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>These consist of fabricated narratives for characters of popular culture franchises, namely “The Witcher” and “Friends”. They were proposed by Kapesh Padia et al.&nbsp;as case studies for <a href="https://doi.org/10.20380/GI2018.05">Yarn: Generating Storyline Visualizations Using HTN Planning</a>. It had <a href="#temporal_event_sequence"><span class="pill-tag pill_color_temporal_event_sequence">temporal event sequences</span></a>and <a href="#layered_graphs"><span class="pill-tag pill_color_layered_graphs">layered</span></a>graphs.</p>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="padia2018">[@padia2018]</span>
</div>
<div class="col-8">
Yarn: Generating Storyline Visualizations Using HTN Planning
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="padia2018">[@padia2018]</span>
</div>
<div class="col-4">
Yarn: generating storyline visualizations using HTN planning
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="padia2019">[@padia2019]</span>
</div>
<div class="col-4">
A system for generating storyline visualizations using hierarchical task network planning
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<p><br></p>
<section id="descriptions-from-literature-35" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-35">Descriptions from Literature</h4>
<p>From ’’<a href="https://doi.org/10.20380/GI2018.05">Yarn: Generating Storyline Visualizations Using HTN Planning</a>”:</p>
<blockquote class="blockquote">
<p>For our first example we use a narrative adapted from the video game Witcher 2 to show the capabilities of Yarn. In our example narrative, the hero, Geralt of Rivia needs to rescue his friend, Triss Merigold, who has mysteriously disappeared. On his journey to find her he must first determine where she was seen last, talk to multiple friends, and resolve a conflict between a troll couple before he can meet her. There are two choice points in this narrative: first, Margot can either choose to help Geralt or refuse to provide help; second, the troll couple can either accept or reject Geralt’s solution to their conflict. Each of these choices affects the outcome of the narrative creating three possible timelines. The first timeline contains events favorable to Geralt. In this timeline, Margot agrees to help Geralt and the troll couple accept Geralt’s solution leading him to find Triss. Fig. 5 shows a visualization of this outcome as the reality timeline for our narrative. We can see that Geralt was able to successfully track Triss’s last known location, gather information from friends, resolve a conflict between the trolls, and meet Triss. Note that in this visualization we are showing the default, reality timeline and all links are drawn with 100% opacity. We can also observe all single-entity events in the narrative in this visualization. In the second timeline, the trolls reject Geralt’s solution to their conflict. Fig. 6 shows a visualization of this unfavorable outcome as diegetic timeline overlaid on top of the reality timeline. While most of the events are same in both timelines, the diegetic timeline ends at the event “Speak to She-Troll.” The reduced opacity of the reality timeline allows the user to compare common sections of the two timelines, even when one of them is smaller than the other. In the third timeline, Margot refuses to help Geralt preventing him from progressing further on his quest. This is the shortest timeline in our narrative.</p>
</blockquote>
<p>From ’’<a href="https://doi.org/10.20380/GI2018.05">Yarn: Generating Storyline Visualizations Using HTN Planning</a>”:</p>
<blockquote class="blockquote">
<p>For our second example we use a fictional narrative scenario based on the popular sitcom Friends. In this narrative, Ross’s plan is to take Rachel on a date. To do so, he must acquire more information about her, find some way of talking to her, ensure she is positively disposed towards him, and eventually ask her out. He can acquire more information about her in a number of ways including calling her mother, or asking Phoebe about her by either call or text. To ensure she is in a positive mood, he can either give her a gift or say nice things to her, and finally he can either ask her out himself or ask for Phoebe’s help. To illustrate causality of events in the narrative timeline, we have set up the operator functions for Rachel such that she is more inclined to say yes to Ross if he talks to her mother rather than requesting Phoebe’s help. Additionally, Rachel’s probability of accepting Ross’s proposal is also dependent on how impressed she is by him when he asks her. There are four choice points in this narrative: three for Ross, and one for Rachel. The choices made by Ross create 12 alternate timelines, each illustrating a different way in which he can ask Rachel out. Towards the end of each timeline Rachel can decide to either reply yes to Ross, or reply no, creating a total of 24 timelines. Fig. 7 shows a visualization where Rachel agrees to go out with Ross in both timelines. While the timelines result in the same outcome, and visually look the same, we can inspect the labels on top of the events in Ross’s timeline to identify the differences. Lighter labels correspond to the reality timeline, while darker labels correspond to the diegetic timeline. In the reality timeline, Ross decided to give a gift to Rachel after talking to her mom, while in the diegetic timeline he decided to act friendly with her. Fig. 8 shows another visualization for the same narrative. In this example we can see the effect of causal conditions set in the operator functions for Rachel. In the reality timeline, Rachel decided not to go out with Ross because he asked Phoebe for help by messaging her. On the other hand, she agreed to go out with him in the diegetic timeline because in this case Ross decided to talk to her mother.</p>
</blockquote>
</section>
<section id="example-figures-45" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-45">Example Figures</h4>
<p>From ’’<a href="https://doi.org/10.20380/GI2018.05">Yarn: Generating Storyline Visualizations Using HTN Planning</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Padia Stories c02a38b9ea5e40bfb682ae77e2ac23af/Untitled.png" width="80%">
</div>
</section>
</div>
</div>
</div>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Internet Mapping Project">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-44-contents" aria-controls="callout-44" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Internet Mapping Project
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-44" class="callout-44-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>The original Internet Mapping Project started at Bell lab in 1998. It was then transferred to the Lumeta Corporation, however, links to this corporation are broken. A gallery of images still remains publicly available, hosted by Bill Cheswick on this personal website.</p>
<div>
<b>Originally found at: </b><a href="http://www.cheswick.com/ches/map/
">http://www.cheswick.com/ches/map/ </a><a href="http://www.lumeta.com/company/mapping.html ">http://www.lumeta.com/company/mapping.html </a>(broken)
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="burch1999">[@burch1999]</span>
</div>
<div class="col-8">
Mapping the Internet
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Gansner2005b">[@Gansner2005b]</span>
</div>
<div class="col-4">
Topological fisheye views for visualizing large graphs
</div>
<div class="col-4">
<a href="#Generic"><span class="pill-tag pill_color_Generic">Generic</span></a><a href="#Other"><span class="pill-tag pill_color_Other">High degree</span></a><a href="#Large"><span class="pill-tag pill_color_Large">Large</span></a>
</div>
</div>
<p><br></p>
<section id="description-from-literature-8" class="level4">
<h4 class="anchored" data-anchor-id="description-from-literature-8">Description from Literature</h4>
<p>From “<a href="https://doi.org/10.1109/TVCG.2005.66">Topological fisheye views for visualizing large graphs</a>”:</p>
<blockquote class="blockquote">
<p>Fig. 7 shows two topological fisheye views of a published Internet map [4]. It is a large tree (|V|=87,931, |E|=87,930) made by tracing connections from a central probe to all reachable IP subnets. The full layout of this graph in Fig. 7a is too dense to read. In contrast, our approach, seen in Fig. 7b and Fig. 7c, yields useful views by focusing on different portions of the graph.</p>
</blockquote>
</section>
<section id="example-figures-46" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-46">Example Figures</h4>
<p>From “<a href="http://www.cheswick.com/ches/map/gallery/index.html">Internet Mapping Project: Map Gallery</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Internet Mapping Project fe59286f353d4841a2a328fdd6160b2f/isp-ss.gif" width="80%">
</div>
<p>Layout showing the major ISPs.</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Internet Mapping Project fe59286f353d4841a2a328fdd6160b2f/Untitled.png" width="80%">
</div>
<p>From “<a href="https://doi.org/10.1109/TVCG.2005.66">Topological fisheye views for visualizing large graphs</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Internet Mapping Project fe59286f353d4841a2a328fdd6160b2f/Untitled 1.png" width="80%">
</div>
<p><strong>Fig. 7.</strong> This Internet map ((|V|=87,931, |E|=87,930)) is too large to visualize as a flat structure. Two topological fisheye views are shown. The focused sections in orange are the original graph. Peripheral areas, in blue, are simplified.</p>
</section>
</div>
</div>
</div>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Medical Patient Records">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-45-contents" aria-controls="callout-45" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Medical Patient Records
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-45" class="callout-45-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Dataset was not found. It contained <a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a> <a href="#Other"><span class="pill-tag pill_color_Other">Multivariate</span></a>data and <a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Pre-existing clusters</span></a></p>
<div>
<b>Download formatted data: </b>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="NanCao2011">[@NanCao2011]</span>
</div>
<div class="col-4">
DICON: Interactive Visual Analysis of Multidimensional Clusters
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Other"><span class="pill-tag pill_color_Other">Multivariate</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<p><br></p>
<section id="description-from-literature-9" class="level4">
<h4 class="anchored" data-anchor-id="description-from-literature-9">Description from Literature</h4>
<p>From “<a href="https://doi.org/10.1109/TVCG.2011.188">DICON: Interactive Visual Analysis of Multidimensional Clusters</a>”:</p>
<blockquote class="blockquote">
<p>We also applied DICON within the healthcare domain to visualize a dataset containing more than 10,000 patient records. The data includes claims, labs, pharmacy, and patient profile information. To augment this data, we applied a patient similarity algorithm to compute patient similarity scores across multiple dimensions (e.g., diagnoses, lab results, etc.). We also indexed the patient records to make the data searchable.</p>
</blockquote>
</section>
<section id="example-figures-47" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-47">Example Figures</h4>
<p>From “<a href="https://doi.org/10.1109/TVCG.2011.188">DICON: Interactive Visual Analysis of Multidimensional Clusters</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Medical Patient Records 9bb4f4fc1e8242b2905eb286a0e00143/Untitled.png" width="80%">
</div>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Medical Patient Records 9bb4f4fc1e8242b2905eb286a0e00143/Untitled 1.png" width="80%">
</div>
<!-- === STOP RENDERING ===

Reached out to authors? ← saw this note but not Connor nor I
_site:150:21 -->
</section>
</div>
</div>
</div>
<p><a name="Assorted_Fiscal_Network"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Assorted Fiscal Network">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-46-contents" aria-controls="callout-46" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Assorted Fiscal Network
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-46" class="callout-46-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>The dataset is cited as being from an evaluation of TaxNet from Didimo et al.&nbsp;The authors claim how this system was adopted by the Italian Revenue Agency on a pilot program in Tuscany, where all the evaluations happened. The dataset is implied to be internal, with nodes corresponding to individual tax payers and the authors mentioning anonymizing data themselves for security reasons. It contained <a href="#Categorical_nodes"><span class="pill-tag pill_color_Categorical_nodes">Categorical nodes</span></a> <a href="#Other"><span class="pill-tag pill_color_Other">High degree</span></a>and <a href="#Labeled_nodes"><span class="pill-tag pill_color_Labeled_nodes">Labeled nodes</span></a></p>
<div>
<b>Originally found at: </b>cannot find
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="DIDIMO201871">[@DIDIMO201871]</span>
</div>
<div class="col-8">
A visual analytics system to support tax evasion discovery
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="Angori2019">[@Angori2019]</span>
</div>
<div class="col-4">
ChordLink: A New Hybrid Visualization Model0
</div>
<div class="col-4">
<a href="#Categorical_nodes"><span class="pill-tag pill_color_Categorical_nodes">Categorical nodes</span></a><a href="#Other"><span class="pill-tag pill_color_Other">High degree</span></a><a href="#Labeled_nodes"><span class="pill-tag pill_color_Labeled_nodes">Labeled nodes</span></a>
</div>
</div>
<p><br></p>
<section id="descriptions-from-the-literature-2" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-the-literature-2">Descriptions from the Literature</h4>
<p>From “<a href="https://link.springer.com/chapter/10.1007/978-3-030-35802-0_22">CHORDLINK: A New Hybrid Visualization Model</a>”:</p>
<blockquote class="blockquote">
<p>We considered a real network of taxpayers and their economic transactions. The network is provided by the IRV (Italian Revenue Agency) and refers to a portion of data for the fiscal year 2014, consisting of 174 subjects with high fiscal risk and 200 economic transactions between them [<a href="https://link.springer.com/chapter/10.1007/978-3-030-35802-0_22#ref-CR11">11</a>].</p>
</blockquote>
<p>From “<a href="https://www.sciencedirect.com/science/article/pii/S0167923618300563?via%3Dihub#s0015">A visual analytics system to support tax evasion discovery</a>”:</p>
<blockquote class="blockquote">
<p>The data sources queried by the IRV officers are modeled in TaxNet as a unified network <em>G</em>. Each node <em>v</em> of <em>G</em> is a single taxpayer, which can be either an individual or a legal person, like a private company or a public institution. Many attributes are associated with <em>v</em>, including the type of economic activity, the geographic location and territorial scope, the declared income, the amount of VAT credits/debts and of VAT refunded/paid, and the amount of economic exchange within the European Union. The edges of <em>G</em> are directed edges. An edge (<em>u</em>,<em>v</em>) can model different types of relationships between <em>u</em> and <em>v</em>.</p>
</blockquote>
</section>
<section id="example-figures-48" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-48">Example Figures</h4>
<p>From <a href="https://link.springer.com/chapter/10.1007/978-3-030-35802-0_22">CHORDLINK: A New Hybrid Visualization Model</a></p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Assorted Fiscal Network 6610534d6d3b48c5a119b04659153258/Untitled.png" width="80%">
</div>
<p><strong>Fig. 4.</strong> A visualization obtained by selecting some communities in a node-link diagram.</p>
<!-- == STOP RENDERING ==

[https://www.sciencedirect.com/science/article/abs/pii/S0167923618300563?via%3Dihub](https://www.sciencedirect.com/science/article/abs/pii/S0167923618300563?via%3Dihub) ← italian revenue agency
_site:150:21 -->
</section>
</div>
</div>
</div>
<!-- <a name="Debates"></a>

::: {.callout-note title="Debates" collapse=true appearance="minimal"}

Created by researchers at the University of Konstanz for the [LingVis](https://lingvis.io/) framework and [VisArgue](http://www.visargue.uni-konstanz.de/en/) project. First appeared in [NEREx: Named-Entity Relationship Exploration in Multi-Party Conversations](https://doi.org/10.1111/cgf.13181).
The authors did not respond to requests to make this dataset available. While the debate transcripts are publicly available, the entity relationship data generated in [NEREx: Named-Entity Relationship Exploration in Multi-Party Conversations](https://doi.org/10.1111/cgf.13181) is not. It has <a href="#Categorical_nodes"><span class="pill-tag pill_color_Categorical_nodes">Categorical nodes</span></a> <a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a> <a href="#Labeled_nodes"><span class="pill-tag pill_color_Labeled_nodes">Labeled nodes</span></a>and <a href="#Weighted_edges"><span class="pill-tag pill_color_Weighted_edges">Weighted edges</span></a>.

Note: we found the Debates dataset published as a student assignment for a course at the University of Konstanz, now available on OSF [at this link](https://osf.io/rp483/metadata/?format=datacite-json). However, additional non-trivial processing was made to the data to generate the entity relationship data used in the paper. Because of this, we still consider the dataset as lost.

<div><b>Origin paper: </b></div>
<div class="row"><div class="col-4">[@elassady2017]</div><div class="col-8">NEREx: Named-Entity Relationship Exploration in Multi-Party Conversations</div></div><br>
<div><b>Usage examples: </b></div>
<div class="row"><div class="col-4">[@elassady2017]</div><div class="col-4">NEREx: Named-Entity Relationship Exploration in Multi-Party Conversations</div><div class="col-4"><a href="#Categorical_nodes"><span class="pill-tag pill_color_Categorical_nodes">Categorical nodes</span></a><a href="#Directed_edges"><span class="pill-tag pill_color_Directed_edges">Directed edges</span></a><a href="#Labeled_nodes"><span class="pill-tag pill_color_Labeled_nodes">Labeled nodes</span></a><a href="#Weighted_edges"><span class="pill-tag pill_color_Weighted_edges">Weighted edges</span></a></div></div><br>

#### Description from Literature

From “[NEREx: Named-Entity Relationship Exploration in Multi-Party Conversations](https://doi.org/10.1111/cgf.13181)”:

> In order to explore the relations between entities, we use a distance-restricted model for creating entity-pairs. As described in Section 2, prior work considers relations between entities either as semantic relations based on linguistic knowledge or regards two entities as related due to their presence in the same document. The often ungrammatical structure of utterances in verbatim text transcripts (including non-standard lexical items, syntactic patterns, interruptions, repetitions, and crosstalk) requires a simple extralinguistic model. To overcome these limitations, we introduce a model that bounds the scope in which we consider two entities to be related, using a distance threshold (*maxDist*). Our method creates a pair of entities if the entities appear in the same sentences within *maxDist* words of one another.

> However, not only are frequent entity-pairs important, but also pairs that are semantically similar (e.g., *cut $5 trillion*, *cut $4 trillion*). These often present opposing speaker opinions about a given topic. To maximize the chance that entity-pairs represent true semantic relations, we set the default value of *maxDist* to a low value (5 words). To vary the granularity of the analysis, the parameter can be changed interactively depending on the analysis task and data. Nevertheless, some infrequent longer-distance entity-pairs are discovered which do not represent salient relations. To reduce their impact, we calculate the average observed distance (in words) for each entity pair. We use both frequency and average distance in the visualization to reflect the strength of the entity-pair.

#### Example Figures

From “[NEREx: Named-Entity Relationship Exploration in Multi-Party Conversations](https://doi.org/10.1111/cgf.13181)”:

<div style='text-align: center'><img src='../data/Benchmark datasets/Debates%2054eb2cd4ba2c48eb945bd49e5f4bbf4f/cgf13181-fig-0004-m.jpg' width='80%'></div>

Entity graph of the first first presidential debate between Trump and Clinton

<div style='text-align: center'><img src='../data/Benchmark datasets/Debates%2054eb2cd4ba2c48eb945bd49e5f4bbf4f/cgf13181-fig-0006-m.jpg' width='80%'></div>

Entity graph of the same debate, focusing on the concept of taxes

:::  -->
<p><a name="Car_Features"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Car Features">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-47-contents" aria-controls="callout-47" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Car Features
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-47" class="callout-47-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>The origin of this dataset is not disclosed by the author. It is <a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a> has <a href="#clusters"><span class="pill-tag pill_color_clusters">pre-existing clusters</span></a>and it is <a href="#Other"><span class="pill-tag pill_color_Other">Multivariate</span></a></p>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="NanCao2011">[@NanCao2011]</span>
</div>
<div class="col-4">
DICON: Interactive Visual Analysis of Multidimensional Clusters
</div>
<div class="col-4">
<a href="#Clusters_pre-existing"><span class="pill-tag pill_color_Clusters_pre-existing">Clusters (pre-existing)</span></a><a href="#Other"><span class="pill-tag pill_color_Other">Multivariate</span></a><a href="#Spatial"><span class="pill-tag pill_color_Spatial">Spatial</span></a>
</div>
</div>
<p><br></p>
<section id="descriptions-from-literature-36" class="level4">
<h4 class="anchored" data-anchor-id="descriptions-from-literature-36">Descriptions from Literature</h4>
<p>From “<a href="https://doi.org/10.1109/TVCG.2011.188">DICON: Interactive Visual Analysis of Multidimensional Clusters</a>”:</p>
<blockquote class="blockquote">
<p>We applied DICON to a cars dataset which has also been used to evaluate both parallel coordinates (PCP) and scatter plot matrices (SPM) (see Fig. 8(a) and (b)). The cars dataset contains 407 cars described by 7 different dimensions from which we selected 5 quantitative dimensions as features. The two remaining dimensions, year and origin, were used as additional attributes.</p>
</blockquote>
</section>
<section id="example-figures-49" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-49">Example Figures</h4>
<p>From “<a href="https://doi.org/10.1109/TVCG.2011.188">DICON: Interactive Visual Analysis of Multidimensional Clusters</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Car Features 3365100498bc4a1eb77bcadcace2b2fa/IMG_0010.jpeg" width="80%">
</div>
</section>
</div>
</div>
</div>
<p><a name="Business Processes"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Business Processes">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-48-contents" aria-controls="callout-48" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Business Processes
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-48" class="callout-48-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>The Origin Paper collects various graphs from a few of sources. Some of these datasets are from international business process intelligence challenges, and others are specific processes from companies in a certain domain (hospital, road traffic management, pay processes, etc.). Dataset links in supplemental materials are broken. After contacting the author, they mentioned that the other datasets were internal and could not be shared. It had <a href="#Labeled_nodes"><span class="pill-tag pill_color_Labeled_nodes">Labeled nodes</span></a>and <a href="#layered-graphs"><span class="pill-tag pill_color_layered-graphs">layered graphs</span></a></p>
<div>
<b>Originally found at: </b><a href="https://robinmennens.github.io/Portfolio/stableprocessgraphs.html">https://robinmennens.github.io/Portfolio/stableprocessgraphs.html</a>
</div>
<br>
<div>
<b>Size: </b>13 graphs from 11 to 50 nodes and up to 772 edges
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="mennens2019">[@mennens2019]</span>
</div>
<div class="col-8">
A Stable Graph Layout Algorithm for Processes
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="mennens2019">[@mennens2019]</span>
</div>
<div class="col-4">
A stable graph layout algorithm for processes
</div>
<div class="col-4">
<a href="#Labeled_nodes"><span class="pill-tag pill_color_Labeled_nodes">Labeled nodes</span></a><a href="#Layered_graphs"><span class="pill-tag pill_color_Layered_graphs">Layered graphs</span></a><a href="#N-layers"><span class="pill-tag pill_color_N-layers">N-layers</span></a>
</div>
</div>
<p><br></p>
<section id="description-from-literature-10" class="level4">
<h4 class="anchored" data-anchor-id="description-from-literature-10">Description From Literature</h4>
<p>From <a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.13723">A stable graph layout algorithm for processes</a>:</p>
<blockquote class="blockquote">
<p>We use 13 datasets (see supplementary material), which represent real-world processes. For every dataset, we ran 500 tests on both methods. Each test involved a pair of randomly generated graphs&nbsp;&nbsp;<span class="math inline">\(G_1, G_2 \subseteq \bar{G}\)</span> &nbsp;, which were obtained by randomly removing a subset of the edges <span class="math inline">\(\bar{E}\)</span> and then removing all nodes <span class="math inline">\(n \in \bar{V}\)</span> that became disconnected. Consequently,&nbsp;<span class="math inline">\(G_1\)</span> and&nbsp;<span class="math inline">\(G_2\)</span> are random sub-graphs of&nbsp;<span class="math inline">\(\bar{G}\)</span> .</p>
</blockquote>
<p>Table from Supplementary Materials:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Business Processes 9d09430927704fcf8fdf1ee4e00cabbc/Untitled.png" width="80%">
</div>
</section>
<section id="example-figures-50" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-50">Example Figures</h4>
<p>From <a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.13723">A stable graph layout algorithm for processes</a>:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Business Processes 9d09430927704fcf8fdf1ee4e00cabbc/Untitled 1.png" width="80%">
</div>
<!-- == STOP RENDERING ==

“Hi! Thanks for reaching out! The dataset used is an internal one that I cannot share unfortunately. Some other datasets were public datasets and I think I did include urls in my references” 

-urls are broken -->
</section>
</div>
</div>
</div>
<p>This last one is not available as an individual download, but still navigable on the internet and could potentially be crawled:</p>
<p><a name="Mathematics Genealogy"></a></p>
<div class="callout callout-style-simple callout-note no-icon callout-titled" title="Mathematics Genealogy">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-49-contents" aria-controls="callout-49" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Mathematics Genealogy
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-49" class="callout-49-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Database from a project started by Harry Coonce and now supported by the North Dakota State University. Refer to the <a href="https://mathgenealogy.org/contact.php ">https://mathgenealogy.org/contact.php</a> for the data, which consists of doctoral graduates in mathematics and some adjacent fields, alongside properties like the degree granting institution and tittle of thesis. Entries also have relations based on advisors. It has <a href="#Nonplanar"><span class="pill-tag pill_color_Nonplanar">Non-planar</span></a> <a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a>and <a href="#Trees"><span class="pill-tag pill_color_Trees">Tree-like</span></a>graphs.</p>
<div>
<b>Originally found at: </b><a href="https://genealogy.math.ndsu.nodak.edu/">https://genealogy.math.ndsu.nodak.edu/</a>
</div>
<br>
<div>
<b>Size: </b>291014 records as of May 2023. In 2016 the graph consisted of 200037 nodes with 3.8% percent of them being isolated, and 1962 components of size two. The largest component contained 180094 nodes.
</div>
<br>
<div>
<b>Origin paper: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="math2006-mw">[@math2006-mw]</span>
</div>
<div class="col-8">
The Mathematics Genealogy Project
</div>
</div>
<br>
<div>
<b>Usage examples: </b>
</div>
<div class="row">
<div class="col-4">
<span class="citation" data-cites="gray2023">[@gray2023]</span>
</div>
<div class="col-4">
Visualizing Evolving Trees
</div>
<div class="col-4">
<a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic</span></a><a href="#Dynamic"><span class="pill-tag pill_color_Dynamic">Dynamic (discrete)</span></a>
</div>
</div>
<p><br></p>
<section id="description-from-literature-11" class="level4">
<h4 class="anchored" data-anchor-id="description-from-literature-11">Description from Literature</h4>
<p>From “<a href="https://link.springer.com/chapter/10.1007/978-3-031-22203-0_23">Visualizing Evolving Trees</a>”:</p>
<blockquote class="blockquote">
<p><strong>The Mathematics Genealogy:</strong> shows advisor-advisee relationships in the world of mathematics, stretching back to the middle ages [<a href="https://link.springer.com/chapter/10.1007/978-3-031-22203-0_23#ref-CR36">36</a>]. The dataset includes the thesis titles, students, advisors, dates, and number of descendants. The total number of nodes is around 260,000 and is continuously updated. While this data is not quite a tree (or even connected, or planar), we extract a subset to create a tree with 500 nodes. The maximum node degree of this tree is 5 and the radius is 14.</p>
</blockquote>
</section>
<section id="example-figures-51" class="level4">
<h4 class="anchored" data-anchor-id="example-figures-51">Example Figures</h4>
<p>From <a href="https://genealogy.math.ndsu.nodak.edu/">The Mathematics Genealogy Project</a>:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Mathematics Genealogy 081ba4645aac48c19719b16033c637f3/Untitled.png" width="80%">
</div>
<p>From “<a href="https://link.springer.com/chapter/10.1007/978-3-031-22203-0_23">Visualizing Evolving Trees</a>”:</p>
<div style="text-align: center">
<img src="../data/Benchmark datasets/Mathematics Genealogy 081ba4645aac48c19719b16033c637f3/Untitled 1.png" width="80%">
</div>
<p><strong>Fig. 4:</strong> Layouts from DynNoSlice, DynaGraph, Dagre, Radial, ImPrEd, DynaCola and DynaSafe of the same evolving math genealogy tree; each row adds six new nodes.</p>
<!-- == STOP RENDERING ==

[https://github.com/abureyanahmed/evolving_tree/tree/main/dataset](https://github.com/abureyanahmed/evolving_tree/tree/main/dataset)

They make you request the data individually and in the linked form they have a note about not redistributing data:

[mgp-permission-agreement.pdf](Mathematics%20Genealogy%20081ba4645aac48c19719b16033c637f3/mgp-permission-agreement.pdf)

<div style='text-align: center'><img src='../data/Benchmark datasets/Mathematics%20Genealogy%20081ba4645aac48c19719b16033c637f3/Untitled%202.png' width='80%'></div>

reached out to them 5/18/2023

It is non-planar because multiple advisors can advise the same person or a person can have multiple advisors! Infact, there is a K3,3 in it

In July 2016, Cosmin Ionita and Pat Quillen of 
MathWorks used MATLAB to analyze the Math Genealogy Project graph. At 
the time, the genealogy graph contained 200,037 vertices. There were 
7639 (3.8%) isolated vertices and 1962 components of size two 
(advisor-advisee pairs where we have no information about the advisor). 
The largest component of the genealogy graph contained 180,094 vertices,
 accounting for 90% of all vertices in the graph. The main component has
 7323 root vertices (individuals with no advisor) and 137,155 leaves 
(mathematicians with no students), accounting for 76.2% of the vertices 
in this component. The next largest component sizes were 81, 50, 47, 34,
 34, 33, 31, 31, and 30.
For historical comparisonn, we also have data from June 2010, when 
Professor David Joyner of the United States Naval Academy asked for data
 from our database to analyze it as a graph. At the time, the genealogy 
graph had 142,688 vertices. Of these, 7,190 were isolated vertices (5% 
of the total). The largest component had 121,424 vertices (85% of the 
total number). The next largest component had 128 vertices. The next 
largest component sizes were 79, 61, 45, and 42. The most frequent size 
of a nontrivial component was 2; there were 1937 components of size 2. 
The component with 121,424 vertices had 4,639 root verticies, i.e., 
mathematicians for whom the advisor is currently unknown.

The Mathematics Genealogy Project graph is nonplanar. Thanks to 
Professor Ezra Brown of Virginia Tech for assisting in finding the 
subdivision of *K*3,3 depicted below. The green vertices
 form one color class and the yellow ones form the other. Interestingly,
 Gauß is the only vertex that needs to be connected by paths with more 
than one edge.
_site:150:21 -->
</section>
</div>
</div>
</div>
</section>
<section id="sec-custom-made-datasets" class="level3">
<h3 class="anchored" data-anchor-id="sec-custom-made-datasets">Custom-made Datasets</h3>
<p>In the data we collected, we also found several instances of custom-made datasets. We consider custom-made datasets either edits to pre-existing datasets, where the authors found it necessary to either split or modify the dataset in a particular way, or datasets completely made up from scratch using random generators or custom-made code. This can happen in cases where the authors of a paper needed a dataset containing particular characteristics which was not easy to find in the wild, so a new dataset was crafted.</p>
<p>For instance, consider the case where the authors of a paper develop an algorithm that works on hypergraphs. They want to test that the algorithm works, and test its performance on hypergraphs of various sizes, but datasets containing hypergraphs are difficult to find. For this reason, the authors craft one dataset synthetically, or take a pre-existing dataset and edit it so that it now contains hyperedges.</p>
<p>We split custom-made datasets in three categories, with their occurrences in the corpus of papers illustrated below:</p>
<div class="cell">
<div class="sourceCode cell-code hidden" id="cb6" data-startfrom="3068" data-source-offset="0"><pre class="sourceCode js code-with-copy"><code class="sourceCode javascript" style="counter-reset: source-line 3067;"><span id="cb6-3068"><a href="#cb6-3068" aria-hidden="true" tabindex="-1"></a>{ </span>
<span id="cb6-3069"><a href="#cb6-3069" aria-hidden="true" tabindex="-1"></a>  <span class="kw">let</span> d <span class="op">=</span> []</span>
<span id="cb6-3070"><a href="#cb6-3070" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-3071"><a href="#cb6-3071" aria-hidden="true" tabindex="-1"></a>  <span class="kw">let</span> allbenchmarkdatasets <span class="op">=</span> [<span class="op">...</span><span class="kw">new</span> <span class="bu">Set</span>(literature<span class="op">.</span><span class="fu">map</span>(l <span class="kw">=&gt;</span> l[<span class="st">"Dataset tag clean"</span>]<span class="op">.</span><span class="fu">split</span>(<span class="st">","</span>)<span class="op">.</span><span class="fu">map</span>(d <span class="kw">=&gt;</span> d<span class="op">.</span><span class="fu">trim</span>()))<span class="op">.</span><span class="fu">flat</span>())]<span class="op">.</span><span class="fu">filter</span>(a <span class="kw">=&gt;</span> a <span class="op">!=</span> <span class="st">""</span>)</span>
<span id="cb6-3072"><a href="#cb6-3072" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3073"><a href="#cb6-3073" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (<span class="kw">let</span> dataset <span class="kw">of</span> allbenchmarkdatasets){</span>
<span id="cb6-3074"><a href="#cb6-3074" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="op">!</span>dataset<span class="op">.</span><span class="fu">includes</span>(<span class="st">"Custom"</span>)) <span class="cf">continue</span><span class="op">;</span></span>
<span id="cb6-3075"><a href="#cb6-3075" aria-hidden="true" tabindex="-1"></a>    <span class="kw">let</span> literature_entries_with_this_dataset <span class="op">=</span> literature<span class="op">.</span><span class="fu">filter</span>(l <span class="kw">=&gt;</span> l[<span class="st">"Dataset tag clean"</span>]<span class="op">.</span><span class="fu">includes</span>(dataset))</span>
<span id="cb6-3076"><a href="#cb6-3076" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3077"><a href="#cb6-3077" aria-hidden="true" tabindex="-1"></a>    d<span class="op">.</span><span class="fu">push</span>({<span class="st">"dataset"</span><span class="op">:</span> dataset<span class="op">,</span> <span class="st">"count"</span><span class="op">:</span> literature_entries_with_this_dataset<span class="op">.</span><span class="at">length</span>})</span>
<span id="cb6-3078"><a href="#cb6-3078" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb6-3079"><a href="#cb6-3079" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-3080"><a href="#cb6-3080" aria-hidden="true" tabindex="-1"></a>  <span class="cf">return</span> Plot<span class="op">.</span><span class="fu">plot</span>({</span>
<span id="cb6-3081"><a href="#cb6-3081" aria-hidden="true" tabindex="-1"></a>    <span class="dt">color</span><span class="op">:</span> {</span>
<span id="cb6-3082"><a href="#cb6-3082" aria-hidden="true" tabindex="-1"></a>      <span class="co">// legend: true,</span></span>
<span id="cb6-3083"><a href="#cb6-3083" aria-hidden="true" tabindex="-1"></a>      <span class="dt">type</span><span class="op">:</span> <span class="st">"categorical"</span><span class="op">,</span></span>
<span id="cb6-3084"><a href="#cb6-3084" aria-hidden="true" tabindex="-1"></a>      <span class="co">// domain: d3.range(10).map((d, i) =&gt; `Category ${i + 1}`),</span></span>
<span id="cb6-3085"><a href="#cb6-3085" aria-hidden="true" tabindex="-1"></a>      <span class="dt">scheme</span><span class="op">:</span> <span class="st">"Tableau10"</span> <span class="co">// use the "accent" scheme</span></span>
<span id="cb6-3086"><a href="#cb6-3086" aria-hidden="true" tabindex="-1"></a>    }<span class="op">,</span></span>
<span id="cb6-3087"><a href="#cb6-3087" aria-hidden="true" tabindex="-1"></a>    <span class="dt">y</span><span class="op">:</span> {</span>
<span id="cb6-3088"><a href="#cb6-3088" aria-hidden="true" tabindex="-1"></a>      <span class="co">// tickFormat: "s"</span></span>
<span id="cb6-3089"><a href="#cb6-3089" aria-hidden="true" tabindex="-1"></a>    }<span class="op">,</span></span>
<span id="cb6-3090"><a href="#cb6-3090" aria-hidden="true" tabindex="-1"></a>    <span class="dt">marks</span><span class="op">:</span> [</span>
<span id="cb6-3091"><a href="#cb6-3091" aria-hidden="true" tabindex="-1"></a>      Plot<span class="op">.</span><span class="fu">barX</span>(d<span class="op">,</span> {<span class="dt">x</span><span class="op">:</span> <span class="st">"count"</span><span class="op">,</span> <span class="dt">y</span><span class="op">:</span> <span class="st">"dataset"</span><span class="op">,</span> <span class="dt">fill</span><span class="op">:</span> <span class="st">"dataset"</span><span class="op">,</span><span class="dt">inset</span><span class="op">:</span> <span class="dv">2</span><span class="op">,</span> <span class="dt">sort</span><span class="op">:</span> {<span class="dt">y</span><span class="op">:</span> <span class="st">"x"</span><span class="op">,</span> <span class="dt">reverse</span><span class="op">:</span> <span class="kw">true</span>}})<span class="op">,</span></span>
<span id="cb6-3092"><a href="#cb6-3092" aria-hidden="true" tabindex="-1"></a>      Plot<span class="op">.</span><span class="fu">axisY</span>({<span class="dt">label</span><span class="op">:</span> <span class="kw">null</span><span class="op">,</span> <span class="dt">lineWidth</span><span class="op">:</span> <span class="dv">12</span><span class="op">,</span> <span class="dt">marginLeft</span><span class="op">:</span> <span class="dv">150</span>})<span class="op">,</span></span>
<span id="cb6-3093"><a href="#cb6-3093" aria-hidden="true" tabindex="-1"></a>    ]</span>
<span id="cb6-3094"><a href="#cb6-3094" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb6-3095"><a href="#cb6-3095" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="fig-ojs-cell-6" class="cell-output cell-output-display quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ojs-cell-6-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div id="ojs-cell-6" data-nodetype="expression">

</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ojs-cell-6-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: The chart shows what type of edits were made to datasets, or how they were generated. Non-Replicable edits or generation of datasets where the majority in the corpus of papers we collected.
</figcaption>
</figure>
</div>
</div>
<p><strong>Replicable</strong> datasets indicate cases where the authors have given enough information so that the experiment can be replicated exactly as it was run by the authors of a paper, or closely enough that the results obtained reflect the published ones very closely. This includes cases where either the authors published the entire dataset they used, they published the code they used to generate the dataset, or include an exact description of the steps they took to generate it.</p>
<p><strong>Reproducible</strong> datasets are cases where the authors described the steps they took to generate and/or edit their datasets, but not in-depth enough so that the exact same graphs can be reproduced, and did not redistribute it. Results can still be reproduces somewhat closely if the authors took care to report enough information about their graphs.</p>
<p>For <strong>non-replicable</strong> datasets, we indicate cases where the authors did not distribute their datasets and did not include enough information in the paper so that their results could be replicated.</p>
<p>This information is closely tied to the distribution of supplemental material in papers, that is shown in the chart below:</p>
<div class="cell">
<div class="sourceCode cell-code hidden" id="cb7" data-startfrom="1" data-source-offset="0"><pre class="sourceCode js code-with-copy"><code class="sourceCode javascript"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>{ </span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="kw">let</span> d <span class="op">=</span> []</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="kw">let</span> allbenchmarkdatasets <span class="op">=</span> [<span class="op">...</span><span class="kw">new</span> <span class="bu">Set</span>(literature<span class="op">.</span><span class="fu">map</span>(l <span class="kw">=&gt;</span> l[<span class="st">"Supplemental material (Multi-select)"</span>]<span class="op">.</span><span class="fu">split</span>(<span class="st">","</span>)<span class="op">.</span><span class="fu">map</span>(d <span class="kw">=&gt;</span> d<span class="op">.</span><span class="fu">trim</span>()))<span class="op">.</span><span class="fu">flat</span>())]<span class="op">.</span><span class="fu">filter</span>(a <span class="kw">=&gt;</span> a <span class="op">!=</span> <span class="st">""</span>)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (<span class="kw">let</span> dataset <span class="kw">of</span> allbenchmarkdatasets){</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>    <span class="kw">let</span> literature_entries_with_this_dataset <span class="op">=</span> literature<span class="op">.</span><span class="fu">filter</span>(l <span class="kw">=&gt;</span> l[<span class="st">"Supplemental material (Multi-select)"</span>]<span class="op">.</span><span class="fu">includes</span>(dataset))</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (dataset <span class="op">==</span> <span class="st">"Yes"</span>) <span class="cf">continue</span><span class="op">;</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>    <span class="co">// capitalize first letter</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>    dataset <span class="op">=</span> dataset<span class="op">.</span><span class="fu">charAt</span>(<span class="dv">0</span>)<span class="op">.</span><span class="fu">toUpperCase</span>() <span class="op">+</span> dataset<span class="op">.</span><span class="fu">slice</span>(<span class="dv">1</span>)</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>    d<span class="op">.</span><span class="fu">push</span>({<span class="st">"dataset"</span><span class="op">:</span> dataset<span class="op">,</span> <span class="st">"count"</span><span class="op">:</span> literature_entries_with_this_dataset<span class="op">.</span><span class="at">length</span>})</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">return</span> Plot<span class="op">.</span><span class="fu">plot</span>({</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>    <span class="dt">color</span><span class="op">:</span> {</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>      <span class="co">// legend: true,</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>      <span class="dt">type</span><span class="op">:</span> <span class="st">"categorical"</span><span class="op">,</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>      <span class="co">// domain: d3.range(10).map((d, i) =&gt; `Category ${i + 1}`),</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>      <span class="dt">scheme</span><span class="op">:</span> <span class="st">"Tableau10"</span> <span class="co">// use the "accent" scheme</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>    }<span class="op">,</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>    <span class="dt">y</span><span class="op">:</span> {</span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>      <span class="co">// tickFormat: "s"</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>    }<span class="op">,</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>    <span class="dt">marks</span><span class="op">:</span> [</span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>      Plot<span class="op">.</span><span class="fu">barX</span>(d<span class="op">,</span> {<span class="dt">x</span><span class="op">:</span> <span class="st">"count"</span><span class="op">,</span> <span class="dt">y</span><span class="op">:</span> <span class="st">"dataset"</span><span class="op">,</span> <span class="dt">fill</span><span class="op">:</span> <span class="st">"dataset"</span><span class="op">,</span><span class="dt">inset</span><span class="op">:</span> <span class="dv">2</span><span class="op">,</span> <span class="dt">sort</span><span class="op">:</span> {<span class="dt">y</span><span class="op">:</span> <span class="st">"x"</span><span class="op">,</span> <span class="dt">reverse</span><span class="op">:</span> <span class="kw">true</span>}})<span class="op">,</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>      Plot<span class="op">.</span><span class="fu">axisY</span>({<span class="dt">label</span><span class="op">:</span> <span class="kw">null</span><span class="op">,</span> <span class="dt">lineWidth</span><span class="op">:</span> <span class="dv">12</span><span class="op">,</span> <span class="dt">marginLeft</span><span class="op">:</span> <span class="dv">150</span>})<span class="op">,</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>    ]</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="fig-ojs-cell-7" class="cell-output cell-output-display quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-ojs-cell-7-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div id="ojs-cell-7" data-nodetype="expression">

</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-ojs-cell-7-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: What is distributed as supplemental material with papers? Most papers just distribute an extended version of their content, while another large chunk in our analysis did not distribute any.
</figcaption>
</figure>
</div>
</div>
<p>This discussion is part of a larger discourse on research replicability, that is gaining traction in the scientific community. The ACM, for instance, has a policy on artifact review and badging, where authors are encouraged to submit their artifacts for review, and if they pass, they receive a badge that indicates the artifact is available for review. This is a step towards making research more replicable and reproducible, and we hope that our work will contribute to this effort.</p>
<p>See, e.g., ACM’s definitions at <a href="https://www.acm.org/publications/policies/artifact-review-and-badging-current">https://www.acm.org/publications/policies/artifact-review-and-badging-current</a>.</p>


</section>
</section>
</section>

</main> <!-- /main -->
<script type="ojs-module-contents">
{"contents":[{"methodName":"interpret","cellName":"ojs-cell-1","inline":false,"source":"{ \n    let d = []\n    let allbenchmarkdatasets = [...new Set(literature.map(l => l[\"Dataset tag relations\"].split(\"(\")[0].trim()))].filter(a => a != \"\")\n  \n    allbenchmarkdatasets = allbenchmarkdatasets.filter(b => !b.includes(\"Custom\"))\n    allbenchmarkdatasets = allbenchmarkdatasets.slice(0, 20)\n    //console.log(allbenchmarkdatasets)\n\n    for (let dataset of allbenchmarkdatasets){\n        let literature_entries_with_this_dataset = literature.filter(l => l[\"Dataset tag relations\"].includes(dataset))\n        d.push({\"dataset\": dataset, \"count\": literature_entries_with_this_dataset.length})\n    }\n  \n    return Plot.plot({\n        color: {\n      // legend: true,\n      // type: \"categorical\",\n      // domain: d3.range(10).map((d, i) => `Category ${i + 1}`),\n        // range: [\"steelblue\"],\n          scheme: \"Tableau10\" // use the \"accent\" scheme\n        },\n        y: {\n      // tickFormat: \"s\"\n        },\n        marks: [\n            Plot.barX(d, {x: \"count\", y: \"dataset\", fill: \"dataset\", inset: 2, sort: {y: \"x\", reverse: true}}),\n            Plot.axisY({label: null, lineWidth: 12, marginLeft: 150}),\n        ]\n  })\n}\n"},{"methodName":"interpret","cellName":"ojs-cell-2","inline":false,"source":"{ \n  let d = []\n  \n  let collection_type = benchmark_datasets.map(a => a[\"Type of Collection\"]);\n\n  for (let c of [... new Set(collection_type)]){\n    if (c == \"Skip\" || c == \"Missed it\") continue;\n    let count = collection_type.filter(a => a == c).length;\n    if (c.includes(\"No report\")) c = \"Established Network Repository\";\n    if (c.includes(\"Lost\")) continue;\n    d.push({\"collection_type\": c, \"count\": count})\n  }\n  \n  return Plot.plot({\n    color: {\n      scheme: \"Tableau10\" \n    },\n    marks: [\n      Plot.barX(d, {x: \"count\", y: \"collection_type\", fill: \"collection_type\", inset: 2, sort: {y: \"x\", reverse: true}}),\n      Plot.axisY({label: null, lineWidth: 12, marginLeft: 150}),\n    ]\n  })\n}\n"},{"methodName":"interpret","cellName":"ojs-cell-3","inline":false,"source":"{ \n  let d = []\n  \n  let collection_type = benchmark_datasets.map(a => a[\"Type of Collection\"]);\n\n  for (let c of [... new Set(collection_type)]){\n    if (c == \"Skip\" || c == \"Missed it\") continue;\n    let count = collection_type.filter(a => a == c).length;\n    if (c.includes(\"No report\")) c = \"Established Network Repository\";\n    if (!c.includes(\"Lost\")) continue;\n    d.push({\"collection_type\": c, \"count\": count})\n  }\n  \n  return Plot.plot({\n    color: {\n      scheme: \"PiYg\" \n    },\n    marks: [\n      Plot.barX(d, {x: \"count\", y: \"collection_type\", fill: \"collection_type\", inset: 2, sort: {y: \"x\", reverse: true}}),\n      Plot.axisY({label: null, lineWidth: 12, marginLeft: 150}),\n    ]\n  })\n}\n"},{"methodName":"interpret","cellName":"ojs-cell-4","inline":false,"source":"{\n    //console.log(benchmark_datasets)\n    \n    // sorry for all the vars instead of let but this was split before into different cells\n    var uniqueDataTagSet = new Set(); // set for the tags\n    var uniquePaperTagSet = new Set();\n    var dataSetNames = []; // list of names of datasets\n    var paperNames = []\n    var dataMap = {} // maps from the tags to the individual datasets\n    var paperMap = {} // maps from the tags to the individual papers\n    var nameMap = {} // maps from the cleaned name to the actual name containing potentially special characters\n\n    //console.log(literature)\n    // \"Citation name\": \"bekos2022\"\n    // \"Graph feature\": \"hypergraphs, spatial\"\n    \n    const tagMapping = {\n    \"hypergraphs\": \"Hypergraphs\",\n    \"spatial\": \"Spatial\",\n    \"layered graphs\": \"N-layers\",\n    \"generic\": \"Generic\",\n    \"large\": \"Large\",\n    \"dynamic (discrete)\": \"Dynamic\",\n    \"dynamic\": \"Dynamic\",\n    \"dynamic (continuous)\": \"Dynamic\",\n    \"clusters (generated)\": \"Clusters (generated)\",\n    \"clusters (pre-existing)\": \"Clusters (pre-existing)\",\n    \"bundled edges (generated)\": \"Bundled edges (generated)\",\n    \"directed edges\": \"Directed edges\",\n    \"weighted edges\": \"Weighted edges\",\n    \"weighted nodes\": \"Weighted nodes\",\n    \"hierarchical\": \"Hierarchical\",\n    \"categorical nodes\": \"Categorical nodes\",\n    \"trees\": \"Trees\",\n    \"labeled nodes\": \"Labeled nodes\",\n    \"bipartite\": \"Bipartite\",\n    \"tripartite\": \"Bipartite\",\n    \"labeled edges\": \"Labeled edges\",\n    \"planar\": \"Planar\",\n    \"sparse\": \"Sparse\",\n    \"K-planar\": \"Planar\",\n    \"2-planar\": \"Planar\",\n    \"acyclic\": \"Acyclic\",\n};\n\n    \n    benchmark_datasets.forEach((entry) => {\n        let name = name_cleanup(entry.Name)\n        nameMap[name] = entry.Name\n        dataSetNames.push(name);\n        var parts = entry[\"graph features handled\"].split(\",\").map(item => item.trim())\n        for (var part of parts) {\n            if (part !== \"\") {\n                uniqueDataTagSet.add(part);\n                if (!dataMap[part]) {\n                    dataMap[part] = new Set();\n                }\n                dataMap[part].add(name);\n            }\n        }\n    })\n     \n    \n     \n  \n    literature.forEach((paper) => {\n\n      paperNames.push(paper[\"Citation name\"]);\n      var parts = paper[\"Graph feature\"].split(\",\").map(item => item.trim())\n      for (var part of parts) {\n        if (part !== \"\") {\n          uniquePaperTagSet.add(part);\n          \n          if (part in tagMapping) {\n              part = tagMapping[part]\n          }\n          if (!paperMap[part]) {\n              paperMap[part] = new Set();\n          }\n          \n          paperMap[part].add(paper[\"Citation name\"]);\n        }\n      }\n    })\n  \n    // Add keys not present in the paperTags \n    Object.keys(dataMap).forEach((key) => {\n      if (!paperMap[key]) {\n        paperMap[key] = new Set();\n      }\n    })\n  \n  \n    \n    var category_tags = {\n        0: \"Layered\",\n        2: \"Hierarchical\",\n        4: \"Attributes\",\n        11: \"Structures\",\n        21: \"Other\",\n    }\n    \n    // if you change this ordering or move around values also change the category map above!!!\n    //var uniqueDataTags = [...uniqueDataTagSet]\n    var uniqueDataTags = [\n        // Layered\n        \"Bipartite\",\n        \"N-layers\",\n        \n        // Hierarchical\n        \"Trees\",\n        \"Hierarchical\",\n\n        // Attributes\n        \"Dynamic\",\n        \"Weighted edges\",\n        \"Weighted nodes\",\n        \"Labeled nodes\",\n        \"Categorical nodes\",\n        \"Signed\",\n        \"Spatial\",\n        \n        // Structures\n        \"Clusters (pre-existing)\",\n        \"Clusters (generated)\",\n        \"Directed edges\",\n        \"Hypergraphs\",\n        \"Bundled edges (generated)\",\n        \"Partition\",\n        \"Multigraph\",\n        \"Acyclic\",\n        \"Mesh\",\n\n        // Other\n        \"Generic\",\n        \"Planar\",\n        \"Nonplanar\",\n        \"Known crossing number\",\n        \"Large\",\n        \"Dense\",\n        \"Sparse\",\n    ]\n    var uniquePaperTags = [...uniquePaperTagSet]  \n    //console.log(uniqueDataTags)\n    //console.log(uniquePaperTagSet)\n    //console.log(dataMap)\n    //console.log(paperMap)\n    //console.log(dataSetNames)\n    //console.log(paperNames)\n\n    // remove empty entries\n    paperNames = paperNames.filter(str => str.trim() !== \"\")\n    \n\n\n    var checkBoxContainer = document.getElementById('checkbox-container');\n    var linksListContainer = document.getElementById('datasetlinkslist-container')\n    var paperLinksContainer = document.getElementById('paperlinkslist-container')\n    var checkboxStates = [];  // Array to keep track of checkbox states (booleans)\n\n    // needs to be in here since we access so many \"global\" variables\n    function displayDataSetLinksList() {\n        //console.log(checkboxStates); // current states to check which dataset links to display (index-based)\n\n        const filteredTags = uniqueDataTags.map((item, index) => checkboxStates[index] ? item : undefined)\n            .filter(item => item !== undefined);\n\n        //console.log(filteredTags)\n\n        var final_datasets = []\n        var final_data_inverse = []\n\n        var final_papers = []\n        var final_paper_inverse = []\n\n        if (filteredTags.length == 0) {\n            final_datasets = dataSetNames;\n            final_papers = paperNames;\n        } else {\n            // Retrieve an array of sets from the provided keys\n            const dataSets = filteredTags.map(key => dataMap[key]);\n            const paperSets = filteredTags.map(key => paperMap[key]);\n            \n            // Calculate the intersection of all sets\n            const dataIntersection = dataSets.reduce((acc, set) => new Set([...acc].filter(x => set.has(x))));\n            const paperIntersection = paperSets.reduce((acc, set) => new Set([...acc].filter(x => set.has(x))));\n            \n            final_datasets = [...dataIntersection] // white list\n            final_papers = [...paperIntersection]\n\n            // Now, compute the difference between dataSetNames and final_datasets\n            const dataDifference = new Set(dataSetNames.filter(x => !final_datasets.includes(x)));\n            const paperDifference = new Set(paperNames.filter(x => !final_papers.includes(x)));\n\n            final_data_inverse = [...dataDifference]; // black list\n            final_paper_inverse = [...paperDifference]\n        }\n\n        //console.log(\"show: \" + final_datasets) // white list \n        //console.log(\"show: \" + final_papers)\n\n        final_datasets.forEach((name) => {\n            //console.log(name + '-link')\n            let currLinkDiv = document.getElementById(name + '-link')\n            currLinkDiv.style.display = 'block';  // Clear the display property to default ('block')\n        })\n\n        final_data_inverse.forEach((name) => {\n            //console.log(name + '-link')\n            let currLinkDiv = document.getElementById(name + '-link')\n            currLinkDiv.style.display = 'none';  // Hide the div\n        })\n\n        final_papers.forEach((name) => {\n            //console.log(name + '-link')\n            let currLinkDiv = document.getElementById(name + '-link')\n            currLinkDiv.style.display = 'block';  // Clear the display property to default ('block')\n        })\n\n        final_paper_inverse.forEach((name) => {\n            //console.log(name + '-link')\n            let currLinkDiv = document.getElementById(name + '-link')\n            currLinkDiv.style.display = 'none';  // Hide the div\n        })\n\n      /*\n      // this is how the it ran earlier and it was used to generate the static html links above this\n      \n      if (final_datasets.length == 0) {\n      // display message that no dataset contains all these tags\n      linksListContainer.innerHTML = '';\n      } else {\n      \n      linksListContainer.innerHTML = '';\n      \n      final_datasets.forEach((currDataName) => {\n      const anchor = document.createElement('a');\n      anchor.href = '#' + currDataName\n      anchor.classList.add('dataset-link')\n      anchor.textContent = nameMap[currDataName]\n      \n      const div = document.createElement('div');\n      div.id = currDataName + '-link'\n      div.appendChild(anchor);\n      \n      linksListContainer.appendChild(div);\n      })\n      }\n      \n      // and here for the paper names\n      paperNames = paperNames.filter(str => str.trim() !== \"\").sort();\n      paperNames.forEach((currPaperName) => {\n\n          const div = document.createElement('div');\n          div.id = currPaperName + '-link'\n          div.textContent = `[@${currPaperName}]`\n\n          paperLinksContainer.appendChild(div);\n      });*/\n      \n    }\n   \n  // Function to add a new checkbox with a styled span label\n  function addCheckbox(index, name) {\n      const newCheckbox = document.createElement('input');\n      newCheckbox.type = 'checkbox';\n      newCheckbox.id = 'checkbox' + index;\n      newCheckbox.style.marginRight = '8px'; \n      newCheckbox.classList.add('form-check-input');\n\n      // Creating the span that will act as a label\n      const span = document.createElement('span');\n      span.classList.add('pill-tag');\n      span.classList.add(`pill_color_${name_cleanup(name)}`);\n      //span.style.backgroundColor = get_css_color_for_tag(name)\n      //span.style.backgroundColor = graph_feature_tags_colors[index % graph_feature_tags_colors.length];  \n      span.textContent = name;\n\n      const label = document.createElement('label');\n      label.className = 'd-flex align-items-center justify-content-start';\n      label.htmlFor = newCheckbox.id;\n      label.style.cursor = 'pointer';  // Change the cursor to a pointer on hover\n      label.appendChild(newCheckbox);\n      label.appendChild(span);  // Append the span to the label\n\n      const div = document.createElement('div');\n      div.className = 'form-check d-flex align-items-center';  // Ensure the checkbox and label are aligned\n      div.appendChild(label);\n\n      checkBoxContainer.appendChild(div);\n\n      // Update state array and add event listener\n      checkboxStates.push(false);\n      newCheckbox.addEventListener('change', () => {\n          checkboxStates[index] = newCheckbox.checked;\n          displayDataSetLinksList()\n      });\n  }\n\n  // Automatically add checkboxes on load for each tag, we cannot reorder after this!\n  uniqueDataTags.forEach((entry, index) => {\n      if (category_tags.hasOwnProperty(index)) {\n          const span = document.createElement('span');\n          span.style.opacity = \"50%\";  // Setting it to be less obnoxious\n          span.textContent = \"\\u2012 \" + category_tags[index] + \"\\u2012 \";\n\n          const div = document.createElement('div');\n          div.className = 'd-flex align-items-center';  \n          div.appendChild(span);\n\n          checkBoxContainer.appendChild(div);\n      }\n      addCheckbox(index, entry);\n  })\n\n  // initial call to display all datasets (now a bit redundant but adds the \"display\" style-attribute)\n  displayDataSetLinksList()\n\n}\n"},{"methodName":"interpret","cellName":"ojs-cell-5","inline":false,"source":"\nmake_sparkline(\"Pert_DAG\")\n"},{"methodName":"interpret","cellName":"ojs-cell-6","inline":false,"source":"{ \n  let d = []\n  \n  let allbenchmarkdatasets = [...new Set(literature.map(l => l[\"Dataset tag clean\"].split(\",\").map(d => d.trim())).flat())].filter(a => a != \"\")\n\n  for (let dataset of allbenchmarkdatasets){\n    if (!dataset.includes(\"Custom\")) continue;\n    let literature_entries_with_this_dataset = literature.filter(l => l[\"Dataset tag clean\"].includes(dataset))\n\n    d.push({\"dataset\": dataset, \"count\": literature_entries_with_this_dataset.length})\n  }\n  \n  return Plot.plot({\n    color: {\n      // legend: true,\n      type: \"categorical\",\n      // domain: d3.range(10).map((d, i) => `Category ${i + 1}`),\n      scheme: \"Tableau10\" // use the \"accent\" scheme\n    },\n    y: {\n      // tickFormat: \"s\"\n    },\n    marks: [\n      Plot.barX(d, {x: \"count\", y: \"dataset\", fill: \"dataset\",inset: 2, sort: {y: \"x\", reverse: true}}),\n      Plot.axisY({label: null, lineWidth: 12, marginLeft: 150}),\n    ]\n  })\n}\n"},{"methodName":"interpret","cellName":"ojs-cell-7","inline":false,"source":"{ \n  let d = []\n  \n  let allbenchmarkdatasets = [...new Set(literature.map(l => l[\"Supplemental material (Multi-select)\"].split(\",\").map(d => d.trim())).flat())].filter(a => a != \"\")\n\n  for (let dataset of allbenchmarkdatasets){\n    let literature_entries_with_this_dataset = literature.filter(l => l[\"Supplemental material (Multi-select)\"].includes(dataset))\n\n    if (dataset == \"Yes\") continue;\n    // capitalize first letter\n    dataset = dataset.charAt(0).toUpperCase() + dataset.slice(1)\n\n    d.push({\"dataset\": dataset, \"count\": literature_entries_with_this_dataset.length})\n  }\n  \n  return Plot.plot({\n    color: {\n      // legend: true,\n      type: \"categorical\",\n      // domain: d3.range(10).map((d, i) => `Category ${i + 1}`),\n      scheme: \"Tableau10\" // use the \"accent\" scheme\n    },\n    y: {\n      // tickFormat: \"s\"\n    },\n    marks: [\n      Plot.barX(d, {x: \"count\", y: \"dataset\", fill: \"dataset\",inset: 2, sort: {y: \"x\", reverse: true}}),\n      Plot.axisY({label: null, lineWidth: 12, marginLeft: 150}),\n    ]\n  })\n}\n"}]}
</script>
<script type="module">
if (window.location.protocol === "file:") { alert("The OJS runtime does not work with file:// URLs. Please use a web server to view this document."); }
window._ojs.paths.runtimeToDoc = "../../chapters";
window._ojs.paths.runtimeToRoot = "../..";
window._ojs.paths.docToRoot = "..";
window._ojs.selfContained = false;
window._ojs.runtime.interpretFromScriptTags();
</script>
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/VisDunneRight\.github\.io\/jovi-benchmark-sets\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>